// @dhis2-ui/button/build/cjs/dropdown-button/dropdown-button.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnClick = exports.WithInitialFocus = exports.WithIcon = exports.WithComponent = exports.WithChildren = exports.DisabledWithOnClick = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onClick = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'DropdownButton'
};
const Default = () => /*#__PURE__*/_react.default.createElement(_index.DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Content")
}, "Button");
exports.Default = Default;
const WithOnClick = () => /*#__PURE__*/_react.default.createElement(_index.DropdownButton, {
  name: "Button",
  value: "default",
  onClick: window.onClick,
  component: /*#__PURE__*/_react.default.createElement("div", null, "Content")
}, "Button");
exports.WithOnClick = WithOnClick;
const WithChildren = () => /*#__PURE__*/_react.default.createElement(_index.DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component")
}, "I am a child");
exports.WithChildren = WithChildren;
const WithComponent = () => /*#__PURE__*/_react.default.createElement(_index.DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "I am a component")
});
exports.WithComponent = WithComponent;
const WithIcon = () => /*#__PURE__*/_react.default.createElement(_index.DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "I am a component"),
  icon: 'Icon'
});
exports.WithIcon = WithIcon;
const WithInitialFocus = () => /*#__PURE__*/_react.default.createElement(_index.DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Content"),
  initialFocus: true
});
exports.WithInitialFocus = WithInitialFocus;
const DisabledWithOnClick = () => /*#__PURE__*/_react.default.createElement(_index.DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Content"),
  onClick: window.onClick,
  disabled: true
});
exports.DisabledWithOnClick = DisabledWithOnClick;
