// @dhis2-ui/select/build/cjs/select/input-placeholder.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputPlaceholder = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const InputPlaceholder = _ref => {
  let {
    placeholder,
    className,
    dataTest
  } = _ref;
  if (!placeholder) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["1027033961", [_uiConstants.colors.grey600]]]) + " " + (className || "")
  }, placeholder, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1027033961",
    dynamic: [_uiConstants.colors.grey600]
  }, [`div.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey600};-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}`]));
};
exports.InputPlaceholder = InputPlaceholder;
InputPlaceholder.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  className: _propTypes.default.string,
  placeholder: _propTypes.default.string
};
