// @dhis2-ui/select/build/cjs/select/input-clear-button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputClearButton = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _tooltip = require("@dhis2-ui/tooltip");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const InputClearButton = _ref => {
  let {
    onClear,
    clearText,
    className,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
    openDelay: 500,
    content: clearText
  }, /*#__PURE__*/_react.default.createElement("button", {
    "data-test": dataTest,
    onClick: onClear,
    type: "button",
    className: _style.default.dynamic([["2606883952", [_uiConstants.colors.grey500, _uiConstants.colors.grey800, _uiConstants.colors.grey200, _uiConstants.theme.focus]]]) + " " + (className || "")
  }, /*#__PURE__*/_react.default.createElement("svg", {
    width: "16",
    height: "16",
    viewBox: "0 0 16 16",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    className: _style.default.dynamic([["2606883952", [_uiConstants.colors.grey500, _uiConstants.colors.grey800, _uiConstants.colors.grey200, _uiConstants.theme.focus]]])
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16Zm3.536-5.879a1 1 0 1 1-1.415 1.414l-2.12-2.12-2.122 2.121a1 1 0 1 1-1.415-1.414L6.586 8 4.464 5.878A1 1 0 1 1 5.88 4.464L8 6.586l2.121-2.121a1 1 0 1 1 1.415 1.414L9.415 8l2.12 2.121Z",
    fill: "none",
    className: _style.default.dynamic([["2606883952", [_uiConstants.colors.grey500, _uiConstants.colors.grey800, _uiConstants.colors.grey200, _uiConstants.theme.focus]]])
  })), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2606883952",
    dynamic: [_uiConstants.colors.grey500, _uiConstants.colors.grey800, _uiConstants.colors.grey200, _uiConstants.theme.focus]
  }, ["button.__jsx-style-dynamic-selector{background:none;border:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;cursor:pointer;height:24px;width:24px;border-radius:3px;}", `button.__jsx-style-dynamic-selector svg.__jsx-style-dynamic-selector path.__jsx-style-dynamic-selector{fill:${_uiConstants.colors.grey500};}`, `button.__jsx-style-dynamic-selector:hover svg.__jsx-style-dynamic-selector path.__jsx-style-dynamic-selector{fill:${_uiConstants.colors.grey800};}`, `button.__jsx-style-dynamic-selector:hover{background:${_uiConstants.colors.grey200};}`, `button.__jsx-style-dynamic-selector:focus{outline:2px solid ${_uiConstants.theme.focus};}`])));
};
exports.InputClearButton = InputClearButton;
InputClearButton.propTypes = {
  clearText: _propTypes.default.string.isRequired,
  dataTest: _propTypes.default.string.isRequired,
  onClear: _propTypes.default.func.isRequired,
  className: _propTypes.default.string
};
