// @dhis2-ui/select/build/cjs/select/menu-wrapper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuWrapper = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _layer = require("@dhis2-ui/layer");
var _popper = require("@dhis2-ui/popper");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const MenuWrapper = _ref => {
  let {
    children,
    dataTest,
    maxHeight = '280px',
    menuWidth,
    onClick,
    selectRef
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_layer.Layer, {
    onBackdropClick: onClick,
    transparent: true,
    disablePortal: true
  }, /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    reference: selectRef,
    placement: "bottom-start",
    observeReferenceResize: true
  }, /*#__PURE__*/_react.default.createElement("div", {
    "data-test": `${dataTest}-menuwrapper`,
    className: _style.default.dynamic([["1363246346", [menuWidth, maxHeight, _uiConstants.colors.white, _uiConstants.colors.grey200, _uiConstants.elevations.e300]]])
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1363246346",
    dynamic: [menuWidth, maxHeight, _uiConstants.colors.white, _uiConstants.colors.grey200, _uiConstants.elevations.e300]
  }, [`div.__jsx-style-dynamic-selector{width:${menuWidth};height:auto;max-height:${maxHeight};overflow:auto;background:${_uiConstants.colors.white};border:1px solid ${_uiConstants.colors.grey200};border-radius:3px;box-shadow:${_uiConstants.elevations.e300};}`]))));
};
exports.MenuWrapper = MenuWrapper;
MenuWrapper.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  menuWidth: _propTypes.default.string.isRequired,
  selectRef: _propTypes.default.object.isRequired,
  children: _propTypes.default.node,
  maxHeight: _propTypes.default.string,
  onClick: _propTypes.default.func
};
