// @dhis2-ui/popper/build/cjs/popper.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.VirtualElementRef = exports.TopStart = exports.TopEnd = exports.Top = exports.RightStart = exports.RightEnd = exports.Right = exports.RTL = exports.LeftStart = exports.LeftEnd = exports.Left = exports.ElementRef = exports.BottomStart = exports.BottomEnd = exports.Bottom = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _react = _interopRequireWildcard(require("react"));
var _popper = require("./popper.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const description = `
A tool for adding additional information or content outside of the document flow, used for example in the Tooltip or Popover components.

Since it's built using [Popper.js](https://popper.js.org/docs/v2/) and [react-popper](https://popper.js.org/react-popper/), some of that functionality can be accessed through the props of this component, like modifiers.

\`\`\`js
import { Popper } from '@dhis2/ui'
\`\`\`

_**Note**: Some of the stories may not look right on this page. View those examples in the 'Canvas' tab instead._
`;
var _default = exports.default = {
  title: 'Popper',
  component: _popper.Popper,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    placement: {
      ..._uiConstants.sharedPropTypes.popperPlacementArgType
    },
    reference: {
      ..._uiConstants.sharedPropTypes.popperReferenceArgType
    }
  }
};
const boxStyle = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  width: 400,
  height: 400,
  backgroundColor: 'aliceblue'
};
const referenceElementStyle = {
  width: 130,
  height: 50,
  backgroundColor: 'cadetblue',
  textAlign: 'center',
  padding: 6
};
const popperStyle = {
  width: 110,
  height: 30,
  backgroundColor: 'lightblue',
  textAlign: 'center',
  padding: 6
};
const Template = args => {
  const ref = (0, _react.useRef)(null);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "box",
    style: boxStyle
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "reference-element",
    style: referenceElementStyle,
    ref: ref
  }, "Reference Element"), /*#__PURE__*/_react.default.createElement(_popper.Popper, _extends({}, args, {
    reference: ref
  }), /*#__PURE__*/_react.default.createElement("div", {
    style: popperStyle
  }, args.placement)));
};
const Top = exports.Top = Template.bind({});
Top.args = {
  placement: 'top'
};
const TopStart = exports.TopStart = Template.bind({});
TopStart.args = {
  placement: 'top-start'
};
const TopEnd = exports.TopEnd = Template.bind({});
TopEnd.args = {
  placement: 'top-end'
};
const Bottom = exports.Bottom = Template.bind({});
Bottom.args = {
  placement: 'bottom'
};
const BottomStart = exports.BottomStart = Template.bind({});
BottomStart.args = {
  placement: 'bottom-start'
};
const BottomEnd = exports.BottomEnd = Template.bind({});
BottomEnd.args = {
  placement: 'bottom-end'
};
const Right = exports.Right = Template.bind({});
Right.args = {
  placement: 'right'
};
const RightStart = exports.RightStart = Template.bind({});
RightStart.args = {
  placement: 'right-start'
};
const RightEnd = exports.RightEnd = Template.bind({});
RightEnd.args = {
  placement: 'right-end'
};
const Left = exports.Left = Template.bind({});
Left.args = {
  placement: 'left'
};
const LeftStart = exports.LeftStart = Template.bind({});
LeftStart.args = {
  placement: 'left-start'
};
const LeftEnd = exports.LeftEnd = Template.bind({});
LeftEnd.args = {
  placement: 'left-end'
};
const ElementRef = args => {
  const anchor = document.createElement('div');
  document.body.appendChild(anchor);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "box",
    style: {
      ...boxStyle,
      marginBottom: '500px'
    }
  }, /*#__PURE__*/_react.default.createElement(_popper.Popper, _extends({}, args, {
    reference: anchor
  }), /*#__PURE__*/_react.default.createElement("div", {
    style: popperStyle
  }, args.placement)));
};
exports.ElementRef = ElementRef;
ElementRef.args = {
  placement: 'left-end'
};
ElementRef.parameters = {
  docs: {
    source: {
      type: 'code'
    }
  }
};
const VirtualElementRef = args => {
  const virtualElement = {
    getBoundingClientRect: () => ({
      width: 0,
      height: 0,
      top: 100,
      right: 0,
      bottom: 0,
      left: 200,
      x: 200,
      y: 100
    })
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "box",
    style: {
      ...boxStyle,
      marginBottom: '500px'
    }
  }, /*#__PURE__*/_react.default.createElement(_popper.Popper, _extends({}, args, {
    reference: virtualElement
  }), /*#__PURE__*/_react.default.createElement("div", {
    style: popperStyle
  }, args.placement)));
};
exports.VirtualElementRef = VirtualElementRef;
VirtualElementRef.args = {
  placement: 'left-end'
};
VirtualElementRef.parameters = {
  docs: {
    source: {
      type: 'code'
    }
  }
};
const RTL = args => {
  (0, _react.useEffect)(() => {
    document.documentElement.setAttribute('dir', 'rtl');
    return () => {
      document.documentElement.setAttribute('dir', 'ltr');
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement("div", {
    dir: "rtl"
  }, /*#__PURE__*/_react.default.createElement("span", null, "If dir=rtl, `left` and `right` placement are reversed"), /*#__PURE__*/_react.default.createElement(Template, _extends({}, args, {
    placement: "left"
  })), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(Template, _extends({}, args, {
    placement: "right-start"
  })));
};
exports.RTL = RTL;
