// @dhis2-ui/table/build/cjs/data-table/data-table.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Toolbars = exports.SelectableRows = exports.ScrollingDataTableWithToolbars = exports.RTL = exports.LongCellContentLargeCells = exports.LongCellContent = exports.Loading = exports.LargeCells = exports.InlineFiltering = exports.FixedHeaderAndTwoColumns = exports.FixedHeader = exports.FixedFirstColumn = exports.ExpandableContent = exports.DraggableRows = exports.Default = exports.ColumnHeaderSorting = exports.CellStyling = exports.BorderedCells = void 0;
var _box = require("@dhis2-ui/box");
var _checkbox = require("@dhis2-ui/checkbox");
var _input = require("@dhis2-ui/input");
var _react = _interopRequireWildcard(require("react"));
var _dataTableBody = require("./data-table-body.js");
var _dataTableCell = require("./data-table-cell.js");
var _dataTableColumnHeader = require("./data-table-column-header/data-table-column-header.js");
var _dataTableFoot = require("./data-table-foot.js");
var _dataTableHead = require("./data-table-head.js");
var _dataTableRow = require("./data-table-row/data-table-row.js");
var _dataTableToolbar = require("./data-table-toolbar.js");
var _dataTable = require("./data-table.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const subtitle = 'Used to display information in a standard, effective way.';
const description = `
Should be used with multiple DataTable-related child components - see the datatable and examples below.

\`\`\`js
import {
    DataTable,
    DataTableToolbar,
    DataTableHead,
    DataTableBody,
    DataTableFoot,
    DataTableRow,
    DataTableCell,
    DataTableColumnHeader,
} from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'DataTable',
  component: _dataTable.DataTable,
  // Add subcomponents to the args datatable
  subcomponents: {
    DataTableToolbar: _dataTableToolbar.DataTableToolbar,
    DataTableHead: _dataTableHead.DataTableHead,
    DataTableBody: _dataTableBody.DataTableBody,
    DataTableFoot: _dataTableFoot.DataTableFoot,
    DataTableRow: _dataTableRow.DataTableRow,
    DataTableCell: _dataTableCell.DataTableCell,
    DataTableColumnHeader: _dataTableColumnHeader.DataTableColumnHeader
  },
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    // hide control from `children` because that can't be live edited
    children: {
      control: {
        disable: true
      }
    },
    // hide child component controls completely to avoid confusion
    bordered: {
      datatable: {
        disable: true
      }
    },
    large: {
      datatable: {
        disable: true
      }
    },
    draggable: {
      datatable: {
        disable: true
      }
    }
  }
};
const BasicTemplate = _ref => {
  let {
    bordered,
    large,
    draggable,
    bodyProps,
    ...args
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, draggable && /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    large: large
  }), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    large: large
  }, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    large: large
  }, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    large: large
  }, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    large: large
  }, "Last updated"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, bodyProps, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    draggable: draggable
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    draggable: draggable
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    draggable: draggable
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large,
    bordered: bordered
  }, "02/06/2007"))), /*#__PURE__*/_react.default.createElement(_dataTableFoot.DataTableFoot, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    colSpan: draggable ? '5' : '4'
  }, "Footer content"))));
};
const Default = exports.Default = BasicTemplate.bind({});
Default.args = {};
const Loading = exports.Loading = BasicTemplate.bind({});
Loading.args = {
  bodyProps: {
    loading: true
  }
};
const BorderedCells = exports.BorderedCells = BasicTemplate.bind({});
BorderedCells.args = {
  bordered: true
};
const LargeCells = exports.LargeCells = BasicTemplate.bind({});
LargeCells.args = {
  large: true
};
const DraggableRows = exports.DraggableRows = BasicTemplate.bind({});
DraggableRows.args = {
  draggable: true
};
const IndividualCellTemplate = args => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Header"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Header"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Header"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Header"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  tag: "th"
}, "Active Cells"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Inactive cell"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  active: true,
  onClick: () => {
    console.log('You clicked on a DataTableCell');
  }
}, "Acive Cell with onClick"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Inactive cell")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  tag: "th"
}, "Alignment"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  align: "left"
}, "Left"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  align: "center"
}, "Center"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  align: "right"
}, "Right")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  tag: "th"
}, "Built in formatting"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  error: true
}, "Error"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  valid: true
}, "Valid"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  muted: true
}, "Muted")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  tag: "th",
  backgroundColor: "lightgreen"
}, "Custom background (lightgreen)"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  backgroundColor: "#eff"
}, "#eff"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  active: true,
  backgroundColor: "yellow"
}, "yellow"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "None")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
  selected: true
}, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  tag: "th"
}, "Custom background (on selected row )"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  backgroundColor: "#eff"
}, "#eff"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  backgroundColor: "yellow"
}, "yellow"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "None"))));
const CellStyling = exports.CellStyling = IndividualCellTemplate.bind({});
CellStyling.args = {};
const ToolbarsTemplate = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_dataTableToolbar.DataTableToolbar, null, /*#__PURE__*/_react.default.createElement("p", null, "Content")), /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last updated"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007")))), /*#__PURE__*/_react.default.createElement(_dataTableToolbar.DataTableToolbar, {
  position: "bottom"
}, /*#__PURE__*/_react.default.createElement("p", null, "Content (bottom)")));
const Toolbars = exports.Toolbars = ToolbarsTemplate.bind({});
Toolbars.args = {};
const ExpandableContentTemplate = args => {
  const [openRowIndex, setOpenRowIndex] = (0, _react.useState)(null);
  const toggleOpenRow = index => setOpenRowIndex(openRowIndex === index ? null : index);
  const style = {
    margin: 8,
    padding: 4,
    backgroundColor: 'lightblue'
  };
  const expandableContent = /*#__PURE__*/_react.default.createElement("div", {
    style: style
  }, "More info about this row!");
  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last updated"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    expanded: openRowIndex === 0,
    onExpandToggle: payload => {
      console.log(payload);
      toggleOpenRow(0);
    },
    expandableContent: expandableContent
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    expanded: openRowIndex === 1,
    onExpandToggle: () => toggleOpenRow(1),
    expandableContent: expandableContent
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    expanded: openRowIndex === 2,
    onExpandToggle: () => toggleOpenRow(2),
    expandableContent: expandableContent
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"))));
};
const ExpandableContent = exports.ExpandableContent = ExpandableContentTemplate.bind({});
ExpandableContent.args = {};
const SelectableRowsTemplate = args => {
  const [selected, setSelected] = (0, _react.useState)({
    id_2: true
  });
  const toggleSelected = _ref2 => {
    let {
      value,
      checked
    } = _ref2;
    setSelected({
      ...selected,
      [value]: checked
    });
  };
  const toggleAll = _ref3 => {
    let {
      checked
    } = _ref3;
    setSelected({
      id_1: checked,
      id_2: checked,
      id_3: checked
    });
  };
  const allSelected = () => Object.values(selected).filter(value => value).length === 3;
  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    checked: allSelected(),
    onChange: toggleAll
  })), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last updated"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    selected: !!selected.id_1
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    value: "id_1",
    checked: !!selected.id_1,
    onChange: toggleSelected
  })), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    selected: !!selected.id_2
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    value: "id_2",
    checked: !!selected.id_2,
    onChange: toggleSelected
  })), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    selected: !!selected.id_3
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    value: "id_3",
    checked: !!selected.id_3,
    onChange: toggleSelected
  })), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"))));
};
const SelectableRows = exports.SelectableRows = SelectableRowsTemplate.bind({});
SelectableRows.args = {};
const FixedHeaderTemplate = args => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "66"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "38"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "98"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "2"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mtendere"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Afolayan"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "37"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Inyene"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "70"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Amaka"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Pretorius"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "32"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Meti"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "8"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Eshe"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "63"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Obi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "28"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete"))));
const FixedHeader = exports.FixedHeader = FixedHeaderTemplate.bind({});
FixedHeader.args = {
  scrollHeight: '350px'
};
const FixedFirstColumnTemplate = args => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0"
}, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last updated"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Age"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Registering unit"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Assigned user"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Status"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "66"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "38"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "98"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "2"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Mtendere"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Afolayan"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "37"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Inyene"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "70"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Amaka"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Pretorius"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "32"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Meti"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "8"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Eshe"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "63"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Obi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "28"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete"))));
const FixedFirstColumn = exports.FixedFirstColumn = FixedFirstColumnTemplate.bind({});
FixedFirstColumn.args = {
  scrollWidth: '500px'
};
const FixedHeaderAndTwoColumnsTemplate = args => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "66"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "38"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "98"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "2"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "37"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "70"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "32"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "8"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "63"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "28"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete"))));
const FixedHeaderAndTwoColumns = exports.FixedHeaderAndTwoColumns = FixedHeaderAndTwoColumnsTemplate.bind({});
FixedHeaderAndTwoColumns.args = {
  layout: 'fixed',
  width: '1000px',
  scrollWidth: '500px',
  scrollHeight: '400px'
};
const ScrollingDataTableWithToolbarsTemplate = args => /*#__PURE__*/_react.default.createElement(_box.Box, {
  width: "500px"
}, /*#__PURE__*/_react.default.createElement(_dataTableToolbar.DataTableToolbar, null, /*#__PURE__*/_react.default.createElement("p", null, "Content")), /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, _extends({}, args, {
  layout: "fixed",
  width: "1000px",
  scrollWidth: "500px",
  scrollHeight: "400px"
}), /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "66"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "38"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "98"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "2"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "37"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "70"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "32"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "8"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "63"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "28"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")))), /*#__PURE__*/_react.default.createElement(_dataTableToolbar.DataTableToolbar, {
  position: "bottom"
}, /*#__PURE__*/_react.default.createElement("p", null, "Content (bottom)")));
const ScrollingDataTableWithToolbars = exports.ScrollingDataTableWithToolbars = ScrollingDataTableWithToolbarsTemplate.bind({});
ScrollingDataTableWithToolbars.args = {
  layout: 'fixed',
  width: '1000px',
  scrollWidth: '500px',
  scrollHeight: '400px'
};
const ColumnHeaderSortingTemplate = args => {
  const rows = [{
    firstName: 'Onyekachukwu',
    lastName: 'Kariuki'
  }, {
    firstName: 'Kwasi',
    lastName: 'Okafor'
  }, {
    firstName: 'Siyabonga',
    lastName: 'Abiodun'
  }];
  const [{
    column,
    direction
  }, setSortInstructions] = (0, _react.useState)({
    column: 'firstName',
    direction: 'default'
  });
  const getSortDirection = columnName => columnName === column ? direction : 'default';
  const onSortIconClick = _ref4 => {
    let {
      name,
      direction
    } = _ref4;
    setSortInstructions({
      column: name,
      direction
    });
  };
  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    onSortIconClick: onSortIconClick,
    sortDirection: getSortDirection('firstName'),
    name: 'firstName',
    sortIconTitle: "Sort by first name"
  }, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    onSortIconClick: onSortIconClick,
    sortDirection: getSortDirection('lastName'),
    name: 'lastName',
    sortIconTitle: "Sort by last name"
  }, "Last name"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, rows.sort((a, b) => {
    const strA = a[column];
    const strB = b[column];
    if (direction === 'asc' && strA < strB || direction === 'desc' && strA > strB) {
      return -1;
    }
    if (direction === 'desc' && strA < strB || direction === 'asc' && strA > strB) {
      return 1;
    }
    return 0;
  }).map(_ref5 => {
    let {
      firstName,
      lastName
    } = _ref5;
    return /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
      key: firstName + lastName
    }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, firstName), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, lastName));
  })));
};
const ColumnHeaderSorting = exports.ColumnHeaderSorting = ColumnHeaderSortingTemplate.bind({});
ColumnHeaderSorting.args = {};
const InlineFilteringTemplate = args => {
  const rows = [{
    firstName: 'Onyekachukwu',
    lastName: 'Kariuki'
  }, {
    firstName: 'Kwasi',
    lastName: 'Okafor'
  }, {
    firstName: 'Siyabonga',
    lastName: 'Abiodun'
  }];
  const [{
    column,
    value
  }, setFilter] = (0, _react.useState)({
    column: null,
    value: ''
  });
  const onFilterIconClick = _ref6 => {
    let {
      name,
      show
    } = _ref6;
    setFilter({
      column: show ? name : null,
      value: ''
    });
  };
  const onFilterInputChange = _ref7 => {
    let {
      value
    } = _ref7;
    setFilter({
      column: column,
      value
    });
  };
  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    onFilterIconClick: onFilterIconClick,
    name: "firstName",
    showFilter: column === 'firstName',
    filter: /*#__PURE__*/_react.default.createElement(_input.Input, {
      dense: true,
      onChange: onFilterInputChange,
      name: "firstName",
      value: value
    })
  }, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    onFilterIconClick: onFilterIconClick,
    name: "lastName",
    showFilter: column === 'lastName',
    filter: /*#__PURE__*/_react.default.createElement(_input.Input, {
      dense: true,
      onChange: onFilterInputChange,
      name: "lastName",
      value: value
    })
  }, "Last name"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, rows.filter(row => {
    if (!column || !value) {
      return true;
    }
    return row[column].toUpperCase().includes(value.toUpperCase());
  }).map(_ref8 => {
    let {
      firstName,
      lastName
    } = _ref8;
    return /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
      key: firstName + lastName
    }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, firstName), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, lastName));
  })));
};
const InlineFiltering = exports.InlineFiltering = InlineFilteringTemplate.bind({});
InlineFiltering.args = {
  layout: 'fixed'
};
const LongCellContentTemplate = _ref9 => {
  let {
    large
  } = _ref9;
  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, null, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    large: large,
    onSortIconClick: () => {},
    sortDirection: "asc",
    name: "first"
  }, "FIRST - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    large: large,
    onFilterIconClick: () => {},
    name: "firstName",
    showFilter: true,
    filter: /*#__PURE__*/_react.default.createElement(_input.Input, {
      dense: true,
      onChange: () => {},
      name: "firstName",
      value: "Filter value"
    })
  }, "SECOND - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    large: large,
    onSortIconClick: () => {},
    sortDirection: "asc",
    name: "third"
  }, "Third (short)"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    large: large,
    onSortIconClick: () => {},
    sortDirection: "asc",
    name: "fourth"
  }, "Fourth - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large
  }, "FIRST - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large
  }, "SECOND - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large
  }, "Third (short)"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    large: large
  }, "Fourth - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"))));
};
const LongCellContent = exports.LongCellContent = LongCellContentTemplate.bind({});
const LongCellContentLargeCells = exports.LongCellContentLargeCells = LongCellContentTemplate.bind({});
LongCellContentLargeCells.args = {
  large: 'true'
};
const RTL = args => /*#__PURE__*/_react.default.createElement("div", {
  dir: "rtl"
}, /*#__PURE__*/_react.default.createElement(BasicTemplate, args));
exports.RTL = RTL;
