// @dhis2-ui/table/build/cjs/data-table/data-table-cell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DataTableCell = void 0;
var _propTypes = require("@dhis2/prop-types");
var _propTypes2 = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _index = require("./table-elements/index.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const DataTableCell = exports.DataTableCell = /*#__PURE__*/(0, _react.forwardRef)((_ref, ref) => {
  let {
    active,
    align = 'left',
    backgroundColor,
    bordered,
    children,
    className,
    colSpan,
    dataTest = 'dhis2-uicore-datatablecell',
    error,
    fixed,
    large,
    left = 'auto',
    muted,
    rowSpan,
    role,
    scope,
    staticStyle,
    tag,
    valid,
    width = 'auto',
    onClick,
    ...props
  } = _ref;
  const TableCell = !tag && fixed || tag && tag === 'th' ? _index.TableHeaderCell : _index.TableDataCell;
  return /*#__PURE__*/_react.default.createElement(TableCell, _extends({}, props, {
    active: active,
    align: align,
    backgroundColor: backgroundColor,
    bordered: bordered,
    className: className,
    colSpan: colSpan,
    dataTest: dataTest,
    error: error,
    fixed: fixed,
    large: large,
    left: left,
    muted: muted,
    rowSpan: rowSpan,
    ref: ref,
    role: role,
    staticStyle: staticStyle,
    scope: scope,
    valid: valid,
    width: width,
    onClick: onClick
  }), children);
});
DataTableCell.displayName = 'DataTableCell';
const stylePropType = (0, _propTypes.mutuallyExclusive)(['valid', 'error', 'muted'], _propTypes2.default.bool);
const requiredIfFixedPropType = (0, _propTypes.requiredIf)(props => props.fixed, _propTypes2.default.string);
DataTableCell.propTypes = {
  /** To toggle border color, for example for editing */
  active: _propTypes2.default.bool,
  align: _propTypes2.default.oneOf(['left', 'center', 'right']),
  /** Sets background color of the cell. Disables dynamic background colors from active, hover, and selected states */
  backgroundColor: _propTypes2.default.string,
  bordered: _propTypes2.default.bool,
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  colSpan: _propTypes2.default.string,
  dataTest: _propTypes2.default.string,
  /** Mutually exclusive with muted and valid */
  error: stylePropType,
  /** When true a TableHeaderCell with sticky positioning will be rendered */
  fixed: _propTypes2.default.bool,
  large: _propTypes2.default.bool,
  /** Required when fixed */
  left: requiredIfFixedPropType,
  /** Mutually exclusive with error and valid */
  muted: stylePropType,
  role: _propTypes2.default.string,
  rowSpan: _propTypes2.default.string,
  scope: _propTypes2.default.string,
  /** Surpress hover and active event styles */
  staticStyle: _propTypes2.default.bool,
  /** Render a TableDataCell or TableHeaderCell respectively */
  tag: _propTypes2.default.oneOf(['td', 'th']),
  /** Mutually exclusive with error and muted */
  valid: stylePropType,
  /** Required when fixed */
  width: requiredIfFixedPropType,
  onClick: _propTypes2.default.func
};
