// @dhis2-ui/css/build/cjs/css-variables/css-variables.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithTheme = exports.WithSpacers = exports.WithLayers = exports.WithElevations = exports.WithColors = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'CssVariables'
};
const WithColors = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.CssVariables, {
  colors: true
}), /*#__PURE__*/_react.default.createElement("div", {
  id: "custom",
  style: {
    backgroundColor: 'var(--colors-blue900)'
  }
}));
exports.WithColors = WithColors;
const WithTheme = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.CssVariables, {
  theme: true
}), /*#__PURE__*/_react.default.createElement("div", {
  id: "custom",
  style: {
    backgroundColor: 'var(--theme-primary900)'
  }
}));
exports.WithTheme = WithTheme;
const WithLayers = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.CssVariables, {
  layers: true
}), /*#__PURE__*/_react.default.createElement("div", {
  id: "custom",
  style: {
    zIndex: 'var(--layers-alert)',
    position: 'absolute'
  }
}));
exports.WithLayers = WithLayers;
const WithSpacers = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.CssVariables, {
  spacers: true
}), /*#__PURE__*/_react.default.createElement("div", {
  id: "custom",
  style: {
    margin: 'var(--spacers-dp4)'
  }
}));
exports.WithSpacers = WithSpacers;
const WithElevations = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.CssVariables, {
  elevations: true
}), /*#__PURE__*/_react.default.createElement("div", {
  id: "custom",
  style: {
    boxShadow: 'var(--elevations-e100)'
  }
}));
exports.WithElevations = WithElevations;
