// @dhis2-ui/organisation-unit-tree/build/cjs/organisation-unit-node/error-message.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ErrorMessage = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ErrorMessage = _ref => {
  let {
    children,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("span", {
    "data-test": `${dataTest}-error`,
    className: _style.default.dynamic([["3198585030", [_uiConstants.colors.grey800, _uiConstants.spacers.dp4, _uiConstants.theme.error]]])
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3198585030",
    dynamic: [_uiConstants.colors.grey800, _uiConstants.spacers.dp4, _uiConstants.theme.error]
  }, [`span.__jsx-style-dynamic-selector{font-size:14px;color:${_uiConstants.colors.grey800};margin:${_uiConstants.spacers.dp4};color:${_uiConstants.theme.error};}`]));
};
exports.ErrorMessage = ErrorMessage;
ErrorMessage.propTypes = {
  children: _propTypes.default.any.isRequired,
  dataTest: _propTypes.default.string.isRequired
};
