// @dhis2-ui/sharing-dialog/build/cjs/access-list/access-list.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AccessList = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _constants = require("../constants.js");
var _index = _interopRequireDefault(require("../locales/index.js"));
var _index2 = require("../text/index.js");
var _listItem = require("./list-item.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const AccessList = _ref => {
  let {
    onChange,
    onRemove,
    publicAccess,
    allowPublicAccess,
    users,
    groups
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index2.Title, null, _index.default.t('Users and groups that currently have access')), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["805822186", [_uiConstants.colors.grey200, _uiConstants.colors.grey900]]]) + " " + "header"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["805822186", [_uiConstants.colors.grey200, _uiConstants.colors.grey900]]]) + " " + "header-left-column"
  }, _index.default.t('User / Group')), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["805822186", [_uiConstants.colors.grey200, _uiConstants.colors.grey900]]]) + " " + "header-right-column"
  }, _index.default.t('Access level'))), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["805822186", [_uiConstants.colors.grey200, _uiConstants.colors.grey900]]]) + " " + "list"
  }, /*#__PURE__*/_react.default.createElement(_listItem.ListItem, {
    name: _index.default.t('All users'),
    target: _constants.SHARE_TARGET_PUBLIC,
    access: publicAccess,
    accessOptions: [_constants.ACCESS_NONE, _constants.ACCESS_VIEW_ONLY, _constants.ACCESS_VIEW_AND_EDIT],
    disabled: !allowPublicAccess,
    onChange: newAccess => onChange({
      type: 'public',
      access: newAccess
    })
  }), groups.map(_ref2 => {
    let {
      id,
      name,
      access
    } = _ref2;
    return /*#__PURE__*/_react.default.createElement(_listItem.ListItem, {
      key: id,
      name: name,
      target: _constants.SHARE_TARGET_GROUP,
      access: access,
      accessOptions: [_constants.ACCESS_VIEW_ONLY, _constants.ACCESS_VIEW_AND_EDIT],
      onChange: newAccess => onChange({
        type: 'group',
        id,
        access: newAccess
      }),
      onRemove: () => onRemove({
        type: 'group',
        id
      })
    });
  }), users.map(_ref3 => {
    let {
      id,
      name,
      access
    } = _ref3;
    return access && /*#__PURE__*/_react.default.createElement(_listItem.ListItem, {
      key: id,
      name: name,
      target: _constants.SHARE_TARGET_USER,
      access: access,
      accessOptions: [_constants.ACCESS_VIEW_ONLY, _constants.ACCESS_VIEW_AND_EDIT],
      onChange: newAccess => onChange({
        type: 'user',
        id,
        access: newAccess
      }),
      onRemove: () => onRemove({
        type: 'user',
        id
      })
    });
  })), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "805822186",
    dynamic: [_uiConstants.colors.grey200, _uiConstants.colors.grey900]
  }, [`.header.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:10px 8px;margin:0 0 8px 0;background-color:${_uiConstants.colors.grey200};color:${_uiConstants.colors.grey900};font-size:13px;}`, ".header-left-column.__jsx-style-dynamic-selector{-webkit-flex:2;-ms-flex:2;flex:2;}", ".header-right-column.__jsx-style-dynamic-selector{-webkit-flex:1;-ms-flex:1;flex:1;}", ".list.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;overflow-y:auto;}"]));
};
exports.AccessList = AccessList;
AccessList.propTypes = {
  allowPublicAccess: _propTypes.default.bool.isRequired,
  groups: _propTypes.default.arrayOf(_propTypes.default.shape({
    access: _propTypes.default.oneOf([_constants.ACCESS_NONE, _constants.ACCESS_VIEW_ONLY, _constants.ACCESS_VIEW_AND_EDIT]).isRequired,
    id: _propTypes.default.string.isRequired,
    name: _propTypes.default.string.isRequired
  })).isRequired,
  publicAccess: _propTypes.default.oneOf([_constants.ACCESS_NONE, _constants.ACCESS_VIEW_ONLY, _constants.ACCESS_VIEW_AND_EDIT]).isRequired,
  users: _propTypes.default.arrayOf(_propTypes.default.shape({
    access: _propTypes.default.oneOf([_constants.ACCESS_NONE, _constants.ACCESS_VIEW_ONLY, _constants.ACCESS_VIEW_AND_EDIT]).isRequired,
    id: _propTypes.default.string.isRequired,
    name: _propTypes.default.string.isRequired
  })).isRequired,
  onChange: _propTypes.default.func.isRequired,
  onRemove: _propTypes.default.func.isRequired
};
