// @dhis2-ui/sharing-dialog/build/cjs/access-list/list-item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListItem = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _appRuntime = require("@dhis2/app-runtime");
var _uiConstants = require("@dhis2/ui-constants");
var _divider = require("@dhis2-ui/divider");
var _select = require("@dhis2-ui/select");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _constants = require("../constants.js");
var _index = require("../destructive-select-option/index.js");
var _index2 = require("../fetching-context/index.js");
var _index3 = require("../helpers/index.js");
var _index4 = _interopRequireDefault(require("../locales/index.js"));
var _listItemContext = require("./list-item-context.js");
var _listItemIcon = require("./list-item-icon.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ListItem = _ref => {
  let {
    name,
    target,
    access,
    accessOptions,
    disabled,
    onChange,
    onRemove
  } = _ref;
  const isFetching = (0, _react.useContext)(_index2.FetchingContext);
  const {
    isDisconnected: offline
  } = (0, _appRuntime.useDhis2ConnectionStatus)();
  const valueToLabel = {
    ACCESS_NONE: _index4.default.t('No access'),
    ACCESS_VIEW_ONLY: _index4.default.t('View only'),
    ACCESS_VIEW_AND_EDIT: _index4.default.t('View and edit')
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["1581000493", [_uiConstants.colors.grey900]]]) + " " + "wrapper"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["1581000493", [_uiConstants.colors.grey900]]]) + " " + "details"
  }, /*#__PURE__*/_react.default.createElement(_listItemIcon.ListItemIcon, {
    target: target,
    name: name
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["1581000493", [_uiConstants.colors.grey900]]]) + " " + "details-text"
  }, /*#__PURE__*/_react.default.createElement("p", {
    className: _style.default.dynamic([["1581000493", [_uiConstants.colors.grey900]]]) + " " + "details-name"
  }, name), /*#__PURE__*/_react.default.createElement(_listItemContext.ListItemContext, {
    access: access
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["1581000493", [_uiConstants.colors.grey900]]]) + " " + "select"
  }, /*#__PURE__*/_react.default.createElement(_select.SingleSelectField, {
    disabled: disabled || offline || isFetching,
    prefix: _index4.default.t('Metadata'),
    selected: access,
    helpText: offline ? _index4.default.t('Not available offline') : '',
    onChange: _ref2 => {
      let {
        selected
      } = _ref2;
      return onChange(selected);
    }
  }, accessOptions.map(value => /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
    key: value,
    label: valueToLabel[value],
    value: value,
    active: value === access
  })), (0, _index3.isRemovableTarget)(target) && /*#__PURE__*/_react.default.createElement(_index.DestructiveSelectOption, {
    onClick: onRemove,
    label: _index4.default.t('Remove access')
  })))), /*#__PURE__*/_react.default.createElement(_divider.Divider, null), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1581000493",
    dynamic: [_uiConstants.colors.grey900]
  }, [".wrapper.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:4px 8px;}", ".details.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:2;-ms-flex:2;flex:2;}", ".details-text.__jsx-style-dynamic-selector{margin-inline-start:8px;}", `.details-name.__jsx-style-dynamic-selector{font-size:15px;font-weight:500;color:${_uiConstants.colors.grey900};margin:0;padding:0;}`, ".select.__jsx-style-dynamic-selector{-webkit-flex:1;-ms-flex:1;flex:1;}"]));
};
exports.ListItem = ListItem;
ListItem.propTypes = {
  access: _propTypes.default.oneOf([_constants.ACCESS_NONE, _constants.ACCESS_VIEW_ONLY, _constants.ACCESS_VIEW_AND_EDIT]).isRequired,
  accessOptions: _propTypes.default.arrayOf(_propTypes.default.oneOf([_constants.ACCESS_NONE, _constants.ACCESS_VIEW_ONLY, _constants.ACCESS_VIEW_AND_EDIT])).isRequired,
  name: _propTypes.default.string.isRequired,
  target: _propTypes.default.oneOf([_constants.SHARE_TARGET_PUBLIC, _constants.SHARE_TARGET_GROUP, _constants.SHARE_TARGET_USER]).isRequired,
  onChange: _propTypes.default.func.isRequired,
  disabled: _propTypes.default.bool,
  onRemove: _propTypes.default.func
};
