// @dhis2-ui/transfer/build/es/__e2e__/common/stateful-decorator.js
/* eslint-disable react/display-name */

import React, { useState } from 'react';
const WithState = _ref => {
  let {
    fn,
    initialState,
    initialSearchTerm,
    controlFilter
  } = _ref;
  const [selected, setSelected] = useState(initialState);
  const [searchTerm, setSearchTerm] = useState(initialSearchTerm);
  return fn({
    selected,
    searchTerm: controlFilter ? searchTerm : undefined,
    onChange: payload => setSelected(payload.selected),
    onFilterChange: controlFilter ? _ref2 => {
      let {
        value
      } = _ref2;
      return setSearchTerm(value);
    } : undefined
  });
};
export const statefulDecorator = function () {
  let {
    initialState = [],
    controlFilter = false,
    initialSearchTerm = ''
  } = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return fn => /*#__PURE__*/React.createElement(WithState, {
    initialState: initialState,
    initialSearchTerm: initialSearchTerm,
    controlFilter: controlFilter,
    fn: fn
  });
};
