// @dhis2-ui/organisation-unit-tree/build/cjs/__stories__/development-stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DxWithRealBackend = exports.DxSingleSelection = exports.DxNoSelection = exports.DxMultiSelection = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireWildcard(require("react"));
var _index = require("../index.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const DX_onChange = (selected, setSelected, singleSelection) => _ref => {
  let {
    id,
    path,
    checked
  } = _ref;
  console.log('onChange', {
    path,
    id,
    checked
  });
  const pathIndex = selected.indexOf(path);
  let nextSelected;
  if (checked) {
    nextSelected = singleSelection ? [path] : [...selected, path];
  } else {
    nextSelected = singleSelection ? [] : [...selected.slice(0, pathIndex), ...selected.slice(pathIndex + 1)];
  }
  console.log('> nextSelected', nextSelected);
  setSelected(nextSelected);
};
const Wrapper = props => {
  const [selected, setSelected] = (0, _react.useState)([]);
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, _extends({
    name: "Root org unit",
    roots: ['A0000000000'],
    selected: selected,
    onChange: DX_onChange(selected, setSelected, props.singleSelection),
    initiallyExpanded: ['A0000000001/A0000000002']
  }, props));
};
const DxMultiSelection = () => /*#__PURE__*/_react.default.createElement(Wrapper, null);
exports.DxMultiSelection = DxMultiSelection;
DxMultiSelection.storyName = 'DX: Multi selection';
const DxSingleSelection = () => /*#__PURE__*/_react.default.createElement(Wrapper, {
  singleSelection: true
});
exports.DxSingleSelection = DxSingleSelection;
DxSingleSelection.storyName = 'DX: Single selection';
const DxNoSelection = () => /*#__PURE__*/_react.default.createElement(Wrapper, {
  disableSelection: true
});
exports.DxNoSelection = DxNoSelection;
DxNoSelection.storyName = 'DX: No selection';
const DxWithRealBackend = () => /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("div", {
  style: {
    marginBottom: 20,
    lineHeight: '28px'
  }
}, /*#__PURE__*/_react.default.createElement("b", null, "This story doesn't work on netlify for some reason, just run it locally."), /*#__PURE__*/_react.default.createElement("br", null), "You need to log in to", ' ', /*#__PURE__*/_react.default.createElement("a", {
  href: "https://debug.dhis2.org/dev",
  target: "_blank",
  rel: "noopener noreferrer"
}, "https://debug.dhis2.org/dev"), /*#__PURE__*/_react.default.createElement("br", null), "Make sure the", ' ', /*#__PURE__*/_react.default.createElement("code", {
  style: {
    background: '#ccc'
  }
}, "localhost:[PORT]"), " is part of the accepted list:", ' ', /*#__PURE__*/_react.default.createElement("a", {
  href: "https://debug.dhis2.org/dev/dhis-web-settings/#/access",
  target: "_blank",
  rel: "noopener noreferrer"
}, "Settings app / Access")), /*#__PURE__*/_react.default.createElement(_appRuntime.DataProvider, {
  baseUrl: "https://debug.dhis2.org/dev",
  apiVersion: ""
}, /*#__PURE__*/_react.default.createElement(Wrapper, {
  roots: ['ImspTQPwCqd', 'O6uvpzGd5pu', 'fdc6uOvgoji']
})));
exports.DxWithRealBackend = DxWithRealBackend;
DxWithRealBackend.storyName = 'DX: With real backend';
