// @dhis2-ui/organisation-unit-tree/build/cjs/__stories__/loading.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Loading = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _shared = require("./shared.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Loading = () => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
  data: {
    ..._shared.customData,
    organisationUnits: function () {
      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }
      const [, {
        id
      }] = args;
      if (id === 'A0000000001') {
        return new Promise(() => null);
      }
      return _shared.customData.organisationUnits(...args);
    }
  }
}, /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
  onChange: _shared.onChange,
  name: "Root org unit",
  roots: ['A0000000000'],
  initiallyExpanded: ['/A0000000000/A0000000001']
}));
exports.Loading = Loading;
