// @dhis2-ui/organisation-unit-tree/build/cjs/__stories__/loading-error-grandchild.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LoadingErrorGrandchild = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _shared = require("./shared.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const LoadingErrorGrandchild = () => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
  data: {
    ..._shared.customData,
    organisationUnits: function () {
      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }
      const [, {
        id
      }] = args;
      if (id === 'A0000000003') {
        return Promise.reject('Loading org unit 4 and 5 failed');
      }
      return _shared.customData.organisationUnits(...args);
    }
  }
}, /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
  autoExpandLoadingError: true,
  name: "Root org unit",
  roots: ['A0000000000'],
  onChange: _shared.onChange,
  onExpand: _shared.onExpand,
  onCollapse: _shared.onCollapse,
  onChildrenLoaded: _shared.onChildrenLoaded,
  initiallyExpanded: ['/A0000000000/A0000000001']
}));
exports.LoadingErrorGrandchild = LoadingErrorGrandchild;
LoadingErrorGrandchild.storyName = 'Loading error grandchild';
