// @dhis2-ui/select/build/cjs/multi-select-field/multi-select-field.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithHelpText = exports.StatusWarning = exports.StatusValid = exports.StatusError = exports.Required = exports.InputWidth = exports.DefaultLoadingText = exports.DefaultFilterPlaceholderAndNoMatchText = exports.DefaultEmpty = exports.DefaultClearText = exports.Default = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _index2 = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
\`MultiSelectField\` is a wrapper around a \`MultiSelect\` component that adds a label, help text, validation text, and other features.

See the MultiSelect for more information, and read more at [Design System: Select](https://github.com/dhis2/design-system/blob/master/molecules/select.md#multiple-selection).

\`\`\`js
import { MultiSelectField, MultiSelectOption } from '@dhis2/ui'
\`\`\`

_**Note**: The dropdowns in some of the following stories won't appear correctly on this page. View these demos in the 'Canvas' tab._
`;
const onChange = selected => alert(`Selected changed to: ${JSON.stringify(selected, null, 2)}`);
const options = [/*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "4",
  value: "4",
  label: "four"
}), /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "5",
  value: "5",
  label: "five"
}), /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "6",
  value: "6",
  label: "six"
}), /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "7",
  value: "7",
  label: "seven"
}), /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "8",
  value: "8",
  label: "eight"
}), /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "9",
  value: "9",
  label: "nine"
}), /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
  key: "10",
  value: "10",
  label: "ten"
})];
var _default = exports.default = {
  title: 'Multi Select Field',
  component: _index2.MultiSelectField,
  subcomponents: {
    MultiSelectOption: _index.MultiSelectOption
  },
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  // default args for stories
  args: {
    // Fix default prop issues - causes 'i18n is not defined' error
    dataTest: 'dhis2-uiwidgets-multiselectfield',
    clearText: 'Clear',
    empty: 'No data found',
    filterPlaceholder: 'Type to filter options',
    loadingText: 'Loading options',
    noMatchText: 'No options found',
    label: 'Default label',
    selected: ['1'],
    children: options,
    onChange: onChange
  },
  argTypes: {
    valid: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: {
      ..._uiConstants.sharedPropTypes.statusArgType
    }
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_index2.MultiSelectField, args);
const Default = exports.Default = Template.bind({});
const WithHelpText = exports.WithHelpText = Template.bind({});
WithHelpText.args = {
  helpText: 'Helpful text.'
};
WithHelpText.parameters = {
  docs: {
    description: {
      story: `_**Note**: The dropdowns in the following stories won't appear correctly on this page. View these demos in the 'Canvas' tab._`
    }
  }
};
const StatusValid = exports.StatusValid = Template.bind({});
StatusValid.args = {
  valid: true,
  validationText: 'Totally valid!',
  ...WithHelpText.args
};
StatusValid.storyName = 'Status: Valid';
const StatusWarning = exports.StatusWarning = Template.bind({});
StatusWarning.args = {
  warning: true,
  validationText: 'Hm, not quite, I warn thee!',
  ...WithHelpText.args
};
StatusWarning.storyName = 'Status: Warning';
const StatusError = exports.StatusError = Template.bind({});
StatusError.args = {
  error: true,
  validationText: 'That value is wrong. Sorry!',
  ...WithHelpText.args
};
StatusError.storyName = 'Status: Error';
const Required = exports.Required = Template.bind({});
Required.args = {
  required: true
};
const InputWidth = exports.InputWidth = Template.bind({});
InputWidth.args = {
  inputWidth: '200px',
  label: 'A very long label indeed, well at least longer than the input field to show how it looks and works and stuff',
  required: true
};
const DefaultClearText = exports.DefaultClearText = Template.bind({});
DefaultClearText.args = {
  clearable: true,
  children: /*#__PURE__*/_react.default.createElement(_index.MultiSelectOption, {
    key: "1",
    value: "1",
    label: "Not translated, just for showing clear button"
  }),
  label: null
};
DefaultClearText.storyName = 'Default: clearText';
const DefaultEmpty = exports.DefaultEmpty = Template.bind({});
DefaultEmpty.args = {
  children: null,
  selected: [],
  label: null
};
DefaultEmpty.storyName = 'Default: empty';
const DefaultFilterPlaceholderAndNoMatchText = exports.DefaultFilterPlaceholderAndNoMatchText = Template.bind({});
DefaultFilterPlaceholderAndNoMatchText.args = {
  filterable: true,
  ...DefaultEmpty.args
};
DefaultFilterPlaceholderAndNoMatchText.storyName = 'Default: filterPlaceholder and noMatchText';
const DefaultLoadingText = exports.DefaultLoadingText = Template.bind({});
DefaultLoadingText.args = {
  loading: true,
  ...DefaultEmpty.args
};
DefaultLoadingText.storyName = 'Default: loadingText';
