// @dhis2-ui/tooltip/build/es/tooltip.e2e.stories.js
import _JSXStyle from "styled-jsx/style";
import { Modal } from '@dhis2-ui/modal';
import React from 'react';
import { Tooltip } from './tooltip.js';
import { HidesWhenOutOfFrame } from './tooltip.prod.stories.js';
const noPaddingStyles = ["#root{padding:0;}"];
noPaddingStyles.__hash = "249208768";
export default {
  title: 'Tooltip',
  component: Tooltip
};
export const Default = () => /*#__PURE__*/React.createElement("p", {
  className: "jsx-4042709097 " + `jsx-${noPaddingStyles.__hash}`
}, "I am a ", /*#__PURE__*/React.createElement(Tooltip, {
  content: "Short content"
}, "paragraph"), " that contains a tooltip.", /*#__PURE__*/React.createElement(_JSXStyle, {
  id: noPaddingStyles.__hash
}, noPaddingStyles), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "4042709097"
}, ["p.jsx-4042709097{margin-top:40px;}"]));
export const FlippedVertically = () => /*#__PURE__*/React.createElement("p", {
  className: "jsx-2658853518 " + `jsx-${noPaddingStyles.__hash}`
}, "I am a ", /*#__PURE__*/React.createElement(Tooltip, {
  content: "Short content"
}, "paragraph"), " that contains a tooltip.", /*#__PURE__*/React.createElement(_JSXStyle, {
  id: noPaddingStyles.__hash
}, noPaddingStyles), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "2658853518"
}, ["div#root{padding-top:0;}", "p.jsx-2658853518{margin-top:0;}"]));
export const AdjustedHorizontalPosition = () => /*#__PURE__*/React.createElement("p", {
  className: "jsx-4042709097 " + `jsx-${noPaddingStyles.__hash}`
}, "I am a", ' ', /*#__PURE__*/React.createElement(Tooltip, {
  content: "Longer content will force the tooltip popper to shift"
}, "paragraph"), ' ', "that contains a tooltip.", /*#__PURE__*/React.createElement(_JSXStyle, {
  id: noPaddingStyles.__hash
}, noPaddingStyles), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "4042709097"
}, ["p.jsx-4042709097{margin-top:40px;}"]));
export const ShiftingPosition = () => /*#__PURE__*/React.createElement("p", {
  className: "jsx-823038592 " + `jsx-${noPaddingStyles.__hash}`
}, "I am a ", /*#__PURE__*/React.createElement(Tooltip, {
  content: "Short content"
}, "paragraph"), " that contains a tooltip.", /*#__PURE__*/React.createElement(_JSXStyle, {
  id: noPaddingStyles.__hash
}, noPaddingStyles), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "823038592"
}, ["html,body,div#storybook-root{position:relative;height:40px;max-height:40px;overflow:hidden;}"]));
export const ModalWithTooltip = () => /*#__PURE__*/React.createElement(Modal, null, /*#__PURE__*/React.createElement("p", {
  className: "jsx-3185463749"
}, "I am a ", /*#__PURE__*/React.createElement(Tooltip, {
  content: "Short content"
}, "paragraph"), " that contains a tooltip."), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "3185463749"
}, ["[data-test='dhis2-uicore-popper']{pointer-events:all;}"]));
export const ScrollingContainers = HidesWhenOutOfFrame.bind({});
ScrollingContainers.args = {
  content: 'Tooltip content'
};
ScrollingContainers.decorators = [story => /*#__PURE__*/React.createElement("div", {
  style: {
    marginTop: '150px'
  }
}, story())];
