// @dhis2/prop-types/build/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "arrayWithLength", {
  enumerable: true,
  get: function () {
    return _arrayWithLength.arrayWithLength;
  }
});
Object.defineProperty(exports, "conditional", {
  enumerable: true,
  get: function () {
    return _conditional.conditional;
  }
});
Object.defineProperty(exports, "deprecated", {
  enumerable: true,
  get: function () {
    return _deprecated.deprecated;
  }
});
Object.defineProperty(exports, "instanceOfComponent", {
  enumerable: true,
  get: function () {
    return _instanceOfComponent.instanceOfComponent;
  }
});
Object.defineProperty(exports, "mutuallyExclusive", {
  enumerable: true,
  get: function () {
    return _mutuallyExclusive.mutuallyExclusive;
  }
});
Object.defineProperty(exports, "requiredIf", {
  enumerable: true,
  get: function () {
    return _requiredIf.requiredIf;
  }
});

var _arrayWithLength = require("./arrayWithLength.js");

var _conditional = require("./conditional.js");

var _deprecated = require("./deprecated.js");

var _instanceOfComponent = require("./instanceOfComponent.js");

var _mutuallyExclusive = require("./mutuallyExclusive.js");

var _requiredIf = require("./requiredIf.js");
