// @dhis2-ui/header-bar/build/cjs/__e2e__/header-bar.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CustomApplicationTitle", {
  enumerable: true,
  get: function () {
    return _customApplicationTitle.CustomApplicationTitle;
  }
});
Object.defineProperty(exports, "Default", {
  enumerable: true,
  get: function () {
    return _default2.Default;
  }
});
Object.defineProperty(exports, "MeWithAvatar", {
  enumerable: true,
  get: function () {
    return _meWithAvatar.MeWithAvatar;
  }
});
Object.defineProperty(exports, "OnlineStatusMessagingWithPwaEnabled", {
  enumerable: true,
  get: function () {
    return _onlineStatusMessage.OnlineStatusMessagingWithPwaEnabled;
  }
});
Object.defineProperty(exports, "PWAEnabled", {
  enumerable: true,
  get: function () {
    return _pwaEnabled.PWAEnabled;
  }
});
Object.defineProperty(exports, "UserHasAllAuthority", {
  enumerable: true,
  get: function () {
    return _userHasAllAuthority.UserHasAllAuthority;
  }
});
Object.defineProperty(exports, "UserHasNoAuthorities", {
  enumerable: true,
  get: function () {
    return _userHasNoAuthorities.UserHasNoAuthorities;
  }
});
Object.defineProperty(exports, "UserHasWebInterpretationAndMessagingAuthority", {
  enumerable: true,
  get: function () {
    return _userHasWebInterpretationAndMessagingAuthority.UserHasWebInterpretationAndMessagingAuthority;
  }
});
Object.defineProperty(exports, "UserHasWebInterpretationAuthority", {
  enumerable: true,
  get: function () {
    return _userHasWebInterpretationAuthority.UserHasWebInterpretationAuthority;
  }
});
Object.defineProperty(exports, "UserHasWebMessagingAuthority", {
  enumerable: true,
  get: function () {
    return _userHasWebMessagingAuthority.UserHasWebMessagingAuthority;
  }
});
Object.defineProperty(exports, "WithSpecialAppNameCharacters", {
  enumerable: true,
  get: function () {
    return _withSpecialAppNameCharacter.WithSpecialAppNameCharacters;
  }
});
Object.defineProperty(exports, "WithUnknownAppName", {
  enumerable: true,
  get: function () {
    return _withDebugInfoEdgeCases.WithUnknownAppName;
  }
});
Object.defineProperty(exports, "WithUnknownAppNameAndVersion", {
  enumerable: true,
  get: function () {
    return _withDebugInfoEdgeCases.WithUnknownAppNameAndVersion;
  }
});
Object.defineProperty(exports, "WithUnknownAppVersion", {
  enumerable: true,
  get: function () {
    return _withDebugInfoEdgeCases.WithUnknownAppVersion;
  }
});
Object.defineProperty(exports, "WithUnknownInstanceVersion", {
  enumerable: true,
  get: function () {
    return _withDebugInfoEdgeCases.WithUnknownInstanceVersion;
  }
});
Object.defineProperty(exports, "WithUpdateAvailableNotification", {
  enumerable: true,
  get: function () {
    return _withUpdateAvailableNotification.WithUpdateAvailableNotification;
  }
});
Object.defineProperty(exports, "WithUpdateAvailableNotificationNoAppName", {
  enumerable: true,
  get: function () {
    return _withUpdateAvailableNotification.WithUpdateAvailableNotificationNoAppName;
  }
});
Object.defineProperty(exports, "ZeroUnreadInterpretations", {
  enumerable: true,
  get: function () {
    return _zeroUnreadInterpretations.ZeroUnreadInterpretations;
  }
});
Object.defineProperty(exports, "ZeroUnreadMessages", {
  enumerable: true,
  get: function () {
    return _zeroUnreadMessages.ZeroUnreadMessages;
  }
});
exports.default = void 0;
var _index = require("../index.js");
var _default2 = require("./stories/default.js");
var _pwaEnabled = require("./stories/pwa-enabled.js");
var _onlineStatusMessage = require("./stories/online-status-message.js");
var _meWithAvatar = require("./stories/me-with-avatar.js");
var _withSpecialAppNameCharacter = require("./stories/with-special-app-name-character.js");
var _customApplicationTitle = require("./stories/custom-application-title.js");
var _userHasAllAuthority = require("./stories/user-has-all-authority.js");
var _userHasWebInterpretationAndMessagingAuthority = require("./stories/user-has-web-interpretation-and-messaging-authority.js");
var _userHasWebInterpretationAuthority = require("./stories/user-has-web-interpretation-authority.js");
var _userHasWebMessagingAuthority = require("./stories/user-has-web-messaging-authority.js");
var _userHasNoAuthorities = require("./stories/user-has-no-authorities.js");
var _zeroUnreadInterpretations = require("./stories/zero-unread-interpretations.js");
var _zeroUnreadMessages = require("./stories/zero-unread-messages.js");
var _withUpdateAvailableNotification = require("./stories/with-update-available-notification.js");
var _withDebugInfoEdgeCases = require("./stories/with-debug-info-edge-cases.js");
var _default = exports.default = {
  title: 'HeaderBarTesting',
  component: _index.HeaderBar
};
