// @dhis2-ui/center/build/cjs/center.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Top = exports.Default = exports.Bottom = void 0;
var _react = _interopRequireDefault(require("react"));
var _center = require("./center.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
Centers children horizontally, and by default, vertically.
Use the \`top\` or \`bottom\` props to change vertical alignment.

\`\`\`js
import { CenteredContent } from '@dhis2/ui'
\`\`\`
`;
const Wrapper = story => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    height: '150px'
  }
}, story());
var _default = exports.default = {
  title: 'Center',
  component: _center.Center,
  decorators: [Wrapper],
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_center.Center, args, /*#__PURE__*/_react.default.createElement("span", null, "Center me"));
const Default = exports.Default = Template.bind({});
const Top = exports.Top = Template.bind({});
Top.args = {
  position: 'top'
};
const Bottom = exports.Bottom = Template.bind({});
Bottom.args = {
  position: 'bottom'
};
