// @dhis2-ui/alert/build/cjs/alert-stack/alert-stack.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AlertStack = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _portal = require("@dhis2-ui/portal");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const AlertStack = _ref => {
  let {
    className,
    children,
    dataTest = 'dhis2-uicore-alertstack'
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_portal.Portal, null, /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["1009591311", [_uiConstants.layers.alert]]]) + " " + ((0, _classnames.default)(className) || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1009591311",
    dynamic: [_uiConstants.layers.alert]
  }, [`div.__jsx-style-dynamic-selector{position:fixed;top:auto;inset-inline-end:auto;bottom:0;inset-inline-start:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);z-index:${_uiConstants.layers.alert};display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column-reverse;-ms-flex-direction:column-reverse;flex-direction:column-reverse;pointer-events:none;}`, "div.__jsx-style-dynamic-selector:dir(rtl){-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}"])));
};
exports.AlertStack = AlertStack;
AlertStack.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
