// @dhis2-ui/sharing-dialog/build/es/autocomplete/menu-popup.js
import _JSXStyle from "styled-jsx/style";
import { Card } from '@dhis2-ui/card';
import { Layer } from '@dhis2-ui/layer';
import { Popper } from '@dhis2-ui/popper';
import PropTypes from 'prop-types';
import React from 'react';
export const MenuPopup = _ref => {
  let {
    children,
    dataTest,
    maxHeight = '280px',
    menuWidth,
    onClick,
    menuRef
  } = _ref;
  return /*#__PURE__*/React.createElement(Layer, {
    onBackdropClick: onClick,
    transparent: true
  }, /*#__PURE__*/React.createElement(Popper, {
    reference: menuRef,
    placement: "bottom",
    observeReferenceResize: true
  }, /*#__PURE__*/React.createElement("div", {
    "data-test": `${dataTest}-menuwrapper`,
    className: _JSXStyle.dynamic([["1686499909", [menuWidth, maxHeight]]]) + " " + "card"
  }, /*#__PURE__*/React.createElement(Card, null, children))), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "1686499909",
    dynamic: [menuWidth, maxHeight]
  }, [`.card.__jsx-style-dynamic-selector{width:${menuWidth};max-height:${maxHeight};}`]));
};
MenuPopup.propTypes = {
  dataTest: PropTypes.string.isRequired,
  menuRef: PropTypes.object.isRequired,
  menuWidth: PropTypes.string.isRequired,
  children: PropTypes.node,
  maxHeight: PropTypes.string,
  onClick: PropTypes.func
};
