// @dhis2-ui/table/build/es/data-table/table-elements/table-toolbar.js
import _JSXStyle from "styled-jsx/style";
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import { colors, spacers } from '@dhis2/ui-constants';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React, { forwardRef } from 'react';
export const TableToolbar = /*#__PURE__*/forwardRef((_ref, ref) => {
  let {
    children,
    className,
    dataTest = 'dhis2-uicore-tabletoolbar',
    position = 'top',
    ...props
  } = _ref;
  return /*#__PURE__*/React.createElement("div", _extends({}, props, {
    "data-test": dataTest,
    ref: ref,
    className: _JSXStyle.dynamic([["247244135", [colors.grey300, spacers.dp12]]]) + " " + (cx(className, position) || "")
  }), children, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "247244135",
    dynamic: [colors.grey300, spacers.dp12]
  }, [`div.__jsx-style-dynamic-selector{width:100%;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border:1px solid ${colors.grey300};padding:${spacers.dp12};}`, "div.top.__jsx-style-dynamic-selector{border-bottom:none;}", "div.bottom.__jsx-style-dynamic-selector{border-top:none;}"]));
});
TableToolbar.displayName = 'TableToolbar';
TableToolbar.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  position: PropTypes.oneOf(['top', 'bottom'])
};
