// @dhis2-ui/table/build/es/data-table/table-elements/table.e2e.stories.js
import React from 'react';
import { TableBody } from './table-body.js';
import { TableDataCell } from './table-data-cell/table-data-cell.js';
import { TableHead } from './table-head.js';
import { TableHeaderCell } from './table-header-cell/table-header-cell.js';
import { TableRow } from './table-row.js';
import { TableScrollBox } from './table-scroll-box.js';
import { Table } from './table.js';
export default {
  title: 'Table',
  component: Table
};
export const FixedHeaderAndColumns = () => /*#__PURE__*/React.createElement(TableScrollBox, {
  maxHeight: "350px",
  maxWidth: "500px"
}, /*#__PURE__*/React.createElement(Table, {
  layout: "fixed",
  borderless: true,
  width: "1000px"
}, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "Header fixed-left 1"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Header fixed-left 2"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "scrolling top-left"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "scrolling top-right")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "...")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "...")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "...")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "...")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "...")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "...")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "...")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "...")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(TableDataCell, null, "scrolling bottom-left"), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "..."), /*#__PURE__*/React.createElement(TableDataCell, null, "scrolling bottom-right")))));
