// @dhis2-ui/table/build/es/data-table/table-elements/table.stories.internal.js
/* eslint-disable react/prop-types */
import { IconAttachment16 } from '@dhis2/ui-icons';
import { Box } from '@dhis2-ui/box';
import { Checkbox } from '@dhis2-ui/checkbox';
import React, { useState } from 'react';
import { TableBody } from './table-body.js';
import { TableDataCell } from './table-data-cell/table-data-cell.js';
import { TableFoot } from './table-foot.js';
import { TableHead } from './table-head.js';
import { TableHeaderCell } from './table-header-cell/table-header-cell.js';
import { TableHeaderCellAction } from './table-header-cell-action.js';
import { TableRow } from './table-row.js';
import { TableScrollBox } from './table-scroll-box.js';
import { TableToolbar } from './table-toolbar.js';
import { Table } from './table.js';
const subtitle = 'Used to display information in a standard, effective way.';
const description = `
Should be used with multiple Table-related child components - see the table and examples below.

\`\`\`js
import {
    Table,
    TableToolbar,
    TableHead,
    TableBody,
    TableFoot,
    TableRow,
    TableDataCell,
    TableHeaderCell,
    TableScrollBox,
} from '@dhis2/ui'
\`\`\`
`;
export default {
  title: 'Data Display/Table',
  component: Table,
  // Add subcomponents to the args table
  subcomponents: {
    TableToolbar,
    TableHead,
    TableBody,
    TableFoot,
    TableRow,
    TableDataCell,
    TableHeaderCell
  },
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    // hide control from `children` because that can't be live edited
    children: {
      control: {
        disable: true
      }
    },
    // hide child component controls completely to avoid confusion
    borderedCells: {
      table: {
        disable: true
      }
    },
    largeCells: {
      table: {
        disable: true
      }
    },
    staticCells: {
      table: {
        disable: true
      }
    },
    draggableRows: {
      table: {
        disable: true
      }
    }
  }
};
const BasicTemplate = _ref => {
  let {
    borderedCells,
    largeCells,
    staticCells,
    draggableRows,
    ...args
  } = _ref;
  return /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
    large: largeCells
  }, "First name"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    large: largeCells
  }, "Last name"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    large: largeCells
  }, "Incident date"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    large: largeCells
  }, "Last updated"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, {
    draggable: draggableRows
  }, /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "Onyekachukwu"), /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "Kariuki"), /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "05/25/1972")), /*#__PURE__*/React.createElement(TableRow, {
    draggable: draggableRows
  }, /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "Kwasi"), /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "08/11/2010"), /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "02/26/1991")), /*#__PURE__*/React.createElement(TableRow, {
    draggable: draggableRows
  }, /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "Siyabonga"), /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "07/21/1981"), /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells
  }, "02/06/2007"))), /*#__PURE__*/React.createElement(TableFoot, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, {
    staticStyle: staticCells,
    large: largeCells,
    bordered: borderedCells,
    colSpan: "4"
  }, "Footer content"))));
};
export const Default = BasicTemplate.bind({});
Default.args = {};
export const BorderedCells = BasicTemplate.bind({});
BorderedCells.args = {
  borderedCells: true
};
export const LargeCells = BasicTemplate.bind({});
LargeCells.args = {
  largeCells: true
};
export const StaticCells = BasicTemplate.bind({});
StaticCells.args = {
  staticCells: true
};
export const DraggableRows = BasicTemplate.bind({});
DraggableRows.args = {
  draggableRows: true
};
const IndividualCellTemplate = args => /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, null, "Header"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Header"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Header"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Header"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, {
  tag: "th"
}, "Active Cells"), /*#__PURE__*/React.createElement(TableDataCell, null, "Inactive cell"), /*#__PURE__*/React.createElement(TableDataCell, {
  active: true,
  onClick: () => {
    console.log('You clicked on a TableDataCell');
  }
}, "Acive Cell with onClick"), /*#__PURE__*/React.createElement(TableDataCell, null, "Inactive cell")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, {
  tag: "th"
}, "Alignment"), /*#__PURE__*/React.createElement(TableDataCell, {
  align: "left"
}, "Left"), /*#__PURE__*/React.createElement(TableDataCell, {
  align: "center"
}, "Center"), /*#__PURE__*/React.createElement(TableDataCell, {
  align: "right"
}, "Right")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, {
  tag: "th"
}, "Built in formatting"), /*#__PURE__*/React.createElement(TableDataCell, {
  error: true
}, "Error"), /*#__PURE__*/React.createElement(TableDataCell, {
  valid: true
}, "Valid"), /*#__PURE__*/React.createElement(TableDataCell, {
  muted: true
}, "Muted"))));
export const CellStyling = IndividualCellTemplate.bind({});
CellStyling.args = {};
const IrregularHeadersTemplate = args => /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement("colgroup", null), /*#__PURE__*/React.createElement("colgroup", {
  span: "2"
}), /*#__PURE__*/React.createElement("colgroup", {
  span: "2"
}), /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  rowSpan: "2",
  colSpan: "2",
  bordered: true
}), /*#__PURE__*/React.createElement(TableHeaderCell, {
  align: "center",
  colSpan: "2",
  scope: "colgroup"
}, "Mars"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  align: "center",
  colSpan: "2",
  scope: "colgroup"
}, "Venus")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  scope: "col"
}, "Produced"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  scope: "col"
}, "Sold"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  scope: "col"
}, "Produced"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  scope: "col"
}, "Sold"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  staticStyle: true,
  bordered: true,
  scope: "row",
  rowSpan: "2"
}, "Type"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  staticStyle: true,
  bordered: true,
  scope: "row"
}, "Teddy Bears"), /*#__PURE__*/React.createElement(TableDataCell, {
  staticStyle: true
}, "50,000"), /*#__PURE__*/React.createElement(TableDataCell, {
  staticStyle: true
}, "30,000"), /*#__PURE__*/React.createElement(TableDataCell, {
  staticStyle: true
}, "100,000"), /*#__PURE__*/React.createElement(TableDataCell, {
  staticStyle: true
}, "80,000")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  staticStyle: true,
  bordered: true,
  scope: "row"
}, "Board Games"), /*#__PURE__*/React.createElement(TableDataCell, {
  staticStyle: true
}, "10,000"), /*#__PURE__*/React.createElement(TableDataCell, {
  staticStyle: true
}, "5,000"), /*#__PURE__*/React.createElement(TableDataCell, {
  staticStyle: true
}, "12,000"), /*#__PURE__*/React.createElement(TableDataCell, {
  staticStyle: true
}, "9,000"))));
export const IrregularHeaders = IrregularHeadersTemplate.bind({});
IrregularHeaders.args = {};
const ToolbarsTemplate = args => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(TableToolbar, null, /*#__PURE__*/React.createElement("p", null, "Content")), /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, null, "First name"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Last name"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Incident date"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Last updated"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Onyekachukwu"), /*#__PURE__*/React.createElement(TableDataCell, null, "Kariuki"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/25/1972")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Kwasi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/26/1991")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Siyabonga"), /*#__PURE__*/React.createElement(TableDataCell, null, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007")))), /*#__PURE__*/React.createElement(TableToolbar, {
  position: "bottom"
}, /*#__PURE__*/React.createElement("p", null, "Content (bottom)")));
export const Toolbars = ToolbarsTemplate.bind({});
Toolbars.args = {};
const SelectableRowsTemplate = args => {
  const [selected, setSelected] = useState({
    id_2: true
  });
  const toggleSelected = _ref2 => {
    let {
      value,
      checked
    } = _ref2;
    setSelected({
      ...selected,
      [value]: checked
    });
  };
  const toggleAll = _ref3 => {
    let {
      checked
    } = _ref3;
    setSelected({
      id_1: checked,
      id_2: checked,
      id_3: checked
    });
  };
  const allSelected = () => Object.values(selected).filter(value => value).length === 3;
  return /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
    width: "48px"
  }, /*#__PURE__*/React.createElement(Checkbox, {
    checked: allSelected(),
    onChange: toggleAll
  })), /*#__PURE__*/React.createElement(TableHeaderCell, null, "First name"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Last name"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Incident date"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Last updated"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, {
    selected: !!selected.id_1
  }, /*#__PURE__*/React.createElement(TableDataCell, {
    width: "48px"
  }, /*#__PURE__*/React.createElement(Checkbox, {
    value: "id_1",
    checked: !!selected.id_1,
    onChange: toggleSelected
  })), /*#__PURE__*/React.createElement(TableDataCell, null, "Onyekachukwu"), /*#__PURE__*/React.createElement(TableDataCell, null, "Kariuki"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/25/1972")), /*#__PURE__*/React.createElement(TableRow, {
    selected: !!selected.id_2
  }, /*#__PURE__*/React.createElement(TableDataCell, {
    width: "48px"
  }, /*#__PURE__*/React.createElement(Checkbox, {
    value: "id_2",
    checked: !!selected.id_2,
    onChange: toggleSelected
  })), /*#__PURE__*/React.createElement(TableDataCell, null, "Kwasi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/26/1991")), /*#__PURE__*/React.createElement(TableRow, {
    selected: !!selected.id_3
  }, /*#__PURE__*/React.createElement(TableDataCell, {
    width: "48px"
  }, /*#__PURE__*/React.createElement(Checkbox, {
    value: "id_3",
    checked: !!selected.id_3,
    onChange: toggleSelected
  })), /*#__PURE__*/React.createElement(TableDataCell, null, "Siyabonga"), /*#__PURE__*/React.createElement(TableDataCell, null, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"))));
};
export const SelectableRows = SelectableRowsTemplate.bind({});
SelectableRows.args = {};
const FixedHeaderTemplate = _ref4 => {
  let {
    ...args
  } = _ref4;
  return /*#__PURE__*/React.createElement(TableScrollBox, {
    maxHeight: "350px"
  }, /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
    fixed: true,
    top: "0"
  }, "First name"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    fixed: true,
    top: "0"
  }, "Last name"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    fixed: true,
    top: "0"
  }, "Incident date"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    fixed: true,
    top: "0"
  }, "Last updated"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    fixed: true,
    top: "0"
  }, "Age"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    fixed: true,
    top: "0"
  }, "Registering unit"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    fixed: true,
    top: "0"
  }, "Assigned user"), /*#__PURE__*/React.createElement(TableHeaderCell, {
    fixed: true,
    top: "0"
  }, "Status"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Onyekachukwu"), /*#__PURE__*/React.createElement(TableDataCell, null, "Kariuki"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/25/1972"), /*#__PURE__*/React.createElement(TableDataCell, null, "66"), /*#__PURE__*/React.createElement(TableDataCell, null, "Jawi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Sofie Hubert"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Kwasi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/26/1991"), /*#__PURE__*/React.createElement(TableDataCell, null, "38"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mokassie MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Dashonte Clarke"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Siyabonga"), /*#__PURE__*/React.createElement(TableDataCell, null, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "98"), /*#__PURE__*/React.createElement(TableDataCell, null, "Bathurst MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Unassigned"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Chiyembekezo"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okeke"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/23/1982"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/15/2003"), /*#__PURE__*/React.createElement(TableDataCell, null, "2"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mayolla MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Wan Gengxin"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Mtendere"), /*#__PURE__*/React.createElement(TableDataCell, null, "Afolayan"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/12/1994"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/12/1972"), /*#__PURE__*/React.createElement(TableDataCell, null, "37"), /*#__PURE__*/React.createElement(TableDataCell, null, "Gbangadu MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Gvozden Boskovsky"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Inyene"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okonkwo"), /*#__PURE__*/React.createElement(TableDataCell, null, "04/01/1971"), /*#__PURE__*/React.createElement(TableDataCell, null, "03/16/2000"), /*#__PURE__*/React.createElement(TableDataCell, null, "70"), /*#__PURE__*/React.createElement(TableDataCell, null, "Kunike Barina"), /*#__PURE__*/React.createElement(TableDataCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Amaka"), /*#__PURE__*/React.createElement(TableDataCell, null, "Pretorius"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/25/1996"), /*#__PURE__*/React.createElement(TableDataCell, null, "09/15/1986"), /*#__PURE__*/React.createElement(TableDataCell, null, "32"), /*#__PURE__*/React.createElement(TableDataCell, null, "Bargbo"), /*#__PURE__*/React.createElement(TableDataCell, null, "Alberto Raya"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Meti"), /*#__PURE__*/React.createElement(TableDataCell, null, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "10/24/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/26/1989"), /*#__PURE__*/React.createElement(TableDataCell, null, "8"), /*#__PURE__*/React.createElement(TableDataCell, null, "Majihun MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Unassigned"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Eshe"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okeke"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(TableDataCell, null, "63"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mambiama CHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Shadrias Pearson"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, null, "Obi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "06/07/1990"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/03/2006"), /*#__PURE__*/React.createElement(TableDataCell, null, "28"), /*#__PURE__*/React.createElement(TableDataCell, null, "Dalakuru CHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")))));
};
export const FixedHeader = FixedHeaderTemplate.bind({});
FixedHeader.args = {
  borderless: true
};
const FixedFirstColumnTemplate = args => /*#__PURE__*/React.createElement(TableScrollBox, {
  maxWidth: "500px"
}, /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "0"
}, "First name"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Last name"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Incident date"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Last updated"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Age"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Registering unit"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Assigned user"), /*#__PURE__*/React.createElement(TableHeaderCell, null, "Status"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Onyekachukwu"), /*#__PURE__*/React.createElement(TableDataCell, null, "Kariuki"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/25/1972"), /*#__PURE__*/React.createElement(TableDataCell, null, "66"), /*#__PURE__*/React.createElement(TableDataCell, null, "Jawi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Sofie Hubert"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Kwasi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/26/1991"), /*#__PURE__*/React.createElement(TableDataCell, null, "38"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mokassie MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Dashonte Clarke"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Siyabonga"), /*#__PURE__*/React.createElement(TableDataCell, null, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "98"), /*#__PURE__*/React.createElement(TableDataCell, null, "Bathurst MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Unassigned"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Chiyembekezo"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okeke"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/23/1982"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/15/2003"), /*#__PURE__*/React.createElement(TableDataCell, null, "2"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mayolla MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Wan Gengxin"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Mtendere"), /*#__PURE__*/React.createElement(TableDataCell, null, "Afolayan"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/12/1994"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/12/1972"), /*#__PURE__*/React.createElement(TableDataCell, null, "37"), /*#__PURE__*/React.createElement(TableDataCell, null, "Gbangadu MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Gvozden Boskovsky"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Inyene"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okonkwo"), /*#__PURE__*/React.createElement(TableDataCell, null, "04/01/1971"), /*#__PURE__*/React.createElement(TableDataCell, null, "03/16/2000"), /*#__PURE__*/React.createElement(TableDataCell, null, "70"), /*#__PURE__*/React.createElement(TableDataCell, null, "Kunike Barina"), /*#__PURE__*/React.createElement(TableDataCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Amaka"), /*#__PURE__*/React.createElement(TableDataCell, null, "Pretorius"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/25/1996"), /*#__PURE__*/React.createElement(TableDataCell, null, "09/15/1986"), /*#__PURE__*/React.createElement(TableDataCell, null, "32"), /*#__PURE__*/React.createElement(TableDataCell, null, "Bargbo"), /*#__PURE__*/React.createElement(TableDataCell, null, "Alberto Raya"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Meti"), /*#__PURE__*/React.createElement(TableDataCell, null, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "10/24/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/26/1989"), /*#__PURE__*/React.createElement(TableDataCell, null, "8"), /*#__PURE__*/React.createElement(TableDataCell, null, "Majihun MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Unassigned"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Eshe"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okeke"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(TableDataCell, null, "63"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mambiama CHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Shadrias Pearson"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Obi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "06/07/1990"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/03/2006"), /*#__PURE__*/React.createElement(TableDataCell, null, "28"), /*#__PURE__*/React.createElement(TableDataCell, null, "Dalakuru CHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")))));
export const FixedFirstColumn = FixedFirstColumnTemplate.bind({});
FixedFirstColumn.args = {
  borderless: true,
  width: '1000px'
};
const FixedHeaderAndTwoColumnsTemplate = args => /*#__PURE__*/React.createElement(TableScrollBox, {
  maxHeight: "350px",
  maxWidth: "500px"
}, /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/25/1972"), /*#__PURE__*/React.createElement(TableDataCell, null, "66"), /*#__PURE__*/React.createElement(TableDataCell, null, "Jawi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Sofie Hubert"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/26/1991"), /*#__PURE__*/React.createElement(TableDataCell, null, "38"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mokassie MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Dashonte Clarke"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "98"), /*#__PURE__*/React.createElement(TableDataCell, null, "Bathurst MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Unassigned"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/23/1982"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/15/2003"), /*#__PURE__*/React.createElement(TableDataCell, null, "2"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mayolla MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Wan Gengxin"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/12/1994"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/12/1972"), /*#__PURE__*/React.createElement(TableDataCell, null, "37"), /*#__PURE__*/React.createElement(TableDataCell, null, "Gbangadu MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Gvozden Boskovsky"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/React.createElement(TableDataCell, null, "04/01/1971"), /*#__PURE__*/React.createElement(TableDataCell, null, "03/16/2000"), /*#__PURE__*/React.createElement(TableDataCell, null, "70"), /*#__PURE__*/React.createElement(TableDataCell, null, "Kunike Barina"), /*#__PURE__*/React.createElement(TableDataCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/25/1996"), /*#__PURE__*/React.createElement(TableDataCell, null, "09/15/1986"), /*#__PURE__*/React.createElement(TableDataCell, null, "32"), /*#__PURE__*/React.createElement(TableDataCell, null, "Bargbo"), /*#__PURE__*/React.createElement(TableDataCell, null, "Alberto Raya"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "10/24/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/26/1989"), /*#__PURE__*/React.createElement(TableDataCell, null, "8"), /*#__PURE__*/React.createElement(TableDataCell, null, "Majihun MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Unassigned"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(TableDataCell, null, "63"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mambiama CHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Shadrias Pearson"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "06/07/1990"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/03/2006"), /*#__PURE__*/React.createElement(TableDataCell, null, "28"), /*#__PURE__*/React.createElement(TableDataCell, null, "Dalakuru CHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")))));
export const FixedHeaderAndTwoColumns = FixedHeaderAndTwoColumnsTemplate.bind({});
FixedHeaderAndTwoColumns.args = {
  layout: 'fixed',
  borderless: true,
  width: '1000px'
};
const ScrollingTableWithToolbarsTemplate = args => /*#__PURE__*/React.createElement(Box, {
  width: "500px"
}, /*#__PURE__*/React.createElement(TableToolbar, null, /*#__PURE__*/React.createElement("p", null, "Content")), /*#__PURE__*/React.createElement(TableScrollBox, {
  maxHeight: "350px",
  maxWidth: "500px"
}, /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/25/1972"), /*#__PURE__*/React.createElement(TableDataCell, null, "66"), /*#__PURE__*/React.createElement(TableDataCell, null, "Jawi"), /*#__PURE__*/React.createElement(TableDataCell, null, "Sofie Hubert"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/26/1991"), /*#__PURE__*/React.createElement(TableDataCell, null, "38"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mokassie MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Dashonte Clarke"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(TableDataCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(TableDataCell, null, "98"), /*#__PURE__*/React.createElement(TableDataCell, null, "Bathurst MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Unassigned"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/23/1982"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/15/2003"), /*#__PURE__*/React.createElement(TableDataCell, null, "2"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mayolla MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Wan Gengxin"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/React.createElement(TableDataCell, null, "08/12/1994"), /*#__PURE__*/React.createElement(TableDataCell, null, "05/12/1972"), /*#__PURE__*/React.createElement(TableDataCell, null, "37"), /*#__PURE__*/React.createElement(TableDataCell, null, "Gbangadu MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Gvozden Boskovsky"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/React.createElement(TableDataCell, null, "04/01/1971"), /*#__PURE__*/React.createElement(TableDataCell, null, "03/16/2000"), /*#__PURE__*/React.createElement(TableDataCell, null, "70"), /*#__PURE__*/React.createElement(TableDataCell, null, "Kunike Barina"), /*#__PURE__*/React.createElement(TableDataCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/25/1996"), /*#__PURE__*/React.createElement(TableDataCell, null, "09/15/1986"), /*#__PURE__*/React.createElement(TableDataCell, null, "32"), /*#__PURE__*/React.createElement(TableDataCell, null, "Bargbo"), /*#__PURE__*/React.createElement(TableDataCell, null, "Alberto Raya"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/React.createElement(TableDataCell, null, "10/24/2010"), /*#__PURE__*/React.createElement(TableDataCell, null, "07/26/1989"), /*#__PURE__*/React.createElement(TableDataCell, null, "8"), /*#__PURE__*/React.createElement(TableDataCell, null, "Majihun MCHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Unassigned"), /*#__PURE__*/React.createElement(TableDataCell, null, "Complete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(TableDataCell, null, "63"), /*#__PURE__*/React.createElement(TableDataCell, null, "Mambiama CHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Shadrias Pearson"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete")), /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/React.createElement(TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/React.createElement(TableDataCell, null, "06/07/1990"), /*#__PURE__*/React.createElement(TableDataCell, null, "01/03/2006"), /*#__PURE__*/React.createElement(TableDataCell, null, "28"), /*#__PURE__*/React.createElement(TableDataCell, null, "Dalakuru CHP"), /*#__PURE__*/React.createElement(TableDataCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/React.createElement(TableDataCell, null, "Incomplete"))))), /*#__PURE__*/React.createElement(TableToolbar, {
  position: "bottom"
}, /*#__PURE__*/React.createElement("p", null, "Content (bottom)")));
export const ScrollingTableWithToolbars = ScrollingTableWithToolbarsTemplate.bind({});
ScrollingTableWithToolbars.args = {
  layout: 'fixed',
  borderless: true,
  width: '1000px'
};
const HeaderActionsTemplate = args => /*#__PURE__*/React.createElement(Table, args, /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableHeaderCell, null, /*#__PURE__*/React.createElement("span", {
  style: {
    display: 'flex',
    height: '100%',
    alignItems: 'center'
  }
}, "First name", /*#__PURE__*/React.createElement(TableHeaderCellAction, null, /*#__PURE__*/React.createElement(IconAttachment16, {
  color: "green"
})))), /*#__PURE__*/React.createElement(TableHeaderCell, null, /*#__PURE__*/React.createElement("span", {
  style: {
    display: 'flex',
    height: '100%',
    alignItems: 'center'
  }
}, "Last name", /*#__PURE__*/React.createElement(TableHeaderCellAction, null, /*#__PURE__*/React.createElement(IconAttachment16, {
  color: "red"
})))))), /*#__PURE__*/React.createElement(TableBody, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableDataCell, {
  colSpan: "3"
}))));
export const HeaderActions = HeaderActionsTemplate.bind({});
HeaderActions.args = {};
