// @dhis2-ui/node/build/cjs/leaves.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Leaves = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Leaves = _ref => {
  let {
    children,
    open,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: "jsx-3677374542" + " " + ((0, _classnames.default)({
      open
    }) || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3677374542"
  }, ["div.jsx-3677374542{height:0;overflow:hidden;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}", ".open.jsx-3677374542{height:auto;}"]));
};
exports.Leaves = Leaves;
Leaves.propTypes = {
  children: _propTypes.default.node,
  dataTest: _propTypes.default.string,
  open: _propTypes.default.bool
};
