// @dhis2-ui/node/build/cjs/toggle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Toggle = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ArrowDown = () => /*#__PURE__*/_react.default.createElement("div", {
  className: "nodeArrow"
}, /*#__PURE__*/_react.default.createElement("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 24 24",
  className: "jsx-4098137027"
}, /*#__PURE__*/_react.default.createElement("path", {
  d: "M15.7041 10.3329L12.1495 14.3318C12.0699 14.4213 11.9301 14.4213 11.8505 14.3318L8.29589 10.3329C8.18124 10.2039 8.2728 10 8.44537 10H15.5546C15.7272 10 15.8188 10.2039 15.7041 10.3329Z",
  className: "jsx-4098137027"
}), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "4098137027"
}, ["svg.jsx-4098137027{fill:inherit;height:24px;width:24px;vertical-align:middle;pointer-events:none;}"])));
const Toggle = _ref => {
  let {
    open,
    onOpen,
    onClose,
    dataTest
  } = _ref;
  const onClick = open ? onClose : onOpen;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    onClick: event => onClick && onClick({
      open: !open
    }, event),
    className: _style.default.dynamic([["375729792", [_uiConstants.colors.grey400, _uiConstants.colors.grey900, _uiConstants.colors.grey100]]]) + " " + ((0, _classnames.default)({
      open
    }) || "")
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: _style.default.dynamic([["375729792", [_uiConstants.colors.grey400, _uiConstants.colors.grey900, _uiConstants.colors.grey100]]])
  }, /*#__PURE__*/_react.default.createElement(ArrowDown, null)), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "375729792",
    dynamic: [_uiConstants.colors.grey400, _uiConstants.colors.grey900, _uiConstants.colors.grey100]
  }, ["div.__jsx-style-dynamic-selector{width:24px;position:relative;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;}", `div.__jsx-style-dynamic-selector:after{background:${_uiConstants.colors.grey400};height:calc(100% - 24px);inset-inline-start:12px;position:absolute;top:15px;width:1px;z-index:1;}`, ".open.__jsx-style-dynamic-selector:after{content:'';}", `span.__jsx-style-dynamic-selector{display:block;position:relative;z-index:2;fill:${_uiConstants.colors.grey900};}`, "div.__jsx-style-dynamic-selector .nodeArrow{vertical-align:top;-webkit-transform:rotate(-90deg);-ms-transform:rotate(-90deg);transform:rotate(-90deg);-webkit-transition:-webkit-transform 0.1s ease-out;-webkit-transition:transform 0.1s ease-out;transition:transform 0.1s ease-out;}", "div.__jsx-style-dynamic-selector .nodeArrow:dir(rtl){-webkit-transform:rotate(90deg);-ms-transform:rotate(90deg);transform:rotate(90deg);}", ".open.__jsx-style-dynamic-selector .nodeArrow{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}", ".open.__jsx-style-dynamic-selector .nodeArrow:dir(rtl){-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}", `div.__jsx-style-dynamic-selector:hover{cursor:pointer;border-radius:3px;background:${_uiConstants.colors.grey100};}`, "div.__jsx-style-dynamic-selector:hover.__jsx-style-dynamic-selector:after{background:#bcc8d4;}"]));
};
exports.Toggle = Toggle;
Toggle.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  open: _propTypes.default.bool,
  onClose: _propTypes.default.func,
  onOpen: _propTypes.default.func
};
