// @dhis2/ui-forms/build/cjs/FieldGroupFF/FieldGroupFF.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FieldGroupFF = void 0;
var _field = require("@dhis2-ui/field");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const FieldGroupFF = _ref => {
  let {
    name,
    label,
    children,
    required
  } = _ref;
  const {
    meta: {
      error,
      touched
    }
  } = (0, _reactFinalForm.useField)(name, {
    subscription: {
      error: true,
      touched: true
    }
  });
  const isError = !!error && !!touched;
  let errorText;
  if (isError) {
    if (typeof error === 'string') {
      errorText = error;
    } else if (typeof error === 'object' && error[name]) {
      errorText = error[name];
    } else {
      errorText = null;
    }
  }
  return /*#__PURE__*/_react.default.createElement(_field.FieldGroup, {
    label: label,
    required: required,
    error: isError,
    name: name,
    validationText: errorText
  }, children);
};
exports.FieldGroupFF = FieldGroupFF;
FieldGroupFF.propTypes = {
  children: _propTypes.default.node,
  label: _propTypes.default.string,
  name: _propTypes.default.string,
  required: _propTypes.default.bool
};
