// @dhis2-ui/divider/build/es/divider.prod.stories.js
import React from 'react';
import { Divider } from './divider.js';
const description = `
A light divider to separate content.

\`\`\`js
import { Divider } from '@dhis2/ui'
\`\`\`
`;
export default {
  title: 'Divider',
  component: Divider,
  argTypes: {
    margin: {
      table: {
        defaultValue: {
          summary: '8px 0'
        }
      }
    }
  },
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Template = args => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", null, "Content above"), /*#__PURE__*/React.createElement(Divider, args), /*#__PURE__*/React.createElement("div", null, "Content below"));
export const Default = Template.bind({});
export const Dense = Template.bind({});
Dense.args = {
  dense: true
};
export const Margin = Template.bind({});
Margin.args = {
  margin: '20px 20px 20px 20px'
};
export const RTLUnevenMargin = args => /*#__PURE__*/React.createElement("div", {
  dir: "rtl"
}, /*#__PURE__*/React.createElement(Template, args));
RTLUnevenMargin.args = {
  margin: '10px 5px 40px 50px'
};
