// @dhis2-ui/header-bar/build/cjs/apps.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _appRuntime = require("@dhis2/app-runtime");
var _uiConstants = require("@dhis2/ui-constants");
var _uiIcons = require("@dhis2/ui-icons");
var _card = require("@dhis2-ui/card");
var _input = require("@dhis2-ui/input");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _joinPath = require("./join-path.js");
var _index = _interopRequireDefault(require("./locales/index.js"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/**
 * Copied from here:
 * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Escaping
 */
function escapeRegExpCharacters(text) {
  return text.replace(/[/.*+?^${}()|[\]\\]/g, '\\$&');
}
function Search(_ref) {
  let {
    value,
    onChange
  } = _ref;
  const {
    baseUrl
  } = (0, _appRuntime.useConfig)();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-4264724627"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-4264724627"
  }, /*#__PURE__*/_react.default.createElement(_input.InputField, {
    value: value,
    name: "filter",
    placeholder: _index.default.t('Search apps'),
    onChange: onChange,
    initialFocus: true
  })), /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-4264724627"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: (0, _joinPath.joinPath)(baseUrl, 'dhis-web-menu-management'),
    className: "jsx-4264724627"
  }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconSettings24, {
    color: _uiConstants.colors.grey700
  }))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4264724627"
  }, ["div.jsx-4264724627{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-flex-wrap:nowrap;-ms-flex-wrap:nowrap;flex-wrap:nowrap;height:52px;margin:8px;}", "span.jsx-4264724627{-webkit-flex:1 100%;-ms-flex:1 100%;flex:1 100%;}", "span.jsx-4264724627:last-child{-webkit-flex:1 auto;-ms-flex:1 auto;flex:1 auto;margin:8px;}"]));
}
Search.propTypes = {
  value: _propTypes.default.string.isRequired,
  onChange: _propTypes.default.func.isRequired
};
function Item(_ref2) {
  let {
    name,
    path,
    img
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement("a", {
    href: path,
    className: _style.default.dynamic([["1412616027", [_uiConstants.theme.primary050]]])
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: img,
    alt: "app logo",
    className: _style.default.dynamic([["1412616027", [_uiConstants.theme.primary050]]])
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["1412616027", [_uiConstants.theme.primary050]]])
  }, name), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1412616027",
    dynamic: [_uiConstants.theme.primary050]
  }, ["a.__jsx-style-dynamic-selector{display:inline-block;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;width:96px;margin:8px;border-radius:12px;-webkit-text-decoration:none;text-decoration:none;cursor:pointer;}", `a.__jsx-style-dynamic-selector:hover,a.__jsx-style-dynamic-selector:focus{background-color:${_uiConstants.theme.primary050};cursor:pointer;}`, "a.__jsx-style-dynamic-selector:hover>div.__jsx-style-dynamic-selector{font-weight:500;cursor:pointer;}", "img.__jsx-style-dynamic-selector{width:48px;height:48px;margin:8px;cursor:pointer;}", "div.__jsx-style-dynamic-selector{overflow-wrap:anywhere;margin-top:14px;color:rgba(0,0,0,0.87);font-size:12px;-webkit-letter-spacing:0.01em;-moz-letter-spacing:0.01em;-ms-letter-spacing:0.01em;letter-spacing:0.01em;line-height:14px;text-align:center;cursor:pointer;}"]));
}
Item.propTypes = {
  img: _propTypes.default.string,
  name: _propTypes.default.string,
  path: _propTypes.default.string
};
function List(_ref3) {
  let {
    apps,
    filter
  } = _ref3;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": "headerbar-apps-menu-list",
    className: "jsx-2424821176"
  }, apps.filter(_ref4 => {
    let {
      displayName,
      name
    } = _ref4;
    const appName = displayName || name;
    const formattedAppName = appName.toLowerCase();
    const formattedFilter = escapeRegExpCharacters(filter).toLowerCase();
    return filter.length > 0 ? formattedAppName.match(formattedFilter) : true;
  }).map((_ref5, idx) => {
    let {
      displayName,
      name,
      defaultAction,
      icon
    } = _ref5;
    return /*#__PURE__*/_react.default.createElement(Item, {
      key: `app-${name}-${idx}`,
      name: displayName || name,
      path: defaultAction,
      img: icon
    });
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2424821176"
  }, ["div.jsx-2424821176{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-align-content:flex-start;-ms-flex-line-pack:start;align-content:flex-start;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;width:30vw;min-width:300px;max-width:560px;min-height:200px;max-height:465px;margin-block-start:0;margin-block-end:8px;margin-inline:8px;overflow:auto;overflow-x:hidden;}"]));
}
List.propTypes = {
  apps: _propTypes.default.array,
  filter: _propTypes.default.string
};
const AppMenu = _ref6 => {
  let {
    apps,
    filter,
    onFilterChange
  } = _ref6;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": "headerbar-apps-menu",
    className: "jsx-3826299842"
  }, /*#__PURE__*/_react.default.createElement(_card.Card, null, /*#__PURE__*/_react.default.createElement(Search, {
    value: filter,
    onChange: onFilterChange
  }), /*#__PURE__*/_react.default.createElement(List, {
    apps: apps,
    filter: filter
  })), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3826299842"
  }, ["div.jsx-3826299842{z-index:10000;position:absolute;inset-inline-end:-4px;}"]));
};
AppMenu.propTypes = {
  apps: _propTypes.default.array.isRequired,
  onFilterChange: _propTypes.default.func.isRequired,
  filter: _propTypes.default.string
};
const Apps = _ref7 => {
  let {
    apps
  } = _ref7;
  const [show, setShow] = (0, _react.useState)(false);
  const [filter, setFilter] = (0, _react.useState)('');
  const handleVisibilityToggle = (0, _react.useCallback)(() => setShow(!show), [show]);
  const handleFilterChange = (0, _react.useCallback)(_ref8 => {
    let {
      value
    } = _ref8;
    return setFilter(value);
  }, []);
  const containerEl = (0, _react.useRef)(null);
  const onDocClick = (0, _react.useCallback)(evt => {
    if (containerEl.current && !containerEl.current.contains(evt.target)) {
      setShow(false);
    }
  }, []);
  (0, _react.useEffect)(() => {
    document.addEventListener('click', onDocClick);
    return () => document.removeEventListener('click', onDocClick);
  }, [onDocClick]);
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: containerEl,
    "data-test": "headerbar-apps",
    className: _style.default.dynamic([["325245423", [_uiConstants.spacers.dp4, _uiConstants.spacers.dp12]]])
  }, /*#__PURE__*/_react.default.createElement("button", {
    onClick: handleVisibilityToggle,
    "data-test": "headerbar-apps-icon",
    className: _style.default.dynamic([["325245423", [_uiConstants.spacers.dp4, _uiConstants.spacers.dp12]]])
  }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconApps24, {
    color: _uiConstants.colors.white
  })), show ? /*#__PURE__*/_react.default.createElement(AppMenu, {
    apps: apps,
    filter: filter,
    onFilterChange: handleFilterChange
  }) : null, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "325245423",
    dynamic: [_uiConstants.spacers.dp4, _uiConstants.spacers.dp12]
  }, [`button.__jsx-style-dynamic-selector{display:block;background:transparent;padding-block-start:${_uiConstants.spacers.dp4};padding-block-end:0;padding-inline:${_uiConstants.spacers.dp12};border:0;cursor:pointer;height:100%;}`, "button.__jsx-style-dynamic-selector:focus{outline:2px solid white;outline-offset:-2px;}", "button.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector:not(:focus-visible){outline:none;}", "button.__jsx-style-dynamic-selector:hover{background:#1a557f;}", "button.__jsx-style-dynamic-selector:active{background:#104067;}", "div.__jsx-style-dynamic-selector{position:relative;height:100%;}"]));
};
Apps.propTypes = {
  apps: _propTypes.default.array.isRequired
};
var _default = exports.default = Apps;
