// @dhis2-ui/header-bar/build/cjs/header-bar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HeaderBar = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _appRuntime = require("@dhis2/app-runtime");
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _apps = _interopRequireDefault(require("./apps.js"));
var _headerBarContext = require("./header-bar-context.js");
var _joinPath = require("./join-path.js");
var _index = _interopRequireDefault(require("./locales/index.js"));
var _logo = require("./logo.js");
var _notifications = require("./notifications.js");
var _onlineStatus = require("./online-status.js");
var _profile = _interopRequireDefault(require("./profile.js"));
var _title = require("./title.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const query = {
  title: {
    resource: 'systemSettings/applicationTitle'
  },
  help: {
    resource: 'systemSettings/helpPageLink'
  },
  user: {
    resource: 'me',
    params: {
      fields: ['authorities', 'avatar', 'email', 'name', 'settings']
    }
  },
  apps: {
    resource: 'action::menu/getModules'
  },
  notifications: {
    resource: 'me/dashboard'
  }
};
const HeaderBar = _ref => {
  var _data$user$avatar;
  let {
    appName,
    className,
    updateAvailable,
    onApplyAvailableUpdate
  } = _ref;
  const {
    appName: configAppName,
    baseUrl,
    pwaEnabled
  } = (0, _appRuntime.useConfig)();
  const {
    loading,
    error,
    data
  } = (0, _appRuntime.useDataQuery)(query);
  const apps = (0, _react.useMemo)(() => {
    const getPath = path => path.startsWith('http:') || path.startsWith('https:') ? path : (0, _joinPath.joinPath)(baseUrl, 'api', path);
    return data === null || data === void 0 ? void 0 : data.apps.modules.map(app => ({
      ...app,
      icon: getPath(app.icon),
      defaultAction: getPath(app.defaultAction)
    }));
  }, [data, baseUrl]);

  // See https://jira.dhis2.org/browse/LIBS-180
  if (!loading && !error) {
    // TODO: This will run every render which is probably wrong!
    // Also, setting the global locale shouldn't be done in the headerbar
    const locale = data.user.settings.keyUiLocale || 'en';
    _index.default.setDefaultNamespace('default');
    _index.default.changeLanguage(locale);
  }
  return /*#__PURE__*/_react.default.createElement(_headerBarContext.HeaderBarContextProvider, {
    updateAvailable: updateAvailable,
    onApplyAvailableUpdate: onApplyAvailableUpdate
  }, /*#__PURE__*/_react.default.createElement("header", {
    className: _style.default.dynamic([["4230161118", [_uiConstants.colors.white]]]) + " " + (className || "")
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["4230161118", [_uiConstants.colors.white]]]) + " " + "main"
  }, !loading && !error && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_logo.Logo, null), /*#__PURE__*/_react.default.createElement(_title.Title, {
    app: appName || configAppName,
    instance: data.title.applicationTitle
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["4230161118", [_uiConstants.colors.white]]]) + " " + "right-control-spacer"
  }), pwaEnabled && /*#__PURE__*/_react.default.createElement(_onlineStatus.OnlineStatus, null), /*#__PURE__*/_react.default.createElement(_notifications.Notifications, {
    interpretations: data.notifications.unreadInterpretations,
    messages: data.notifications.unreadMessageConversations,
    userAuthorities: data.user.authorities
  }), /*#__PURE__*/_react.default.createElement(_apps.default, {
    apps: apps
  }), /*#__PURE__*/_react.default.createElement(_profile.default, {
    name: data.user.name,
    email: data.user.email,
    avatarId: (_data$user$avatar = data.user.avatar) === null || _data$user$avatar === void 0 ? void 0 : _data$user$avatar.id,
    helpUrl: data.help.helpPageLink
  }))), pwaEnabled && !loading && !error && /*#__PURE__*/_react.default.createElement(_onlineStatus.OnlineStatus, {
    dense: true
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4230161118",
    dynamic: [_uiConstants.colors.white]
  }, [`.main.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;background-color:#2c6693;color:${_uiConstants.colors.white};height:48px;}`, ".right-control-spacer.__jsx-style-dynamic-selector{margin-inline-start:auto;}"])));
};
exports.HeaderBar = HeaderBar;
HeaderBar.propTypes = {
  appName: _propTypes.default.string,
  className: _propTypes.default.string,
  updateAvailable: _propTypes.default.bool,
  onApplyAvailableUpdate: _propTypes.default.func
};
