// @dhis2-ui/header-bar/build/cjs/logo-image.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LogoImage = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _appRuntime = require("@dhis2/app-runtime");
var _logo = require("@dhis2-ui/logo");
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const logoStyles = {
  styles: /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3467673193"
  }, ["svg.jsx-3467673193{height:25px;width:27px;}", "img.jsx-3467673193{max-height:100%;min-height:auto;width:auto;}"]),
  className: "jsx-3467673193"
};
const query = {
  customLogo: {
    resource: 'staticContent/logo_banner'
  }
};
const pathExists = data => data && data.customLogo && data.customLogo.images && data.customLogo.images.png;
const LogoImage = () => {
  const {
    loading,
    error,
    data
  } = (0, _appRuntime.useDataQuery)(query);
  if (loading) {
    return null;
  }
  let Logo;
  if (!error && pathExists(data)) {
    Logo = /*#__PURE__*/_react.default.createElement("img", {
      alt: "Headerbar Logo",
      src: data.customLogo.images.png,
      className: logoStyles.className
    });
  } else {
    Logo = /*#__PURE__*/_react.default.createElement(_logo.LogoIconWhite, {
      className: logoStyles.className
    });
  }
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-3930434724"
  }, Logo, logoStyles.styles, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3930434724"
  }, ["div.jsx-3930434724{padding:4px;min-width:48px;max-width:250px;height:48px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;overflow:hidden;}"]));
};
exports.LogoImage = LogoImage;
