// @dhis2-ui/calendar/build/es/calendar-input/calendar-input.js
import _JSXStyle from "styled-jsx/style";
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import { Button } from '@dhis2-ui/button';
import { Card } from '@dhis2-ui/card';
import { InputField, InputFieldProps } from '@dhis2-ui/input';
import { Layer } from '@dhis2-ui/layer';
import { Popper } from '@dhis2-ui/popper';
import cx from 'classnames';
import React, { useRef, useState } from 'react';
import { Calendar, CalendarProps } from '../calendar/calendar.js';
import i18n from '../locales/index.js';
const offsetModifier = {
  name: 'offset',
  options: {
    offset: [0, 2]
  }
};
export const CalendarInput = function () {
  let {
    onDateSelect,
    calendar,
    date,
    dir,
    locale,
    numberingSystem,
    weekDayFormat,
    timeZone,
    width,
    cellSize,
    clearable,
    dataTest = 'dhis2-uiwidgets-calendar-inputfield',
    ...rest
  } = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  const ref = useRef();
  const [open, setOpen] = useState(false);
  const calendarProps = React.useMemo(() => {
    const onDateSelectWrapper = selectedDate => {
      setOpen(false);
      onDateSelect === null || onDateSelect === void 0 ? void 0 : onDateSelect(selectedDate);
    };
    return {
      onDateSelect: onDateSelectWrapper,
      calendar,
      date,
      dir,
      locale,
      numberingSystem,
      weekDayFormat,
      timeZone,
      width,
      cellSize
    };
  }, [calendar, cellSize, date, dir, locale, numberingSystem, onDateSelect, timeZone, weekDayFormat, width]);
  const onFocus = () => {
    setOpen(true);
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    ref: ref,
    className: "jsx-862452676" + " " + "calendar-input-wrapper"
  }, /*#__PURE__*/React.createElement(InputField, _extends({
    label: i18n.t('Pick a date')
  }, rest, {
    dataTest: dataTest,
    type: "text",
    onFocus: onFocus,
    value: date
  })), clearable && /*#__PURE__*/React.createElement("div", {
    className: "jsx-862452676" + " " + (cx('calendar-clear-button', {
      // ToDo: this is a workaround to show the clear button in the correct place when an icon is shown.
      // Long-term, we should abstract and share the logic multi-select uses for the input-wrapper
      // https://dhis2.atlassian.net/browse/DHIS2-14848
      'with-icon': rest.valid || rest.error || rest.warning || rest.loading,
      'with-dense-wrapper': rest.dense
    }) || "")
  }, /*#__PURE__*/React.createElement(Button, {
    dataTest: "calendar-clear-button",
    secondary: true,
    small: true,
    onClick: () => calendarProps.onDateSelect(null),
    type: "button"
  }, i18n.t('Clear')))), open && /*#__PURE__*/React.createElement(Layer, {
    onBackdropClick: () => {
      setOpen(false);
    }
  }, /*#__PURE__*/React.createElement(Popper, {
    reference: ref,
    placement: "bottom-start",
    modifiers: [offsetModifier]
  }, /*#__PURE__*/React.createElement(Card, null, /*#__PURE__*/React.createElement(Calendar, _extends({}, calendarProps, {
    date: date
  }))))), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "862452676"
  }, [".calendar-input-wrapper.jsx-862452676{position:relative;}", ".calendar-clear-button.jsx-862452676{position:absolute;inset-inline-end:6px;-webkit-inset-block-start:27px;-ms-intb-rlock-start:27px;inset-block-start:27px;}", ".calendar-clear-button.with-icon.jsx-862452676{inset-inline-end:36px;}", ".calendar-clear-button.with-dense-wrapper.jsx-862452676{-webkit-inset-block-start:23px;-ms-intb-rlock-start:23px;inset-block-start:23px;}"]));
};
CalendarInput.propTypes = {
  ...CalendarProps,
  ...InputFieldProps
};
