// @dhis2-ui/button/build/es/split-button/split-button.e2e.stories.js
import React from 'react';
import { SplitButton } from './split-button.js';
window.onClick = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'SplitButton'
};
export const Default = () => /*#__PURE__*/React.createElement(SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Component")
}, "Label me!");
export const WithOnClick = () => /*#__PURE__*/React.createElement(SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Component"),
  onClick: window.onClick
}, "Label me!");
export const WithChildren = () => /*#__PURE__*/React.createElement(SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Component")
}, "I am a child");
export const WithIcon = () => /*#__PURE__*/React.createElement(SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Component"),
  icon: /*#__PURE__*/React.createElement("div", null, "Icon")
}, "Children");
export const WithInitialFocus = () => /*#__PURE__*/React.createElement(SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Component"),
  initialFocus: true
}, "Children");
export const WithDisabled = () => /*#__PURE__*/React.createElement(SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Component"),
  disabled: true
}, "Children");
