// @dhis2-ui/button/build/es/split-button/split-button.js
import _JSXStyle from "styled-jsx/style";
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { spacers, sharedPropTypes } from '@dhis2/ui-constants';
import { IconChevronUp16, IconChevronDown16 } from '@dhis2/ui-icons';
import { Layer } from '@dhis2-ui/layer';
import { Popper } from '@dhis2-ui/popper';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Button } from '../button/index.js';
import i18n from '../locales/index.js';
const rightButton = {
  styles: /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "2502148942"
  }, [`button.jsx-2502148942{padding:0 ${spacers.dp12};}`]),
  className: "jsx-2502148942"
};
class SplitButton extends Component {
  constructor() {
    super(...arguments);
    _defineProperty(this, "state", {
      open: false
    });
    _defineProperty(this, "anchorRef", /*#__PURE__*/React.createRef());
    _defineProperty(this, "handleKeyDown", event => {
      if (event.key === 'Escape' && this.state.open) {
        event.preventDefault();
        event.stopPropagation();
        this.setState({
          open: false
        });
        this.anchorRef.current && this.anchorRef.current.focus();
      }
    });
    _defineProperty(this, "onClick", (payload, event) => {
      if (this.props.onClick) {
        this.props.onClick({
          name: payload.name,
          value: payload.value,
          open: this.state.open
        }, event);
      }
    });
    _defineProperty(this, "onToggle", () => {
      this.setState(prevState => ({
        open: !prevState.open
      }));
    });
  }
  componentDidMount() {
    document.addEventListener('keydown', this.handleKeyDown);
  }
  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleKeyDown);
  }
  render() {
    const {
      open
    } = this.state;
    const {
      component,
      children,
      className,
      name,
      value,
      icon,
      small,
      large,
      primary,
      secondary,
      destructive,
      disabled,
      type,
      tabIndex,
      dataTest = 'dhis2-uicore-splitbutton',
      initialFocus
    } = this.props;
    const arrow = open ? /*#__PURE__*/React.createElement(IconChevronUp16, null) : /*#__PURE__*/React.createElement(IconChevronDown16, null);
    return /*#__PURE__*/React.createElement("div", {
      ref: this.anchorRef,
      "data-test": dataTest,
      className: "jsx-2116647326"
    }, /*#__PURE__*/React.createElement(Button, {
      name: name,
      value: value,
      icon: icon,
      small: small,
      large: large,
      primary: primary,
      secondary: secondary,
      destructive: destructive,
      disabled: disabled,
      onClick: this.onClick,
      type: type,
      tabIndex: tabIndex,
      className: cx(className),
      initialFocus: initialFocus,
      dataTest: `${dataTest}-button`
    }, children), /*#__PURE__*/React.createElement(Button, {
      name: name,
      value: value,
      small: small,
      large: large,
      primary: primary,
      secondary: secondary,
      destructive: destructive,
      disabled: disabled,
      onClick: this.onToggle,
      type: type,
      tabIndex: tabIndex,
      className: cx(className, rightButton.className),
      dataTest: `${dataTest}-toggle`,
      title: i18n.t('Toggle dropdown'),
      "aria-label": i18n.t('Toggle dropdown')
    }, arrow), open && /*#__PURE__*/React.createElement(Layer, {
      onBackdropClick: this.onToggle,
      transparent: true
    }, /*#__PURE__*/React.createElement(Popper, {
      dataTest: `${dataTest}-menu`,
      placement: "bottom-end",
      reference: this.anchorRef
    }, component)), rightButton.styles, /*#__PURE__*/React.createElement(_JSXStyle, {
      id: "2116647326"
    }, ["div.jsx-2116647326{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:inherit;white-space:nowrap;position:relative;z-index:0;}", "div.jsx-2116647326>button:first-child{border-start-end-radius:0;border-end-end-radius:0;border-inline-end:0;}", "div.jsx-2116647326>button:last-child{border-start-start-radius:0;border-end-start-radius:0;}"]));
  }
}
_defineProperty(SplitButton, "defaultProps", {
  dataTest: 'dhis2-uicore-splitbutton'
});
SplitButton.propTypes = {
  children: PropTypes.string,
  className: PropTypes.string,
  /** Component to render when the dropdown is opened */
  component: PropTypes.element,
  dataTest: PropTypes.string,
  /**
   * Applies 'destructive' button appearance, implying a dangerous action.
   */
  destructive: PropTypes.bool,
  /** Disables the button and makes it uninteractive */
  disabled: PropTypes.bool,
  /** An icon to add inside the button */
  icon: PropTypes.element,
  /** Grants the button the initial focus */
  initialFocus: PropTypes.bool,
  /** Changes button size. Mutually exclusive with `small` prop */
  large: sharedPropTypes.sizePropType,
  name: PropTypes.string,
  /**
   * Applies 'primary' button appearance, implying the most important action.
   */
  primary: PropTypes.bool,
  /**
   * Applies 'secondary' button appearance.
   */
  secondary: PropTypes.bool,
  /** Changes button size. Mutually exclusive with `large` prop */
  small: sharedPropTypes.sizePropType,
  tabIndex: PropTypes.string,
  /** Type of button. Applied to html `button` element */
  type: PropTypes.oneOf(['submit', 'reset', 'button']),
  /** Value associated with the button. Passed in object to onClick handler */
  value: PropTypes.string,
  onClick: PropTypes.func
};
export { SplitButton };
