// @dhis2-ui/table/build/cjs/table/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function () {
    return _table.Table;
  }
});
Object.defineProperty(exports, "TableBody", {
  enumerable: true,
  get: function () {
    return _tableBody.TableBody;
  }
});
Object.defineProperty(exports, "TableCell", {
  enumerable: true,
  get: function () {
    return _tableCell.TableCell;
  }
});
Object.defineProperty(exports, "TableCellHead", {
  enumerable: true,
  get: function () {
    return _tableCellHead.TableCellHead;
  }
});
Object.defineProperty(exports, "TableFoot", {
  enumerable: true,
  get: function () {
    return _tableFoot.TableFoot;
  }
});
Object.defineProperty(exports, "TableHead", {
  enumerable: true,
  get: function () {
    return _tableHead.TableHead;
  }
});
Object.defineProperty(exports, "TableRow", {
  enumerable: true,
  get: function () {
    return _tableRow.TableRow;
  }
});
Object.defineProperty(exports, "TableRowHead", {
  enumerable: true,
  get: function () {
    return _tableRowHead.TableRowHead;
  }
});
var _table = require("./table.js");
var _tableFoot = require("./table-foot.js");
var _tableBody = require("./table-body.js");
var _tableHead = require("./table-head.js");
var _tableCell = require("./table-cell.js");
var _tableCellHead = require("./table-cell-head.js");
var _tableRow = require("./table-row.js");
var _tableRowHead = require("./table-row-head.js");
