// @dhis2-ui/table/build/cjs/table/table-cell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableCell = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const tableCellStyles = ["td.jsx-2229340108{border-bottom:1px solid #e8edf2;font-size:14px;line-height:18px;padding:13px 12px;height:45px;}", "td.jsx-2229340108:dir(rtl){text-align:right;}", ".dense.jsx-2229340108{padding:9px 12px;height:36px;}"];
tableCellStyles.__hash = "2229340108";
const TableCell = _ref => {
  let {
    role,
    className,
    children,
    colSpan,
    rowSpan,
    dense,
    dataTest = 'dhis2-uicore-tablecell'
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("td", {
    colSpan: colSpan,
    rowSpan: rowSpan,
    "data-test": dataTest,
    role: role,
    className: `jsx-${tableCellStyles.__hash}` + " " + ((0, _classnames.default)({
      dense
    }, className) || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: tableCellStyles.__hash
  }, tableCellStyles));
};
exports.TableCell = TableCell;
TableCell.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  colSpan: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  /** Usees less padding and height for information-dense layouts */
  dense: _propTypes.default.bool,
  role: _propTypes.default.string,
  rowSpan: _propTypes.default.string
};
