// @dhis2-ui/table/build/cjs/table/table-row-head.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableRowHead = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _tableRow = require("./table-row.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const TableRowHead = _ref => {
  let {
    role,
    children,
    className,
    dataTest = 'dhis2-uicore-tablerowhead',
    suppressZebraStriping
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, {
    className: className,
    dataTest: dataTest,
    suppressZebraStriping: suppressZebraStriping,
    role: role
  }, children);
};
exports.TableRowHead = TableRowHead;
TableRowHead.propTypes = {
  /** Should be `<TableCellHead>` components */
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  role: _propTypes.default.string,
  /** Disables the default row striping for this row */
  suppressZebraStriping: _propTypes.default.bool
};
