// @dhis2-ui/table/build/cjs/table/table.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Table = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _tableContext = require("./table-context.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const tableStyles = ["table.jsx-2430604489{border:1px solid #e8edf2;background-color:#ffffff;min-width:100%;text-align:left;border-collapse:collapse;vertical-align:top;}"];
tableStyles.__hash = "2430604489";
const Table = _ref => {
  let {
    role,
    children,
    className,
    dataTest = 'dhis2-uicore-table',
    suppressZebraStriping
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_tableContext.Provider, {
    value: {
      suppressZebraStriping
    }
  }, /*#__PURE__*/_react.default.createElement("table", {
    "data-test": dataTest,
    role: role,
    className: `jsx-${tableStyles.__hash}` + " " + (className || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: tableStyles.__hash
  }, tableStyles)));
};
exports.Table = Table;
Table.propTypes = {
  /** Should be `<TableHead>`, `<TableBody>`, and `<TableFoot>` components */
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  role: _propTypes.default.string,
  /** Remove the default striping on alternating rows */
  suppressZebraStriping: _propTypes.default.bool
};
