// @dhis2/ui-forms/build/cjs/FileInputFieldFF/FileInputFieldFF.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithValues = exports.Required = exports.PreventPlaceholder = exports.Multifile = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
var _formDecorator = require("../formDecorator.js");
var _propTypes = require("../shared/propTypes.js");
var _index = require("../validators/index.js");
var _FileInputFieldFF = require("./FileInputFieldFF.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
The \`FileInputFieldFF\` is a wrapper around a \`FileInputField\` that enables it to work with Final Form, the preferred library in DHIS 2 apps for form validation and utilities.

#### Final Form

See how to use Final Form at [Final Form - Getting Started](https://final-form.org/docs/react-final-form/getting-started).

Inside a Final Form \`<Form>\` component, these 'FF' UI components are intended to be used in the \`component\` prop of the [Final Form \`<Field>\` components](https://final-form.org/docs/react-final-form/api/Field) where they will receive some props from the Field, e.g. \`<Field component={FileInputFieldFF} />\`.  See the code samples below for examples.

#### Props

The props shown in the table below are generally provided to the \`FileInputFieldFF\` wrapper by the Final Form \`Field\`.

Note that any props beyond the API of the \`Field\` component will be spread to the \`FileInputFieldFF\`, which passes any extra props to the underlying \`FileInputField\` using \`{...rest}\`.

Therefore, to add any props to the \`FileInputFieldFF\` or \`FileInputField\`, add those props to the parent Final Form \`Field\` component.

Also see \`FileInput\` and \`FileInputField\` for notes about props and implementation.

\`\`\`js
import { FileInputFieldFF } from '@dhis2/ui'
\`\`\`

Press **Submit** to see the form values logged to the console.
`;
const files = [new File([], 'file1.txt'), new File([], 'file2.txt')];
var _default = exports.default = {
  title: 'File Input Field (Final Form)',
  component: _FileInputFieldFF.FileInputFieldFF,
  decorators: [_formDecorator.formDecorator],
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    input: {
      ..._propTypes.inputArgType
    },
    meta: {
      ..._propTypes.metaArgType
    }
  }
};
const Default = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _FileInputFieldFF.FileInputFieldFF,
  name: "upload",
  label: "This is a file upload"
});
exports.Default = Default;
const Required = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _FileInputFieldFF.FileInputFieldFF,
  name: "upload",
  label: "This is a file upload",
  required: true,
  validate: _index.hasValue
});
exports.Required = Required;
const Multifile = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _FileInputFieldFF.FileInputFieldFF,
  name: "upload",
  label: "This is a file upload",
  multifile: true
});
exports.Multifile = Multifile;
const WithValues = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _FileInputFieldFF.FileInputFieldFF,
  name: "upload",
  label: "This is a file upload",
  required: true,
  multifile: true,
  initialValue: files,
  validate: _index.hasValue
});
exports.WithValues = WithValues;
WithValues.storyName = 'With values';
const PreventPlaceholder = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _FileInputFieldFF.FileInputFieldFF,
  name: "upload",
  label: "This is a file upload",
  placeholder: ""
});
exports.PreventPlaceholder = PreventPlaceholder;
PreventPlaceholder.storyName = 'Prevent placeholder';
