// @dhis2-ui/calendar/build/cjs/calendar/calendar-table.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CalendarTable = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _calendarTableCell = require("./calendar-table-cell.js");
var _calendarTableDaysHeader = require("./calendar-table-days-header.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const CalendarTable = _ref => {
  let {
    weekDayLabels,
    calendarWeekDays,
    width,
    cellSize,
    selectedDate
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["452536960", [_uiConstants.spacers.dp4, _uiConstants.spacers.dp4]]]) + " " + "calendar-table-wrapper"
  }, /*#__PURE__*/_react.default.createElement("table", {
    className: _style.default.dynamic([["452536960", [_uiConstants.spacers.dp4, _uiConstants.spacers.dp4]]]) + " " + "calendar-table"
  }, /*#__PURE__*/_react.default.createElement(_calendarTableDaysHeader.CalendarTableDaysHeader, {
    weekDayLabels: weekDayLabels
  }), /*#__PURE__*/_react.default.createElement("tbody", {
    className: _style.default.dynamic([["452536960", [_uiConstants.spacers.dp4, _uiConstants.spacers.dp4]]])
  }, calendarWeekDays.map((week, weekIndex) => /*#__PURE__*/_react.default.createElement("tr", {
    key: `week-${weekIndex + 1}`,
    className: _style.default.dynamic([["452536960", [_uiConstants.spacers.dp4, _uiConstants.spacers.dp4]]])
  }, week.map(day => /*#__PURE__*/_react.default.createElement(_calendarTableCell.CalendarTableCell, {
    selectedDate: selectedDate,
    day: day,
    key: day === null || day === void 0 ? void 0 : day.calendarDate,
    cellSize: cellSize,
    width: width
  })))))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "452536960",
    dynamic: [_uiConstants.spacers.dp4, _uiConstants.spacers.dp4]
  }, [`.calendar-table.__jsx-style-dynamic-selector{border:none;border-collapse:collapse;width:100%;margin-block:${_uiConstants.spacers.dp4};}`, ".calendar-table.__jsx-style-dynamic-selector tr.__jsx-style-dynamic-selector,.calendar-table.__jsx-style-dynamic-selector td.__jsx-style-dynamic-selector{border:none;}", `.calendar-table-wrapper.__jsx-style-dynamic-selector{padding-inline:${_uiConstants.spacers.dp4};}`]));
};
exports.CalendarTable = CalendarTable;
CalendarTable.propTypes = {
  calendarWeekDays: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.shape({
    calendarDate: _propTypes.default.string,
    isInCurrentMonth: _propTypes.default.bool,
    isSelected: _propTypes.default.bool,
    isToday: _propTypes.default.bool,
    label: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),
    zdt: _propTypes.default.object,
    onClick: _propTypes.default.func
  }).isRequired).isRequired).isRequired,
  cellSize: _propTypes.default.string,
  selectedDate: _propTypes.default.string,
  weekDayLabels: _propTypes.default.arrayOf(_propTypes.default.string),
  width: _propTypes.default.string
};
