// @dhis2-ui/calendar/build/cjs/calendar/calendar-table-cell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CalendarTableCell = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const CalendarTableCell = _ref => {
  let {
    day,
    cellSize,
    selectedDate
  } = _ref;
  const dayHoverBackgroundColor = _uiConstants.colors.grey200;
  const selectedDayBackgroundColor = _uiConstants.colors.teal700;
  return /*#__PURE__*/_react.default.createElement("td", {
    "data-test": day === null || day === void 0 ? void 0 : day.calendarDate,
    onClick: day.onClick,
    className: _style.default.dynamic([["2052411850", [cellSize, cellSize, cellSize, cellSize, _uiConstants.colors.grey900, dayHoverBackgroundColor, _uiConstants.colors.grey300, selectedDayBackgroundColor, _uiConstants.colors.teal600, _uiConstants.colors.teal200, _uiConstants.colors.grey600]]])
  }, /*#__PURE__*/_react.default.createElement("button", {
    name: "day",
    className: _style.default.dynamic([["2052411850", [cellSize, cellSize, cellSize, cellSize, _uiConstants.colors.grey900, dayHoverBackgroundColor, _uiConstants.colors.grey300, selectedDayBackgroundColor, _uiConstants.colors.teal600, _uiConstants.colors.teal200, _uiConstants.colors.grey600]]]) + " " + ((0, _classnames.default)('day', {
      isSelected: selectedDate === (day === null || day === void 0 ? void 0 : day.calendarDate),
      isToday: day.isToday,
      otherMonth: !day.isInCurrentMonth
    }) || "")
  }, day.label), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2052411850",
    dynamic: [cellSize, cellSize, cellSize, cellSize, _uiConstants.colors.grey900, dayHoverBackgroundColor, _uiConstants.colors.grey300, selectedDayBackgroundColor, _uiConstants.colors.teal600, _uiConstants.colors.teal200, _uiConstants.colors.grey600]
  }, [`td.__jsx-style-dynamic-selector{width:${cellSize};height:100%;height:${cellSize};text-align:center;border:2px solid transparent;padding:0;}`, "td.__jsx-style-dynamic-selector span.__jsx-style-dynamic-selector{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;padding:2px 2px;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;border:2px solid transparent;min-width:16px;}", `button.__jsx-style-dynamic-selector{border:0;overflow:hidden;width:${cellSize};height:${cellSize};border:2px solid transparent;border-radius:3px;background:none;color:${_uiConstants.colors.grey900};}`, `button.__jsx-style-dynamic-selector:hover{background-color:${dayHoverBackgroundColor};-webkit-text-decoration:underline;text-decoration:underline;cursor:pointer;}`, `button.__jsx-style-dynamic-selector:active{background-color:${_uiConstants.colors.grey300};}`, `button.isSelected.__jsx-style-dynamic-selector,button.otherMonth.isSelected.__jsx-style-dynamic-selector{background-color:${selectedDayBackgroundColor};color:white;}`, "button.isToday.__jsx-style-dynamic-selector{position:relative;}", `button.isToday.__jsx-style-dynamic-selector::after{content:'';position:absolute;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);height:4px;width:4px;bottom:2px;left:50%;border-radius:100%;background-color:${_uiConstants.colors.teal600};}`, `button.isSelected.isToday.__jsx-style-dynamic-selector::after{background-color:${_uiConstants.colors.teal200};}`, `button.otherMonth.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey600};}`]));
};
exports.CalendarTableCell = CalendarTableCell;
CalendarTableCell.propTypes = {
  cellSize: _propTypes.default.string,
  day: _propTypes.default.shape({
    calendarDate: _propTypes.default.string,
    isInCurrentMonth: _propTypes.default.bool,
    isSelected: _propTypes.default.bool,
    isToday: _propTypes.default.bool,
    label: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),
    onClick: _propTypes.default.func
  }),
  selectedDate: _propTypes.default.string
};
