// @dhis2-ui/calendar/build/cjs/calendar/calendar-table-days-header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CalendarTableDaysHeader = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const CalendarTableDaysHeader = _ref => {
  let {
    weekDayLabels
  } = _ref;
  const dayNamesColor = _uiConstants.colors.grey700;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("thead", {
    className: _style.default.dynamic([["1577512909", [dayNamesColor]]])
  }, /*#__PURE__*/_react.default.createElement("tr", {
    className: _style.default.dynamic([["1577512909", [dayNamesColor]]])
  }, weekDayLabels.map((label, i) => /*#__PURE__*/_react.default.createElement("th", {
    scope: "col",
    key: `weekday-${i}`,
    className: _style.default.dynamic([["1577512909", [dayNamesColor]]])
  }, label)))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1577512909",
    dynamic: [dayNamesColor]
  }, [`th.__jsx-style-dynamic-selector{color:${dayNamesColor};font-weight:300;font-style:normal;padding:8px 8px;background:none;font-size:0.85em;border:none;cursor:default;}`, "tr.__jsx-style-dynamic-selector{border:none;}"]));
};
exports.CalendarTableDaysHeader = CalendarTableDaysHeader;
CalendarTableDaysHeader.propTypes = {
  weekDayLabels: _propTypes.default.arrayOf(_propTypes.default.string)
};
