// @dhis2-ui/table/build/cjs/stacked-table/stacked-table-cell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StackedTableCell = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _contentWithTitle = require("./content-with-title.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const defaultHeaderLabels = [];
const StackedTableCell = _ref => {
  let {
    children,
    className,
    colSpan,
    column,
    dataTest = 'dhis2-uicore-stackedtablecell',
    headerLabels = defaultHeaderLabels,
    hideTitle,
    rowSpan,
    title
  } = _ref;
  const cellTitle = title || headerLabels[column] || '';
  const realTitle = hideTitle ? '' : cellTitle;
  return /*#__PURE__*/_react.default.createElement("td", {
    colSpan: colSpan,
    rowSpan: rowSpan,
    "data-test": dataTest,
    className: _style.default.dynamic([["1923758240", [_uiConstants.colors.grey300]]]) + " " + (className || "")
  }, /*#__PURE__*/_react.default.createElement(_contentWithTitle.ContentWithTitle, {
    title: realTitle
  }, children), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1923758240",
    dynamic: [_uiConstants.colors.grey300]
  }, [`td.__jsx-style-dynamic-selector{border-bottom:1px solid ${_uiConstants.colors.grey300};padding:0 12px;font-size:14px;width:100%;display:block;}`, "td.__jsx-style-dynamic-selector:dir(rtl){text-align:right;}", "td.__jsx-style-dynamic-selector:last-child{border-bottom:0;}"]));
};
exports.StackedTableCell = StackedTableCell;
StackedTableCell.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  colSpan: _propTypes.default.string,
  column: _propTypes.default.number,
  dataTest: _propTypes.default.string,
  headerLabels: _propTypes.default.arrayOf(_propTypes.default.string),
  hideTitle: _propTypes.default.bool,
  rowSpan: _propTypes.default.string,
  title: _propTypes.default.string
};
