// @dhis2-ui/table/build/cjs/stacked-table/table.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Table = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Table = _ref => {
  let {
    children,
    className,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("table", {
    "data-test": dataTest,
    className: _style.default.dynamic([["4278890465", [_uiConstants.colors.white, _uiConstants.colors.grey900]]]) + " " + (className || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4278890465",
    dynamic: [_uiConstants.colors.white, _uiConstants.colors.grey900]
  }, [`table.__jsx-style-dynamic-selector{display:block;border:0;background-color:${_uiConstants.colors.white};min-width:100%;text-align:left;border-collapse:collapse;vertical-align:top;color:${_uiConstants.colors.grey900};}`]));
};
exports.Table = Table;
Table.propTypes = {
  children: _propTypes.default.node.isRequired,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
