// @dhis2/ui-forms/build/cjs/InputFieldFF/InputFieldFF.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Required = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
var _formDecorator = require("../formDecorator.js");
var _index = require("../validators/index.js");
var _InputFieldFF = require("./InputFieldFF.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'Testing:InputFieldFF',
  decorators: [_formDecorator.formDecorator]
};
const Default = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _InputFieldFF.InputFieldFF,
  name: "agree",
  label: "Do you agree?"
});
exports.Default = Default;
const Required = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  name: "agree",
  component: _InputFieldFF.InputFieldFF,
  required: true,
  validate: _index.hasValue,
  label: "Do you agree?"
});
exports.Required = Required;
