// @dhis2-ui/intersection-detector/build/cjs/intersection-detector.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.OutOfView = exports.InView = void 0;
var _react = _interopRequireWildcard(require("react"));
var _intersectionDetector = require("./intersection-detector.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
window.onChange = window.Cypress ? window.Cypress.cy.stub() : () => null;
var _default = exports.default = {
  title: 'IntersectionDetector'
};
const OutOfView = () => {
  const rootRef = (0, _react.useRef)();
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: rootRef,
    "data-test": "scroll-container",
    style: {
      width: 200,
      height: 300,
      overflow: 'auto'
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    // spacer to push indicator out of the intersecting area
    style: {
      width: 200,
      height: 310,
      overflow: 'auto',
      background: 'blue'
    }
  }), /*#__PURE__*/_react.default.createElement(_intersectionDetector.IntersectionDetector, {
    rootRef: rootRef,
    onChange: window.onChange
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      background: 'red',
      height: 200
    }
  })));
};
exports.OutOfView = OutOfView;
const InView = () => {
  const rootRef = (0, _react.useRef)();
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: rootRef,
    "data-test": "scroll-container",
    style: {
      width: 200,
      height: 300,
      overflow: 'auto'
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      background: 'red',
      height: 200
    }
  }, /*#__PURE__*/_react.default.createElement(_intersectionDetector.IntersectionDetector, {
    rootRef: rootRef,
    onChange: window.onChange
  })), /*#__PURE__*/_react.default.createElement("div", {
    // spacer to push indicator out of the intersecting area
    style: {
      width: 200,
      height: 310,
      overflow: 'auto',
      background: 'blue'
    }
  }));
};
exports.InView = InView;
