// @dhis2-ui/header-bar/build/cjs/debug-info/debug-info-menu-item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DebugInfoMenuItem = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _menu = require("@dhis2-ui/menu");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = _interopRequireDefault(require("../locales/index.js"));
var _useDebugInfo = require("./use-debug-info.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const DebugInfoMenuItem = _ref => {
  let {
    hideProfileMenu,
    showDebugInfoModal
  } = _ref;
  const debugInfo = (0, _useDebugInfo.useDebugInfo)();
  const openDebugModal = () => {
    hideProfileMenu();
    showDebugInfoModal();
  };
  const debugInfoLabel = /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["3534786758", [_uiConstants.colors.grey700]]]) + " " + "root"
  }, /*#__PURE__*/_react.default.createElement("div", {
    "data-test": "dhis2-ui-headerbar-instanceinfo",
    className: _style.default.dynamic([["3534786758", [_uiConstants.colors.grey700]]]) + " " + "instance-info version"
  }, debugInfo.dhis2_version ? _index.default.t('DHIS2 {{dhis2Version}}', {
    dhis2Version: debugInfo.dhis2_version
  }) : _index.default.t('DHIS2 version unknown')), /*#__PURE__*/_react.default.createElement("div", {
    "data-test": "dhis2-ui-headerbar-appinfo",
    className: _style.default.dynamic([["3534786758", [_uiConstants.colors.grey700]]]) + " " + "version"
  }, debugInfo.app_name ? debugInfo.app_version ? `${debugInfo.app_name} ${debugInfo.app_version}` : _index.default.t('{{appName}} version unknown', {
    appName: debugInfo.app_name
  }) : debugInfo.app_version ? _index.default.t('App {{appVersion}}', {
    appVersion: debugInfo.app_version
  }) : _index.default.t('App version unknown')), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3534786758",
    dynamic: [_uiConstants.colors.grey700]
  }, [`.root.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey700};font-style:italic;font-size:14px;line-height:17px;}`, ".instance-info.__jsx-style-dynamic-selector{margin-bottom:4px;}", ".version.__jsx-style-dynamic-selector{white-space:no-wrap;}"]));
  return /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    dense: true,
    onClick: openDebugModal,
    label: debugInfoLabel,
    dataTest: "dhis2-ui-headerbar-debuginfo"
  });
};
exports.DebugInfoMenuItem = DebugInfoMenuItem;
DebugInfoMenuItem.propTypes = {
  hideProfileMenu: _propTypes.default.func.isRequired,
  showDebugInfoModal: _propTypes.default.func.isRequired
};
