// @dhis2-ui/select/build/cjs/multi-select/filterable-menu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FilterableMenu = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = require("../select/index.js");
var _menu = require("./menu.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const FilterableMenu = _ref => {
  let {
    dataTest,
    options,
    onChange,
    selected,
    empty,
    handleClose,
    handleFocusInput,
    placeholder,
    noMatchText
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_index.FilterableMenu, {
    dataTest: dataTest,
    options: options,
    onChange: onChange,
    selected: selected,
    empty: empty,
    handleClose: handleClose,
    handleFocusInput: handleFocusInput,
    placeholder: placeholder,
    noMatchText: noMatchText,
    Menu: _menu.Menu
  });
};
exports.FilterableMenu = FilterableMenu;
FilterableMenu.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  noMatchText: _propTypes.default.string.isRequired,
  empty: _propTypes.default.node,
  handleClose: _propTypes.default.func,
  handleFocusInput: _propTypes.default.func,
  options: _propTypes.default.node,
  placeholder: _propTypes.default.string,
  selected: _propTypes.default.arrayOf(_propTypes.default.string),
  onChange: _propTypes.default.func
};
