// @dhis2-ui/select/build/cjs/multi-select/menu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Menu = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = require("../select/index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const onDisabledClick = (_, e) => {
  e.stopPropagation();
  e.preventDefault();
};
const createHandler = _ref => {
  let {
    isActive,
    onChange,
    selected,
    value
  } = _ref;
  return (_, e) => {
    e.stopPropagation();

    // If the option is currently selected remove it from the array of selected options
    if (isActive) {
      const filtered = (0, _index.removeOption)(value, selected);
      const data = {
        selected: filtered
      };
      return onChange(data, e);
    }

    // Otherwise, add it to selected
    const data = {
      selected: selected.concat([value])
    };
    return onChange(data, e);
  };
};
const Menu = _ref2 => {
  let {
    options,
    onChange,
    selected,
    empty = '',
    dataTest
  } = _ref2;
  const renderedOptions = (0, _index.filterIgnored)(options);
  if (_react.default.Children.count(renderedOptions) === 0) {
    // If it's a string, supply it to our <Empty> component so it looks better
    if (typeof empty === 'string') {
      return /*#__PURE__*/_react.default.createElement(_index.Empty, {
        message: empty,
        dataTest: `${dataTest}-empty`
      });
    }

    // Otherwise just render the supplied markup
    return empty;
  }
  const children = _react.default.Children.map(options, child => {
    const isValidOption = (0, _index.checkIfValidOption)(child);

    // Return early if the child isn't an option, to prevent attaching handlers etc.
    if (!isValidOption) {
      return child;
    }
    const {
      value,
      label,
      disabled: isDisabled
    } = child.props;

    // Active means the option is currently selected
    const isActive = !!(0, _index.findOption)(value, selected);

    // Create the appropriate click handler for the option
    const onClick = isDisabled ? onDisabledClick : createHandler({
      isActive,
      onChange,
      selected,
      value,
      label
    });
    return /*#__PURE__*/_react.default.cloneElement(child, {
      ...child.props,
      onClick,
      active: isActive
    });
  });
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, children);
};
exports.Menu = Menu;
Menu.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  empty: _propTypes.default.node,
  options: _propTypes.default.node,
  selected: _propTypes.default.arrayOf(_propTypes.default.string),
  onChange: _propTypes.default.func
};
