// @dhis2-ui/node/build/es/node.e2e.stories.js
import React from 'react';
import { Node } from './node.js';
window.onClose = window.Cypress && window.Cypress.cy.stub();
window.onOpen = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'Node'
};
export const OpenWithOnClose = () => /*#__PURE__*/React.createElement(Node, {
  open: true,
  onClose: window.onClose,
  component: /*#__PURE__*/React.createElement("div", null, "Component")
}, "Children");
export const ClosedWithOnOpen = () => /*#__PURE__*/React.createElement(Node, {
  onOpen: window.onOpen,
  component: /*#__PURE__*/React.createElement("div", null, "Component")
}, "Children");
export const ClosedWithChildren = () => /*#__PURE__*/React.createElement(Node, {
  component: /*#__PURE__*/React.createElement("div", null, "Component")
}, "I am a child");
export const OpenWithChildren = () => /*#__PURE__*/React.createElement(Node, {
  open: true,
  component: /*#__PURE__*/React.createElement("div", null, "Component")
}, "I am a child");
export const WithComponent = () => /*#__PURE__*/React.createElement(Node, {
  component: /*#__PURE__*/React.createElement("div", null, "I am a component")
}, "Children");
export const WithIcon = () => /*#__PURE__*/React.createElement(Node, {
  component: /*#__PURE__*/React.createElement("div", null, "Component"),
  icon: /*#__PURE__*/React.createElement("div", null, "Icon")
}, "Children");
