// @dhis2-ui/transfer/build/cjs/__e2e__/common/stateful-decorator.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.statefulDecorator = void 0;
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/* eslint-disable react/display-name */

const WithState = _ref => {
  let {
    fn,
    initialState,
    initialSearchTerm,
    controlFilter
  } = _ref;
  const [selected, setSelected] = (0, _react.useState)(initialState);
  const [searchTerm, setSearchTerm] = (0, _react.useState)(initialSearchTerm);
  return fn({
    selected,
    searchTerm: controlFilter ? searchTerm : undefined,
    onChange: payload => setSelected(payload.selected),
    onFilterChange: controlFilter ? _ref2 => {
      let {
        value
      } = _ref2;
      return setSearchTerm(value);
    } : undefined
  });
};
const statefulDecorator = function () {
  let {
    initialState = [],
    controlFilter = false,
    initialSearchTerm = ''
  } = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return fn => /*#__PURE__*/_react.default.createElement(WithState, {
    initialState: initialState,
    initialSearchTerm: initialSearchTerm,
    controlFilter: controlFilter,
    fn: fn
  });
};
exports.statefulDecorator = statefulDecorator;
