// @dhis2/ui-forms/build/cjs/shared/propTypes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.toggleGroupOptionsProp = exports.metaPropType = exports.metaArgType = exports.inputPropType = exports.inputArgType = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const toggleGroupOptionsProp = exports.toggleGroupOptionsProp = _propTypes.default.arrayOf(_propTypes.default.shape({
  label: _propTypes.default.string.isRequired,
  value: _propTypes.default.string.isRequired
}));
const inputPropType = exports.inputPropType = _propTypes.default.shape({
  name: _propTypes.default.string.isRequired,
  onChange: _propTypes.default.func.isRequired,
  value: _propTypes.default.any,
  onBlur: _propTypes.default.func,
  onFocus: _propTypes.default.func
});
const inputArgType = exports.inputArgType = {
  table: {
    type: {
      summary: '{ name: string (required), onChange: func (required), value: any, onBlur: func, onFocus: func }'
    }
  }
};
const metaPropType = exports.metaPropType = _propTypes.default.shape({
  error: _propTypes.default.string,
  invalid: _propTypes.default.bool,
  touched: _propTypes.default.bool,
  valid: _propTypes.default.bool,
  validating: _propTypes.default.bool
});
const metaArgType = exports.metaArgType = {
  table: {
    type: {
      summary: '{ error: string, invalid: bool, touched: bool, valid: bool, validating: bool }'
    }
  }
};
