// @dhis2-ui/select/build/es/multi-select-field/multi-select-field.e2e.stories.js
import React from 'react';
import { MultiSelectOption } from '../index.js';
import { MultiSelectField } from './index.js';
export default {
  title: 'MultiSelectField'
};
export const WithLabel = () => /*#__PURE__*/React.createElement(MultiSelectField, {
  label: "The label"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "three"
}));
export const WithHelpText = () => /*#__PURE__*/React.createElement(MultiSelectField, {
  helpText: "The help text"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "three"
}));
export const WithValidationText = () => /*#__PURE__*/React.createElement(MultiSelectField, {
  validationText: "The validation text"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "three"
}));
export const WithLabelAndRequiredStatus = () => /*#__PURE__*/React.createElement(MultiSelectField, {
  label: "The label",
  required: true
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "three"
}));
export const WithClearableAndSelectedOption = () => /*#__PURE__*/React.createElement(MultiSelectField, {
  selected: ['1'],
  clearable: true
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  key: "1",
  value: "1",
  label: "One"
}));
export const WithFilterable = () => /*#__PURE__*/React.createElement(MultiSelectField, {
  filterable: true
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  key: "1",
  value: "1",
  label: "One"
}));
export const WithLoading = () => /*#__PURE__*/React.createElement(MultiSelectField, {
  loading: true
});
export const WithoutOptions = () => /*#__PURE__*/React.createElement(MultiSelectField, null);
