// @dhis2-ui/card/build/cjs/card.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _box = require("@dhis2-ui/box");
var _react = _interopRequireDefault(require("react"));
var _card = require("./card.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const subtitle = `
A card is a container element for grouping together
and separating blocks of content.
`;
const description = `
Use a card where there is content that can be grouped together.
Cards are most often useful when this grouped content may be repeated,
for example with items on a dashboard, or different sections of patient
information displayed in a profile.

Note that it requires a parent, like [Box](../?path=/docs/layout-box--default), to define its size.

\`\`\`js
import { Card } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Card',
  component: _card.Card,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    children: {
      control: {
        type: 'text'
      }
    }
  }
};
const Default = args => /*#__PURE__*/_react.default.createElement(_box.Box, {
  width: "200px",
  height: "75px"
}, /*#__PURE__*/_react.default.createElement(_card.Card, args));
exports.Default = Default;
Default.args = {
  children: 'Card content'
};
