// @dhis2-ui/checkbox/build/cjs/checkbox/checkbox.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnFocus = exports.WithOnChange = exports.WithLabel = exports.WithInitialFocusAndOnBlur = exports.WithInitialFocus = exports.NoIndeterminateProp = exports.IndeterminateProp = exports.DisabledWithOnClick = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onClick = window.Cypress && window.Cypress.cy.stub();
window.onChange = window.Cypress && window.Cypress.cy.stub();
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'Checkbox'
};
const WithOnChange = () => /*#__PURE__*/_react.default.createElement(_index.Checkbox, {
  name: "Ex",
  label: "Checkbox",
  value: "default",
  onChange: window.onChange
});
exports.WithOnChange = WithOnChange;
const WithInitialFocusAndOnBlur = () => /*#__PURE__*/_react.default.createElement(_index.Checkbox, {
  initialFocus: true,
  name: "Ex",
  label: "Checkbox",
  value: "default",
  onBlur: window.onBlur
});
exports.WithInitialFocusAndOnBlur = WithInitialFocusAndOnBlur;
const WithOnFocus = () => /*#__PURE__*/_react.default.createElement(_index.Checkbox, {
  name: "Ex",
  label: "Checkbox",
  value: "default",
  onFocus: window.onFocus
});
exports.WithOnFocus = WithOnFocus;
const DisabledWithOnClick = () => /*#__PURE__*/_react.default.createElement(_index.Checkbox, {
  name: "Ex",
  label: "Checkbox",
  value: "default",
  disabled: true,
  onClick: window.onClick
});
exports.DisabledWithOnClick = DisabledWithOnClick;
const WithLabel = () => /*#__PURE__*/_react.default.createElement(_index.Checkbox, {
  name: "Ex",
  label: "The label",
  value: "default"
});
exports.WithLabel = WithLabel;
const WithInitialFocus = () => /*#__PURE__*/_react.default.createElement(_index.Checkbox, {
  name: "Ex",
  label: "The label",
  value: "default",
  initialFocus: true
});
exports.WithInitialFocus = WithInitialFocus;
const IndeterminateProp = () => /*#__PURE__*/_react.default.createElement(_index.Checkbox, {
  name: "Ex",
  label: "The label",
  value: "default",
  indeterminate: true
});
exports.IndeterminateProp = IndeterminateProp;
const NoIndeterminateProp = () => /*#__PURE__*/_react.default.createElement(_index.Checkbox, {
  name: "Ex",
  label: "The label",
  value: "default"
});
exports.NoIndeterminateProp = NoIndeterminateProp;
