// @dhis2-ui/alert/build/cjs/alert-bar/alert-bar.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnHidden = exports.WithMessage = exports.PermanentWithActions = exports.Permanent = exports.HiddenProp = exports.DisabledIcon = exports.Default = exports.CustomIcon = exports.CustomDuration = void 0;
var _react = _interopRequireWildcard(require("react"));
var _index = require("./index.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
window.onHidden = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'AlertBar'
};
const Default = () => /*#__PURE__*/_react.default.createElement(_index.AlertBar, null, "Default");
exports.Default = Default;
const CustomDuration = () => /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  duration: 2000
}, "Custom duration");
exports.CustomDuration = CustomDuration;
const PermanentWithActions = () => /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  actions: [{
    label: 'Save',
    onClick: () => {}
  }, {
    label: 'Cancel',
    onClick: () => {}
  }]
}, "With Actions");
exports.PermanentWithActions = PermanentWithActions;
const DisabledIcon = () => /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  icon: false
}, "Message");
exports.DisabledIcon = DisabledIcon;
const CustomIcon = () => /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  icon: /*#__PURE__*/_react.default.createElement("span", null, "Custom icon")
}, "Message");
exports.CustomIcon = CustomIcon;
const WithMessage = () => /*#__PURE__*/_react.default.createElement(_index.AlertBar, null, "With a message");
exports.WithMessage = WithMessage;
const WithOnHidden = () => /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  onHidden: window.onHidden
}, "Message");
exports.WithOnHidden = WithOnHidden;
const Permanent = () => /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true
}, "Message");
exports.Permanent = Permanent;
const HiddenProp = () => {
  const [hidden, setHidden] = (0, _react.useState)(true);
  const toggleVisibility = () => setHidden(prevHidden => !prevHidden);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
    style: {
      display: 'block',
      position: 'fixed',
      bottom: 150,
      left: 10
    },
    onClick: toggleVisibility
  }, hidden ? 'Show' : 'Hide'), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
    permanent: true,
    hidden: hidden,
    onHidden: window.onHidden
  }, "Short text"));
};
exports.HiddenProp = HiddenProp;
