// @dhis2-ui/alert/build/cjs/alert-bar/dismiss.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Dismiss = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _uiIcons = require("@dhis2/ui-icons");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Dismiss = _ref => {
  let {
    onClick,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    onClick: onClick,
    "data-test": dataTest,
    className: _style.default.dynamic([["1892313381", [_uiConstants.spacers.dp16]]])
  }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconCross24, null), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1892313381",
    dynamic: [_uiConstants.spacers.dp16]
  }, [`div.__jsx-style-dynamic-selector{margin-inline-start:${_uiConstants.spacers.dp16};min-height:32px;min-width:32px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;border-radius:5px;}`, "div.__jsx-style-dynamic-selector:hover{cursor:pointer;background:rgba(0,0,0,0.15);}", "div.__jsx-style-dynamic-selector:active{background:rgba(0,0,0,0.25);}", "div.__jsx-style-dynamic-selector svg{width:18px;height:18px;}"]));
};
exports.Dismiss = Dismiss;
Dismiss.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  onClick: _propTypes.default.func.isRequired
};
