// @dhis2-ui/legend/build/es/legend.js
import _JSXStyle from "styled-jsx/style";
import { colors } from '@dhis2/ui-constants';
import { Required } from '@dhis2-ui/required';
import PropTypes from 'prop-types';
import React from 'react';
const Legend = _ref => {
  let {
    className,
    children,
    required,
    dataTest = 'dhis2-uicore-legend'
  } = _ref;
  return /*#__PURE__*/React.createElement("legend", {
    "data-test": dataTest,
    className: _JSXStyle.dynamic([["945530629", [colors.grey900]]]) + " " + (className || "")
  }, children, required && /*#__PURE__*/React.createElement(Required, {
    dataTest: `${dataTest}-required`
  }), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "945530629",
    dynamic: [colors.grey900]
  }, [`legend.__jsx-style-dynamic-selector{font-size:14px;line-height:16px;color:${colors.grey900};}`]));
};
Legend.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  /** Indicates the associated field set is required */
  required: PropTypes.bool
};
export { Legend };
