// @dhis2-ui/pagination/build/cjs/page-size-select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PageSizeSelect = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _select = require("@dhis2-ui/select");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
// TODO: i18n translate
const translate = (prop, interpolationObject) => {
  if (typeof prop === 'function') {
    return prop(interpolationObject);
  }
  return prop;
};
const PageSizeSelect = _ref => {
  let {
    dataTest,
    disabled,
    pageSizeSelectText,
    pageSize,
    pageSizes,
    onChange
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": `${dataTest}-pagesize`,
    className: _style.default.dynamic([["3539890457", [_uiConstants.spacers.dp12]]])
  }, /*#__PURE__*/_react.default.createElement(_select.SingleSelect, {
    dense: true,
    disabled: disabled,
    selected: pageSize.toString(),
    onChange: _ref2 => {
      let {
        selected
      } = _ref2;
      return onChange(parseInt(selected, 10));
    },
    className: "select",
    dataTest: `${dataTest}-pagesize-select`,
    prefix: translate(pageSizeSelectText)
  }, pageSizes.map(length => /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
    key: length,
    value: length,
    label: length
  }))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3539890457",
    dynamic: [_uiConstants.spacers.dp12]
  }, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;min-height:32px;margin-inline-end:${_uiConstants.spacers.dp12};-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}`]));
};
exports.PageSizeSelect = PageSizeSelect;
PageSizeSelect.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  pageSize: _propTypes.default.number.isRequired,
  pageSizeSelectText: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.func]).isRequired,
  pageSizes: _propTypes.default.arrayOf(_propTypes.default.string).isRequired,
  onChange: _propTypes.default.func.isRequired,
  disabled: _propTypes.default.bool
};
