// @dhis2-ui/pagination/build/cjs/page-controls.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PageControls = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _uiIcons = require("@dhis2/ui-icons");
var _button = require("@dhis2-ui/button");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const translate = (prop, interpolationObject) => {
  if (typeof prop === 'function') {
    return prop(interpolationObject);
  }
  return prop;
};
const PageControls = _ref => {
  let {
    dataTest,
    isNextDisabled,
    isPreviousDisabled,
    nextPageText,
    page,
    previousPageText,
    onClick
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": `${dataTest}-pagecontrols`,
    className: _style.default.dynamic([["3252926163", [_uiConstants.spacers.dp4]]])
  }, /*#__PURE__*/_react.default.createElement(_button.Button, {
    secondary: true,
    className: "button-previous",
    small: true,
    disabled: isPreviousDisabled,
    onClick: () => onClick(page - 1),
    dataTest: `${dataTest}-page-previous`
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["3252926163", [_uiConstants.spacers.dp4]]]) + " " + "navigator-icon"
  }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconChevronLeft16, null)), translate(previousPageText)), /*#__PURE__*/_react.default.createElement(_button.Button, {
    secondary: true,
    className: "button-next",
    small: true,
    disabled: isNextDisabled,
    onClick: () => onClick(page + 1),
    dataTest: `${dataTest}-page-next`
  }, translate(nextPageText), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["3252926163", [_uiConstants.spacers.dp4]]]) + " " + "navigator-icon"
  }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconChevronRight16, null))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3252926163",
    dynamic: [_uiConstants.spacers.dp4]
  }, ["div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", "div.__jsx-style-dynamic-selector .button-previous{height:32px;padding-inline-start:0;}", `div.__jsx-style-dynamic-selector .button-next{height:32px;padding-inline-end:0;margin-inline-start:${_uiConstants.spacers.dp4};}`, "div.__jsx-style-dynamic-selector .navigator-icon.__jsx-style-dynamic-selector:dir(rtl){-webkit-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}"]));
};
exports.PageControls = PageControls;
PageControls.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  nextPageText: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.func]).isRequired,
  page: _propTypes.default.number.isRequired,
  previousPageText: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.func]).isRequired,
  onClick: _propTypes.default.func.isRequired,
  isNextDisabled: _propTypes.default.bool,
  isPreviousDisabled: _propTypes.default.bool
};
