// @dhis2-ui/pagination/build/cjs/page-summary.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PageSummary = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const translate = (prop, interpolationObject) => {
  if (typeof prop === 'function') {
    return prop(interpolationObject);
  }
  return prop;
};
const PageSummary = _ref => {
  let {
    dataTest,
    firstItem,
    inactive,
    lastItem,
    page,
    pageCount,
    pageSummaryText,
    total
  } = _ref;
  const summary = translate(pageSummaryText, {
    firstItem,
    lastItem,
    page,
    pageCount,
    total
  });
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": `${dataTest}-summary`,
    className: _style.default.dynamic([["3973865131", [_uiConstants.spacers.dp12, _uiConstants.colors.grey700, _uiConstants.colors.grey500]]])
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: _style.default.dynamic([["3973865131", [_uiConstants.spacers.dp12, _uiConstants.colors.grey700, _uiConstants.colors.grey500]]]) + " " + ((0, _classnames.default)({
      inactive
    }) || "")
  }, summary), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3973865131",
    dynamic: [_uiConstants.spacers.dp12, _uiConstants.colors.grey700, _uiConstants.colors.grey500]
  }, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;min-height:32px;margin-inline-end:${_uiConstants.spacers.dp12};}`, `span.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey700};font-size:14px;}`, `span.inactive.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey500};}`]));
};
exports.PageSummary = PageSummary;
PageSummary.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  page: _propTypes.default.number.isRequired,
  pageSummaryText: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.func]).isRequired,
  firstItem: _propTypes.default.number,
  inactive: _propTypes.default.bool,
  lastItem: _propTypes.default.number,
  pageCount: _propTypes.default.number,
  total: _propTypes.default.number
};
