// @dhis2-ui/legend/build/cjs/legend.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Required = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _legend = require("./legend.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
Legends are to be used in a Field Set to describe the form fields.  They may indicate that the fields are required.

See the [Field Set](../?path=/docs/forms-field-set-field-set--usage-example-a-radio-button-group-with-error-status) for a usage example.

\`\`\`js
import { Legend } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Legend',
  component: _legend.Legend,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_legend.Legend, args, "I am wrapped in a legend which has some styling");
const Default = exports.Default = Template.bind({});
const Required = exports.Required = Template.bind({});
Required.args = {
  required: true
};
