// @dhis2/ui-forms/build/cjs/SingleSelectFieldFF/SingleSelectFieldFF.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Required = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
var _formDecorator = require("../formDecorator.js");
var _index = require("../validators/index.js");
var _SingleSelectFieldFF = require("./SingleSelectFieldFF.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const defaultOptions = [{
  value: 'initial',
  label: 'Initial'
}];
var _default = exports.default = {
  title: 'Testing:SingleSelectFieldFF',
  decorators: [_formDecorator.formDecorator],
  parameters: {
    options: {
      showPanel: false
    }
  }
};
const Required = (_, _ref) => {
  let {
    cypressProps
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
    required: true,
    name: "singleSelect",
    label: "Single select",
    component: _SingleSelectFieldFF.SingleSelectFieldFF,
    validate: _index.hasValue,
    options: cypressProps.options || defaultOptions
  });
};
exports.Required = Required;
