// @dhis2/ui-forms/build/es/SwitchFieldFF/SwitchFieldFF.e2e.stories.js
import React from 'react';
import { Field } from 'react-final-form';
import { formDecorator } from '../formDecorator.js';
import { hasValue } from '../validators/index.js';
import { SwitchFieldFF } from './SwitchFieldFF.js';

// https://github.com/final-form/react-final-form-arrays/issues/111
const initialValue = ['yes'];
export default {
  title: 'Testing:SwitchFieldFF',
  decorators: [formDecorator]
};
export const Unchecked = () => /*#__PURE__*/React.createElement(Field, {
  component: SwitchFieldFF,
  className: "switch",
  name: "switch",
  label: "Label text",
  validate: hasValue,
  required: true,
  type: "checkbox"
});
export const Checked = () => /*#__PURE__*/React.createElement(Field, {
  component: SwitchFieldFF,
  className: "switch",
  name: "switch",
  label: "Label text",
  initialValue: true,
  type: "checkbox"
});
export const UncheckedWithValue = () => /*#__PURE__*/React.createElement(Field, {
  component: SwitchFieldFF,
  className: "switch",
  name: "switch",
  label: "Label text",
  value: "yes",
  type: "checkbox"
});
export const CheckedWithValue = () => /*#__PURE__*/React.createElement(Field, {
  component: SwitchFieldFF,
  className: "switch",
  name: "switch",
  label: "Label text",
  value: "yes",
  initialValue: initialValue,
  type: "checkbox"
});
