// @dhis2-ui/selector-bar/build/es/selector-bar/__stories__/with-right-hand-side-content.js
import _JSXStyle from "styled-jsx/style";
import { Button } from '@dhis2-ui/button';
import React from 'react';
import { SelectorBar, SelectorBarItem } from '../../index.js';
import { OrgUnitSelect, MenuSelect, createDecoratorCustomDataProvider, createStatefulDecorator, decoratorCommonStyles, workflows } from './common.js';
const RHSContent = () => /*#__PURE__*/React.createElement("div", {
  className: "jsx-876392447" + " " + "rhs-content"
}, /*#__PURE__*/React.createElement(Button, {
  small: true
}, "Options"), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "876392447"
}, [".rhs-content.jsx-876392447{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;height:40px;padding:0 16px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}"]));
export const WithAdditionalContent = (_, _ref) => {
  let {
    workflow,
    setWorkflow,
    workflowOpen,
    setWorkflowOpen,
    orgUnit,
    setOrgUnit,
    orgUnitOpen,
    setOrgUnitOpen
  } = _ref;
  return /*#__PURE__*/React.createElement(SelectorBar, {
    disableClearSelections: !workflow || !orgUnit,
    onClearSelectionClick: () => {
      setWorkflow(null);
      setOrgUnit(null);
    },
    additionalContent: /*#__PURE__*/React.createElement(RHSContent, null)
  }, /*#__PURE__*/React.createElement(SelectorBarItem, {
    disabled: false,
    label: "Workflow",
    value: workflow === null || workflow === void 0 ? void 0 : workflow.label,
    noValueMessage: "Choose a workflow",
    open: workflowOpen,
    setOpen: setWorkflowOpen,
    onClearSelectionClick: () => {
      setWorkflow(null);
    }
  }, /*#__PURE__*/React.createElement(MenuSelect, {
    values: workflows,
    selected: workflow === null || workflow === void 0 ? void 0 : workflow.value,
    onChange: _ref2 => {
      let {
        selected
      } = _ref2;
      setWorkflow(workflows.find(currentWorkflow => selected === currentWorkflow.value));
      setWorkflowOpen(false);
    }
  })), /*#__PURE__*/React.createElement(SelectorBarItem, {
    disabled: !workflow,
    label: "Org unit",
    value: orgUnit === null || orgUnit === void 0 ? void 0 : orgUnit.displayName,
    noValueMessage: "Choose an organisation unit",
    open: orgUnitOpen,
    setOpen: setOrgUnitOpen,
    onClearSelectionClick: () => {
      setOrgUnit(null);
    }
  }, /*#__PURE__*/React.createElement(OrgUnitSelect, {
    selected: orgUnit ? [orgUnit.path] : [],
    onChange: (nextOrgUnit, evt) => {
      evt.stopPropagation();
      setOrgUnit(nextOrgUnit);
      setOrgUnitOpen(false);
    }
  })));
};
WithAdditionalContent.decorators = [decoratorCommonStyles, createStatefulDecorator(), createDecoratorCustomDataProvider()];
