// @dhis2-ui/cover/build/es/cover.prod.stories.js
import { Center } from '@dhis2-ui/center';
import { CircularLoader } from '@dhis2-ui/loader';
import React from 'react';
import { Cover } from './cover.js';
const description = `
Covers sibling components.  Useful for covering a component while it is loading.

\`\`\`js
import { Cover } from '@dhis2/ui'
\`\`\`
`;
export default {
  title: 'Component Cover',
  component: Cover,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Template = args => /*#__PURE__*/React.createElement("div", {
  style: {
    width: '400px',
    height: '200px',
    position: 'relative',
    border: '1px dotted grey'
  }
}, /*#__PURE__*/React.createElement(Cover, args), /*#__PURE__*/React.createElement("h1", null, "Text behind the cover"), /*#__PURE__*/React.createElement("p", null, "Lorem ipsum"));
export const Default = Template.bind({});
export const Translucent = Template.bind({});
Translucent.args = {
  translucent: true
};
export const WithClickHandler = Template.bind({});
WithClickHandler.args = {
  onClick: () => alert('Cover was clicked')
};
export const WithCenteredContentCircularLoader = Template.bind({});
WithCenteredContentCircularLoader.args = {
  translucent: true,
  children: /*#__PURE__*/React.createElement(Center, null, /*#__PURE__*/React.createElement(CircularLoader, null))
};
