// @dhis2-ui/cover/build/es/cover.e2e.stories.js
import _JSXStyle from "styled-jsx/style";
import React from 'react';
import { Cover } from './cover.js';
window.onButtonClick = window.Cypress && window.Cypress.cy.stub();
window.onCover = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'Cover',
  component: Cover,
  decorators: [story => /*#__PURE__*/React.createElement("div", {
    className: "jsx-1235544169"
  }, story(), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "1235544169"
  }, ["div.jsx-1235544169{width:400px;height:400px;position:relative;border:1px dotted grey;}"]))]
};
export const WithChildren = () => /*#__PURE__*/React.createElement(Cover, null, /*#__PURE__*/React.createElement("p", null, "I am a child"));
export const Blocking = () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("button", {
  "data-test": "button",
  onClick: window.onButtonClick
}, "Test"), /*#__PURE__*/React.createElement(Cover, null));
export const WithClickHandler = () => /*#__PURE__*/React.createElement(Cover, {
  onClick: window.onCover
}, /*#__PURE__*/React.createElement("button", {
  "data-test": "button",
  onClick: window.onButtonClick
}, "Test"));
