// @dhis2-ui/radio/build/es/radio.e2e.stories.js
import React from 'react';
import { Radio } from './radio.js';
window.onChange = window.Cypress && window.Cypress.cy.stub();
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'Radio'
};
export const WithOnChange = () => /*#__PURE__*/React.createElement(Radio, {
  name: "Ex",
  label: "Radio",
  value: "default",
  onChange: window.onChange
});
export const WithInitialFocusAndOnBlur = () => /*#__PURE__*/React.createElement(Radio, {
  name: "Ex",
  label: "Radio",
  value: "default",
  initialFocus: true,
  onBlur: window.onBlur
});
export const WithOnFocus = () => /*#__PURE__*/React.createElement(Radio, {
  initialFocus: true,
  name: "Ex",
  label: "Radio",
  value: "default",
  onFocus: window.onFocus
});
export const WithDisabled = () => /*#__PURE__*/React.createElement(Radio, {
  name: "Ex",
  label: "Radio",
  value: "default",
  disabled: true
});
export const WithLabel = () => /*#__PURE__*/React.createElement(Radio, {
  name: "Ex",
  label: "The label",
  value: "default"
});
export const WithInitialFocus = () => /*#__PURE__*/React.createElement(Radio, {
  name: "Ex",
  label: "The label",
  value: "default",
  initialFocus: true
});
