// @dhis2/ui-forms/build/cjs/MultiSelectFieldFF/MultiSelectFieldFF.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Required = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
var _formDecorator = require("../formDecorator.js");
var _index = require("../validators/index.js");
var _MultiSelectFieldFF = require("./MultiSelectFieldFF.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const defaultOptions = [{
  value: 'first',
  label: 'First'
}, {
  value: 'second',
  label: 'Second'
}];
var _default = exports.default = {
  title: 'MultiSelectFieldFF',
  component: _MultiSelectFieldFF.MultiSelectFieldFF,
  decorators: [_formDecorator.formDecorator]
};
const Required = (_, _ref) => {
  let {
    cypressProps
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
    required: true,
    name: "multiSelect",
    label: "Multi select",
    component: _MultiSelectFieldFF.MultiSelectFieldFF,
    validate: _index.hasValue,
    options: cypressProps.options || defaultOptions
  });
};
exports.Required = Required;
