// @dhis2/ui-forms/build/cjs/MultiSelectFieldFF/MultiSelectFieldFF.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MultiSelectFieldFF = void 0;
var _select = require("@dhis2-ui/select");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _helpers = require("../shared/helpers.js");
var _propTypes2 = require("../shared/propTypes.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const MultiSelectFieldFF = _ref => {
  let {
    error,
    input,
    loading,
    meta,
    onBlur,
    onFocus,
    options = [],
    showLoadingStatus,
    showValidStatus,
    valid,
    validationText,
    ...rest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_select.MultiSelectField, _extends({}, rest, {
    name: input.name,
    error: (0, _helpers.hasError)(meta, error),
    valid: (0, _helpers.isValid)(meta, valid, showValidStatus),
    loading: (0, _helpers.isLoading)(meta, loading, showLoadingStatus),
    validationText: (0, _helpers.getValidationText)(meta, validationText, error),
    onFocus: (0, _helpers.createFocusHandler)(input, onFocus),
    onChange: (0, _helpers.createSelectChangeHandler)(input),
    onBlur: (0, _helpers.createBlurHandler)(input, onBlur),
    selected: input.value || [] /* input.value is an empty string initially, so we're providing an empty array if falsey */
  }), options.map(option => /*#__PURE__*/_react.default.createElement(_select.MultiSelectOption, _extends({
    key: option.value
  }, option))));
};
exports.MultiSelectFieldFF = MultiSelectFieldFF;
MultiSelectFieldFF.propTypes = {
  /** `input` props provided by Final Form `Field` */
  input: _propTypes2.inputPropType.isRequired,
  /** `meta` props provided by Final Form `Field` */
  meta: _propTypes2.metaPropType.isRequired,
  error: _propTypes.default.bool,
  loading: _propTypes.default.bool,
  options: _propTypes.default.arrayOf(_propTypes.default.shape({
    label: _propTypes.default.string,
    value: _propTypes.default.string
  })),
  showLoadingStatus: _propTypes.default.bool,
  showValidStatus: _propTypes.default.bool,
  valid: _propTypes.default.bool,
  validationText: _propTypes.default.string,
  onBlur: _propTypes.default.func,
  onFocus: _propTypes.default.func
};
