// @dhis2/ui-forms/build/es/CheckboxFieldFF/CheckboxFieldFF.e2e.stories.js
import React from 'react';
import { Field } from 'react-final-form';
import { formDecorator } from '../formDecorator.js';
import { hasValue } from '../validators/index.js';
import { CheckboxFieldFF } from './CheckboxFieldFF.js';

// https://github.com/final-form/react-final-form-arrays/issues/111
const initialValue = ['yes'];
export default {
  title: 'Testing:Checkbox',
  decorators: [formDecorator]
};
export const Unchecked = () => /*#__PURE__*/React.createElement(Field, {
  type: "checkbox",
  component: CheckboxFieldFF,
  className: "checkbox",
  name: "checkbox",
  label: "Label text",
  validate: hasValue,
  required: true
});
export const Checked = () => /*#__PURE__*/React.createElement(Field, {
  type: "checkbox",
  component: CheckboxFieldFF,
  className: "checkbox",
  name: "checkbox",
  label: "Label text",
  initialValue: true
});
export const UncheckedWithValue = () => /*#__PURE__*/React.createElement(Field, {
  type: "checkbox",
  component: CheckboxFieldFF,
  className: "checkbox",
  name: "checkbox",
  label: "Label text",
  value: "yes"
});
export const CheckedWithValue = () => /*#__PURE__*/React.createElement(Field, {
  type: "checkbox",
  component: CheckboxFieldFF,
  className: "checkbox",
  name: "checkbox",
  label: "Label text",
  value: "yes",
  initialValue: initialValue
});
