// @dhis2-ui/organisation-unit-tree/build/cjs/organisation-unit-node/use-org-data/use-org-data.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useOrgData = void 0;
var _appRuntime = require("@dhis2/app-runtime");
const ORG_DATA_QUERY = {
  orgUnit: {
    resource: `organisationUnits`,
    id: _ref => {
      let {
        id
      } = _ref;
      return id;
    },
    params: _ref2 => {
      let {
        isUserDataViewFallback
      } = _ref2;
      return {
        isUserDataViewFallback,
        fields: ['path', 'children::size']
      };
    }
  }
};

/**
 * @param {string[]} ids
 * @param {Object} options
 * @param {string} options.displayName
 * @param {bool} options.isUserDataViewFallback
 * @returns {Object}
 */
const useOrgData = (id, _ref3) => {
  let {
    displayName,
    isUserDataViewFallback
  } = _ref3;
  if (!displayName) {
    throw new Error('"displayName" is required');
  }
  const variables = {
    id,
    isUserDataViewFallback
  };
  const {
    loading,
    error,
    data = {}
  } = (0, _appRuntime.useDataQuery)(ORG_DATA_QUERY, {
    variables
  });
  return {
    loading,
    error: error || null,
    data: {
      id,
      displayName,
      ...data.orgUnit
    }
  };
};
exports.useOrgData = useOrgData;
