// @dhis2-ui/table/build/cjs/data-table/data-table-column-header/data-table-column-header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DataTableColumnHeader = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = require("@dhis2/prop-types");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes2 = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _index = require("../table-elements/index.js");
var _dataTableColumnHeaderStyles = require("./data-table-column-header.styles.js");
var _filterHandle = require("./filter-handle.js");
var _sorter = require("./sorter.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const flexboxAlignLookup = {
  left: 'flex-start',
  center: 'center',
  right: 'flex-end'
};
const DataTableColumnHeader = exports.DataTableColumnHeader = /*#__PURE__*/(0, _react.forwardRef)((_ref, ref) => {
  let {
    align,
    children,
    className,
    colSpan,
    dataTest = 'dhis2-uicore-datatablecellhead',
    filter,
    fixed,
    large,
    left,
    name,
    role,
    rowSpan,
    scope,
    showFilter,
    sortDirection,
    sortIconTitle,
    top,
    width,
    onFilterIconClick,
    onSortIconClick
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_index.TableHeaderCell, {
    align: align,
    className: (0, _classnames.default)(className, 'DataTableColumnHeader', _dataTableColumnHeaderStyles.resolvedTableHeaderCss.className),
    colSpan: colSpan,
    dataTest: dataTest,
    fixed: fixed,
    large: large,
    left: left,
    ref: ref,
    rowSpan: rowSpan,
    role: role,
    scope: scope,
    top: top,
    width: width
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-3463223249 " + `jsx-${_dataTableColumnHeaderStyles.styles.__hash}` + " " + "container"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-3463223249 " + `jsx-${_dataTableColumnHeaderStyles.styles.__hash}` + " " + ((0, _classnames.default)('top', {
      large
    }) || "")
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-3463223249 " + `jsx-${_dataTableColumnHeaderStyles.styles.__hash}` + " " + "content"
  }, children), sortDirection && /*#__PURE__*/_react.default.createElement(_sorter.Sorter, {
    name: name,
    sortDirection: sortDirection,
    onClick: onSortIconClick,
    title: sortIconTitle
  }), filter && /*#__PURE__*/_react.default.createElement(_filterHandle.FilterHandle, {
    name: name,
    active: showFilter,
    onClick: onFilterIconClick
  })), showFilter && filter), _dataTableColumnHeaderStyles.resolvedTableHeaderCss.styles, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: _dataTableColumnHeaderStyles.styles.__hash
  }, _dataTableColumnHeaderStyles.styles), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3463223249"
  }, [`.label.jsx-3463223249{-webkit-box-pack:${flexboxAlignLookup[align]};-webkit-justify-content:${flexboxAlignLookup[align]};-ms-flex-pack:${flexboxAlignLookup[align]};justify-content:${flexboxAlignLookup[align]};}`]));
});
DataTableColumnHeader.displayName = 'DataTableColumnHeader';
DataTableColumnHeader.propTypes = {
  align: _propTypes2.default.oneOf(['left', 'center', 'right']),
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  colSpan: _propTypes2.default.string,
  dataTest: _propTypes2.default.string,
  /** The filter element (JSX), required when onFilterIconClick or showFilter are present */
  filter: (0, _propTypes.requiredIf)(props => props.onFilterIconClick || props.showFilter, _propTypes2.default.node),
  fixed: _propTypes2.default.bool,
  large: _propTypes2.default.bool,
  /** Left or top required when fixed */
  left: (0, _propTypes.requiredIf)(props => props.fixed && !props.top, _propTypes2.default.string),
  /** Can be used to match a column with a property name */
  name: _propTypes2.default.string,
  role: _propTypes2.default.string,
  rowSpan: _propTypes2.default.string,
  scope: _propTypes2.default.string,
  showFilter: (0, _propTypes.requiredIf)(props => props.filter, _propTypes2.default.bool),
  sortDirection: (0, _propTypes.requiredIf)(props => props.onSortIconClick, _propTypes2.default.oneOf(_sorter.SORT_DIRECTIONS)),
  sortIconTitle: _propTypes2.default.string,
  /** Left or top required when fixed */
  top: (0, _propTypes.requiredIf)(props => props.fixed && !props.left, _propTypes2.default.string),
  width: _propTypes2.default.string,
  onFilterIconClick: (0, _propTypes.requiredIf)(props => props.filter, _propTypes2.default.func),
  /** Sort icon click callback with `nextSortDirection` and `name` in payload */
  onSortIconClick: (0, _propTypes.requiredIf)(props => props.sortDirection, _propTypes2.default.func)
};
