// @dhis2-ui/table/build/cjs/data-table/data-table-column-header/sorter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getNextSortDirection = exports.Sorter = exports.SORT_DIRECTIONS = exports.DESC = exports.DEFAULT = exports.ASC = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = require("@dhis2/prop-types");
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes2 = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = _interopRequireDefault(require("../../locales/index.js"));
var _index2 = require("../table-elements/index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ASC = exports.ASC = 'asc';
const DESC = exports.DESC = 'desc';
const DEFAULT = exports.DEFAULT = 'default';
const SORT_DIRECTIONS = exports.SORT_DIRECTIONS = [DEFAULT, ASC, DESC];
const getNextSortDirection = currentDirection => {
  const currentIndex = SORT_DIRECTIONS.indexOf(currentDirection);
  const nextIndex = (currentIndex + 1) % SORT_DIRECTIONS.length;
  return SORT_DIRECTIONS[nextIndex];
};
exports.getNextSortDirection = getNextSortDirection;
const Sorter = _ref => {
  let {
    name,
    sortDirection,
    title,
    onClick
  } = _ref;
  const nextSortDirection = getNextSortDirection(sortDirection);
  const clickHandler = onClick ? event => {
    onClick({
      name,
      direction: nextSortDirection
    }, event);
  } : undefined;
  return /*#__PURE__*/_react.default.createElement(_index2.TableHeaderCellAction, {
    onClick: clickHandler,
    title: title || _index.default.t('Sort items')
  }, /*#__PURE__*/_react.default.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    width: "16",
    height: "16",
    viewBox: "0 0 16 16",
    className: _style.default.dynamic([["884818973", [_uiConstants.colors.grey500, _uiConstants.colors.blue700]]]) + " " + (sortDirection || "")
  }, /*#__PURE__*/_react.default.createElement("g", {
    fill: "none",
    fillRule: "evenodd",
    className: _style.default.dynamic([["884818973", [_uiConstants.colors.grey500, _uiConstants.colors.blue700]]])
  }, /*#__PURE__*/_react.default.createElement("polygon", {
    points: "4 9 12 9 8 14",
    className: _style.default.dynamic([["884818973", [_uiConstants.colors.grey500, _uiConstants.colors.blue700]]]) + " " + "top"
  }), /*#__PURE__*/_react.default.createElement("polygon", {
    points: "4 7 12 7 8 2",
    className: _style.default.dynamic([["884818973", [_uiConstants.colors.grey500, _uiConstants.colors.blue700]]]) + " " + "bottom"
  }))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "884818973",
    dynamic: [_uiConstants.colors.grey500, _uiConstants.colors.blue700]
  }, [`svg.__jsx-style-dynamic-selector .top.__jsx-style-dynamic-selector,svg.__jsx-style-dynamic-selector .bottom.__jsx-style-dynamic-selector{fill:${_uiConstants.colors.grey500};}`, `svg.asc.__jsx-style-dynamic-selector .top.__jsx-style-dynamic-selector,svg.desc.__jsx-style-dynamic-selector .bottom.__jsx-style-dynamic-selector{fill:${_uiConstants.colors.blue700};}`]));
};
exports.Sorter = Sorter;
Sorter.propTypes = {
  name: _propTypes2.default.string,
  sortDirection: (0, _propTypes.requiredIf)(props => props.onClick, _propTypes2.default.oneOf(SORT_DIRECTIONS)),
  title: _propTypes2.default.string,
  onClick: _propTypes2.default.func
};
