// @dhis2-ui/file-input/build/cjs/file-input/file-input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FileInput = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _uiIcons = require("@dhis2/ui-icons");
var _button = require("@dhis2-ui/button");
var _statusIcon = require("@dhis2-ui/status-icon");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
class FileInput extends _react.Component {
  constructor() {
    super(...arguments);
    _defineProperty(this, "ref", /*#__PURE__*/(0, _react.createRef)());
    _defineProperty(this, "handleClick", () => {
      this.ref.current.click();
    });
    _defineProperty(this, "handleChange", e => {
      if (this.props.onChange) {
        this.props.onChange(this.createHandlerPayload(), e);
      }

      // reset the file input so it won't prevent on-change
      // if the same file was added in a second attempt
      this.ref.current.value = '';
    });
    _defineProperty(this, "handleBlur", e => {
      if (this.props.onBlur) {
        this.props.onBlur(this.createHandlerPayload(), e);
      }
    });
    _defineProperty(this, "handleFocus", e => {
      if (this.props.onFocus) {
        this.props.onFocus(this.createHandlerPayload(), e);
      }
    });
    _defineProperty(this, "handleKeyDown", e => {
      if (this.props.onKeyDown) {
        this.props.onKeyDown(this.createHandlerPayload(), e);
      }
    });
  }
  createHandlerPayload() {
    return {
      files: this.ref.current.files,
      name: this.props.name
    };
  }
  render() {
    const {
      accept = '*',
      buttonLabel,
      className,
      dataTest = 'dhis2-uicore-fileinput',
      disabled,
      error,
      initialFocus,
      large,
      multiple,
      name,
      small,
      tabIndex,
      valid,
      warning
    } = this.props;
    return /*#__PURE__*/_react.default.createElement("div", {
      "data-test": dataTest,
      className: _style.default.dynamic([["1746996489", [_uiConstants.spacers.dp8, _uiConstants.spacers.dp4]]]) + " " + ((0, _classnames.default)('file-input', className) || "")
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: _style.default.dynamic([["1746996489", [_uiConstants.spacers.dp8, _uiConstants.spacers.dp4]]])
    }, /*#__PURE__*/_react.default.createElement("input", {
      id: name,
      name: name,
      type: "file",
      ref: this.ref,
      onChange: this.handleChange,
      accept: accept,
      multiple: multiple,
      disabled: disabled,
      "data-test": `${dataTest}-input`,
      className: _style.default.dynamic([["1746996489", [_uiConstants.spacers.dp8, _uiConstants.spacers.dp4]]])
    }), /*#__PURE__*/_react.default.createElement(_button.Button, {
      disabled: disabled,
      icon: /*#__PURE__*/_react.default.createElement(_uiIcons.IconUpload24, {
        color: _uiConstants.colors.grey700
      }),
      initialFocus: initialFocus,
      large: large,
      onBlur: this.handleBlur,
      onClick: this.handleClick,
      onFocus: this.handleFocus,
      onKeyDown: this.handleKeyDown,
      small: small,
      tabIndex: tabIndex,
      type: "button"
    }, buttonLabel)), /*#__PURE__*/_react.default.createElement(_statusIcon.StatusIcon, {
      error: error,
      valid: valid,
      warning: warning
    }), /*#__PURE__*/_react.default.createElement(_style.default, {
      id: "1746996489",
      dynamic: [_uiConstants.spacers.dp8, _uiConstants.spacers.dp4]
    }, ["input.__jsx-style-dynamic-selector{display:none;}", `.file-input.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:${_uiConstants.spacers.dp8};padding-bottom:${_uiConstants.spacers.dp4};}`]));
  }
}
exports.FileInput = FileInput;
_defineProperty(FileInput, "defaultProps", {
  accept: '*',
  dataTest: 'dhis2-uicore-fileinput'
});
FileInput.propTypes = {
  /**
   * The `accept` attribute of the [native file input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept)
   */
  accept: _propTypes.default.string,
  buttonLabel: _propTypes.default.string,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  disabled: _propTypes.default.bool,
  /** Input status. Mutually exclusive with `warning` and `valid` */
  error: _uiConstants.sharedPropTypes.statusPropType,
  initialFocus: _propTypes.default.bool,
  /** Button size. Mutually exclusive with `small` */
  large: _uiConstants.sharedPropTypes.sizePropType,
  /**
   * The `multiple` attribute of the [native file input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#multiple)
   */
  multiple: _propTypes.default.bool,
  name: _propTypes.default.string,
  /** Button size. Mutually exclusive with `large` */
  small: _uiConstants.sharedPropTypes.sizePropType,
  tabIndex: _propTypes.default.string,
  /** Input status. Mutually exclusive with `warning` and `error` */
  valid: _uiConstants.sharedPropTypes.statusPropType,
  /** Input status. Mutually exclusive with `valid` and `error` */
  warning: _uiConstants.sharedPropTypes.statusPropType,
  /** Called with signature `(object, event)` */
  onBlur: _propTypes.default.func,
  /** Called with signature `(object, event)` */
  onChange: _propTypes.default.func,
  /** Called with signature `(object, event)` */
  onFocus: _propTypes.default.func,
  /** Called with signature `(object, event)` */
  onKeyDown: _propTypes.default.func
};
