// @dhis2-ui/file-input/build/cjs/file-input/file-input.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnFocus = exports.WithOnChangeAndMultiple = exports.WithOnChange = exports.WithInitialFocusAndOnBlur = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
window.onChange = window.Cypress && window.Cypress.cy.stub();
const onChange = (payload, event) => {
  // NOTE: if files is not transformed into an array,
  // cypress will get an empty file list!
  window.onChange({
    ...payload,
    files: [...payload.files]
  }, event);
};
var _default = exports.default = {
  title: 'FileInput'
};
const WithOnChange = () => /*#__PURE__*/_react.default.createElement(_index.FileInput, {
  onChange: onChange,
  buttonLabel: "Upload file",
  name: "upload"
});
exports.WithOnChange = WithOnChange;
const WithOnChangeAndMultiple = () => /*#__PURE__*/_react.default.createElement(_index.FileInput, {
  name: "upload",
  onChange: onChange,
  buttonLabel: "Upload files",
  multiple: true
});
exports.WithOnChangeAndMultiple = WithOnChangeAndMultiple;
const WithInitialFocusAndOnBlur = () => /*#__PURE__*/_react.default.createElement(_index.FileInput, {
  buttonLabel: "Upload file",
  name: "upload",
  initialFocus: true,
  onBlur: window.onBlur
});
exports.WithInitialFocusAndOnBlur = WithInitialFocusAndOnBlur;
const WithOnFocus = () => /*#__PURE__*/_react.default.createElement(_index.FileInput, {
  buttonLabel: "Upload file",
  name: "upload",
  onFocus: window.onFocus
});
exports.WithOnFocus = WithOnFocus;
