// @dhis2-ui/logo/build/cjs/logo.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports._LogoWhite = exports._LogoIconWhite = exports._LogoIcon = exports._Logo = void 0;
var _react = _interopRequireDefault(require("react"));
var _logo = require("./logo.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
The master DHIS2 logo should be used overall, whenever possible. The master logo is a blue colored icon with black colored dhis2 wordmark.

\`\`\`js
import { Logo, LogoWhite, LogoIcon, LogoIconWhite } from '@dhis2/ui'
\`\`\`
`;
const Wrapper = fn => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    width: '358px'
  }
}, fn());
const Background = _ref => {
  let {
    children
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: '#276696'
    }
  }, children);
};
var _default = exports.default = {
  title: 'Logo',
  component: _logo.Logo,
  decorators: [Wrapper],
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const _Logo = args => /*#__PURE__*/_react.default.createElement(_logo.Logo, args);
exports._Logo = _Logo;
const _LogoWhite = args => /*#__PURE__*/_react.default.createElement(Background, null, /*#__PURE__*/_react.default.createElement(_logo.LogoWhite, args));
exports._LogoWhite = _LogoWhite;
_LogoWhite.parameters = {
  docs: {
    description: {
      story: 'When placing the DHIS2 logo on a dark background, the reversed version can be used. The icon and wordmark are white in this version. The reversed logo can be placed on any colored background, but ideally blue would be used to remain consistent with the DHIS2 brand.'
    }
  }
};
const _LogoIcon = args => /*#__PURE__*/_react.default.createElement(_logo.LogoIcon, args);
exports._LogoIcon = _LogoIcon;
_LogoIcon.parameters = {
  docs: {
    description: {
      story: 'There are times when it makes sense to use only the DHIS2 icon, such as in headers, toolbars and app icons. There are both blue colored and white colored icons, and these should be used as the main logo is used – blue colored where possible and white colored on darker backgrounds.'
    }
  }
};
const _LogoIconWhite = args => /*#__PURE__*/_react.default.createElement(Background, null, /*#__PURE__*/_react.default.createElement(_logo.LogoIconWhite, args));
exports._LogoIconWhite = _LogoIconWhite;
