// @dhis2-ui/modal/build/cjs/modal/modal.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithStatefulChildren = exports.WithOnClose = exports.WithChildren = exports.BottomAlignedWithOnClose = void 0;
var _button = require("@dhis2-ui/button");
var _react = _interopRequireWildcard(require("react"));
var _index = require("../index.js");
var _modal = require("./modal.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
window.onClose = window.Cypress && window.Cypress.cy.stub();
const StatefuleComponent = () => {
  const [counter, setCounter] = (0, _react.useState)(0);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("p", null, "Current counter:", /*#__PURE__*/_react.default.createElement("span", {
    id: "counter-value"
  }, counter)), /*#__PURE__*/_react.default.createElement("button", {
    id: "increment-counter",
    onClick: () => setCounter(counter + 1)
  }, "Add 1 to counter"));
};
var _default = exports.default = {
  title: 'Modal'
};
const WithOnClose = () => /*#__PURE__*/_react.default.createElement(_modal.Modal, {
  small: true,
  onClose: window.onClose
}, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Title"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Content"), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  primary: true
}, "Primary action"))));
exports.WithOnClose = WithOnClose;
const BottomAlignedWithOnClose = () => /*#__PURE__*/_react.default.createElement(_modal.Modal, {
  small: true,
  onClose: window.onClose,
  position: "bottom"
}, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Title"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Content"), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  primary: true
}, "Primary action"))));
exports.BottomAlignedWithOnClose = BottomAlignedWithOnClose;
const WithChildren = () => /*#__PURE__*/_react.default.createElement(_modal.Modal, null, "I am a child");
exports.WithChildren = WithChildren;
const WithStatefulChildren = () => {
  const [hide, setHide] = (0, _react.useState)(false);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("button", {
    id: "show-modal",
    onClick: () => setHide(false)
  }, "Show"), /*#__PURE__*/_react.default.createElement(_modal.Modal, {
    hide: hide
  }, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Can be hidden"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(StatefuleComponent, null)), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement("button", {
    id: "hide-modal",
    onClick: () => setHide(true)
  }, "Hide modal"))));
};
exports.WithStatefulChildren = WithStatefulChildren;
