// @dhis2-ui/modal/build/cjs/modal/modal.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TopScrollable = exports.SmallTitleContentAction = exports.SmallLongTitle = exports.SmallDestructivePrimary = exports.SmallContentPrimary = exports.SmallClickableScreenCover = exports.RTL = exports.ModalWithLoadingState = exports.ModalWithErrorState = exports.ModalThatHidesWithStatefulComponens = exports.MiddleScrollable = exports.MediumTitleContentAction = exports.LargeWithSelectComponent = exports.LargeTitleContentPrimary = exports.LargeModalWithMoreNestedModals = exports.FluidTop = exports.FluidTitleContentPrimary = exports.FluidMiddle = exports.FluidBottom = exports.DefaultContent = exports.BottomScrollable = exports.AlignmentMiddle = exports.AlignmentBottom = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _box = require("@dhis2-ui/box");
var _button = require("@dhis2-ui/button");
var _center = require("@dhis2-ui/center");
var _input = require("@dhis2-ui/input");
var _loader = require("@dhis2-ui/loader");
var _menu = require("@dhis2-ui/menu");
var _noticeBox = require("@dhis2-ui/notice-box");
var _select = require("@dhis2-ui/select");
var _tooltip = require("@dhis2-ui/tooltip");
var _react = _interopRequireWildcard(require("react"));
var _index = require("../index.js");
var _modal = require("./modal.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
_**Note**: For performance reasons, only one representative example is shown here. For more (interactive) examples, see individual stories in the Canvas tab._

A modal focuses user attention on a single task or piece of information inside a container. A modal blocks the rest of the application until it is dismissed.

#### Usage

A modal should be used to focus user attention on a single task or piece of information. Modals take over the entire screen and should be used sparingly to avoid interrupting a user's flow too often.

Use a modal in the following cases:

- to collect short, focussed user input that is blocking progress
- to present critical information that a user needs to acknowledge before continuing
- to ask the user to confirm a destructive action that cannot be undone

Do not use a modal in the following cases:

- to display unrelated or non-critical information; use an alert bar or a notice box instead
- to display complex, workflows that span multiple screens; navigate to a new, full page in the app instead

#### Children

Modals should be used with children \`<ModalTitle>\` (optional), \`<ModalContent>\` (required), and \`<ModalActions>\` (recommended)

See more about modal usage, including alignment and sizing, at [Design System: Modals](https://github.com/dhis2/design-system/blob/master/molecules/modal.md#usage).

\`\`\`js
import { Modal } from '@dhis2/ui'
\`\`\`
`;
const say = something => () => alert(something);
window.onClose = (payload, event) => {
  console.log('onClose payload', payload);
  console.log('onClose event', event);
};
const onClose = function () {
  return window.onClose(...arguments);
};
var _default = exports.default = {
  title: 'Modal',
  component: _modal.Modal,
  parameters: {
    docs: {
      // Use iframes to contain modals in docs page (otherwise chaos ensues)
      inlineStories: false,
      iframeHeight: '500px',
      /**
       * Due to iframes being very slow, disable stories on the docs page by default and
       * make one representative story as the primary ('SmallTitleContentAction')
       */
      disable: true,
      description: {
        component: description
      }
    }
  },
  argTypes: {
    small: {
      ..._uiConstants.sharedPropTypes.sizeArgType
    },
    large: {
      ..._uiConstants.sharedPropTypes.sizeArgType
    },
    position: {
      ..._uiConstants.sharedPropTypes.insideAlignmentArgType
    }
  }
};
const DefaultContent = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."));
exports.DefaultContent = DefaultContent;
DefaultContent.args = {
  onClose: onClose
};
DefaultContent.storyName = 'Default: Content';
const AlignmentMiddle = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."));
exports.AlignmentMiddle = AlignmentMiddle;
AlignmentMiddle.args = {
  onClose,
  position: 'middle'
};
AlignmentMiddle.storyName = 'Alignment: Middle';
const AlignmentBottom = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."));
exports.AlignmentBottom = AlignmentBottom;
AlignmentBottom.args = {
  onClose,
  position: 'bottom'
};
AlignmentBottom.storyName = 'Position: Bottom';
const SmallTitleContentAction = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a small modal with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
// Have this be the primary story on the docs page
exports.SmallTitleContentAction = SmallTitleContentAction;
SmallTitleContentAction.parameters = {
  docs: {
    disable: false,
    source: {
      type: 'dynamic'
    }
  }
};
SmallTitleContentAction.args = {
  small: true,
  onClose
};
SmallTitleContentAction.storyName = 'Small: Title, Content, Action';
const MediumTitleContentAction = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a medium modal with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
exports.MediumTitleContentAction = MediumTitleContentAction;
MediumTitleContentAction.storyName = 'Medium: Title, Content, Action';
const LargeTitleContentPrimary = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a large modal with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
exports.LargeTitleContentPrimary = LargeTitleContentPrimary;
LargeTitleContentPrimary.args = {
  large: true
};
LargeTitleContentPrimary.storyName = 'Large: Title, Content, Primary';
const FluidTitleContentPrimary = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal using custom dimensions, with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement("div", {
  style: {
    display: 'flex',
    flexFlow: 'row wrap'
  }
}, /*#__PURE__*/_react.default.createElement("div", {
  style: {
    width: '600px',
    backgroundColor: '#fea'
  }
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    width: '300px',
    backgroundColor: '#eaf'
  }
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
exports.FluidTitleContentPrimary = FluidTitleContentPrimary;
FluidTitleContentPrimary.args = {
  fluid: true,
  position: 'top'
};
FluidTitleContentPrimary.storyName = 'Fluid (Custom sizes): Title, Content, Primary';
const SmallContentPrimary = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
exports.SmallContentPrimary = SmallContentPrimary;
SmallContentPrimary.args = {
  small: true
};
SmallContentPrimary.storyName = 'Small: Content & Primary';
const SmallDestructivePrimary = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));
exports.SmallDestructivePrimary = SmallDestructivePrimary;
SmallDestructivePrimary.args = {
  small: true
};
SmallDestructivePrimary.storyName = 'Small: Destructive Primary';
const SmallClickableScreenCover = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal with clickable screen cover"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  primary: true,
  onClick: say('Button primary')
}, "Primary action"))));
exports.SmallClickableScreenCover = SmallClickableScreenCover;
SmallClickableScreenCover.args = {
  small: true,
  onClose: args => {
    onClose(args);
    say('Clickable screen cover')();
  }
};
SmallClickableScreenCover.storyName = 'Small: Clickable screen cover';
const TopScrollable = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal with scrollable content"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));
exports.TopScrollable = TopScrollable;
TopScrollable.args = {
  ...SmallClickableScreenCover.args
};
TopScrollable.storyName = 'Top: scrollable';
const MiddleScrollable = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal with scrollable content"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));
exports.MiddleScrollable = MiddleScrollable;
MiddleScrollable.args = {
  ...TopScrollable.args,
  position: 'middle'
};
MiddleScrollable.storyName = 'Middle: scrollable';
const BottomScrollable = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal with scrollable content"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));
exports.BottomScrollable = BottomScrollable;
BottomScrollable.args = {
  ...TopScrollable.args,
  position: 'bottom'
};
BottomScrollable.storyName = 'Bottom: scrollable';
const SmallLongTitle = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This headline should break into multiple lines because it's way too long for one!"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
exports.SmallLongTitle = SmallLongTitle;
SmallLongTitle.args = {
  small: true
};
SmallLongTitle.storyName = 'Small: Long title';
const LargeWithSelectComponent = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Select opens on top of the Modal"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelect, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
}))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
exports.LargeWithSelectComponent = LargeWithSelectComponent;
LargeWithSelectComponent.args = {
  large: true
};
LargeWithSelectComponent.storyName = 'Large: with Select component';
const ModalWithErrorState = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Update profile"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_input.InputField, {
  label: "First name",
  required: true,
  error: true,
  validationText: "First name can't be empty.",
  inputWidth: "320px"
}), /*#__PURE__*/_react.default.createElement(_input.InputField, {
  label: "Last name",
  value: "Olefeme",
  required: true,
  inputWidth: "320px"
}), /*#__PURE__*/_react.default.createElement(_box.Box, {
  marginTop: "16px"
}, /*#__PURE__*/_react.default.createElement(_noticeBox.NoticeBox, {
  error: true
}, "There is a problem with this form."))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  secondary: true
}, "Cancel"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  disabled: true,
  primary: true
}, "Save changes"))));
exports.ModalWithErrorState = ModalWithErrorState;
ModalWithErrorState.storyName = 'With error state';
const ModalWithLoadingState = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Modal title"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_box.Box, {
  minHeight: "240px"
}, /*#__PURE__*/_react.default.createElement(_center.Center, null, /*#__PURE__*/_react.default.createElement(_loader.CircularLoader, null)))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  secondary: true
}, "Cancel"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  primary: true
}, "Save changes"))));
exports.ModalWithLoadingState = ModalWithLoadingState;
ModalWithLoadingState.storyName = 'With loading state';
const LargeModalWithMoreNestedModals = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Select opens on top of the Modal - Level 1"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelect, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
}))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))), /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Select opens on top of the Modal - Level 2"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelect, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
}))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))), /*#__PURE__*/_react.default.createElement(_modal.Modal, {
  small: true
}, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Select opens on top of the Modal - Level 3"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelect, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
})), /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
  content: "Some extra info"
}, _ref => {
  let {
    ref,
    onMouseOver,
    onMouseOut
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("span", {
    ref: ref,
    onMouseOver: onMouseOver,
    onMouseOut: onMouseOut,
    className: "jsx-2364308943"
  }, /*#__PURE__*/_react.default.createElement("button", {
    className: "jsx-2364308943"
  }, "Tooltip on button"), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2364308943"
  }, ["span.jsx-2364308943{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}"]));
}), /*#__PURE__*/_react.default.createElement(_menu.FlyoutMenu, null, /*#__PURE__*/_react.default.createElement(_menu.MenuSectionHeader, {
  label: "Section with sub-menus"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 a"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b i"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b ii"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 c"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 3"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 a"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b i"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b ii"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 c"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 5"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuSectionHeader, {
  label: "Section with dividers between menu items"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuDivider, null), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuDivider, null), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 3"
}))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))));
exports.LargeModalWithMoreNestedModals = LargeModalWithMoreNestedModals;
LargeModalWithMoreNestedModals.args = {
  large: true
};
LargeModalWithMoreNestedModals.storyName = 'Large: modal with more nested modals';
const StatefuleComponent = () => {
  const [counter, setCounter] = (0, _react.useState)(0);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("p", null, "Current counter: ", counter), /*#__PURE__*/_react.default.createElement("button", {
    onClick: () => setCounter(counter + 1)
  }, "Add 1 to counter"));
};
const ModalThatHidesWithStatefulComponens = () => {
  const [render, setRender] = (0, _react.useState)(false);
  const [hide, setHide] = (0, _react.useState)(false);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, null, /*#__PURE__*/_react.default.createElement(_button.Button, {
    onClick: () => setHide(false),
    disabled: !render
  }, "Show the hidden modal"), /*#__PURE__*/_react.default.createElement(_button.Button, {
    onClick: () => setRender(true),
    disabled: render
  }, "Render Modal")), render && /*#__PURE__*/_react.default.createElement(_modal.Modal, {
    hide: hide
  }, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Can be hidden"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(StatefuleComponent, null)), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
    end: true
  }, /*#__PURE__*/_react.default.createElement(_button.Button, {
    onClick: () => setRender(false),
    secondary: true
  }, "Close modal"), /*#__PURE__*/_react.default.createElement(_button.Button, {
    onClick: () => setHide(true),
    primary: true
  }, "Hide modal")))));
};
exports.ModalThatHidesWithStatefulComponens = ModalThatHidesWithStatefulComponens;
const FluidTop = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal using custom dimensions, with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_box.Box, {
  width: "666px",
  height: "666px"
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.")), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
exports.FluidTop = FluidTop;
FluidTop.args = {
  fluid: true,
  position: 'top'
};
FluidTop.storyName = 'Fluid (Top)';
const FluidMiddle = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal using custom dimensions, with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_box.Box, {
  width: "666px",
  height: "666px"
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.")), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
exports.FluidMiddle = FluidMiddle;
FluidMiddle.args = {
  fluid: true,
  position: 'middle'
};
FluidMiddle.storyName = 'Fluid (Middle)';
const FluidBottom = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal using custom dimensions, with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_box.Box, {
  width: "666px",
  height: "666px"
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.")), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
exports.FluidBottom = FluidBottom;
FluidBottom.args = {
  fluid: true,
  position: 'bottom'
};
FluidBottom.storyName = 'Fluid (Bottom)';
const RTL = args => {
  (0, _react.useEffect)(() => {
    document.body.dir = 'rtl';
    return () => {
      document.body.dir = 'ltr';
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement("div", {
    dir: "rtl"
  }, /*#__PURE__*/_react.default.createElement(SmallTitleContentAction, args));
};
exports.RTL = RTL;
RTL.args = {
  small: true,
  onClose
};
