// @dhis2-ui/modal/build/cjs/modal/modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Modal = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _card = require("@dhis2-ui/card");
var _center = require("@dhis2-ui/center");
var _layer = require("@dhis2-ui/layer");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _closeButton = require("./close-button.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const resolveLayerStyles = hide => ({
  styles: /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1884450060",
    dynamic: [_uiConstants.spacers.dp64, hide ? 'none' : 'block']
  }, [`div.__jsx-style-dynamic-selector{padding:${_uiConstants.spacers.dp64};display:${hide ? 'none' : 'block'};}`]),
  className: _style.default.dynamic([["1884450060", [_uiConstants.spacers.dp64, hide ? 'none' : 'block']]])
});
const Modal = _ref => {
  let {
    children,
    className,
    dataTest = 'dhis2-uicore-modal',
    hide,
    fluid,
    large,
    onClose,
    position = 'top',
    small
  } = _ref;
  const layerStyles = resolveLayerStyles(hide);
  (0, _react.useEffect)(() => {
    if (hide) {
      return;
    }
    const handleKeyDown = event => {
      if (event.key === 'Escape' && onClose) {
        event.preventDefault();
        event.stopPropagation();
        onClose();
      }
    };
    document.addEventListener('keydown', handleKeyDown);
    return () => {
      document.removeEventListener('keydown', handleKeyDown);
    };
  }, [hide, onClose]);
  return /*#__PURE__*/_react.default.createElement(_layer.Layer, {
    onBackdropClick: onClose,
    className: layerStyles.className,
    translucent: !hide
  }, /*#__PURE__*/_react.default.createElement(_center.Center, {
    position: position
  }, /*#__PURE__*/_react.default.createElement("aside", {
    role: "dialog",
    "aria-modal": "true",
    "data-test": dataTest,
    className: _style.default.dynamic([["3655640549", [2 * _uiConstants.spacersNum.dp64, 2 * _uiConstants.spacersNum.dp64, 2 * _uiConstants.spacersNum.dp64, 2 * _uiConstants.spacersNum.dp64]]]) + " " + ((0, _classnames.default)(className, {
      small,
      large,
      fluid
    }) || "")
  }, /*#__PURE__*/_react.default.createElement(_card.Card, null, onClose && /*#__PURE__*/_react.default.createElement(_closeButton.CloseButton, {
    onClick: onClose
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["3655640549", [2 * _uiConstants.spacersNum.dp64, 2 * _uiConstants.spacersNum.dp64, 2 * _uiConstants.spacersNum.dp64, 2 * _uiConstants.spacersNum.dp64]]])
  }, children))), layerStyles.styles), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3655640549",
    dynamic: [2 * _uiConstants.spacersNum.dp64, 2 * _uiConstants.spacersNum.dp64, 2 * _uiConstants.spacersNum.dp64, 2 * _uiConstants.spacersNum.dp64]
  }, [`aside.__jsx-style-dynamic-selector{height:auto;width:600px;max-width:calc(100vw - ${2 * _uiConstants.spacersNum.dp64}px);max-height:calc(100vh - ${2 * _uiConstants.spacersNum.dp64}px);}`, "aside.small.__jsx-style-dynamic-selector{width:400px;}", "aside.large.__jsx-style-dynamic-selector{width:800px;}", "aside.fluid.__jsx-style-dynamic-selector{width:auto;}", `div.__jsx-style-dynamic-selector{padding:24px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-flow:column;-ms-flex-flow:column;flex-flow:column;max-width:calc(100vw - ${2 * _uiConstants.spacersNum.dp64}px);max-height:calc(100vh - ${2 * _uiConstants.spacersNum.dp64}px);}`]));
};
exports.Modal = Modal;
Modal.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  fluid: _propTypes.default.bool,
  hide: _propTypes.default.bool,
  large: _uiConstants.sharedPropTypes.sizePropType,
  position: _uiConstants.sharedPropTypes.insideAlignmentPropType,
  small: _uiConstants.sharedPropTypes.sizePropType,
  /** Callback used when the Modal closes */
  onClose: _propTypes.default.func
};
