// @dhis2-ui/transfer/build/es/__e2e__/highlight-range-of-options.e2e.stories.js
import React from 'react';
import { Transfer } from '../transfer.js';
import { options } from './common/options.js';
import { statefulDecorator } from './common/stateful-decorator.js';
export default {
  title: 'Transfer highlight range of options',
  decorators: [statefulDecorator()]
};
export const HasOptions = (_, _ref) => {
  let {
    onChange,
    selected
  } = _ref;
  return /*#__PURE__*/React.createElement(Transfer, {
    filterable: true,
    selected: selected,
    onChange: onChange,
    options: options
  });
};
export const HasSelected = (_, _ref2) => {
  let {
    onChange,
    selected
  } = _ref2;
  return /*#__PURE__*/React.createElement(Transfer, {
    filterable: true,
    onChange: onChange,
    selected: selected,
    options: options
  });
};
HasSelected.story = {
  decorators: [statefulDecorator({
    initialState: options.slice(0, 4).map(_ref3 => {
      let {
        value
      } = _ref3;
      return value;
    })
  })]
};
export const AllSelected = (_, _ref4) => {
  let {
    onChange,
    selected
  } = _ref4;
  return /*#__PURE__*/React.createElement(Transfer, {
    filterable: true,
    onChange: onChange,
    selected: selected,
    options: options
  });
};
AllSelected.story = {
  decorators: [statefulDecorator({
    initialState: options.map(_ref5 => {
      let {
        value
      } = _ref5;
      return value;
    })
  })]
};
