// @dhis2-ui/tab/build/cjs/tab-bar/tab-bar.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithChildren = exports.ScrollableWithSomeTabs = exports.ScrollableWithChildren = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("../tab/index.js");
var _index2 = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'TabBar'
};
const WithChildren = () => /*#__PURE__*/_react.default.createElement(_index2.TabBar, null, "I am a child");
exports.WithChildren = WithChildren;
const ScrollableWithChildren = () => /*#__PURE__*/_react.default.createElement(_index2.TabBar, {
  scrollable: true
}, "I am a child");
exports.ScrollableWithChildren = ScrollableWithChildren;
const ScrollableWithSomeTabs = () => {
  const TabStaticWidth = () => /*#__PURE__*/_react.default.createElement(_index.Tab, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 100,
      border: '1px solid black'
    }
  }, "Foo"));
  return /*#__PURE__*/_react.default.createElement(_index2.TabBar, {
    scrollable: true
  }, /*#__PURE__*/_react.default.createElement(TabStaticWidth, null), /*#__PURE__*/_react.default.createElement(TabStaticWidth, null), /*#__PURE__*/_react.default.createElement(TabStaticWidth, null));
};
exports.ScrollableWithSomeTabs = ScrollableWithSomeTabs;
