// @dhis2-ui/tab/build/cjs/tab-bar/tabs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tabs = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Tabs = _ref => {
  let {
    children,
    fixed,
    dataTest
  } = _ref;
  const tabContainer = (0, _react.useRef)(null);
  const childrenRefs = (0, _react.useMemo)(() => _react.default.Children.map(children, () => /*#__PURE__*/_react.default.createRef()), [children]);
  const handleKeyDown = event => {
    const currentFocus = document.activeElement;
    if (tabContainer.current && tabContainer.current === currentFocus) {
      if (childrenRefs.length > 0 && childrenRefs[0].current) {
        childrenRefs[0].current.focus();
      }
      return;
    }
    const currentIndex = childrenRefs.findIndex(ref => ref.current === currentFocus);
    if (currentIndex === -1) {
      return;
    }
    if (event.key === 'ArrowRight') {
      event.preventDefault();
      const nextIndex = (currentIndex + 1) % childrenRefs.length;
      childrenRefs[nextIndex].current.focus();
    }
    if (event.key === 'ArrowLeft') {
      event.preventDefault();
      const prevIndex = (currentIndex - 1 + childrenRefs.length) % childrenRefs.length;
      childrenRefs[prevIndex].current.focus();
    }
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: tabContainer,
    "data-test": dataTest,
    role: "tablist",
    tabIndex: 0,
    onKeyDown: handleKeyDown,
    className: _style.default.dynamic([["771129571", [_uiConstants.colors.grey400, _uiConstants.colors.white]]]) + " " + ((0, _classnames.default)({
      fixed
    }) || "")
  }, _react.default.Children.map(children, (child, index) => {
    if ( /*#__PURE__*/_react.default.isValidElement(child)) {
      return /*#__PURE__*/_react.default.cloneElement(child, {
        ref: childrenRefs[index]
      });
    }
    // Wrap non-element children e.g string in a <span>
    return /*#__PURE__*/_react.default.createElement("span", {
      ref: childrenRefs[index],
      tabIndex: -1,
      className: _style.default.dynamic([["771129571", [_uiConstants.colors.grey400, _uiConstants.colors.white]]])
    }, child);
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "771129571",
    dynamic: [_uiConstants.colors.grey400, _uiConstants.colors.white]
  }, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;position:relative;overflow:hidden;box-shadow:inset 0 -1px 0 0 ${_uiConstants.colors.grey400};-webkit-flex-wrap:nowrap;-ms-flex-wrap:nowrap;flex-wrap:nowrap;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;background:${_uiConstants.colors.white};}`]));
};
exports.Tabs = Tabs;
Tabs.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  children: _propTypes.default.node,
  fixed: _propTypes.default.bool
};
