// @dhis2-ui/card/build/cjs/card.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Card = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Card = _ref => {
  let {
    className,
    children,
    dataTest = 'dhis2-uicore-card'
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["3454125090", [_uiConstants.colors.white, _uiConstants.elevations.e100]]]) + " " + ((0, _classnames.default)(className) || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3454125090",
    dynamic: [_uiConstants.colors.white, _uiConstants.elevations.e100]
  }, [`div.__jsx-style-dynamic-selector{display:inline-block;position:relative;width:100%;height:100%;border-radius:3px;background:${_uiConstants.colors.white};box-shadow:${_uiConstants.elevations.e100};}`]));
};
exports.Card = Card;
Card.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
