// @dhis2-ui/box/build/cjs/box.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Width = exports.Overflow = exports.MinWidth = exports.MinHeight = exports.MaxWidth = exports.MaxHeight = exports.Height = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _box = require("./box.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
A box for creating some layout on the page.

\`\`\`js
import { Box } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Box',
  component: _box.Box,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_box.Box, args);
const Default = exports.Default = Template.bind({});
Default.args = {
  children: 'I am a child in a Box.'
};
const Height = exports.Height = Template.bind({});
Height.args = {
  ...Default.args,
  height: '250px'
};
const MaxHeight = exports.MaxHeight = Template.bind({});
MaxHeight.args = {
  children: /*#__PURE__*/_react.default.createElement("p", {
    style: {
      height: '500px'
    }
  }, "I am a tall child in a low Box."),
  maxHeight: '250px'
};
const MinHeight = exports.MinHeight = Template.bind({});
MinHeight.args = {
  ...Default.args,
  minHeight: '50vh'
};
const Width = exports.Width = Template.bind({});
Width.args = {
  ...Default.args,
  width: '250px'
};
const MinWidth = exports.MinWidth = Template.bind({});
MinWidth.args = {
  ...Default.args,
  minWidth: '50vh'
};
const MaxWidth = exports.MaxWidth = Template.bind({});
MaxWidth.args = {
  ...Default.args,
  maxWidth: '50vh'
};
const Overflow = exports.Overflow = Template.bind({});
Overflow.args = {
  maxHeight: '250px',
  overflow: 'scroll',
  children: /*#__PURE__*/_react.default.createElement("p", {
    style: {
      height: '500px'
    }
  }, "I am a tall child in a low Box, and my parent clips me")
};
