// @dhis2-ui/organisation-unit-tree/build/es/__stories__/force-reload-all.js
import _JSXStyle from "styled-jsx/style";
import { Button } from '@dhis2/ui';
import React, { useState } from 'react';
import { OrganisationUnitTree } from '../index.js';
import { onChange } from './shared.js';
export const ForceReloadAll = () => {
  const [forceReload, _setForceReload] = useState(false);
  const setForceReload = v => console.log('setForceReload', v) || _setForceReload(v);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Button, {
    disabled: forceReload,
    onClick: () => setForceReload(true)
  }, "Reload org unit tree"), ' ', /*#__PURE__*/React.createElement("span", {
    className: "jsx-2109967648"
  }, "(Force reload: ", forceReload ? 'true' : 'false', ")"), /*#__PURE__*/React.createElement("div", {
    className: "jsx-2109967648"
  }, /*#__PURE__*/React.createElement(OrganisationUnitTree, {
    onChange: onChange,
    forceReload: forceReload,
    name: "Root org unit",
    roots: ['A0000000000'],
    initiallyExpanded: ['/A0000000000/A0000000001'],
    selected: ['/A0000000000/A0000000001/A0000000003'],
    onChildrenLoaded: data => {
      const {
        id
      } = data;
      if (id === 'A0000000000') {
        setForceReload(false);
      }
    }
  })), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "2109967648"
  }, ["div.jsx-2109967648{width:400px;border:1px solid black;margin-top:32px;padding:16px;min-height:200px;}"]));
};
ForceReloadAll.storyName = 'Force reload all';
