// @dhis2-ui/organisation-unit-tree/build/es/__stories__/root-error.js
import { CustomDataProvider } from '@dhis2/app-runtime';
import React from 'react';
import { OrganisationUnitTree } from '../index.js';
import { customData, onChange } from './shared.js';
export const RootError = () => /*#__PURE__*/React.createElement(CustomDataProvider, {
  data: {
    ...customData,
    organisationUnits: function () {
      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }
      const [, {
        id
      }] = args;
      if (id === 'A0000000000') {
        return Promise.reject('This is a custom error message, it could be anything');
      }
      return customData.organisationUnits(...args);
    }
  }
}, /*#__PURE__*/React.createElement("fieldset", {
  style: {
    maxWidth: 600
  }
}, /*#__PURE__*/React.createElement("legend", {
  style: {
    padding: '0 10px'
  }
}, "Custom container (max-width: 600px)"), /*#__PURE__*/React.createElement(OrganisationUnitTree, {
  onChange: onChange,
  name: "Root org unit",
  roots: ['A0000000000'],
  initiallyExpanded: ['/A0000000000/A0000000001']
})));
RootError.storyName = 'Root error';
