// @dhis2-ui/table/build/es/stacked-table/stacked-table-cell-head.js
import _JSXStyle from "styled-jsx/style";
import { colors } from '@dhis2/ui-constants';
import PropTypes from 'prop-types';
import React from 'react';
export const StackedTableCellHead = _ref => {
  let {
    children = '',
    className,
    colSpan,
    dataTest = 'dhis2-uicore-stackedtablecellhead',
    rowSpan
  } = _ref;
  return /*#__PURE__*/React.createElement("th", {
    colSpan: colSpan,
    rowSpan: rowSpan,
    "data-test": dataTest,
    className: _JSXStyle.dynamic([["178822310", [colors.grey300]]]) + " " + (className || "")
  }, children && /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["178822310", [colors.grey300]]])
  }, children), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "178822310",
    dynamic: [colors.grey300]
  }, [`th.__jsx-style-dynamic-selector{border-bottom:1px solid ${colors.grey300};padding:0 12px;}`, "div.__jsx-style-dynamic-selector{min-height:36px;}"]));
};
StackedTableCellHead.propTypes = {
  children: PropTypes.string,
  className: PropTypes.string,
  colSpan: PropTypes.string,
  dataTest: PropTypes.string,
  rowSpan: PropTypes.string
};
