// @dhis2-ui/checkbox/build/cjs/checkbox-field/checkbox-field.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithValidationText = exports.WithLabelAndRequired = exports.WithLabel = exports.WithHelpText = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'CheckboxField'
};
const WithLabelAndRequired = () => /*#__PURE__*/_react.default.createElement(_index.CheckboxField, {
  name: "Ex",
  label: "CheckboxField",
  required: true,
  value: "checked"
});
exports.WithLabelAndRequired = WithLabelAndRequired;
const WithHelpText = () => /*#__PURE__*/_react.default.createElement(_index.CheckboxField, {
  name: "Ex",
  label: "CheckboxField",
  value: "checked",
  helpText: "Help text"
});
exports.WithHelpText = WithHelpText;
const WithLabel = () => /*#__PURE__*/_react.default.createElement(_index.CheckboxField, {
  name: "Ex",
  label: "The label",
  value: "checked"
});
exports.WithLabel = WithLabel;
const WithValidationText = () => /*#__PURE__*/_react.default.createElement(_index.CheckboxField, {
  name: "Ex",
  label: "The label",
  value: "checked",
  validationText: "Validation text"
});
exports.WithValidationText = WithValidationText;
