// @dhis2-ui/file-input/build/cjs/file-input-field-with-list/file-input-field-with-list.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithFileAndDefaultTexts = exports.WithDefaultTexts = exports.SingleFileWithFile = exports.MultipleFilesWithFiles = exports.MultipleFilesEmpty = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onChange = window.Cypress && window.Cypress.cy.stub();
const singleFileArray = [new File([], 'dummy.txt')];
window.duplicateFileConstructorArgs = [[], 'duplicate.md', {
  lastModified: 1581348175357,
  type: 'text/markdown'
}];
const multipleFileArray = [new File([], 'test1.md'), new File(...window.duplicateFileConstructorArgs), new File([], 'test2.md')];
const onChange = (payload, event) => {
  // NOTE: if files is not transformed into an array,
  // cypress will get an empty file list!
  window.onChange({
    ...payload,
    files: [...payload.files]
  }, event);
};
var _default = exports.default = {
  title: 'FileInputFieldWithList'
};
const SingleFileWithFile = () => /*#__PURE__*/_react.default.createElement(_index.FileInputFieldWithList, {
  buttonLabel: "Upload file",
  name: "upload",
  files: singleFileArray,
  removeText: "remove",
  onChange: onChange
});
exports.SingleFileWithFile = SingleFileWithFile;
const MultipleFilesEmpty = () => /*#__PURE__*/_react.default.createElement(_index.FileInputFieldWithList, {
  buttonLabel: "Upload file",
  multiple: true,
  name: "upload",
  removeText: "remove",
  onChange: onChange
});
exports.MultipleFilesEmpty = MultipleFilesEmpty;
const MultipleFilesWithFiles = () => /*#__PURE__*/_react.default.createElement(_index.FileInputFieldWithList, {
  buttonLabel: "Upload file",
  multiple: true,
  name: "upload",
  files: multipleFileArray,
  removeText: "remove",
  onChange: onChange
});
exports.MultipleFilesWithFiles = MultipleFilesWithFiles;
const WithFileAndDefaultTexts = () => /*#__PURE__*/_react.default.createElement(_index.FileInputFieldWithList, {
  name: "upload",
  files: singleFileArray,
  onChange: onChange
});
exports.WithFileAndDefaultTexts = WithFileAndDefaultTexts;
const WithDefaultTexts = () => /*#__PURE__*/_react.default.createElement(_index.FileInputFieldWithList, {
  name: "upload",
  onChange: onChange
});
exports.WithDefaultTexts = WithDefaultTexts;
