// @dhis2-ui/transfer/build/cjs/left-header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LeftHeader = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = require("./common/index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const LeftHeader = _ref => {
  let {
    children,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["2175954399", [_index.borderColor, _uiConstants.spacers.dp8]]])
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2175954399",
    dynamic: [_index.borderColor, _uiConstants.spacers.dp8]
  }, [`div.__jsx-style-dynamic-selector{border-bottom:1px solid ${borderColor};-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;padding:0 ${_uiConstants.spacers.dp8};}`]));
};
exports.LeftHeader = LeftHeader;
LeftHeader.propTypes = {
  children: _propTypes.default.node,
  dataTest: _propTypes.default.string
};
