// @dhis2-ui/transfer/build/cjs/picked-options.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PickedOptions = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _endIntersectionDetector = require("./end-intersection-detector.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const PickedOptions = _ref => {
  let {
    children,
    dataTest,
    selectedEmptyComponent,
    pickedOptionsRef,
    onPickedEndReached
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    ref: pickedOptionsRef,
    className: _style.default.dynamic([["392419471", [_uiConstants.spacers.dp4]]]) + " " + "container"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["392419471", [_uiConstants.spacers.dp4]]]) + " " + "content-container"
  }, !_react.default.Children.count(children) && selectedEmptyComponent, children, onPickedEndReached && /*#__PURE__*/_react.default.createElement(_endIntersectionDetector.EndIntersectionDetector, {
    rootRef: pickedOptionsRef,
    onEndReached: onPickedEndReached
  })), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "392419471",
    dynamic: [_uiConstants.spacers.dp4]
  }, [`.container.__jsx-style-dynamic-selector{padding:${_uiConstants.spacers.dp4} 0;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;overflow-y:auto;}`, ".content-container.__jsx-style-dynamic-selector{position:relative;}"]));
};
exports.PickedOptions = PickedOptions;
PickedOptions.propTypes = {
  children: _propTypes.default.node.isRequired,
  dataTest: _propTypes.default.string.isRequired,
  pickedOptionsRef: _propTypes.default.shape({
    current: _propTypes.default.instanceOf(HTMLElement)
  }),
  selectedEmptyComponent: _propTypes.default.node,
  onPickedEndReached: _propTypes.default.func
};
