// @dhis2-ui/transfer/build/cjs/reordering-actions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ReorderingActions = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _button = require("@dhis2-ui/button");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _icons = require("./icons.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ReorderingActions = _ref => {
  let {
    dataTest,
    disabledDown,
    disabledUp,
    onChangeUp,
    onChangeDown
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["874845231", [_uiConstants.spacers.dp8, _uiConstants.spacers.dp8, _uiConstants.spacers.dp8]]])
  }, /*#__PURE__*/_react.default.createElement(_button.Button, {
    small: true,
    disabled: disabledDown,
    onClick: () => !disabledDown && onChangeDown(event),
    dataTest: `${dataTest}-buttonmovedown`,
    icon: /*#__PURE__*/_react.default.createElement(_icons.IconMoveDown, {
      dataTest: `${dataTest}-iconmovedown`,
      disabled: disabledDown
    })
  }), /*#__PURE__*/_react.default.createElement(_button.Button, {
    small: true,
    disabled: disabledUp,
    onClick: () => !disabledUp && onChangeUp(event),
    dataTest: `${dataTest}-buttonmoveup`,
    icon: /*#__PURE__*/_react.default.createElement(_icons.IconMoveUp, {
      dataTest: `${dataTest}-iconmoveup`,
      disabled: disabledUp
    })
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "874845231",
    dynamic: [_uiConstants.spacers.dp8, _uiConstants.spacers.dp8, _uiConstants.spacers.dp8]
  }, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row-reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;padding-top:${_uiConstants.spacers.dp8};}`, `div.__jsx-style-dynamic-selector:last-child{padding-bottom:${_uiConstants.spacers.dp8};}`, `div.__jsx-style-dynamic-selector>button:first-child{margin-inline-start:${_uiConstants.spacers.dp8};}`]));
};
exports.ReorderingActions = ReorderingActions;
ReorderingActions.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  onChangeDown: _propTypes.default.func.isRequired,
  onChangeUp: _propTypes.default.func.isRequired,
  disabledDown: _propTypes.default.bool,
  disabledUp: _propTypes.default.bool
};
