// @dhis2-ui/transfer/build/cjs/right-footer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RightFooter = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = require("./common/index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const RightFooter = _ref => {
  let {
    children,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["3629116366", [_index.borderColor, _uiConstants.spacers.dp8]]])
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3629116366",
    dynamic: [_index.borderColor, _uiConstants.spacers.dp8]
  }, [`div.__jsx-style-dynamic-selector{-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;border-top:1px solid ${borderColor};padding:0 ${_uiConstants.spacers.dp8};}`]));
};
exports.RightFooter = RightFooter;
RightFooter.propTypes = {
  children: _propTypes.default.node,
  dataTest: _propTypes.default.string
};
