// @dhis2-ui/transfer/build/cjs/transfer-option.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TransferOption = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const DOUBLE_CLICK_MAX_DELAY = 500;
const TransferOption = _ref => {
  let {
    className,
    disabled,
    dataTest = 'dhis2-uicore-transferoption',
    highlighted,
    onClick,
    onDoubleClick,
    label,
    value
  } = _ref;
  const doubleClickTimeout = (0, _react.useRef)(null);
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    onClick: event => {
      if (disabled) {
        return;
      }
      if (doubleClickTimeout.current) {
        clearTimeout(doubleClickTimeout.current);
        doubleClickTimeout.current = null;
        onDoubleClick({
          value
        }, event);
      } else {
        doubleClickTimeout.current = setTimeout(() => {
          clearTimeout(doubleClickTimeout.current);
          doubleClickTimeout.current = null;
        }, DOUBLE_CLICK_MAX_DELAY);
        onClick({
          value
        }, event);
      }
    },
    "data-value": value,
    className: _style.default.dynamic([["4216903932", [_uiConstants.colors.grey900, _uiConstants.colors.grey200, _uiConstants.colors.teal700, _uiConstants.colors.white, _uiConstants.colors.grey600]]]) + " " + ((0, _classnames.default)(className, {
      highlighted,
      disabled
    }) || "")
  }, label, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4216903932",
    dynamic: [_uiConstants.colors.grey900, _uiConstants.colors.grey200, _uiConstants.colors.teal700, _uiConstants.colors.white, _uiConstants.colors.grey600]
  }, [`div.__jsx-style-dynamic-selector{font-size:14px;line-height:16px;padding:4px 8px;color:${_uiConstants.colors.grey900};-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}`, `div.__jsx-style-dynamic-selector:hover{background:${_uiConstants.colors.grey200};}`, `div.highlighted.__jsx-style-dynamic-selector{background:${_uiConstants.colors.teal700};color:${_uiConstants.colors.white};}`, `div.disabled.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey600};cursor:not-allowed;}`]));
};
exports.TransferOption = TransferOption;
TransferOption.propTypes = {
  label: _propTypes.default.node.isRequired,
  value: _propTypes.default.string.isRequired,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  disabled: _propTypes.default.bool,
  highlighted: _propTypes.default.bool,
  onClick: _propTypes.default.func,
  onDoubleClick: _propTypes.default.func
};
