// @dhis2-ui/table/build/es/data-table/table-elements/table-scroll-box.js
import _JSXStyle from "styled-jsx/style";
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import { colors } from '@dhis2/ui-constants';
import PropTypes from 'prop-types';
import React, { forwardRef } from 'react';
export const TableScrollBox = /*#__PURE__*/forwardRef((_ref, ref) => {
  let {
    children,
    className,
    dataTest = 'dhis2-uicore-tablescrollbox',
    maxHeight = 'auto',
    maxWidth = 'auto',
    ...props
  } = _ref;
  return /*#__PURE__*/React.createElement("div", _extends({}, props, {
    "data-test": dataTest,
    ref: ref,
    className: _JSXStyle.dynamic([["2859562182", [colors.grey300, maxHeight, maxWidth]]]) + " " + (className || "")
  }), children, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "2859562182",
    dynamic: [colors.grey300, maxHeight, maxWidth]
  }, [`div.__jsx-style-dynamic-selector{border:1px solid ${colors.grey300};border-top:none;box-sizing:border-box;max-height:${maxHeight};max-width:${maxWidth};overflow:auto;}`]));
});
TableScrollBox.displayName = 'TableScrollBox';
TableScrollBox.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  maxHeight: PropTypes.string,
  maxWidth: PropTypes.string
};
