// @dhis2-ui/popover/build/cjs/popover.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.NoArrow = exports.Default = exports.Customization = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _react = _interopRequireWildcard(require("react"));
var _popover = require("./popover.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const subtitle = `Useful to give a user more information or possible actions without disrupting a page or flow`;
const description = `
Popovers are similar to tooltips: they are for displaying extra information, but popovers are intended for richer information and actions.

Popovers are triggered by hovering or tapping on an element. Clicking on a element keeps the popover open until the user clicks or interacts elsewhere on the page.

Popovers can contain information in the form of rich markup, as well as actions. Critical actions, or the only action on a page, should not be hidden inside a popover.

Before using a popover, consider that some users may never see the information contained within. If that is a problem, display the information right on the page instead. Do not use a popover for content that is essential to the user experience or application.

See more about Popovers at [Design System: Popover](https://github.com/dhis2/design-system/blob/master/molecules/popover.md).

\`\`\`js
import { Popover } from '@dhis2/ui'
\`\`\`

_**Note**: Due to the full-page interaction of this component, only one representative example in an iframe sandbox is shown here. See more (interactive) examples in the 'Canvas' tab._
`;
var _default = exports.default = {
  title: 'Popover',
  component: _popover.Popover,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      },
      // Contain the popovers in iframes in the docs page
      inlineStories: false,
      iframeHeight: '500px',
      // Disable stories in docs page by default to use one representative example
      disable: true
    }
  },
  // Handles weird treatment of non-literal args (`elevation: elevations.e200`)
  args: {
    arrow: true,
    dataTest: 'dhis2-uicore-popover',
    elevation: _uiConstants.elevations.e200,
    maxWidth: 360,
    placement: 'top'
  },
  argTypes: {
    reference: {
      ..._uiConstants.sharedPropTypes.popperReferenceArgType
    },
    placement: {
      ..._uiConstants.sharedPropTypes.popperPlacementArgType
    }
  }
};
const boxStyle = {
  display: 'flex',
  justifyContent: 'center',
  width: 400,
  paddingTop: 280,
  backgroundColor: 'aliceblue'
};
const referenceElementStyle = {
  width: 100,
  height: 50,
  backgroundColor: 'cadetblue',
  textAlign: 'center',
  padding: 6
};
const Template = args => {
  const ref = (0, _react.useRef)(null);
  return /*#__PURE__*/_react.default.createElement("div", {
    style: boxStyle
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: referenceElementStyle,
    ref: ref
  }, "Reference element"), /*#__PURE__*/_react.default.createElement(_popover.Popover, _extends({}, args, {
    reference: ref
  }), /*#__PURE__*/_react.default.createElement("div", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Consectetur purus ut faucibus pulvinar elementum. Dignissim diam quis enim lobortis scelerisque fermentum dui faucibus. Rhoncus aenean vel elit scelerisque mauris pellentesque. Non sodales neque sodales ut etiam sit amet. Volutpat sed cras ornare arcu dui. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula. Convallis posuere morbi leo urna molestie at. Mauris cursus mattis molestie a iaculis at.")));
};
const Default = exports.Default = Template.bind({});
Default.parameters = {
  docs: {
    // Enable this story for the docs page
    disable: false,
    // Show source, including 'ref' hooks
    source: {
      type: 'code'
    }
  }
};
const NoArrow = exports.NoArrow = Template.bind({});
NoArrow.args = {
  arrow: false
};
const Customization = exports.Customization = Template.bind({});
Customization.args = {
  arrow: true,
  className: 'custom-classname',
  dataTest: 'custom-data-test-id',
  elevation: _uiConstants.elevations.e200,
  maxWidth: 400,
  placement: 'bottom-start',
  onClickOutside: () => console.log('backdrop was clicked...')
};
