// @dhis2-ui/button/build/es/button-strip/button-strip.prod.stories.js
import React from 'react';
import { Button, SplitButton } from '../index.js';
import { ButtonStrip } from './index.js';
const description = `
A wrapper for buttons to add spacing and alignment.

\`\`\`js
import { ButtonStrip } from '@dhis2/ui'
\`\`\`
`;
const Wrapper = fn => /*#__PURE__*/React.createElement("div", {
  style: {
    width: '100%',
    border: '1px solid #c4c9cc',
    padding: 8
  }
}, fn());
const alignmentArgType = {
  table: {
    type: {
      summary: 'bool',
      detail: "'middle' and 'end' are mutually exclusive props"
    }
  },
  control: {
    type: 'boolean'
  }
};
export default {
  title: 'Button Strip',
  component: ButtonStrip,
  decorators: [Wrapper],
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    middle: {
      ...alignmentArgType
    },
    end: {
      ...alignmentArgType
    }
  }
};
export const Default = args => /*#__PURE__*/React.createElement(ButtonStrip, args, /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(SplitButton, null, "Label?"));
export const DefaultAlignedMiddle = args => /*#__PURE__*/React.createElement(ButtonStrip, args, /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"));
DefaultAlignedMiddle.args = {
  middle: true
};
DefaultAlignedMiddle.storyName = 'Default - aligned middle';
export const DefaultAlignedRight = args => /*#__PURE__*/React.createElement(ButtonStrip, args, /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"));
DefaultAlignedRight.args = {
  end: true
};
DefaultAlignedRight.storyName = 'Default - aligned right';
export const DefaultRTL = args => /*#__PURE__*/React.createElement("div", {
  dir: "rtl"
}, /*#__PURE__*/React.createElement(ButtonStrip, args, /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save"), /*#__PURE__*/React.createElement(Button, null, "Save")));
DefaultRTL.storyName = 'Default - RTL';
