// @dhis2-ui/calendar/build/es/calendar/calendar-table-days-header.js
import _JSXStyle from "styled-jsx/style";
import { colors } from '@dhis2/ui-constants';
import PropTypes from 'prop-types';
import React from 'react';
export const CalendarTableDaysHeader = _ref => {
  let {
    weekDayLabels
  } = _ref;
  const dayNamesColor = colors.grey700;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("thead", {
    className: _JSXStyle.dynamic([["1577512909", [dayNamesColor]]])
  }, /*#__PURE__*/React.createElement("tr", {
    className: _JSXStyle.dynamic([["1577512909", [dayNamesColor]]])
  }, weekDayLabels.map((label, i) => /*#__PURE__*/React.createElement("th", {
    scope: "col",
    key: `weekday-${i}`,
    className: _JSXStyle.dynamic([["1577512909", [dayNamesColor]]])
  }, label)))), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "1577512909",
    dynamic: [dayNamesColor]
  }, [`th.__jsx-style-dynamic-selector{color:${dayNamesColor};font-weight:300;font-style:normal;padding:8px 8px;background:none;font-size:0.85em;border:none;cursor:default;}`, "tr.__jsx-style-dynamic-selector{border:none;}"]));
};
CalendarTableDaysHeader.propTypes = {
  weekDayLabels: PropTypes.arrayOf(PropTypes.string)
};
