// @dhis2-ui/calendar/build/es/calendar/navigation-container.js
import _JSXStyle from "styled-jsx/style";
import { colors, spacers } from '@dhis2/ui-constants';
import { IconChevronLeft16, IconChevronRight16 } from '@dhis2/ui-icons';
import PropTypes from 'prop-types';
import React from 'react';
import i18n from '../locales/index.js';
const wrapperBorderColor = colors.grey300;
const headerBackground = colors.grey050;
export const NavigationContainer = _ref => {
  var _currYear$label;
  let {
    languageDirection,
    pickerOptions
  } = _ref;
  const PreviousIcon = languageDirection === 'ltr' ? IconChevronLeft16 : IconChevronRight16;
  const NextIcon = languageDirection === 'ltr' ? IconChevronRight16 : IconChevronLeft16;
  const {
    currMonth,
    currYear,
    nextMonth,
    nextYear,
    prevMonth,
    prevYear
  } = pickerOptions;

  // Ethiopic years - when localised to English - add the era (i.e. 2015 ERA1), which is redundant in practice (like writing AD for gregorian years)
  // there is an ongoing discussion in JS-Temporal polyfill whether the era should be included or not, but for our case, it's safer to remove it
  const currentYearLabel = (_currYear$label = currYear.label) === null || _currYear$label === void 0 ? void 0 : _currYear$label.toString().replace(/ERA1/, '');
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "navigation-container"
  }, /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "month"
  }, /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "prev"
  }, /*#__PURE__*/React.createElement("button", {
    onClick: prevMonth.navigateTo,
    name: "previous-month",
    "data-test": "calendar-previous-month",
    "aria-label": `${i18n.t(`Go to ${prevMonth.label}`)}`,
    type: "button",
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]])
  }, /*#__PURE__*/React.createElement(PreviousIcon, {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]])
  }))), /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "curr"
  }, /*#__PURE__*/React.createElement("span", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "label"
  }, currMonth.label)), /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "next"
  }, /*#__PURE__*/React.createElement("button", {
    onClick: nextMonth.navigateTo,
    "data-test": "calendar-next-month",
    name: "next-month",
    "aria-label": `${i18n.t(`Go to ${nextMonth.label}`)}`,
    type: "button",
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]])
  }, /*#__PURE__*/React.createElement(NextIcon, {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]])
  })))), /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "year"
  }, /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "prev"
  }, /*#__PURE__*/React.createElement("button", {
    onClick: prevYear.navigateTo,
    name: "previous-year",
    "aria-label": `${i18n.t('Go to previous year')}`,
    type: "button",
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]])
  }, /*#__PURE__*/React.createElement(PreviousIcon, {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]])
  }))), /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "curr"
  }, /*#__PURE__*/React.createElement("span", {
    "data-test": "calendar-current-year",
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "label"
  }, currentYearLabel)), /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]]) + " " + "next"
  }, /*#__PURE__*/React.createElement("button", {
    onClick: nextYear.navigateTo,
    name: "next-year",
    "aria-label": `${i18n.t('Go to next year')}`,
    type: "button",
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]])
  }, /*#__PURE__*/React.createElement(NextIcon, {
    className: _JSXStyle.dynamic([["3883083596", [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]]])
  }))))), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "3883083596",
    dynamic: [spacers.dp4, colors.grey600, colors.grey200, colors.grey900, colors.grey300, spacers.dp8, spacers.dp4, wrapperBorderColor, headerBackground]
  }, ["button.__jsx-style-dynamic-selector{background:none;border:0;}", ".month.__jsx-style-dynamic-selector,.year.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;cursor:default;}", ".month.__jsx-style-dynamic-selector .curr.__jsx-style-dynamic-selector,.year.__jsx-style-dynamic-selector .curr.__jsx-style-dynamic-selector{-webkit-flex:2;-ms-flex:2;flex:2;white-space:nowrap;}", ".prev.__jsx-style-dynamic-selector,.next.__jsx-style-dynamic-selector{-webkit-flex:1;-ms-flex:1;flex:1;text-align:center;}", `.prev.__jsx-style-dynamic-selector button.__jsx-style-dynamic-selector,.next.__jsx-style-dynamic-selector button.__jsx-style-dynamic-selector{padding:${spacers.dp4};height:24px;width:24px;color:${colors.grey600};border-radius:3px;}`, ".prev.__jsx-style-dynamic-selector button.__jsx-style-dynamic-selector svg.__jsx-style-dynamic-selector,.next.__jsx-style-dynamic-selector button.__jsx-style-dynamic-selector svg.__jsx-style-dynamic-selector{width:16px;height:16px;}", `.prev.__jsx-style-dynamic-selector:hover button.__jsx-style-dynamic-selector,.next.__jsx-style-dynamic-selector:hover button.__jsx-style-dynamic-selector{background-color:${colors.grey200};color:${colors.grey900};cursor:pointer;}`, `.prev.__jsx-style-dynamic-selector button.__jsx-style-dynamic-selector:active,.next.__jsx-style-dynamic-selector button.__jsx-style-dynamic-selector:active{background-color:${colors.grey300};}`, ".label.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:4px 8px;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;min-height:16px;}", `.navigation-container.__jsx-style-dynamic-selector{gap:${spacers.dp8};padding:${spacers.dp4};display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;border-bottom:1px solid ${wrapperBorderColor};background-color:${headerBackground};font-size:1.08em;}`]));
};
NavigationContainer.propTypes = {
  languageDirection: PropTypes.oneOf(['ltr', 'rtl']),
  pickerOptions: PropTypes.shape({
    currMonth: PropTypes.shape({
      label: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    }),
    currYear: PropTypes.shape({
      label: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    }),
    nextMonth: PropTypes.shape({
      label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      navigateTo: PropTypes.func
    }),
    nextYear: PropTypes.shape({
      label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      navigateTo: PropTypes.func
    }),
    prevMonth: PropTypes.shape({
      label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      navigateTo: PropTypes.func
    }),
    prevYear: PropTypes.shape({
      label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      navigateTo: PropTypes.func
    })
  })
};
