// @dhis2-ui/menu/build/es/menu-item/menu-item.e2e.stories.js
import React from 'react';
import { Menu } from '../index.js';
import { MenuItem } from './menu-item.js';
window.onClick = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'MenuItem',
  component: MenuItem,
  decorators: [story => /*#__PURE__*/React.createElement(Menu, null, story())]
};
export const WithLabel = () => /*#__PURE__*/React.createElement(MenuItem, {
  label: "label"
});
export const WithOnClickAndValue = () => /*#__PURE__*/React.createElement(MenuItem, {
  value: "Value",
  onClick: window.onClick,
  label: "Menu item"
});
export const WithHref = () => /*#__PURE__*/React.createElement(MenuItem, {
  href: "url.test",
  label: "Menu item"
});
export const WithTarget = () => /*#__PURE__*/React.createElement(MenuItem, {
  href: "url.test",
  target: "_blank",
  label: "Menu item"
});
export const WithIcon = () => /*#__PURE__*/React.createElement(MenuItem, {
  icon: /*#__PURE__*/React.createElement("div", null, "Icon"),
  label: "Menu item"
});
export const WithSuffix = () => /*#__PURE__*/React.createElement(MenuItem, {
  suffix: "Suffix",
  label: "Menu item"
});
