// @dhis2-ui/select/build/cjs/select/filter-input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FilterInput = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _input = require("@dhis2-ui/input");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const FilterInput = _ref => {
  let {
    value,
    onChange,
    placeholder,
    className,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["730881408", [_uiConstants.colors.white, _uiConstants.spacers.dp8, _uiConstants.spacers.dp8, _uiConstants.spacers.dp4, _uiConstants.spacers.dp8]]]) + " " + (className || "")
  }, /*#__PURE__*/_react.default.createElement(_input.Input, {
    dense: true,
    dataTest: dataTest,
    value: value,
    onChange: onChange,
    type: "text",
    name: "filter",
    placeholder: placeholder,
    initialFocus: true
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "730881408",
    dynamic: [_uiConstants.colors.white, _uiConstants.spacers.dp8, _uiConstants.spacers.dp8, _uiConstants.spacers.dp4, _uiConstants.spacers.dp8]
  }, [`div.__jsx-style-dynamic-selector{position:-webkit-sticky;position:sticky;-webkit-inset-block-start:0;-ms-intb-rlock-start:0;inset-block-start:0;background:${_uiConstants.colors.white};padding-block-start:${_uiConstants.spacers.dp8};padding-inline-end:${_uiConstants.spacers.dp8};padding-block-end:${_uiConstants.spacers.dp4};padding-inline-start:${_uiConstants.spacers.dp8};z-index:1;}`]));
};
exports.FilterInput = FilterInput;
FilterInput.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  value: _propTypes.default.string.isRequired,
  className: _propTypes.default.string,
  placeholder: _propTypes.default.string,
  onChange: _propTypes.default.func
};
