// @dhis2-ui/select/build/cjs/select/input-wrapper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputWrapper = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _uiIcons = require("@dhis2/ui-icons");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const InputWrapper = _ref => {
  let {
    dataTest,
    onToggle,
    children,
    tabIndex = '0',
    error,
    warning,
    valid,
    disabled,
    dense,
    className,
    inputRef
  } = _ref;
  const classNames = (0, _classnames.default)(className, 'root', {
    error,
    warning,
    valid,
    disabled,
    dense
  });
  return /*#__PURE__*/_react.default.createElement("div", {
    onClick: onToggle,
    tabIndex: tabIndex,
    ref: inputRef,
    "data-test": dataTest,
    className: _style.default.dynamic([["1568723238", [_uiConstants.colors.grey500, _uiConstants.theme.focus, _uiConstants.theme.focus, _uiConstants.theme.valid, _uiConstants.theme.warning, _uiConstants.theme.error, _uiConstants.colors.grey100, _uiConstants.colors.grey500, _uiConstants.theme.disabled]]]) + " " + (classNames || "")
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["1568723238", [_uiConstants.colors.grey500, _uiConstants.theme.focus, _uiConstants.theme.focus, _uiConstants.theme.valid, _uiConstants.theme.warning, _uiConstants.theme.error, _uiConstants.colors.grey100, _uiConstants.colors.grey500, _uiConstants.theme.disabled]]]) + " " + "root-children"
  }, children), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["1568723238", [_uiConstants.colors.grey500, _uiConstants.theme.focus, _uiConstants.theme.focus, _uiConstants.theme.valid, _uiConstants.theme.warning, _uiConstants.theme.error, _uiConstants.colors.grey100, _uiConstants.colors.grey500, _uiConstants.theme.disabled]]]) + " " + "root-right"
  }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconChevronDown16, null)), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1568723238",
    dynamic: [_uiConstants.colors.grey500, _uiConstants.theme.focus, _uiConstants.theme.focus, _uiConstants.theme.valid, _uiConstants.theme.warning, _uiConstants.theme.error, _uiConstants.colors.grey100, _uiConstants.colors.grey500, _uiConstants.theme.disabled]
  }, [`.root.__jsx-style-dynamic-selector{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;background-color:white;border-radius:3px;border:1px solid ${_uiConstants.colors.grey500};box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;min-height:40px;padding-block:6px;padding-inline:12px;box-shadow:inset 0 0 1px 0 rgba(48,54,60,0.1);}`, `.root.__jsx-style-dynamic-selector:not(.disabled):focus,.root.__jsx-style-dynamic-selector:not(.disabled):active{outline:none;box-shadow:inset 0 0 0 2px ${_uiConstants.theme.focus};border-color:${_uiConstants.theme.focus};}`, `.root.valid.__jsx-style-dynamic-selector{border-color:${_uiConstants.theme.valid};}`, `.root.warning.__jsx-style-dynamic-selector{border-color:${_uiConstants.theme.warning};}`, `.root.error.__jsx-style-dynamic-selector{border-color:${_uiConstants.theme.error};}`, `.root.disabled.__jsx-style-dynamic-selector{background-color:${_uiConstants.colors.grey100};border-color:${_uiConstants.colors.grey500};color:${_uiConstants.theme.disabled};cursor:not-allowed;}`, ".root.dense.__jsx-style-dynamic-selector{padding-block:2px;padding-inline:8px;min-height:32px;}", ".root-children.__jsx-style-dynamic-selector{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}", ".root-right.__jsx-style-dynamic-selector{margin-block-start:4px;margin-inline-start:8px;}"]));
};
exports.InputWrapper = InputWrapper;
InputWrapper.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  inputRef: _propTypes.default.object.isRequired,
  tabIndex: _propTypes.default.string.isRequired,
  onToggle: _propTypes.default.func.isRequired,
  children: _propTypes.default.element,
  className: _propTypes.default.string,
  dense: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  error: _uiConstants.sharedPropTypes.statusPropType,
  valid: _uiConstants.sharedPropTypes.statusPropType,
  warning: _uiConstants.sharedPropTypes.statusPropType
};
