// @dhis2-ui/select/build/cjs/select/input-prefix.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputPrefix = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const InputPrefix = _ref => {
  let {
    prefix,
    className,
    dataTest
  } = _ref;
  if (!prefix) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["1343055987", [_uiConstants.colors.grey600, _uiConstants.spacers.dp12]]]) + " " + (className || "")
  }, prefix, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1343055987",
    dynamic: [_uiConstants.colors.grey600, _uiConstants.spacers.dp12]
  }, [`div.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey600};padding-inline-end:${_uiConstants.spacers.dp12};font-size:14px;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}`]));
};
exports.InputPrefix = InputPrefix;
InputPrefix.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  className: _propTypes.default.string,
  prefix: _propTypes.default.string
};
