// @dhis2/ui-forms/build/cjs/FieldGroupFF/FieldGroupFF.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
var _CheckboxFieldFF = require("../CheckboxFieldFF/CheckboxFieldFF.js");
var _formDecorator = require("../formDecorator.js");
var _index = require("../validators/index.js");
var _FieldGroupFF = require("./FieldGroupFF.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
This component is intended for use with [Final Form](https://final-form.org/docs/react-final-form/getting-started), the preferred library for form validation and utilities in DHIS 2 apps.

\`FieldGroupFF\` groups related fields (using the Final Form \`<Field>\`), like checkboxes, and adds a label and name.

\`\`\`js
import { FieldGroupFF } from '@dhis2/ui'
\`\`\`

Press **Submit** to see the form values logged to the console.
`;
var _default = exports.default = {
  title: 'Field Group (Final Form)',
  component: _FieldGroupFF.FieldGroupFF,
  decorators: [_formDecorator.formDecorator],
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Default = () => /*#__PURE__*/_react.default.createElement(_FieldGroupFF.FieldGroupFF, {
  label: "Group label",
  required: true,
  name: "set"
}, /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  type: "checkbox",
  component: _CheckboxFieldFF.CheckboxFieldFF,
  name: "set",
  label: "One",
  value: "one",
  validate: _index.hasValue,
  validationText: null
}), /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  type: "checkbox",
  component: _CheckboxFieldFF.CheckboxFieldFF,
  name: "set",
  label: "Two",
  validate: _index.hasValue,
  value: "two",
  validationText: null
}));
exports.Default = Default;
