// @dhis2-ui/sharing-dialog/build/cjs/cascade-sharing/update-info.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UpdateInfo = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _noticeBox = require("@dhis2-ui/notice-box");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _helpers = require("./helpers.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const query = {
  dashboard: {
    resource: 'dashboards',
    id: _ref => {
      let {
        id
      } = _ref;
      return id;
    },
    params: {
      fields: 'dashboardItems[type]'
    }
  }
};
const UpdateInfo = _ref2 => {
  let {
    id,
    entityAmount
  } = _ref2;
  const {
    data,
    loading
  } = (0, _appRuntime.useDataQuery)(query, {
    variables: {
      id
    }
  });
  if (loading) {
    return null;
  }
  const {
    dashboard
  } = data;
  const visualizationsCount = (0, _helpers.getVisualizationsCount)(dashboard.dashboardItems);
  if (!visualizationsCount) {
    return null;
  }
  const message = (0, _helpers.getInfoMessage)({
    visualizationsCount,
    entityAmount
  });
  return /*#__PURE__*/_react.default.createElement(_noticeBox.NoticeBox, null, message);
};
exports.UpdateInfo = UpdateInfo;
UpdateInfo.propTypes = {
  entityAmount: _propTypes.default.number.isRequired,
  id: _propTypes.default.string.isRequired
};
