// @dhis2-ui/organisation-unit-tree/build/cjs/organisation-unit-node/label/label-container.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LabelContainer = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/**
 * @param {Object} props
 * @param {bool} props.highlighted
 * @param {React.Component|React.Component[]} props.children
 * @returns {React.Component}
 */
const LabelContainer = _ref => {
  let {
    highlighted,
    children
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["1387769986", [_uiConstants.colors.teal200]]]) + " " + ((0, _classnames.default)({
      highlighted
    }) || "")
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: _style.default.dynamic([["1387769986", [_uiConstants.colors.teal200]]])
  }, children), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1387769986",
    dynamic: [_uiConstants.colors.teal200]
  }, ["div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", "span.__jsx-style-dynamic-selector{display:block;}", `.highlighted.__jsx-style-dynamic-selector{background:${_uiConstants.colors.teal200};padding-inline-end:4px;}`]));
};
exports.LabelContainer = LabelContainer;
LabelContainer.propTypes = {
  children: _propTypes.default.node,
  highlighted: _propTypes.default.bool
};
