// @dhis2-ui/organisation-unit-tree/build/cjs/organisation-unit-node/label/single-selection-label.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SingleSelectionLabel = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/**
 * @param {Object} props
 * @param {string} props.label
 * @param {bool} [props.checked]
 * @param {bool} [props.loading]
 * @param {Function} [props.onChange]
 * @returns {React.Component}
 */
const SingleSelectionLabel = _ref => {
  let {
    checked,
    children,
    onChange,
    loading
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("span", {
    onClick: event => {
      const payload = {
        checked: !checked
      };
      onChange(payload, event);
    },
    className: _style.default.dynamic([["321503310", [_uiConstants.colors.grey900, _uiConstants.colors.teal700]]]) + " " + ((0, _classnames.default)({
      checked,
      loading
    }) || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "321503310",
    dynamic: [_uiConstants.colors.grey900, _uiConstants.colors.teal700]
  }, [`span.__jsx-style-dynamic-selector{background:transparent;border-radius:3px;color:${_uiConstants.colors.grey900};cursor:pointer;display:inline-block;font-size:14px;line-height:24px;padding:0 5px;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;white-space:nowrap;}`, `.checked.__jsx-style-dynamic-selector{background:${_uiConstants.colors.teal700};color:white;}`, ".loading.__jsx-style-dynamic-selector{cursor:auto;}"]));
};
exports.SingleSelectionLabel = SingleSelectionLabel;
SingleSelectionLabel.propTypes = {
  children: _propTypes.default.any.isRequired,
  checked: _propTypes.default.bool,
  loading: _propTypes.default.bool,
  onChange: _propTypes.default.func
};
