// @dhis2/multi-calendar-dates/build/cjs/period-calculation/period-type-groups.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.yearlyFixedPeriodTypes = exports.weeklyFixedPeriodTypes = exports.sixmonthlyFixedPeriodTypes = exports.septemberFixedPeriodTypes = exports.quarterlyFixedPeriodTypes = exports.octoberFixedPeriodTypes = exports.novemberFixedPeriodTypes = exports.multiMonthFixedPeriodTypes = exports.monthlyStandardFixedPeriodTypes = exports.monthlyOffsetFixedPeriodTypes = exports.monthlyFixedPeriodTypes = exports.mayFixedPeriodTypes = exports.marchFixedPeriodTypes = exports.juneFixedPeriodTypes = exports.julyFixedPeriodTypes = exports.januaryFixedPeriodTypes = exports.financialYearFixedPeriodTypes = exports.februaryFixedPeriodTypes = exports.decemberFixedPeriodTypes = exports.augustFixedPeriodTypes = exports.aprilFixedPeriodTypes = void 0;
var _periodTypes = require("./period-types");
const weeklyFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/WEEKLY/));
exports.weeklyFixedPeriodTypes = weeklyFixedPeriodTypes;
const monthlyFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/MONTHLY|QUARTERLY|SIXMONTHLY/));
exports.monthlyFixedPeriodTypes = monthlyFixedPeriodTypes;
const quarterlyFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/QUARTERLY/));
exports.quarterlyFixedPeriodTypes = quarterlyFixedPeriodTypes;
const sixmonthlyFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/SIXMONTHLY/));
exports.sixmonthlyFixedPeriodTypes = sixmonthlyFixedPeriodTypes;
const monthlyStandardFixedPeriodTypes = ['MONTHLY', 'BIMONTHLY', 'QUARTERLY', 'QUARTERLYJAN', 'SIXMONTHLY', 'SIXMONTHLYJAN'];
exports.monthlyStandardFixedPeriodTypes = monthlyStandardFixedPeriodTypes;
const monthlyOffsetFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/(QUARTERLY|SIXMONTHLY)[A-Z]/));
exports.monthlyOffsetFixedPeriodTypes = monthlyOffsetFixedPeriodTypes;
const multiMonthFixedPeriodTypes = monthlyFixedPeriodTypes.filter(periodType => periodType !== 'MONTHLY');
exports.multiMonthFixedPeriodTypes = multiMonthFixedPeriodTypes;
const yearlyFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/^(YEARLY|FY[A-Z]{3})/));
exports.yearlyFixedPeriodTypes = yearlyFixedPeriodTypes;
const financialYearFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/^FY[A-Z]{3}$/));
exports.financialYearFixedPeriodTypes = financialYearFixedPeriodTypes;
const januaryFixedPeriodTypes = [...monthlyStandardFixedPeriodTypes, ..._periodTypes.periodTypes.filter(periodType => periodType.match(/JAN$/))];
exports.januaryFixedPeriodTypes = januaryFixedPeriodTypes;
const februaryFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/FEB$/));
exports.februaryFixedPeriodTypes = februaryFixedPeriodTypes;
const marchFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/MAR$/));
exports.marchFixedPeriodTypes = marchFixedPeriodTypes;
const aprilFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/APR$/));
exports.aprilFixedPeriodTypes = aprilFixedPeriodTypes;
const mayFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/MAY$/));
exports.mayFixedPeriodTypes = mayFixedPeriodTypes;
const juneFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/JUN$/));
exports.juneFixedPeriodTypes = juneFixedPeriodTypes;
const julyFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/JUL$/));
exports.julyFixedPeriodTypes = julyFixedPeriodTypes;
const augustFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/AUG$/));
exports.augustFixedPeriodTypes = augustFixedPeriodTypes;
const septemberFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/SEP$/));
exports.septemberFixedPeriodTypes = septemberFixedPeriodTypes;
const octoberFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/OCT$/));
exports.octoberFixedPeriodTypes = octoberFixedPeriodTypes;
const novemberFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/NOV$/));
exports.novemberFixedPeriodTypes = novemberFixedPeriodTypes;
const decemberFixedPeriodTypes = _periodTypes.periodTypes.filter(periodType => periodType.match(/DEC/));
exports.decemberFixedPeriodTypes = decemberFixedPeriodTypes;
