// @dhis2-ui/organisation-unit-tree/build/cjs/__stories__/force-reload-all.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ForceReloadAll = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _ui = require("@dhis2/ui");
var _react = _interopRequireWildcard(require("react"));
var _index = require("../index.js");
var _shared = require("./shared.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ForceReloadAll = () => {
  const [forceReload, _setForceReload] = (0, _react.useState)(false);
  const setForceReload = v => console.log('setForceReload', v) || _setForceReload(v);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_ui.Button, {
    disabled: forceReload,
    onClick: () => setForceReload(true)
  }, "Reload org unit tree"), ' ', /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-2109967648"
  }, "(Force reload: ", forceReload ? 'true' : 'false', ")"), /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-2109967648"
  }, /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    onChange: _shared.onChange,
    forceReload: forceReload,
    name: "Root org unit",
    roots: ['A0000000000'],
    initiallyExpanded: ['/A0000000000/A0000000001'],
    selected: ['/A0000000000/A0000000001/A0000000003'],
    onChildrenLoaded: data => {
      const {
        id
      } = data;
      if (id === 'A0000000000') {
        setForceReload(false);
      }
    }
  })), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2109967648"
  }, ["div.jsx-2109967648{width:400px;border:1px solid black;margin-top:32px;padding:16px;min-height:200px;}"]));
};
exports.ForceReloadAll = ForceReloadAll;
ForceReloadAll.storyName = 'Force reload all';
