// @dhis2-ui/segmented-control/build/cjs/segmented-control.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SegmentedControl = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = require("prop-types");
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/**
A segmented control is used to select between options that relate to another
area of content. All of the options in a segmented control should be closely related.

Do not use a segmented control as a standalone selection, it should always be
used as a selector for other content. For example, do not use a segmented
control in place of radio buttons when making a single, standalone choice.

See specification: [Design System](https://github.com/dhis2/design-system/blob/master/molecules/segmented-control.md)

```js
import { SegmentedControl } from '@dhis2/ui'
```
*/

const SegmentedControl = _ref => {
  let {
    options,
    selected,
    onChange,
    ariaLabel
  } = _ref;
  if (!options.map(_ref2 => {
    let {
      value
    } = _ref2;
    return value;
  }).includes(selected)) {
    const message = `There is no option with the value: "${selected}". ` + 'Make sure that the value passed to the selected ' + 'prop matches the value of an existing option.';
    throw new Error(message);
  }
  return /*#__PURE__*/_react.default.createElement("ul", {
    "aria-label": ariaLabel,
    className: _style.default.dynamic([["3664972916", [_uiConstants.colors.grey300, _uiConstants.colors.grey700, _uiConstants.spacers.dp12, _uiConstants.theme.focus, _uiConstants.colors.grey400, _uiConstants.colors.grey900, _uiConstants.elevations.e100, _uiConstants.colors.white, _uiConstants.colors.grey900]]]) + " " + "segmented-control"
  }, options.map(_ref3 => {
    let {
      label,
      value,
      disabled
    } = _ref3;
    return /*#__PURE__*/_react.default.createElement("li", {
      key: `option-${value}`,
      className: _style.default.dynamic([["3664972916", [_uiConstants.colors.grey300, _uiConstants.colors.grey700, _uiConstants.spacers.dp12, _uiConstants.theme.focus, _uiConstants.colors.grey400, _uiConstants.colors.grey900, _uiConstants.elevations.e100, _uiConstants.colors.white, _uiConstants.colors.grey900]]])
    }, /*#__PURE__*/_react.default.createElement("button", {
      type: "button",
      onClick: e => onChange({
        value
      }, e),
      disabled: disabled,
      className: _style.default.dynamic([["3664972916", [_uiConstants.colors.grey300, _uiConstants.colors.grey700, _uiConstants.spacers.dp12, _uiConstants.theme.focus, _uiConstants.colors.grey400, _uiConstants.colors.grey900, _uiConstants.elevations.e100, _uiConstants.colors.white, _uiConstants.colors.grey900]]]) + " " + ((0, _classnames.default)('segment', {
        selected: value === selected,
        disabled
      }) || "")
    }, label));
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3664972916",
    dynamic: [_uiConstants.colors.grey300, _uiConstants.colors.grey700, _uiConstants.spacers.dp12, _uiConstants.theme.focus, _uiConstants.colors.grey400, _uiConstants.colors.grey900, _uiConstants.elevations.e100, _uiConstants.colors.white, _uiConstants.colors.grey900]
  }, [`.segmented-control.__jsx-style-dynamic-selector{all:unset;list-style:none;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:stretch;-webkit-box-align:stretch;-ms-flex-align:stretch;align-items:stretch;background:${_uiConstants.colors.grey300};border-radius:5px;padding:2px;}`, `.segment.__jsx-style-dynamic-selector{all:unset;box-sizing:border-box;cursor:pointer;font-size:14px;text-align:center;border-radius:5px;background:transparent;color:${_uiConstants.colors.grey700};min-width:72px;max-width:320px;height:100%;padding:6px ${_uiConstants.spacers.dp12};}`, `.segment.__jsx-style-dynamic-selector:focus{outline:3px solid ${_uiConstants.theme.focus};outline-offset:-3px;}`, ".segment.__jsx-style-dynamic-selector:focus.__jsx-style-dynamic-selector:not(:focus-visible){outline:none;}", `.segment.__jsx-style-dynamic-selector:not(.selected):not(.disabled):hover{background:${_uiConstants.colors.grey400};color:${_uiConstants.colors.grey900};}`, `.segment.selected.__jsx-style-dynamic-selector{cursor:default;box-shadow:${_uiConstants.elevations.e100};background:${_uiConstants.colors.white};color:${_uiConstants.colors.grey900};}`, ".segment.disabled.__jsx-style-dynamic-selector{cursor:not-allowed;opacity:0.5;}"]));
};
exports.SegmentedControl = SegmentedControl;
SegmentedControl.propTypes = {
  /** Options to populate the segmented control */
  options: _propTypes.PropTypes.arrayOf(_propTypes.PropTypes.shape({
    label: _propTypes.PropTypes.string.isRequired,
    value: _propTypes.PropTypes.string.isRequired,
    disabled: _propTypes.PropTypes.bool
  })).isRequired,
  /** An option to select; should match the `value` property of the option to be selected */
  selected: _propTypes.PropTypes.string.isRequired,
  /** Called with the signature `({ value: string }, event)` */
  onChange: _propTypes.PropTypes.func.isRequired,
  /** Used to provide an accessible label to a segmented control without a visible label */
  ariaLabel: _propTypes.PropTypes.string
};
