// @dhis2-ui/help/build/cjs/help.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Help = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Help = _ref => {
  let {
    children,
    valid,
    error,
    warning,
    className,
    dataTest = 'dhis2-uicore-help'
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("p", {
    "data-test": dataTest,
    className: _style.default.dynamic([["1832050605", [_uiConstants.spacers.dp4, _uiConstants.theme.default, _uiConstants.colors.blue700, _uiConstants.theme.error, _uiConstants.colors.yellow800]]]) + " " + ((0, _classnames.default)(className, {
      valid,
      error,
      warning
    }) || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1832050605",
    dynamic: [_uiConstants.spacers.dp4, _uiConstants.theme.default, _uiConstants.colors.blue700, _uiConstants.theme.error, _uiConstants.colors.yellow800]
  }, [`p.__jsx-style-dynamic-selector{margin-top:${_uiConstants.spacers.dp4};margin-right:0;margin-bottom:0;margin-left:0;font-size:12px;line-height:14px;color:${_uiConstants.theme.default};}`, `.valid.__jsx-style-dynamic-selector{color:${_uiConstants.colors.blue700};}`, `.error.__jsx-style-dynamic-selector{color:${_uiConstants.theme.error};}`, `.warning.__jsx-style-dynamic-selector{color:${_uiConstants.colors.yellow800};}`]));
};
exports.Help = Help;
Help.propTypes = {
  children: _propTypes.default.string,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  error: _uiConstants.sharedPropTypes.statusPropType,
  valid: _uiConstants.sharedPropTypes.statusPropType,
  warning: _uiConstants.sharedPropTypes.statusPropType
};
