// @dhis2-ui/help/build/cjs/help.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TextOverflow = exports.StatusWarning = exports.StatusValid = exports.StatusError = exports.Default = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _react = _interopRequireDefault(require("react"));
var _help = require("./help.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
Small text for giving guiding information or feedback, especially for data entry instructions or form validation feedback.

\`\`\`js
import { Help } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Help',
  component: _help.Help,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    valid: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: {
      ..._uiConstants.sharedPropTypes.statusArgType
    }
  },
  args: {
    children: 'Allow me to be of assistance'
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_help.Help, args);
const Default = exports.Default = Template.bind({});
const StatusWarning = exports.StatusWarning = Template.bind({});
StatusWarning.args = {
  warning: true
};
StatusWarning.storyName = 'Status: Warning';
const StatusValid = exports.StatusValid = Template.bind({});
StatusValid.args = {
  valid: true
};
StatusValid.storyName = 'Status: Valid';
const StatusError = exports.StatusError = Template.bind({});
StatusError.args = {
  error: true
};
StatusError.storyName = 'Status: Error';
const TextOverflow = args => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    width: 200
  }
}, /*#__PURE__*/_react.default.createElement(_help.Help, args, "I take up more space than my container"));
exports.TextOverflow = TextOverflow;
