// @dhis2-ui/modal/build/es/modal/modal.e2e.stories.js
import { Button, ButtonStrip } from '@dhis2-ui/button';
import React, { useState } from 'react';
import { ModalActions, ModalContent, ModalTitle } from '../index.js';
import { Modal } from './modal.js';
window.onClose = window.Cypress && window.Cypress.cy.stub();
const StatefuleComponent = () => {
  const [counter, setCounter] = useState(0);
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("p", null, "Current counter:", /*#__PURE__*/React.createElement("span", {
    id: "counter-value"
  }, counter)), /*#__PURE__*/React.createElement("button", {
    id: "increment-counter",
    onClick: () => setCounter(counter + 1)
  }, "Add 1 to counter"));
};
export default {
  title: 'Modal'
};
export const WithOnClose = () => /*#__PURE__*/React.createElement(Modal, {
  small: true,
  onClose: window.onClose
}, /*#__PURE__*/React.createElement(ModalTitle, null, "Title"), /*#__PURE__*/React.createElement(ModalContent, null, "Content"), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  primary: true
}, "Primary action"))));
export const BottomAlignedWithOnClose = () => /*#__PURE__*/React.createElement(Modal, {
  small: true,
  onClose: window.onClose,
  position: "bottom"
}, /*#__PURE__*/React.createElement(ModalTitle, null, "Title"), /*#__PURE__*/React.createElement(ModalContent, null, "Content"), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  primary: true
}, "Primary action"))));
export const WithChildren = () => /*#__PURE__*/React.createElement(Modal, null, "I am a child");
export const WithStatefulChildren = () => {
  const [hide, setHide] = useState(false);
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("button", {
    id: "show-modal",
    onClick: () => setHide(false)
  }, "Show"), /*#__PURE__*/React.createElement(Modal, {
    hide: hide
  }, /*#__PURE__*/React.createElement(ModalTitle, null, "Can be hidden"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(StatefuleComponent, null)), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement("button", {
    id: "hide-modal",
    onClick: () => setHide(true)
  }, "Hide modal"))));
};
