// @dhis2-ui/selector-bar/build/es/selector-bar/__stories__/with-some-inputs.js
import React from 'react';
import { SelectorBar, SelectorBarItem } from '../../index.js';
import { OrgUnitSelect, MenuSelect, createDecoratorCustomDataProvider, createStatefulDecorator, decoratorCommonStyles, workflows } from './common.js';
export const WithSomeInputs = (_, _ref) => {
  let {
    workflow,
    setWorkflow,
    workflowOpen,
    setWorkflowOpen,
    orgUnit,
    setOrgUnit,
    orgUnitOpen,
    setOrgUnitOpen
  } = _ref;
  return /*#__PURE__*/React.createElement(SelectorBar, {
    disableClearSelections: !workflow && !orgUnit,
    onClearSelectionClick: () => {
      setWorkflow(null);
      setOrgUnit(null);
    }
  }, /*#__PURE__*/React.createElement(SelectorBarItem, {
    disabled: false,
    label: "Workflow",
    value: workflow === null || workflow === void 0 ? void 0 : workflow.label,
    noValueMessage: "Choose a workflow",
    open: workflowOpen,
    setOpen: setWorkflowOpen,
    onClearSelectionClick: () => {
      setWorkflow(null);
    }
  }, /*#__PURE__*/React.createElement(MenuSelect, {
    values: workflows,
    selected: workflow === null || workflow === void 0 ? void 0 : workflow.value,
    onChange: _ref2 => {
      let {
        selected
      } = _ref2;
      setWorkflow(workflows.find(currentWorkflow => selected === currentWorkflow.value));
      setWorkflowOpen(false);
    }
  })), /*#__PURE__*/React.createElement(SelectorBarItem, {
    disabled: !workflow,
    label: "Org unit",
    value: orgUnit === null || orgUnit === void 0 ? void 0 : orgUnit.displayName,
    noValueMessage: "Choose an organisation unit",
    open: orgUnitOpen,
    setOpen: setOrgUnitOpen,
    onClearSelectionClick: () => {
      setOrgUnit(null);
    }
  }, /*#__PURE__*/React.createElement(OrgUnitSelect, {
    selected: orgUnit ? [orgUnit.path] : [],
    onChange: (nextOrgUnit, evt) => {
      evt.stopPropagation();
      setOrgUnit(nextOrgUnit);
      setOrgUnitOpen(false);
    }
  })), /*#__PURE__*/React.createElement(SelectorBarItem, {
    displayOnly: true,
    label: "Person",
    value: "John doe",
    onClearSelectionClick: () => {
      alert('This selection would be cleared');
    }
  }));
};
WithSomeInputs.decorators = [decoratorCommonStyles, createStatefulDecorator({
  workflow: workflows[0]
}), createDecoratorCustomDataProvider()];
