// @dhis2-ui/table/build/es/stacked-table/content-with-title.js
import _JSXStyle from "styled-jsx/style";
import { colors } from '@dhis2/ui-constants';
import PropTypes from 'prop-types';
import React, { Fragment } from 'react';
export const ContentWithTitle = _ref => {
  let {
    title,
    children
  } = _ref;
  return /*#__PURE__*/React.createElement(Fragment, null, title && /*#__PURE__*/React.createElement("span", {
    className: _JSXStyle.dynamic([["215664166", [colors.grey700]]]) + " " + "title"
  }, title), /*#__PURE__*/React.createElement("span", {
    className: _JSXStyle.dynamic([["215664166", [colors.grey700]]]) + " " + "content"
  }, children), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "215664166",
    dynamic: [colors.grey700]
  }, [`.title.__jsx-style-dynamic-selector{display:block;white-space:normal;min-height:24px;font-size:13px;line-height:16px;padding:8px 0 4px;font-weight:normal;color:${colors.grey700};}`, ".content.__jsx-style-dynamic-selector{display:block;padding:0 0 8px 0;font-size:14px;line-height:18px;}", ".content.__jsx-style-dynamic-selector:first-child{padding-top:8px;padding-bottom:8px;}"]));
};
ContentWithTitle.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string
};
