// @dhis2-ui/table/build/es/stacked-table/stacked-table-row.js
import _JSXStyle from "styled-jsx/style";
import { colors } from '@dhis2/ui-constants';
import PropTypes from 'prop-types';
import React from 'react';
import { addColNumToChildren } from './add-col-num-to-children.js';
import { supplyHeaderLabelsToChildren } from './supply-header-labels-to-children.js';
import { Consumer } from './table-context.js';
export const StackedTableRow = _ref => {
  let {
    children,
    className,
    dataTest = 'dhis2-uicore-stackedtablerow'
  } = _ref;
  return /*#__PURE__*/React.createElement("tr", {
    "data-test": dataTest,
    className: _JSXStyle.dynamic([["2003961452", [colors.grey300, colors.white]]]) + " " + (className || "")
  }, /*#__PURE__*/React.createElement(Consumer, null, _ref2 => {
    let {
      headerLabels
    } = _ref2;
    return supplyHeaderLabelsToChildren(headerLabels, addColNumToChildren(children));
  }), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "2003961452",
    dynamic: [colors.grey300, colors.white]
  }, [`tr.__jsx-style-dynamic-selector{min-height:45px;display:block;border:1px solid ${colors.grey300};}`, `tr.__jsx-style-dynamic-selector:nth-child(even){background:${colors.white};}`, "thead>tr.__jsx-style-dynamic-selector,tbody>tr.__jsx-style-dynamic-selector{min-height:36px;}", "tr.__jsx-style-dynamic-selector+tr.__jsx-style-dynamic-selector{margin-top:32px;}"]));
};
StackedTableRow.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  dataTest: PropTypes.string
};
