// @dhis2-ui/alert/build/cjs/alert-bar/actions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.actionsPropType = exports.Actions = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = require("@dhis2/prop-types");
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes2 = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _action = require("./action.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const Actions = _ref => {
  let {
    actions,
    hide,
    dataTest
  } = _ref;
  if (!actions) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["793807948", [_uiConstants.spacers.dp48, _uiConstants.spacers.dp12]]])
  }, actions.map(action => /*#__PURE__*/_react.default.createElement(_action.Action, _extends({
    key: action.label
  }, action, {
    hide: hide,
    dataTest: `${dataTest}-action`
  }))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "793807948",
    dynamic: [_uiConstants.spacers.dp48, _uiConstants.spacers.dp12]
  }, [`div.__jsx-style-dynamic-selector{margin-inline-start:${_uiConstants.spacers.dp48};margin-inline-end:-${_uiConstants.spacers.dp12};}`]));
};
exports.Actions = Actions;
const actionsPropType = exports.actionsPropType = (0, _propTypes.arrayWithLength)(0, 2, _propTypes2.default.shape({
  label: _propTypes2.default.string.isRequired,
  onClick: _propTypes2.default.func.isRequired
}));
Actions.propTypes = {
  dataTest: _propTypes2.default.string.isRequired,
  hide: _propTypes2.default.func.isRequired,
  actions: actionsPropType
};
