// @dhis2-ui/transfer/build/cjs/__e2e__/loading_lists.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.NotLoadingSource = exports.NotLoadingPicked = exports.LoadingSource = exports.LoadingPicked = void 0;
var _react = _interopRequireDefault(require("react"));
var _transfer = require("../transfer.js");
var _options = require("./common/options.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'Transfer Loading Lists'
};
const LoadingSource = () => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  loading: true,
  selected: [],
  onChange: () => null,
  options: _options.options
});
exports.LoadingSource = LoadingSource;
const LoadingPicked = () => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  loadingPicked: true,
  selected: [],
  onChange: () => null,
  options: _options.options
});
exports.LoadingPicked = LoadingPicked;
const NotLoadingSource = () => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  selected: [],
  onChange: () => null,
  options: _options.options
});
exports.NotLoadingSource = NotLoadingSource;
const NotLoadingPicked = () => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  selected: [],
  onChange: () => null,
  options: _options.options
});
exports.NotLoadingPicked = NotLoadingPicked;
