// @dhis2-ui/transfer/build/cjs/__e2e__/notify_at_end_of_list.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.PartialSourceList = exports.PartialPickedList = exports.FullSourceList = exports.FullPickedList = void 0;
var _react = _interopRequireDefault(require("react"));
var _transfer = require("../transfer.js");
var _options = require("./common/options.js");
var _statefulDecorator = require("./common/stateful-decorator.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'Transfer End Of List',
  decorators: [(0, _statefulDecorator.statefulDecorator)()]
};
window.onEndReached = window.Cypress ? window.Cypress.cy.stub() : () => console.log('onEndReached');
window.onEndReachedPicked = window.Cypress ? window.Cypress.cy.stub() : () => console.log('onEndReachedPicked');
const FullSourceList = (_, _ref) => {
  let {
    selected,
    onChange
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    options: _options.options,
    selected: selected,
    onChange: onChange,
    onEndReached: window.onEndReached
  });
};
exports.FullSourceList = FullSourceList;
const FullPickedList = (_, _ref2) => {
  let {
    selected,
    onChange
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    options: _options.options,
    selected: selected,
    onChange: onChange,
    onEndReachedPicked: window.onEndReachedPicked
  });
};
exports.FullPickedList = FullPickedList;
FullPickedList.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.map(_ref3 => {
      let {
        value
      } = _ref3;
      return value;
    })
  })]
};
const PartialSourceList = (_, _ref4) => {
  let {
    selected,
    onChange
  } = _ref4;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    options: _options.options.slice(0, 4),
    selected: selected,
    onChange: onChange,
    onEndReached: window.onEndReached
  });
};
exports.PartialSourceList = PartialSourceList;
const PartialPickedList = (_, _ref5) => {
  let {
    selected,
    onChange
  } = _ref5;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    options: _options.options,
    selected: selected,
    onChange: onChange,
    onEndReachedPicked: window.onEndReachedPicked
  });
};
exports.PartialPickedList = PartialPickedList;
PartialPickedList.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.slice(0, 4).map(_ref6 => {
      let {
        value
      } = _ref6;
      return value;
    })
  })]
};
