// @dhis2-ui/table/build/es/data-table/data-table.prod.stories.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import { Box } from '@dhis2-ui/box';
import { Checkbox } from '@dhis2-ui/checkbox';
import { Input } from '@dhis2-ui/input';
import React, { useState } from 'react';
import { DataTableBody } from './data-table-body.js';
import { DataTableCell } from './data-table-cell.js';
import { DataTableColumnHeader } from './data-table-column-header/data-table-column-header.js';
import { DataTableFoot } from './data-table-foot.js';
import { DataTableHead } from './data-table-head.js';
import { DataTableRow } from './data-table-row/data-table-row.js';
import { DataTableToolbar } from './data-table-toolbar.js';
import { DataTable } from './data-table.js';
const subtitle = 'Used to display information in a standard, effective way.';
const description = `
Should be used with multiple DataTable-related child components - see the datatable and examples below.

\`\`\`js
import {
    DataTable,
    DataTableToolbar,
    DataTableHead,
    DataTableBody,
    DataTableFoot,
    DataTableRow,
    DataTableCell,
    DataTableColumnHeader,
} from '@dhis2/ui'
\`\`\`
`;
export default {
  title: 'DataTable',
  component: DataTable,
  // Add subcomponents to the args datatable
  subcomponents: {
    DataTableToolbar,
    DataTableHead,
    DataTableBody,
    DataTableFoot,
    DataTableRow,
    DataTableCell,
    DataTableColumnHeader
  },
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    // hide control from `children` because that can't be live edited
    children: {
      control: {
        disable: true
      }
    },
    // hide child component controls completely to avoid confusion
    bordered: {
      datatable: {
        disable: true
      }
    },
    large: {
      datatable: {
        disable: true
      }
    },
    draggable: {
      datatable: {
        disable: true
      }
    }
  }
};
const BasicTemplate = _ref => {
  let {
    bordered,
    large,
    draggable,
    bodyProps,
    ...args
  } = _ref;
  return /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, draggable && /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    large: large
  }), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    large: large
  }, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    large: large
  }, "Last name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    large: large
  }, "Incident date"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    large: large
  }, "Last updated"))), /*#__PURE__*/React.createElement(DataTableBody, bodyProps, /*#__PURE__*/React.createElement(DataTableRow, {
    draggable: draggable
  }, /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "Onyekachukwu"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "Kariuki"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "05/25/1972")), /*#__PURE__*/React.createElement(DataTableRow, {
    draggable: draggable
  }, /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "Kwasi"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "08/11/2010"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "02/26/1991")), /*#__PURE__*/React.createElement(DataTableRow, {
    draggable: draggable
  }, /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "Siyabonga"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "07/21/1981"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large,
    bordered: bordered
  }, "02/06/2007"))), /*#__PURE__*/React.createElement(DataTableFoot, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
    colSpan: draggable ? '5' : '4'
  }, "Footer content"))));
};
export const Default = BasicTemplate.bind({});
Default.args = {};
export const Loading = BasicTemplate.bind({});
Loading.args = {
  bodyProps: {
    loading: true
  }
};
export const BorderedCells = BasicTemplate.bind({});
BorderedCells.args = {
  bordered: true
};
export const LargeCells = BasicTemplate.bind({});
LargeCells.args = {
  large: true
};
export const DraggableRows = BasicTemplate.bind({});
DraggableRows.args = {
  draggable: true
};
const IndividualCellTemplate = args => /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Header"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Header"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Header"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Header"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  tag: "th"
}, "Active Cells"), /*#__PURE__*/React.createElement(DataTableCell, null, "Inactive cell"), /*#__PURE__*/React.createElement(DataTableCell, {
  active: true,
  onClick: () => {
    console.log('You clicked on a DataTableCell');
  }
}, "Acive Cell with onClick"), /*#__PURE__*/React.createElement(DataTableCell, null, "Inactive cell")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  tag: "th"
}, "Alignment"), /*#__PURE__*/React.createElement(DataTableCell, {
  align: "left"
}, "Left"), /*#__PURE__*/React.createElement(DataTableCell, {
  align: "center"
}, "Center"), /*#__PURE__*/React.createElement(DataTableCell, {
  align: "right"
}, "Right")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  tag: "th"
}, "Built in formatting"), /*#__PURE__*/React.createElement(DataTableCell, {
  error: true
}, "Error"), /*#__PURE__*/React.createElement(DataTableCell, {
  valid: true
}, "Valid"), /*#__PURE__*/React.createElement(DataTableCell, {
  muted: true
}, "Muted")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  tag: "th",
  backgroundColor: "lightgreen"
}, "Custom background (lightgreen)"), /*#__PURE__*/React.createElement(DataTableCell, {
  backgroundColor: "#eff"
}, "#eff"), /*#__PURE__*/React.createElement(DataTableCell, {
  active: true,
  backgroundColor: "yellow"
}, "yellow"), /*#__PURE__*/React.createElement(DataTableCell, null, "None")), /*#__PURE__*/React.createElement(DataTableRow, {
  selected: true
}, /*#__PURE__*/React.createElement(DataTableCell, {
  tag: "th"
}, "Custom background (on selected row )"), /*#__PURE__*/React.createElement(DataTableCell, {
  backgroundColor: "#eff"
}, "#eff"), /*#__PURE__*/React.createElement(DataTableCell, {
  backgroundColor: "yellow"
}, "yellow"), /*#__PURE__*/React.createElement(DataTableCell, null, "None"))));
export const CellStyling = IndividualCellTemplate.bind({});
CellStyling.args = {};
const ToolbarsTemplate = args => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DataTableToolbar, null, /*#__PURE__*/React.createElement("p", null, "Content")), /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Last name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Last updated"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/React.createElement(DataTableCell, null, "Kariuki"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/25/1972")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Kwasi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/26/1991")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Siyabonga"), /*#__PURE__*/React.createElement(DataTableCell, null, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007")))), /*#__PURE__*/React.createElement(DataTableToolbar, {
  position: "bottom"
}, /*#__PURE__*/React.createElement("p", null, "Content (bottom)")));
export const Toolbars = ToolbarsTemplate.bind({});
Toolbars.args = {};
const ExpandableContentTemplate = args => {
  const [openRowIndex, setOpenRowIndex] = useState(null);
  const toggleOpenRow = index => setOpenRowIndex(openRowIndex === index ? null : index);
  const style = {
    margin: 8,
    padding: 4,
    backgroundColor: 'lightblue'
  };
  const expandableContent = /*#__PURE__*/React.createElement("div", {
    style: style
  }, "More info about this row!");
  return /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, null), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Last name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Last updated"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, {
    expanded: openRowIndex === 0,
    onExpandToggle: payload => {
      console.log(payload);
      toggleOpenRow(0);
    },
    expandableContent: expandableContent
  }, /*#__PURE__*/React.createElement(DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/React.createElement(DataTableCell, null, "Kariuki"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/25/1972")), /*#__PURE__*/React.createElement(DataTableRow, {
    expanded: openRowIndex === 1,
    onExpandToggle: () => toggleOpenRow(1),
    expandableContent: expandableContent
  }, /*#__PURE__*/React.createElement(DataTableCell, null, "Kwasi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/26/1991")), /*#__PURE__*/React.createElement(DataTableRow, {
    expanded: openRowIndex === 2,
    onExpandToggle: () => toggleOpenRow(2),
    expandableContent: expandableContent
  }, /*#__PURE__*/React.createElement(DataTableCell, null, "Siyabonga"), /*#__PURE__*/React.createElement(DataTableCell, null, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"))));
};
export const ExpandableContent = ExpandableContentTemplate.bind({});
ExpandableContent.args = {};
const SelectableRowsTemplate = args => {
  const [selected, setSelected] = useState({
    id_2: true
  });
  const toggleSelected = _ref2 => {
    let {
      value,
      checked
    } = _ref2;
    setSelected({
      ...selected,
      [value]: checked
    });
  };
  const toggleAll = _ref3 => {
    let {
      checked
    } = _ref3;
    setSelected({
      id_1: checked,
      id_2: checked,
      id_3: checked
    });
  };
  const allSelected = () => Object.values(selected).filter(value => value).length === 3;
  return /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    width: "48px"
  }, /*#__PURE__*/React.createElement(Checkbox, {
    checked: allSelected(),
    onChange: toggleAll
  })), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Last name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Last updated"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, {
    selected: !!selected.id_1
  }, /*#__PURE__*/React.createElement(DataTableCell, {
    width: "48px"
  }, /*#__PURE__*/React.createElement(Checkbox, {
    value: "id_1",
    checked: !!selected.id_1,
    onChange: toggleSelected
  })), /*#__PURE__*/React.createElement(DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/React.createElement(DataTableCell, null, "Kariuki"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/25/1972")), /*#__PURE__*/React.createElement(DataTableRow, {
    selected: !!selected.id_2
  }, /*#__PURE__*/React.createElement(DataTableCell, {
    width: "48px"
  }, /*#__PURE__*/React.createElement(Checkbox, {
    value: "id_2",
    checked: !!selected.id_2,
    onChange: toggleSelected
  })), /*#__PURE__*/React.createElement(DataTableCell, null, "Kwasi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/26/1991")), /*#__PURE__*/React.createElement(DataTableRow, {
    selected: !!selected.id_3
  }, /*#__PURE__*/React.createElement(DataTableCell, {
    width: "48px"
  }, /*#__PURE__*/React.createElement(Checkbox, {
    value: "id_3",
    checked: !!selected.id_3,
    onChange: toggleSelected
  })), /*#__PURE__*/React.createElement(DataTableCell, null, "Siyabonga"), /*#__PURE__*/React.createElement(DataTableCell, null, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"))));
};
export const SelectableRows = SelectableRowsTemplate.bind({});
SelectableRows.args = {};
const FixedHeaderTemplate = args => /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/React.createElement(DataTableCell, null, "Kariuki"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/25/1972"), /*#__PURE__*/React.createElement(DataTableCell, null, "66"), /*#__PURE__*/React.createElement(DataTableCell, null, "Jawi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Kwasi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/26/1991"), /*#__PURE__*/React.createElement(DataTableCell, null, "38"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Siyabonga"), /*#__PURE__*/React.createElement(DataTableCell, null, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "98"), /*#__PURE__*/React.createElement(DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Unassigned"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Chiyembekezo"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okeke"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/23/1982"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/15/2003"), /*#__PURE__*/React.createElement(DataTableCell, null, "2"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Mtendere"), /*#__PURE__*/React.createElement(DataTableCell, null, "Afolayan"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/12/1994"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/12/1972"), /*#__PURE__*/React.createElement(DataTableCell, null, "37"), /*#__PURE__*/React.createElement(DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Inyene"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okonkwo"), /*#__PURE__*/React.createElement(DataTableCell, null, "04/01/1971"), /*#__PURE__*/React.createElement(DataTableCell, null, "03/16/2000"), /*#__PURE__*/React.createElement(DataTableCell, null, "70"), /*#__PURE__*/React.createElement(DataTableCell, null, "Kunike Barina"), /*#__PURE__*/React.createElement(DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Amaka"), /*#__PURE__*/React.createElement(DataTableCell, null, "Pretorius"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/25/1996"), /*#__PURE__*/React.createElement(DataTableCell, null, "09/15/1986"), /*#__PURE__*/React.createElement(DataTableCell, null, "32"), /*#__PURE__*/React.createElement(DataTableCell, null, "Bargbo"), /*#__PURE__*/React.createElement(DataTableCell, null, "Alberto Raya"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Meti"), /*#__PURE__*/React.createElement(DataTableCell, null, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "10/24/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/26/1989"), /*#__PURE__*/React.createElement(DataTableCell, null, "8"), /*#__PURE__*/React.createElement(DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Unassigned"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Eshe"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okeke"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(DataTableCell, null, "63"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, null, "Obi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "06/07/1990"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/03/2006"), /*#__PURE__*/React.createElement(DataTableCell, null, "28"), /*#__PURE__*/React.createElement(DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete"))));
export const FixedHeader = FixedHeaderTemplate.bind({});
FixedHeader.args = {
  scrollHeight: '350px'
};
const FixedFirstColumnTemplate = args => /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0"
}, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Last name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Last updated"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Age"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Registering unit"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Assigned user"), /*#__PURE__*/React.createElement(DataTableColumnHeader, null, "Status"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Onyekachukwu"), /*#__PURE__*/React.createElement(DataTableCell, null, "Kariuki"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/25/1972"), /*#__PURE__*/React.createElement(DataTableCell, null, "66"), /*#__PURE__*/React.createElement(DataTableCell, null, "Jawi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Kwasi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/26/1991"), /*#__PURE__*/React.createElement(DataTableCell, null, "38"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Siyabonga"), /*#__PURE__*/React.createElement(DataTableCell, null, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "98"), /*#__PURE__*/React.createElement(DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Unassigned"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Chiyembekezo"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okeke"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/23/1982"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/15/2003"), /*#__PURE__*/React.createElement(DataTableCell, null, "2"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Mtendere"), /*#__PURE__*/React.createElement(DataTableCell, null, "Afolayan"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/12/1994"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/12/1972"), /*#__PURE__*/React.createElement(DataTableCell, null, "37"), /*#__PURE__*/React.createElement(DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Inyene"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okonkwo"), /*#__PURE__*/React.createElement(DataTableCell, null, "04/01/1971"), /*#__PURE__*/React.createElement(DataTableCell, null, "03/16/2000"), /*#__PURE__*/React.createElement(DataTableCell, null, "70"), /*#__PURE__*/React.createElement(DataTableCell, null, "Kunike Barina"), /*#__PURE__*/React.createElement(DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Amaka"), /*#__PURE__*/React.createElement(DataTableCell, null, "Pretorius"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/25/1996"), /*#__PURE__*/React.createElement(DataTableCell, null, "09/15/1986"), /*#__PURE__*/React.createElement(DataTableCell, null, "32"), /*#__PURE__*/React.createElement(DataTableCell, null, "Bargbo"), /*#__PURE__*/React.createElement(DataTableCell, null, "Alberto Raya"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Meti"), /*#__PURE__*/React.createElement(DataTableCell, null, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "10/24/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/26/1989"), /*#__PURE__*/React.createElement(DataTableCell, null, "8"), /*#__PURE__*/React.createElement(DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Unassigned"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Eshe"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okeke"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(DataTableCell, null, "63"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0"
}, "Obi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "06/07/1990"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/03/2006"), /*#__PURE__*/React.createElement(DataTableCell, null, "28"), /*#__PURE__*/React.createElement(DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete"))));
export const FixedFirstColumn = FixedFirstColumnTemplate.bind({});
FixedFirstColumn.args = {
  scrollWidth: '500px'
};
const FixedHeaderAndTwoColumnsTemplate = args => /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/25/1972"), /*#__PURE__*/React.createElement(DataTableCell, null, "66"), /*#__PURE__*/React.createElement(DataTableCell, null, "Jawi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/26/1991"), /*#__PURE__*/React.createElement(DataTableCell, null, "38"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "98"), /*#__PURE__*/React.createElement(DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Unassigned"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/23/1982"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/15/2003"), /*#__PURE__*/React.createElement(DataTableCell, null, "2"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/12/1994"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/12/1972"), /*#__PURE__*/React.createElement(DataTableCell, null, "37"), /*#__PURE__*/React.createElement(DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/React.createElement(DataTableCell, null, "04/01/1971"), /*#__PURE__*/React.createElement(DataTableCell, null, "03/16/2000"), /*#__PURE__*/React.createElement(DataTableCell, null, "70"), /*#__PURE__*/React.createElement(DataTableCell, null, "Kunike Barina"), /*#__PURE__*/React.createElement(DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/25/1996"), /*#__PURE__*/React.createElement(DataTableCell, null, "09/15/1986"), /*#__PURE__*/React.createElement(DataTableCell, null, "32"), /*#__PURE__*/React.createElement(DataTableCell, null, "Bargbo"), /*#__PURE__*/React.createElement(DataTableCell, null, "Alberto Raya"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "10/24/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/26/1989"), /*#__PURE__*/React.createElement(DataTableCell, null, "8"), /*#__PURE__*/React.createElement(DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Unassigned"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(DataTableCell, null, "63"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "06/07/1990"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/03/2006"), /*#__PURE__*/React.createElement(DataTableCell, null, "28"), /*#__PURE__*/React.createElement(DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete"))));
export const FixedHeaderAndTwoColumns = FixedHeaderAndTwoColumnsTemplate.bind({});
FixedHeaderAndTwoColumns.args = {
  layout: 'fixed',
  width: '1000px',
  scrollWidth: '500px',
  scrollHeight: '400px'
};
const ScrollingDataTableWithToolbarsTemplate = args => /*#__PURE__*/React.createElement(Box, {
  width: "500px"
}, /*#__PURE__*/React.createElement(DataTableToolbar, null, /*#__PURE__*/React.createElement("p", null, "Content")), /*#__PURE__*/React.createElement(DataTable, _extends({}, args, {
  layout: "fixed",
  width: "1000px",
  scrollWidth: "500px",
  scrollHeight: "400px"
}), /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/25/1972"), /*#__PURE__*/React.createElement(DataTableCell, null, "66"), /*#__PURE__*/React.createElement(DataTableCell, null, "Jawi"), /*#__PURE__*/React.createElement(DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/11/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/26/1991"), /*#__PURE__*/React.createElement(DataTableCell, null, "38"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/21/1981"), /*#__PURE__*/React.createElement(DataTableCell, null, "02/06/2007"), /*#__PURE__*/React.createElement(DataTableCell, null, "98"), /*#__PURE__*/React.createElement(DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Unassigned"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/23/1982"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/15/2003"), /*#__PURE__*/React.createElement(DataTableCell, null, "2"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/React.createElement(DataTableCell, null, "08/12/1994"), /*#__PURE__*/React.createElement(DataTableCell, null, "05/12/1972"), /*#__PURE__*/React.createElement(DataTableCell, null, "37"), /*#__PURE__*/React.createElement(DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/React.createElement(DataTableCell, null, "04/01/1971"), /*#__PURE__*/React.createElement(DataTableCell, null, "03/16/2000"), /*#__PURE__*/React.createElement(DataTableCell, null, "70"), /*#__PURE__*/React.createElement(DataTableCell, null, "Kunike Barina"), /*#__PURE__*/React.createElement(DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/25/1996"), /*#__PURE__*/React.createElement(DataTableCell, null, "09/15/1986"), /*#__PURE__*/React.createElement(DataTableCell, null, "32"), /*#__PURE__*/React.createElement(DataTableCell, null, "Bargbo"), /*#__PURE__*/React.createElement(DataTableCell, null, "Alberto Raya"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/React.createElement(DataTableCell, null, "10/24/2010"), /*#__PURE__*/React.createElement(DataTableCell, null, "07/26/1989"), /*#__PURE__*/React.createElement(DataTableCell, null, "8"), /*#__PURE__*/React.createElement(DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Unassigned"), /*#__PURE__*/React.createElement(DataTableCell, null, "Complete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/31/1995"), /*#__PURE__*/React.createElement(DataTableCell, null, "63"), /*#__PURE__*/React.createElement(DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/React.createElement(DataTableCell, null, "06/07/1990"), /*#__PURE__*/React.createElement(DataTableCell, null, "01/03/2006"), /*#__PURE__*/React.createElement(DataTableCell, null, "28"), /*#__PURE__*/React.createElement(DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/React.createElement(DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/React.createElement(DataTableCell, null, "Incomplete")))), /*#__PURE__*/React.createElement(DataTableToolbar, {
  position: "bottom"
}, /*#__PURE__*/React.createElement("p", null, "Content (bottom)")));
export const ScrollingDataTableWithToolbars = ScrollingDataTableWithToolbarsTemplate.bind({});
ScrollingDataTableWithToolbars.args = {
  layout: 'fixed',
  width: '1000px',
  scrollWidth: '500px',
  scrollHeight: '400px'
};
const ColumnHeaderSortingTemplate = args => {
  const rows = [{
    firstName: 'Onyekachukwu',
    lastName: 'Kariuki'
  }, {
    firstName: 'Kwasi',
    lastName: 'Okafor'
  }, {
    firstName: 'Siyabonga',
    lastName: 'Abiodun'
  }];
  const [{
    column,
    direction
  }, setSortInstructions] = useState({
    column: 'firstName',
    direction: 'default'
  });
  const getSortDirection = columnName => columnName === column ? direction : 'default';
  const onSortIconClick = _ref4 => {
    let {
      name,
      direction
    } = _ref4;
    setSortInstructions({
      column: name,
      direction
    });
  };
  return /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    onSortIconClick: onSortIconClick,
    sortDirection: getSortDirection('firstName'),
    name: 'firstName',
    sortIconTitle: "Sort by first name"
  }, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    onSortIconClick: onSortIconClick,
    sortDirection: getSortDirection('lastName'),
    name: 'lastName',
    sortIconTitle: "Sort by last name"
  }, "Last name"))), /*#__PURE__*/React.createElement(DataTableBody, null, rows.sort((a, b) => {
    const strA = a[column];
    const strB = b[column];
    if (direction === 'asc' && strA < strB || direction === 'desc' && strA > strB) {
      return -1;
    }
    if (direction === 'desc' && strA < strB || direction === 'asc' && strA > strB) {
      return 1;
    }
    return 0;
  }).map(_ref5 => {
    let {
      firstName,
      lastName
    } = _ref5;
    return /*#__PURE__*/React.createElement(DataTableRow, {
      key: firstName + lastName
    }, /*#__PURE__*/React.createElement(DataTableCell, null, firstName), /*#__PURE__*/React.createElement(DataTableCell, null, lastName));
  })));
};
export const ColumnHeaderSorting = ColumnHeaderSortingTemplate.bind({});
ColumnHeaderSorting.args = {};
const InlineFilteringTemplate = args => {
  const rows = [{
    firstName: 'Onyekachukwu',
    lastName: 'Kariuki'
  }, {
    firstName: 'Kwasi',
    lastName: 'Okafor'
  }, {
    firstName: 'Siyabonga',
    lastName: 'Abiodun'
  }];
  const [{
    column,
    value
  }, setFilter] = useState({
    column: null,
    value: ''
  });
  const onFilterIconClick = _ref6 => {
    let {
      name,
      show
    } = _ref6;
    setFilter({
      column: show ? name : null,
      value: ''
    });
  };
  const onFilterInputChange = _ref7 => {
    let {
      value
    } = _ref7;
    setFilter({
      column: column,
      value
    });
  };
  return /*#__PURE__*/React.createElement(DataTable, args, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    onFilterIconClick: onFilterIconClick,
    name: "firstName",
    showFilter: column === 'firstName',
    filter: /*#__PURE__*/React.createElement(Input, {
      dense: true,
      onChange: onFilterInputChange,
      name: "firstName",
      value: value
    })
  }, "First name"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    onFilterIconClick: onFilterIconClick,
    name: "lastName",
    showFilter: column === 'lastName',
    filter: /*#__PURE__*/React.createElement(Input, {
      dense: true,
      onChange: onFilterInputChange,
      name: "lastName",
      value: value
    })
  }, "Last name"))), /*#__PURE__*/React.createElement(DataTableBody, null, rows.filter(row => {
    if (!column || !value) {
      return true;
    }
    return row[column].toUpperCase().includes(value.toUpperCase());
  }).map(_ref8 => {
    let {
      firstName,
      lastName
    } = _ref8;
    return /*#__PURE__*/React.createElement(DataTableRow, {
      key: firstName + lastName
    }, /*#__PURE__*/React.createElement(DataTableCell, null, firstName), /*#__PURE__*/React.createElement(DataTableCell, null, lastName));
  })));
};
export const InlineFiltering = InlineFilteringTemplate.bind({});
InlineFiltering.args = {
  layout: 'fixed'
};
const LongCellContentTemplate = _ref9 => {
  let {
    large
  } = _ref9;
  return /*#__PURE__*/React.createElement(DataTable, null, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    large: large,
    onSortIconClick: () => {},
    sortDirection: "asc",
    name: "first"
  }, "FIRST - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    large: large,
    onFilterIconClick: () => {},
    name: "firstName",
    showFilter: true,
    filter: /*#__PURE__*/React.createElement(Input, {
      dense: true,
      onChange: () => {},
      name: "firstName",
      value: "Filter value"
    })
  }, "SECOND - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    large: large,
    onSortIconClick: () => {},
    sortDirection: "asc",
    name: "third"
  }, "Third (short)"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
    large: large,
    onSortIconClick: () => {},
    sortDirection: "asc",
    name: "fourth"
  }, "Fourth - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
    large: large
  }, "FIRST - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large
  }, "SECOND - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large
  }, "Third (short)"), /*#__PURE__*/React.createElement(DataTableCell, {
    large: large
  }, "Fourth - Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eligendi non quis exercitationem culpa nesciunt nihil aut nostrum explicabo reprehenderit optio amet ab temporibus asperiores quasi cupiditate. Voluptatum ducimus voluptates voluptas?"))));
};
export const LongCellContent = LongCellContentTemplate.bind({});
export const LongCellContentLargeCells = LongCellContentTemplate.bind({});
LongCellContentLargeCells.args = {
  large: 'true'
};
export const RTL = args => /*#__PURE__*/React.createElement("div", {
  dir: "rtl"
}, /*#__PURE__*/React.createElement(BasicTemplate, args));
