// @dhis2-ui/input/build/es/input/input.e2e.stories.js
import React from 'react';
import { Input } from './index.js';
window.onChange = window.Cypress && window.Cypress.cy.stub();
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'Input'
};
export const WithOnChange = () => /*#__PURE__*/React.createElement(Input, {
  label: "Default label",
  name: "Default",
  value: "",
  onChange: window.onChange
});
export const WithInitialFocusAndOnBlur = () => /*#__PURE__*/React.createElement(Input, {
  label: "Default label",
  name: "Default",
  value: "",
  initialFocus: true,
  onBlur: window.onBlur
});
export const WithOnFocus = () => /*#__PURE__*/React.createElement(Input, {
  label: "Default label",
  name: "Default",
  value: "",
  onFocus: window.onFocus
});
export const WithInitialFocus = () => /*#__PURE__*/React.createElement(Input, {
  label: "Default label",
  name: "Default",
  value: "",
  initialFocus: true
});
export const WithDisabled = () => /*#__PURE__*/React.createElement(Input, {
  label: "Default label",
  name: "Default",
  value: "",
  disabled: true
});
