// @dhis2-ui/menu/build/es/flyout-menu/flyout-menu.e2e.stories.js
import _JSXStyle from "styled-jsx/style";
import { MenuItem } from '@dhis2-ui/menu';
import React from 'react';
import { FlyoutMenu } from './flyout-menu.js';
export default {
  title: 'FlyoutMenu',
  component: FlyoutMenu
};
const MenuItemSubMenuPositions = () => /*#__PURE__*/React.createElement(FlyoutMenu, null, /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 1"
}, /*#__PURE__*/React.createElement("div", {
  style: {
    padding: 30
  }
}, "SubMenu 1")));
export const WithChildren = () => /*#__PURE__*/React.createElement(FlyoutMenu, null, "I am a child");
export const TogglesSubMenus = () => /*#__PURE__*/React.createElement(FlyoutMenu, null, /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 1"
}, "SubMenu 1"), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 2"
}, "SubMenu 2"));
export const DefaultPosition = () => /*#__PURE__*/React.createElement(MenuItemSubMenuPositions, null);
export const FlippedPosition = () => /*#__PURE__*/React.createElement("div", {
  className: "jsx-2297277891"
}, /*#__PURE__*/React.createElement(MenuItemSubMenuPositions, null), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "2297277891"
}, ["div.jsx-2297277891{text-align:right;}"]));
export const ShiftIntoView = () => /*#__PURE__*/React.createElement("div", {
  className: "jsx-3052292650"
}, /*#__PURE__*/React.createElement(MenuItemSubMenuPositions, null), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "3052292650"
}, ["html,body,#root{position:relative;width:200px;max-width:200px;overflow:hidden;}", "#root{border:3px dashed grey;}"]));
