// @dhis2-ui/sharing-dialog/build/es/autocomplete/autocomplete.js
import { useDhis2ConnectionStatus } from '@dhis2/app-runtime';
import { InputField } from '@dhis2-ui/input';
import { Menu, MenuItem } from '@dhis2-ui/menu';
import useSize from '@react-hook/size';
import PropTypes from 'prop-types';
import React, { useRef } from 'react';
import i18n from '../locales/index.js';
import { MenuPopup } from './menu-popup.js';
export const Autocomplete = _ref => {
  let {
    dataTest = 'dhis2-sharingdialog-autocomplete',
    inputWidth,
    label,
    loading,
    onClose,
    onSearch,
    onSelect,
    placeholder,
    search,
    searchResults
  } = _ref;
  const wrapper = useRef(null);
  const [menuWidth] = useSize(wrapper);
  const {
    isDisconnected: offline
  } = useDhis2ConnectionStatus();
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    ref: wrapper
  }, /*#__PURE__*/React.createElement(InputField, {
    label: label,
    loading: loading,
    placeholder: placeholder,
    onChange: _ref2 => {
      let {
        value
      } = _ref2;
      return onSearch(value);
    },
    value: search,
    inputWidth: inputWidth,
    disabled: offline,
    helpText: offline ? i18n.t('Not available offline') : ''
  })), searchResults.length > 0 && /*#__PURE__*/React.createElement(MenuPopup, {
    onClick: onClose,
    menuWidth: `${menuWidth}px`,
    menuRef: wrapper,
    dataTest: `${dataTest}-menu`
  }, /*#__PURE__*/React.createElement(Menu, null, searchResults.map(result => /*#__PURE__*/React.createElement(MenuItem, {
    key: result.id,
    label: result.displayName,
    value: result.id,
    onClick: _ref3 => {
      let {
        value
      } = _ref3;
      return onSelect(value);
    }
  })))));
};
Autocomplete.propTypes = {
  searchResults: PropTypes.arrayOf(PropTypes.shape({
    displayName: PropTypes.string,
    id: PropTypes.string
  })).isRequired,
  dataTest: PropTypes.string,
  inputWidth: PropTypes.string,
  label: PropTypes.string,
  loading: PropTypes.bool,
  placeholder: PropTypes.string,
  search: PropTypes.string,
  onClose: PropTypes.func,
  onSearch: PropTypes.func,
  onSelect: PropTypes.func
};
