// @dhis2-ui/file-input/build/cjs/file-list/file-list-placeholder.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FileListPlaceholder = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const FileListPlaceholder = _ref => {
  let {
    children,
    dataTest = 'dhis2-uicore-filelistplaceholder'
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("p", {
    "data-test": dataTest,
    className: _style.default.dynamic([["1287857385", [_uiConstants.spacers.dp4, _uiConstants.colors.grey700]]])
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1287857385",
    dynamic: [_uiConstants.spacers.dp4, _uiConstants.colors.grey700]
  }, [`p.__jsx-style-dynamic-selector{margin:0;padding-top:${_uiConstants.spacers.dp4};font-size:14px;color:${_uiConstants.colors.grey700};}`]));
};
exports.FileListPlaceholder = FileListPlaceholder;
FileListPlaceholder.propTypes = {
  children: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
