// @dhis2-ui/file-input/build/cjs/file-list/file-list-item.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithRemoveText = exports.WithOnRemove = exports.WithLabel = exports.LoadingWithOnCancelAndCancelText = exports.LoadingWithOnCancel = void 0;
var _react = _interopRequireDefault(require("react"));
var _fileListItem = require("./file-list-item.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onRemove = window.Cypress && window.Cypress.cy.stub();
window.onCancel = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'FileListItem'
};
const WithOnRemove = () => /*#__PURE__*/_react.default.createElement(_fileListItem.FileListItem, {
  label: "File list item",
  onRemove: window.onRemove,
  removeText: "Remove"
});
exports.WithOnRemove = WithOnRemove;
const LoadingWithOnCancel = () => /*#__PURE__*/_react.default.createElement(_fileListItem.FileListItem, {
  loading: true,
  label: "File list item",
  removeText: "Remove",
  onCancel: window.onCancel,
  cancelText: "Cancel"
});
exports.LoadingWithOnCancel = LoadingWithOnCancel;
const LoadingWithOnCancelAndCancelText = () => /*#__PURE__*/_react.default.createElement(_fileListItem.FileListItem, {
  loading: true,
  label: "File list item",
  removeText: "Remove",
  onCancel: () => {},
  onRemove: () => {},
  cancelText: "Cancel"
});
exports.LoadingWithOnCancelAndCancelText = LoadingWithOnCancelAndCancelText;
const WithLabel = () => /*#__PURE__*/_react.default.createElement(_fileListItem.FileListItem, {
  label: "Label",
  removeText: "Remove",
  onRemove: () => {}
});
exports.WithLabel = WithLabel;
const WithRemoveText = () => /*#__PURE__*/_react.default.createElement(_fileListItem.FileListItem, {
  label: "Label",
  removeText: "Remove",
  onRemove: () => {}
});
exports.WithRemoveText = WithRemoveText;
