// @dhis2-ui/file-input/build/es/file-list/file-list-item.e2e.stories.js
import React from 'react';
import { FileListItem } from './file-list-item.js';
window.onRemove = window.Cypress && window.Cypress.cy.stub();
window.onCancel = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'FileListItem'
};
export const WithOnRemove = () => /*#__PURE__*/React.createElement(FileListItem, {
  label: "File list item",
  onRemove: window.onRemove,
  removeText: "Remove"
});
export const LoadingWithOnCancel = () => /*#__PURE__*/React.createElement(FileListItem, {
  loading: true,
  label: "File list item",
  removeText: "Remove",
  onCancel: window.onCancel,
  cancelText: "Cancel"
});
export const LoadingWithOnCancelAndCancelText = () => /*#__PURE__*/React.createElement(FileListItem, {
  loading: true,
  label: "File list item",
  removeText: "Remove",
  onCancel: () => {},
  onRemove: () => {},
  cancelText: "Cancel"
});
export const WithLabel = () => /*#__PURE__*/React.createElement(FileListItem, {
  label: "Label",
  removeText: "Remove",
  onRemove: () => {}
});
export const WithRemoveText = () => /*#__PURE__*/React.createElement(FileListItem, {
  label: "Label",
  removeText: "Remove",
  onRemove: () => {}
});
