// @dhis2/multi-calendar-dates/build/cjs/hooks/internal/useCalendarWeekDays.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useCalendarWeekDays = void 0;
var _polyfill = require("@js-temporal/polyfill");
var _react = require("react");
const groupByWeek = (acc, day) => {
  if (day.dayOfWeek === 1) {
    acc.push([]);
  }
  const currentWeekArray = acc[acc.length - 1];
  currentWeekArray.push(day);
  return acc;
};

/**
 * internal hook used by useDatePicker hook to return the week days numbers in a calendar
 *
 * @param dayZdt
 * @returns an array of array of days (each top-level array is a week)
 */
const useCalendarWeekDays = dayZdt => {
  return (0, _react.useMemo)(() => {
    const dateInfo = {
      year: dayZdt.year,
      month: dayZdt.month,
      day: dayZdt.day,
      hour: 0,
      minute: 0,
      second: 0,
      calendar: dayZdt.calendar,
      timeZone: dayZdt.timeZone
    };

    // get first day of the month
    const firstDayOfMonth = _polyfill.Temporal.ZonedDateTime.from({
      ...dateInfo,
      day: 1
    });
    // get first day of first week to display
    const firstDayToDisplay = firstDayOfMonth.subtract({
      days: firstDayOfMonth.dayOfWeek - 1
    });

    // get last day of month
    const lastDayOfMonth = _polyfill.Temporal.ZonedDateTime.from({
      ...dateInfo,
      day: dayZdt.daysInMonth
    });

    // get last day of last week of month
    const lastDayToDisplay = lastDayOfMonth.add({
      days: 7 - lastDayOfMonth.dayOfWeek
    });
    const numberOfDaysInCalendar = lastDayToDisplay.toPlainDate().since(firstDayToDisplay.toPlainDate()).days;
    let date = _polyfill.Temporal.ZonedDateTime.from(firstDayToDisplay);
    const allDates = [];
    for (let i = 0; i <= numberOfDaysInCalendar; i++) {
      allDates.push(date);
      date = date.add({
        days: 1
      });
    }
    return allDates.reduce(groupByWeek, []);
  }, [dayZdt]);
};
exports.useCalendarWeekDays = useCalendarWeekDays;
