// @dhis2-ui/table/build/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "DataTable", {
  enumerable: true,
  get: function () {
    return _index3.DataTable;
  }
});
Object.defineProperty(exports, "DataTableBody", {
  enumerable: true,
  get: function () {
    return _index3.DataTableBody;
  }
});
Object.defineProperty(exports, "DataTableCell", {
  enumerable: true,
  get: function () {
    return _index3.DataTableCell;
  }
});
Object.defineProperty(exports, "DataTableColumnHeader", {
  enumerable: true,
  get: function () {
    return _index3.DataTableColumnHeader;
  }
});
Object.defineProperty(exports, "DataTableFoot", {
  enumerable: true,
  get: function () {
    return _index3.DataTableFoot;
  }
});
Object.defineProperty(exports, "DataTableHead", {
  enumerable: true,
  get: function () {
    return _index3.DataTableHead;
  }
});
Object.defineProperty(exports, "DataTableRow", {
  enumerable: true,
  get: function () {
    return _index3.DataTableRow;
  }
});
Object.defineProperty(exports, "DataTableToolbar", {
  enumerable: true,
  get: function () {
    return _index3.DataTableToolbar;
  }
});
Object.defineProperty(exports, "StackedTable", {
  enumerable: true,
  get: function () {
    return _index2.StackedTable;
  }
});
Object.defineProperty(exports, "StackedTableBody", {
  enumerable: true,
  get: function () {
    return _index2.StackedTableBody;
  }
});
Object.defineProperty(exports, "StackedTableCell", {
  enumerable: true,
  get: function () {
    return _index2.StackedTableCell;
  }
});
Object.defineProperty(exports, "StackedTableCellHead", {
  enumerable: true,
  get: function () {
    return _index2.StackedTableCellHead;
  }
});
Object.defineProperty(exports, "StackedTableFoot", {
  enumerable: true,
  get: function () {
    return _index2.StackedTableFoot;
  }
});
Object.defineProperty(exports, "StackedTableHead", {
  enumerable: true,
  get: function () {
    return _index2.StackedTableHead;
  }
});
Object.defineProperty(exports, "StackedTableRow", {
  enumerable: true,
  get: function () {
    return _index2.StackedTableRow;
  }
});
Object.defineProperty(exports, "StackedTableRowHead", {
  enumerable: true,
  get: function () {
    return _index2.StackedTableRowHead;
  }
});
Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function () {
    return _index.Table;
  }
});
Object.defineProperty(exports, "TableBody", {
  enumerable: true,
  get: function () {
    return _index.TableBody;
  }
});
Object.defineProperty(exports, "TableCell", {
  enumerable: true,
  get: function () {
    return _index.TableCell;
  }
});
Object.defineProperty(exports, "TableCellHead", {
  enumerable: true,
  get: function () {
    return _index.TableCellHead;
  }
});
Object.defineProperty(exports, "TableFoot", {
  enumerable: true,
  get: function () {
    return _index.TableFoot;
  }
});
Object.defineProperty(exports, "TableHead", {
  enumerable: true,
  get: function () {
    return _index.TableHead;
  }
});
Object.defineProperty(exports, "TableRow", {
  enumerable: true,
  get: function () {
    return _index.TableRow;
  }
});
Object.defineProperty(exports, "TableRowHead", {
  enumerable: true,
  get: function () {
    return _index.TableRowHead;
  }
});
var _index = require("./table/index.js");
var _index2 = require("./stacked-table/index.js");
var _index3 = require("./data-table/index.js");
