// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/common.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.delayResponse = exports.dataProviderData = exports.createDecoratorStatefulMultiSelection = exports.createDecoratorCustomDataProvider = exports.StatefulMultiSelectionWrapper = void 0;
Object.defineProperty(exports, "getOrganisationUnitData", {
  enumerable: true,
  get: function () {
    return _getOrganisationUnitData.getOrganisationUnitData;
  }
});
Object.defineProperty(exports, "namespace", {
  enumerable: true,
  get: function () {
    return _namespace.namespace;
  }
});
var _appRuntime = require("@dhis2/app-runtime");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _getOrganisationUnitData = require("./get-organisation-unit-data.js");
var _namespace = require("./namespace.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/* eslint-disable react/display-name */

const delayResponse = (delay, response) => () => new Promise(resolve => setTimeout(() => resolve(response), delay));
exports.delayResponse = delayResponse;
const dataProviderData = exports.dataProviderData = {
  organisationUnits: (_, _ref) => {
    let {
      id,
      params = {}
    } = _ref;
    console.log('id, params', id, params);
    const data = (0, _getOrganisationUnitData.getOrganisationUnitData)(id, params);
    if (!data) {
      return Promise.reject(new Error('404 - Org unit not found'));
    }
    return Promise.resolve(data);
  }
};
const StatefulMultiSelectionWrapper = _ref2 => {
  let {
    children,
    onSelectionChange = () => null
  } = _ref2;
  const [selected, setSelected] = (0, _react.useState)([]);
  return children({
    selected,
    onChange: _ref3 => {
      let {
        selected: newSelected
      } = _ref3;
      setSelected(newSelected);
      onSelectionChange(newSelected);
    }
  });
};
exports.StatefulMultiSelectionWrapper = StatefulMultiSelectionWrapper;
StatefulMultiSelectionWrapper.propTypes = {
  children: _propTypes.default.func.isRequired,
  onSelectionChange: _propTypes.default.func
};
const createDecoratorStatefulMultiSelection = args => {
  return Story => /*#__PURE__*/_react.default.createElement(StatefulMultiSelectionWrapper, {
    onSelectionChange: args === null || args === void 0 ? void 0 : args.onSelectionChange
  }, _ref4 => {
    let {
      selected,
      onChange
    } = _ref4;
    return /*#__PURE__*/_react.default.createElement(Story, {
      selected: selected,
      onChange: onChange
    });
  });
};
exports.createDecoratorStatefulMultiSelection = createDecoratorStatefulMultiSelection;
const createDecoratorCustomDataProvider = args => {
  const data = (args === null || args === void 0 ? void 0 : args.data) || dataProviderData;
  return Story => {
    window.dataProviderData = data;
    return /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
      data: data
    }, /*#__PURE__*/_react.default.createElement(Story, null));
  };
};
exports.createDecoratorCustomDataProvider = createDecoratorCustomDataProvider;
