// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/controlled_expanded.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MissingProps = exports.Controlled = void 0;
var _button = require("@dhis2-ui/button");
var _react = _interopRequireWildcard(require("react"));
var _index = require("../index.js");
var _common = require("./common.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
window.initiallyExpandedPaths = ['/A0000000000/A0000000001'];
window.orgUnitPathToExpand = '/A0000000000/A0000000001';
const Controlled = () => {
  const initiallyExpanded = (0, _index.getAllExpandedOrgUnitPaths)(window.initiallyExpandedPaths);
  const [expanded, setExpanded] = (0, _react.useState)(initiallyExpanded);
  const handleExpand = _ref => {
    let {
      path
    } = _ref;
    if (!expanded.includes(path)) {
      setExpanded([...expanded, path]);
    }
  };
  const handleCollapse = _ref2 => {
    let {
      path
    } = _ref2;
    const pathIndex = expanded.indexOf(path);
    if (pathIndex !== -1) {
      const updatedExpanded = pathIndex === 0 ? expanded.slice(1) : [...expanded.slice(0, pathIndex), ...expanded.slice(pathIndex + 1)];
      setExpanded(updatedExpanded);
    }
  };
  const imperativeToggle = () => {
    if (!expanded.includes('/A0000000000/A0000000001')) {
      // Make sure that all required sub paths are included as well
      const nextPaths = (0, _index.getAllExpandedOrgUnitPaths)([...expanded, window.orgUnitPathToExpand]);
      return setExpanded(nextPaths);
    }
    setExpanded(expanded.filter(v => v !== '/A0000000000/A0000000001'));
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    onChange: () => null,
    name: "Root org unit",
    roots: ['A0000000000'],
    expanded: expanded,
    handleExpand: handleExpand,
    handleCollapse: handleCollapse
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_button.Button, {
    dataTest: "org-unit-toggle",
    onClick: imperativeToggle
  }, "Toggle Org Unit 2"));
};
exports.Controlled = Controlled;
Controlled.decorators = [(0, _common.createDecoratorCustomDataProvider)()];
const MissingProps = () => {
  const expanded = [];
  const handleExpand = () => null;
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    onChange: () => null,
    name: "Root org unit",
    roots: ['A0000000000'],
    expanded: expanded,
    handleExpand: handleExpand
  });
};
exports.MissingProps = MissingProps;
MissingProps.decorators = [(0, _common.createDecoratorCustomDataProvider)()];
