// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/displaying_loading_error.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A0000000001LoadingErrorAutoexpand = exports.A0000000001LoadingError = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _common = require("./common.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const dataProviderDataWithError = {
  organisationUnits: function () {
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    const [, {
      id
    }] = args;
    if (id === 'A0000000001') {
      return new Promise((resolve, reject) => setTimeout(() => {
        reject(new Error('Foobar custom error message'));
      }, 100));
    }
    return _common.dataProviderData.organisationUnits(...args);
  }
};
const A0000000001LoadingError = () => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
  data: dataProviderDataWithError
}, /*#__PURE__*/_react.default.createElement(_common.StatefulMultiSelectionWrapper, null, () => /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
  roots: "A0000000000",
  onChange: () => null
})));
exports.A0000000001LoadingError = A0000000001LoadingError;
const A0000000001LoadingErrorAutoexpand = () => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
  data: dataProviderDataWithError
}, /*#__PURE__*/_react.default.createElement(_common.StatefulMultiSelectionWrapper, null, () => /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
  autoExpandLoadingError: true,
  roots: "A0000000000",
  onChange: () => null
})));
exports.A0000000001LoadingErrorAutoexpand = A0000000001LoadingErrorAutoexpand;
