// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/force_reload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ForceReloading = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireWildcard(require("react"));
var _index = require("../index.js");
var _common = require("./common.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const afterReloadData = {
  organisationUnits: function () {
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    const [, {
      id
    }] = args;
    if (id === 'A0000000000') {
      return (0, _common.delayResponse)(1000, _common.dataProviderData.organisationUnits(...args))();
    }
    if (id === 'A0000000001') {
      return (0, _common.delayResponse)(2200, _common.dataProviderData.organisationUnits(...args))();
    }
    return _common.dataProviderData.organisationUnits(...args);
  }
};
const ForceReloading_ = () => {
  const [forceReload, setForceReload] = (0, _react.useState)(false);
  return /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
    data: afterReloadData
  }, /*#__PURE__*/_react.default.createElement("button", {
    disabled: forceReload,
    "data-test": "reload-all",
    onClick: () => setForceReload(true)
  }, "Force reload tree"), /*#__PURE__*/_react.default.createElement(_common.StatefulMultiSelectionWrapper, null, _ref => {
    let {
      selected,
      onChange
    } = _ref;
    return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
      roots: "A0000000000",
      onChange: onChange,
      selected: selected,
      forceReload: forceReload,
      onChildrenLoaded: data => {
        if (data.A0000000000) {
          setForceReload(false);
        }
      }
    });
  }));
};
var _default = exports.default = {
  title: _common.namespace
};
const ForceReloading = () => /*#__PURE__*/_react.default.createElement(ForceReloading_, null);
exports.ForceReloading = ForceReloading;
