// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/path_based_filtering.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FilteredBy3LevelPathAnd2LevelPath = exports.FilteredBy3LevelPath = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _common = require("./common.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.allUnits = [];
for (let i = 0; i < 7; ++i) {
  window.allUnits.push(`A000000000${i}`);
}
const FilteredBy3LevelPath = () => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
  data: _common.dataProviderData
}, /*#__PURE__*/_react.default.createElement(_common.StatefulMultiSelectionWrapper, null, _ref => {
  let {
    onChange
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    roots: "A0000000000",
    onChange: onChange,
    initiallyExpanded: ['/A0000000000', '/A0000000000/A0000000001', '/A0000000000/A0000000002'],
    filter: ['/A0000000000/A0000000001/A0000000003']
  });
}));
exports.FilteredBy3LevelPath = FilteredBy3LevelPath;
const FilteredBy3LevelPathAnd2LevelPath = () => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
  data: _common.dataProviderData
}, /*#__PURE__*/_react.default.createElement(_common.StatefulMultiSelectionWrapper, null, _ref2 => {
  let {
    onChange
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    roots: "A0000000000",
    onChange: onChange,
    initiallyExpanded: ['/A0000000000', '/A0000000000/A0000000001', '/A0000000000/A0000000002'],
    filter: ['/A0000000000/A0000000001/A0000000003', '/A0000000000/A0000000002']
  });
}));
exports.FilteredBy3LevelPathAnd2LevelPath = FilteredBy3LevelPathAnd2LevelPath;
