// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/single_selection.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SingleSelection = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _common = require("./common.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const data = {
  organisationUnits: function () {
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    const [, {
      id
    }] = args;
    if (id === 'A0000000000') {
      return _common.dataProviderData.organisationUnits(...args).then(response => ({
        ...response,
        children: []
      }));
    }
    if (id === 'A0000000001') {
      return _common.dataProviderData.organisationUnits(...args).then(response => ({
        ...response,
        path: '/A0000000001',
        children: []
      }));
    }
    return Promise.resolve({});
  }
};
const SingleSelection = () => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
  data: data
}, /*#__PURE__*/_react.default.createElement(_common.StatefulMultiSelectionWrapper, null, _ref => {
  let {
    selected,
    onChange
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    singleSelection: true,
    roots: ['A0000000000', 'A0000000001'],
    onChange: onChange,
    selected: selected
  });
}));
exports.SingleSelection = SingleSelection;
