// @dhis2-ui/popover/build/cjs/popover.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnClickOutside = exports.ShiftedArrow = exports.PlacementTop = exports.PlacementRight = exports.PlacementLeft = exports.PlacementBottom = exports.NoArrow = exports.HiddenArrow = exports.Flipped = exports.Default = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _popover = require("./popover.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
const boxStyle = {
  display: 'flex',
  justifyContent: 'center',
  width: 400,
  backgroundColor: 'aliceblue'
};
const referenceElementStyle = {
  width: 100,
  height: 50,
  backgroundColor: 'cadetblue',
  textAlign: 'center',
  padding: 6
};
class PopperInBoxWithCenteredReferenceElement extends _react.Component {
  constructor() {
    super(...arguments);
    _defineProperty(this, "ref", /*#__PURE__*/(0, _react.createRef)());
  }
  render() {
    const {
      paddingTop = 220,
      popoverHeight = 200,
      popoverWidth = 336,
      ...popoverProps
    } = this.props;
    return /*#__PURE__*/_react.default.createElement("div", {
      style: {
        ...boxStyle,
        paddingTop,
        height: paddingTop + 100
      }
    }, /*#__PURE__*/_react.default.createElement("div", {
      style: referenceElementStyle,
      ref: this.ref,
      "data-test": "reference-element"
    }, "Reference element"), /*#__PURE__*/_react.default.createElement(_popover.Popover, _extends({
      reference: this.ref
    }, popoverProps), /*#__PURE__*/_react.default.createElement("div", {
      "data-test": "popover-content",
      style: {
        width: popoverWidth,
        height: popoverHeight
      }
    }, "I am in a box with width: ", popoverWidth, "px and height:", ' ', popoverHeight, "px")));
  }
}
_defineProperty(PopperInBoxWithCenteredReferenceElement, "defaultProps", {
  paddingTop: 220,
  popoverHeight: 200,
  popoverWidth: 336
});
PopperInBoxWithCenteredReferenceElement.propTypes = {
  paddingTop: _propTypes.default.number,
  popoverHeight: _propTypes.default.number,
  popoverWidth: _propTypes.default.number
};
window.onClickOutside = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'Popover',
  component: _popover.Popover
};
const Default = () => /*#__PURE__*/_react.default.createElement(PopperInBoxWithCenteredReferenceElement, null);
exports.Default = Default;
const Flipped = () =>
/*#__PURE__*/
// default viewport-height for flipped popover
// viePort height 400px for diplaced with arrow
_react.default.createElement(PopperInBoxWithCenteredReferenceElement, {
  paddingTop: 160
});
exports.Flipped = Flipped;
const HiddenArrow = () =>
/*#__PURE__*/
// viewPort height 325px
_react.default.createElement(PopperInBoxWithCenteredReferenceElement, {
  paddingTop: 110
});
exports.HiddenArrow = HiddenArrow;
const NoArrow = () => /*#__PURE__*/_react.default.createElement(PopperInBoxWithCenteredReferenceElement, {
  arrow: false
});
exports.NoArrow = NoArrow;
const WithOnClickOutside = () => /*#__PURE__*/_react.default.createElement(PopperInBoxWithCenteredReferenceElement, {
  onClickOutside: window.onClickOutside
});
exports.WithOnClickOutside = WithOnClickOutside;
const PlacementTop = () => /*#__PURE__*/_react.default.createElement(PopperInBoxWithCenteredReferenceElement, {
  popoverHeight: 40,
  popoverWidth: 180,
  paddingTop: 50,
  placement: "top"
});
exports.PlacementTop = PlacementTop;
const PlacementRight = () => /*#__PURE__*/_react.default.createElement(PopperInBoxWithCenteredReferenceElement, {
  popoverHeight: 60,
  popoverWidth: 130,
  paddingTop: 50,
  placement: "right"
});
exports.PlacementRight = PlacementRight;
const PlacementBottom = () => /*#__PURE__*/_react.default.createElement(PopperInBoxWithCenteredReferenceElement, {
  popoverHeight: 40,
  popoverWidth: 180,
  paddingTop: 50,
  placement: "bottom"
});
exports.PlacementBottom = PlacementBottom;
const PlacementLeft = () => /*#__PURE__*/_react.default.createElement(PopperInBoxWithCenteredReferenceElement, {
  popoverHeight: 60,
  popoverWidth: 130,
  paddingTop: 50,
  placement: "left"
});
exports.PlacementLeft = PlacementLeft;
const ShiftedArrow = () => /*#__PURE__*/_react.default.createElement(PopperInBoxWithCenteredReferenceElement, {
  popoverHeight: 160,
  popoverWidth: 130,
  paddingTop: 1,
  placement: "left"
});
exports.ShiftedArrow = ShiftedArrow;
