// @dhis2/ui-forms/build/cjs/SwitchFieldFF/SwitchFieldFF.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.UncheckedWithValue = exports.Unchecked = exports.CheckedWithValue = exports.Checked = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
var _formDecorator = require("../formDecorator.js");
var _index = require("../validators/index.js");
var _SwitchFieldFF = require("./SwitchFieldFF.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
// https://github.com/final-form/react-final-form-arrays/issues/111
const initialValue = ['yes'];
var _default = exports.default = {
  title: 'Testing:SwitchFieldFF',
  decorators: [_formDecorator.formDecorator]
};
const Unchecked = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _SwitchFieldFF.SwitchFieldFF,
  className: "switch",
  name: "switch",
  label: "Label text",
  validate: _index.hasValue,
  required: true,
  type: "checkbox"
});
exports.Unchecked = Unchecked;
const Checked = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _SwitchFieldFF.SwitchFieldFF,
  className: "switch",
  name: "switch",
  label: "Label text",
  initialValue: true,
  type: "checkbox"
});
exports.Checked = Checked;
const UncheckedWithValue = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _SwitchFieldFF.SwitchFieldFF,
  className: "switch",
  name: "switch",
  label: "Label text",
  value: "yes",
  type: "checkbox"
});
exports.UncheckedWithValue = UncheckedWithValue;
const CheckedWithValue = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _SwitchFieldFF.SwitchFieldFF,
  className: "switch",
  name: "switch",
  label: "Label text",
  value: "yes",
  initialValue: initialValue,
  type: "checkbox"
});
exports.CheckedWithValue = CheckedWithValue;
