// @dhis2-ui/header-bar/build/cjs/profile.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _userAvatar = require("@dhis2-ui/user-avatar");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _debugInfoModal = require("./debug-info/debug-info-modal.js");
var _index = _interopRequireDefault(require("./locales/index.js"));
var _useOnDocClick = require("./profile/use-on-doc-click.js");
var _index2 = require("./profile-menu/index.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Profile = _ref => {
  let {
    name,
    email,
    avatarId,
    helpUrl
  } = _ref;
  const [showProfileMenu, setShowProfileMenu] = (0, _react.useState)(false);
  const [showDebugInfoModal, setShowDebugInfoModal] = (0, _react.useState)(false);
  const hideProfileMenu = (0, _react.useCallback)(() => setShowProfileMenu(false), [setShowProfileMenu]);
  const toggleProfileMenu = (0, _react.useCallback)(() => setShowProfileMenu(show => !show), [setShowProfileMenu]);
  const containerRef = (0, _react.useRef)(null);
  (0, _useOnDocClick.useOnDocClick)(containerRef, hideProfileMenu);
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: containerRef,
    "data-test": "headerbar-profile",
    className: "jsx-3442481507" + " " + "headerbar-profile"
  }, /*#__PURE__*/_react.default.createElement("button", {
    onClick: toggleProfileMenu,
    title: _index.default.t('header bar profile'),
    "aria-label": _index.default.t('header bar profile'),
    className: "jsx-3442481507" + " " + "headerbar-profile-btn"
  }, /*#__PURE__*/_react.default.createElement(_userAvatar.UserAvatar, {
    avatarId: avatarId,
    name: name,
    dataTest: "headerbar-profile-icon",
    medium: true
  })), showProfileMenu && /*#__PURE__*/_react.default.createElement(_index2.ProfileMenu, {
    avatarId: avatarId,
    name: name,
    email: email,
    helpUrl: helpUrl,
    hideProfileMenu: hideProfileMenu,
    showDebugInfoModal: () => {
      setShowDebugInfoModal(true);
    }
  }), showDebugInfoModal && /*#__PURE__*/_react.default.createElement(_debugInfoModal.DebugInfoModal, {
    onClose: () => {
      setShowDebugInfoModal(false);
    }
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3442481507"
  }, [".headerbar-profile.jsx-3442481507{position:relative;height:100%;}", ".headerbar-profile-btn.jsx-3442481507{background:transparent;padding:6px;border:0;cursor:pointer;}", ".headerbar-profile-btn.jsx-3442481507:focus{outline:2px solid white;outline-offset:-2px;}", ".headerbar-profile-btn.jsx-3442481507:focus.jsx-3442481507:not(:focus-visible){outline:none;}", ".headerbar-profile-btn.jsx-3442481507:hover{background:#1a557f;}", ".headerbar-profile-btn.jsx-3442481507:active{background:#104067;}"]));
};
Profile.propTypes = {
  name: _propTypes.default.string.isRequired,
  avatarId: _propTypes.default.string,
  email: _propTypes.default.string,
  helpUrl: _propTypes.default.string
};
var _default = exports.default = Profile;
