// @dhis2-ui/css/build/cjs/css-reset/css-reset.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
A tool for adding a global normalization stylesheet into the DOM that applies DHIS2 styles.

- https://github.com/necolas/normalize.css
- https://www.paulirish.com/2012/box-sizing-border-box-ftw/

\`\`\`js
import { CssReset } from '@dhis2/ui'
\`\`\`
`;
const App = _ref => {
  let {
    children
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", null, children);
};
var _default = exports.default = {
  title: 'CSS Reset',
  component: _index.CssReset,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Default = args => /*#__PURE__*/_react.default.createElement(App, null, /*#__PURE__*/_react.default.createElement(_index.CssReset, args), /*#__PURE__*/_react.default.createElement("p", null, "The ", /*#__PURE__*/_react.default.createElement("code", null, "CssReset"), " component injects a global normalization stylesheet into the DOM that sets the DHIS2 application defaults."), /*#__PURE__*/_react.default.createElement("p", null, "This also sets the ", /*#__PURE__*/_react.default.createElement("code", null, "font-family"), " on the ", /*#__PURE__*/_react.default.createElement("code", null, "body"), ' ', "element to the DHIS2 font, which allows it to trickle down to the components as well."), /*#__PURE__*/_react.default.createElement("p", null, "Just include the component in your application, preferably as early as possible to avoid FOUC.", ' '));
exports.Default = Default;
