// @dhis2-ui/button/build/cjs/split-button/split-button.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithIcon = exports.Small = exports.Secondary = exports.RTL = exports.Primary = exports.Large = exports.InitialFocus = exports.Disabled = exports.Destructive = exports.Default = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _react = _interopRequireDefault(require("react"));
var _splitButton = require("./split-button.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
Similar to the dropdown button, but can be triggered independently of opening the contained action list. The main action may be 'Save' and the contained actions may be "Save and add another" and "Save and open".

\`\`\`js
import { SplitButton } from '@dhis2/ui'
\`\`\`
`;
window.onClick = (payload, event) => {
  console.log('onClick payload', payload);
  console.log('onClick event', event);
};
const onClick = function () {
  return window.onClick(...arguments);
};
const DropdownComponent = /*#__PURE__*/_react.default.createElement("span", null, "Dropdown component");
var _default = exports.default = {
  title: 'Split Button',
  component: _splitButton.SplitButton,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  args: {
    name: 'buttonName',
    value: 'buttonValue',
    component: DropdownComponent,
    onClick: onClick,
    children: 'Label me!'
  },
  argTypes: {
    small: {
      ..._uiConstants.sharedPropTypes.sizeArgType
    },
    large: {
      ..._uiConstants.sharedPropTypes.sizeArgType
    },
    primary: {
      ..._uiConstants.sharedPropTypes.buttonVariantArgType
    },
    secondary: {
      ..._uiConstants.sharedPropTypes.buttonVariantArgType
    },
    destructive: {
      ..._uiConstants.sharedPropTypes.buttonVariantArgType
    }
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_splitButton.SplitButton, args);
const Default = exports.Default = Template.bind({});
const Primary = exports.Primary = Template.bind({});
Primary.args = {
  primary: true
};
Primary.parameters = {
  docs: {
    description: {
      story: `_**Note**: The dropdown components in the following examples do not appear in the right place on this page. View the following examples in the 'Canvas' tab for the correct placement._`
    }
  }
};
const Secondary = exports.Secondary = Template.bind({});
Secondary.args = {
  secondary: true
};
const Destructive = exports.Destructive = Template.bind({});
Destructive.props = {
  destructive: true
};
const Disabled = exports.Disabled = Template.bind({});
Disabled.args = {
  disabled: true
};
const Small = exports.Small = Template.bind({});
Small.args = {
  small: true
};
const Large = exports.Large = Template.bind({});
Large.args = {
  large: true
};
const InitialFocus = exports.InitialFocus = Template.bind({});
InitialFocus.args = {
  initialFocus: true
};
// Disable this on docs page because grabbing focus repeatedly is annoying
InitialFocus.parameters = {
  docs: {
    disable: true
  }
};
const WithIcon = exports.WithIcon = Template.bind({});
WithIcon.args = {
  children: 'Children',
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component"),
  icon: /*#__PURE__*/_react.default.createElement("div", null, "Icon")
};
const RTL = args => /*#__PURE__*/_react.default.createElement("div", {
  dir: "rtl"
}, /*#__PURE__*/_react.default.createElement(_splitButton.SplitButton, args));
exports.RTL = RTL;
RTL.args = {
  children: 'RTL text'
};
