// @dhis2-ui/button/build/cjs/split-button/split-button.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnClick = exports.WithInitialFocus = exports.WithIcon = exports.WithDisabled = exports.WithChildren = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _splitButton = require("./split-button.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onClick = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'SplitButton'
};
const Default = () => /*#__PURE__*/_react.default.createElement(_splitButton.SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component")
}, "Label me!");
exports.Default = Default;
const WithOnClick = () => /*#__PURE__*/_react.default.createElement(_splitButton.SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component"),
  onClick: window.onClick
}, "Label me!");
exports.WithOnClick = WithOnClick;
const WithChildren = () => /*#__PURE__*/_react.default.createElement(_splitButton.SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component")
}, "I am a child");
exports.WithChildren = WithChildren;
const WithIcon = () => /*#__PURE__*/_react.default.createElement(_splitButton.SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component"),
  icon: /*#__PURE__*/_react.default.createElement("div", null, "Icon")
}, "Children");
exports.WithIcon = WithIcon;
const WithInitialFocus = () => /*#__PURE__*/_react.default.createElement(_splitButton.SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component"),
  initialFocus: true
}, "Children");
exports.WithInitialFocus = WithInitialFocus;
const WithDisabled = () => /*#__PURE__*/_react.default.createElement(_splitButton.SplitButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component"),
  disabled: true
}, "Children");
exports.WithDisabled = WithDisabled;
