// @dhis2-ui/sharing-dialog/build/es/cascade-sharing/update-info.js
import { useDataQuery } from '@dhis2/app-runtime';
import { NoticeBox } from '@dhis2-ui/notice-box';
import PropTypes from 'prop-types';
import React from 'react';
import { getVisualizationsCount, getInfoMessage } from './helpers.js';
const query = {
  dashboard: {
    resource: 'dashboards',
    id: _ref => {
      let {
        id
      } = _ref;
      return id;
    },
    params: {
      fields: 'dashboardItems[type]'
    }
  }
};
export const UpdateInfo = _ref2 => {
  let {
    id,
    entityAmount
  } = _ref2;
  const {
    data,
    loading
  } = useDataQuery(query, {
    variables: {
      id
    }
  });
  if (loading) {
    return null;
  }
  const {
    dashboard
  } = data;
  const visualizationsCount = getVisualizationsCount(dashboard.dashboardItems);
  if (!visualizationsCount) {
    return null;
  }
  const message = getInfoMessage({
    visualizationsCount,
    entityAmount
  });
  return /*#__PURE__*/React.createElement(NoticeBox, null, message);
};
UpdateInfo.propTypes = {
  entityAmount: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired
};
