// @dhis2/ui-forms/build/es/RadioFieldFF/RadioFieldFF.e2e.stories.js
import React from 'react';
import { Field } from 'react-final-form';
import { FieldGroupFF } from '../FieldGroupFF/FieldGroupFF.js';
import { formDecorator } from '../formDecorator.js';
import { hasValue } from '../validators/index.js';
import { RadioFieldFF } from './RadioFieldFF.js';
export default {
  title: 'Testing:RadioFieldFF',
  decorators: [formDecorator]
};
export const RequiredAndNoSelectedValue = () => /*#__PURE__*/React.createElement(FieldGroupFF, {
  name: "choice"
}, /*#__PURE__*/React.createElement(Field, {
  type: "radio",
  component: RadioFieldFF,
  name: "choice",
  label: "One",
  value: "one",
  validate: hasValue,
  required: true
}), /*#__PURE__*/React.createElement(Field, {
  type: "radio",
  component: RadioFieldFF,
  name: "choice",
  label: "Two",
  value: "two",
  validate: hasValue,
  required: true
}), /*#__PURE__*/React.createElement(Field, {
  type: "radio",
  component: RadioFieldFF,
  name: "choice",
  label: "Three",
  value: "three",
  validate: hasValue,
  required: true
}));
