// @dhis2-ui/radio/build/cjs/radio.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnFocus = exports.WithOnChange = exports.WithLabel = exports.WithInitialFocusAndOnBlur = exports.WithInitialFocus = exports.WithDisabled = void 0;
var _react = _interopRequireDefault(require("react"));
var _radio = require("./radio.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onChange = window.Cypress && window.Cypress.cy.stub();
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'Radio'
};
const WithOnChange = () => /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  name: "Ex",
  label: "Radio",
  value: "default",
  onChange: window.onChange
});
exports.WithOnChange = WithOnChange;
const WithInitialFocusAndOnBlur = () => /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  name: "Ex",
  label: "Radio",
  value: "default",
  initialFocus: true,
  onBlur: window.onBlur
});
exports.WithInitialFocusAndOnBlur = WithInitialFocusAndOnBlur;
const WithOnFocus = () => /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  initialFocus: true,
  name: "Ex",
  label: "Radio",
  value: "default",
  onFocus: window.onFocus
});
exports.WithOnFocus = WithOnFocus;
const WithDisabled = () => /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  name: "Ex",
  label: "Radio",
  value: "default",
  disabled: true
});
exports.WithDisabled = WithDisabled;
const WithLabel = () => /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  name: "Ex",
  label: "The label",
  value: "default"
});
exports.WithLabel = WithLabel;
const WithInitialFocus = () => /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  name: "Ex",
  label: "The label",
  value: "default",
  initialFocus: true
});
exports.WithInitialFocus = WithInitialFocus;
