// @dhis2-ui/table/build/cjs/stacked-table/content-with-title.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ContentWithTitle = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ContentWithTitle = _ref => {
  let {
    title,
    children
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_react.Fragment, null, title && /*#__PURE__*/_react.default.createElement("span", {
    className: _style.default.dynamic([["215664166", [_uiConstants.colors.grey700]]]) + " " + "title"
  }, title), /*#__PURE__*/_react.default.createElement("span", {
    className: _style.default.dynamic([["215664166", [_uiConstants.colors.grey700]]]) + " " + "content"
  }, children), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "215664166",
    dynamic: [_uiConstants.colors.grey700]
  }, [`.title.__jsx-style-dynamic-selector{display:block;white-space:normal;min-height:24px;font-size:13px;line-height:16px;padding:8px 0 4px;font-weight:normal;color:${_uiConstants.colors.grey700};}`, ".content.__jsx-style-dynamic-selector{display:block;padding:0 0 8px 0;font-size:14px;line-height:18px;}", ".content.__jsx-style-dynamic-selector:first-child{padding-top:8px;padding-bottom:8px;}"]));
};
exports.ContentWithTitle = ContentWithTitle;
ContentWithTitle.propTypes = {
  children: _propTypes.default.node.isRequired,
  title: _propTypes.default.string
};
