// @dhis2-ui/table/build/cjs/stacked-table/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "StackedTable", {
  enumerable: true,
  get: function () {
    return _stackedTable.StackedTable;
  }
});
Object.defineProperty(exports, "StackedTableBody", {
  enumerable: true,
  get: function () {
    return _stackedTableBody.StackedTableBody;
  }
});
Object.defineProperty(exports, "StackedTableCell", {
  enumerable: true,
  get: function () {
    return _stackedTableCell.StackedTableCell;
  }
});
Object.defineProperty(exports, "StackedTableCellHead", {
  enumerable: true,
  get: function () {
    return _stackedTableCellHead.StackedTableCellHead;
  }
});
Object.defineProperty(exports, "StackedTableFoot", {
  enumerable: true,
  get: function () {
    return _stackedTableFoot.StackedTableFoot;
  }
});
Object.defineProperty(exports, "StackedTableHead", {
  enumerable: true,
  get: function () {
    return _stackedTableHead.StackedTableHead;
  }
});
Object.defineProperty(exports, "StackedTableRow", {
  enumerable: true,
  get: function () {
    return _stackedTableRow.StackedTableRow;
  }
});
Object.defineProperty(exports, "StackedTableRowHead", {
  enumerable: true,
  get: function () {
    return _stackedTableRowHead.StackedTableRowHead;
  }
});
var _stackedTable = require("./stacked-table.js");
var _stackedTableHead = require("./stacked-table-head.js");
var _stackedTableBody = require("./stacked-table-body.js");
var _stackedTableFoot = require("./stacked-table-foot.js");
var _stackedTableCell = require("./stacked-table-cell.js");
var _stackedTableCellHead = require("./stacked-table-cell-head.js");
var _stackedTableRow = require("./stacked-table-row.js");
var _stackedTableRowHead = require("./stacked-table-row-head.js");
