// @dhis2-ui/table/build/cjs/stacked-table/stacked-table-row.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StackedTableRow = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _addColNumToChildren = require("./add-col-num-to-children.js");
var _supplyHeaderLabelsToChildren = require("./supply-header-labels-to-children.js");
var _tableContext = require("./table-context.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const StackedTableRow = _ref => {
  let {
    children,
    className,
    dataTest = 'dhis2-uicore-stackedtablerow'
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("tr", {
    "data-test": dataTest,
    className: _style.default.dynamic([["2003961452", [_uiConstants.colors.grey300, _uiConstants.colors.white]]]) + " " + (className || "")
  }, /*#__PURE__*/_react.default.createElement(_tableContext.Consumer, null, _ref2 => {
    let {
      headerLabels
    } = _ref2;
    return (0, _supplyHeaderLabelsToChildren.supplyHeaderLabelsToChildren)(headerLabels, (0, _addColNumToChildren.addColNumToChildren)(children));
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2003961452",
    dynamic: [_uiConstants.colors.grey300, _uiConstants.colors.white]
  }, [`tr.__jsx-style-dynamic-selector{min-height:45px;display:block;border:1px solid ${_uiConstants.colors.grey300};}`, `tr.__jsx-style-dynamic-selector:nth-child(even){background:${_uiConstants.colors.white};}`, "thead>tr.__jsx-style-dynamic-selector,tbody>tr.__jsx-style-dynamic-selector{min-height:36px;}", "tr.__jsx-style-dynamic-selector+tr.__jsx-style-dynamic-selector{margin-top:32px;}"]));
};
exports.StackedTableRow = StackedTableRow;
StackedTableRow.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
