// @dhis2-ui/table/build/cjs/stacked-table/stacked-table-cell-head.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StackedTableCellHead = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const StackedTableCellHead = _ref => {
  let {
    children = '',
    className,
    colSpan,
    dataTest = 'dhis2-uicore-stackedtablecellhead',
    rowSpan
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("th", {
    colSpan: colSpan,
    rowSpan: rowSpan,
    "data-test": dataTest,
    className: _style.default.dynamic([["178822310", [_uiConstants.colors.grey300]]]) + " " + (className || "")
  }, children && /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["178822310", [_uiConstants.colors.grey300]]])
  }, children), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "178822310",
    dynamic: [_uiConstants.colors.grey300]
  }, [`th.__jsx-style-dynamic-selector{border-bottom:1px solid ${_uiConstants.colors.grey300};padding:0 12px;}`, "div.__jsx-style-dynamic-selector{min-height:36px;}"]));
};
exports.StackedTableCellHead = StackedTableCellHead;
StackedTableCellHead.propTypes = {
  children: _propTypes.default.string,
  className: _propTypes.default.string,
  colSpan: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  rowSpan: _propTypes.default.string
};
