// @dhis2/ui-forms/build/cjs/RadioFieldFF/RadioFieldFF.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RequiredAndNoSelectedValue = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
var _FieldGroupFF = require("../FieldGroupFF/FieldGroupFF.js");
var _formDecorator = require("../formDecorator.js");
var _index = require("../validators/index.js");
var _RadioFieldFF = require("./RadioFieldFF.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'Testing:RadioFieldFF',
  decorators: [_formDecorator.formDecorator]
};
const RequiredAndNoSelectedValue = () => /*#__PURE__*/_react.default.createElement(_FieldGroupFF.FieldGroupFF, {
  name: "choice"
}, /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  type: "radio",
  component: _RadioFieldFF.RadioFieldFF,
  name: "choice",
  label: "One",
  value: "one",
  validate: _index.hasValue,
  required: true
}), /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  type: "radio",
  component: _RadioFieldFF.RadioFieldFF,
  name: "choice",
  label: "Two",
  value: "two",
  validate: _index.hasValue,
  required: true
}), /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  type: "radio",
  component: _RadioFieldFF.RadioFieldFF,
  name: "choice",
  label: "Three",
  value: "three",
  validate: _index.hasValue,
  required: true
}));
exports.RequiredAndNoSelectedValue = RequiredAndNoSelectedValue;
