// @dhis2-ui/select/build/es/single-select-field/single-select-field.e2e.stories.js
import _JSXStyle from "styled-jsx/style";
import React from 'react';
import { SingleSelectOption } from '../index.js';
import { SingleSelectField } from './index.js';
export default {
  title: 'SingleSelectField'
};
export const WithLabel = () => /*#__PURE__*/React.createElement(SingleSelectField, {
  label: "The label"
}, /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "3",
  label: "three"
}));
export const WithHelpText = () => /*#__PURE__*/React.createElement(SingleSelectField, {
  helpText: "The help text"
}, /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "3",
  label: "three"
}));
export const WithValidationText = () => /*#__PURE__*/React.createElement(SingleSelectField, {
  validationText: "The validation text"
}, /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "3",
  label: "three"
}));
export const WithLabelAndRequiredStatus = () => /*#__PURE__*/React.createElement(SingleSelectField, {
  label: "The label",
  required: true
}, /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  value: "3",
  label: "three"
}));
export const WithClearableAndSelectedOption = () => /*#__PURE__*/React.createElement(SingleSelectField, {
  selected: "1",
  clearable: true
}, /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "1",
  value: "1",
  label: "One"
}));
export const WithFilterable = () => /*#__PURE__*/React.createElement(SingleSelectField, {
  filterable: true
}, /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "1",
  value: "1",
  label: "One"
}));
export const WithLoading = () => /*#__PURE__*/React.createElement(SingleSelectField, {
  loading: true
});
export const WithoutOptions = () => /*#__PURE__*/React.createElement(SingleSelectField, null);
export const WithHiddenParent = () => {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    className: "jsx-3657609608" + " " + "container"
  }, /*#__PURE__*/React.createElement("div", {
    className: "jsx-3657609608" + " " + "hoverSpan"
  }, /*#__PURE__*/React.createElement("span", {
    className: "jsx-3657609608"
  }, "Hover over me"), /*#__PURE__*/React.createElement("div", {
    className: "jsx-3657609608" + " " + "hiddenSelect"
  }, /*#__PURE__*/React.createElement(SingleSelectField, {
    dataTest: "hoverable-select-field",
    label: "Choose something",
    onChange: () => {}
  }, /*#__PURE__*/React.createElement(SingleSelectOption, {
    value: "1",
    label: "one"
  }), /*#__PURE__*/React.createElement(SingleSelectOption, {
    value: "2",
    label: "two"
  }))))), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "3657609608"
  }, [".container.jsx-3657609608{width:100%;height:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;font-size:1rem;}", ".hiddenSelect.jsx-3657609608{display:none;}", ".hoverSpan.jsx-3657609608:hover .hiddenSelect.jsx-3657609608{display:block;}"]));
};
