// @dhis2/multi-calendar-dates/build/cjs/utils/helpers.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.padWithZeroes = exports.isCustomCalendar = exports.getCustomCalendarIfExists = exports.formatDate = exports.extractAndValidateDateString = exports.capitalize = void 0;
var _customCalendars2 = require("../custom-calendars");
var _extractDatePartsFromDateString = require("./extract-date-parts-from-date-string");
var _getNowInCalendar = _interopRequireDefault(require("./getNowInCalendar"));
var _validateDateString = require("./validate-date-string");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const isCustomCalendar = calendar => !!_customCalendars2.customCalendars[calendar];
exports.isCustomCalendar = isCustomCalendar;
const padWithZeroes = function (number) {
  let count = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 2;
  return String(number).padStart(count, '0');
};
exports.padWithZeroes = padWithZeroes;
const formatDate = (date, dayType, format) => {
  var _date$eraYear;
  const year = (_date$eraYear = date.eraYear) !== null && _date$eraYear !== void 0 ? _date$eraYear : date.year;
  const month = padWithZeroes(date.month);
  let day = date.day;
  if (dayType === 'endOfMonth') {
    day = date.daysInMonth;
  } else if (dayType === 'startOfMonth') {
    day = 1;
  }
  const dayString = padWithZeroes(day);
  return format === 'DD-MM-YYYY' ? `${dayString}-${month}-${year}` : `${year}-${month}-${dayString}`;
};

// capitalize method taking into account locales that have different way of lower/upper case
// based on https://stackoverflow.com/a/53930826
exports.formatDate = formatDate;
const capitalize = function (_ref) {
  let [firstLetter = '', ...rest] = _ref;
  let locale = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'en';
  return [firstLetter.toLocaleUpperCase(locale), ...rest].join('');
};
exports.capitalize = capitalize;
const getCustomCalendarIfExists = calendar => {
  var _customCalendars;
  const isCustom = isCustomCalendar(calendar);
  if (!isCustom) {
    return calendar;
  }
  const customCalendar = (_customCalendars = _customCalendars2.customCalendars[calendar]) === null || _customCalendars === void 0 ? void 0 : _customCalendars.calendar;
  if (!customCalendar) {
    throw new Error(`No implemenation found for custom calendar ${calendar}`);
  }
  return customCalendar;
};
exports.getCustomCalendarIfExists = getCustomCalendarIfExists;
const extractAndValidateDateString = (date, options) => {
  if (!date) {
    return getCurrentDateResult(options);
  }
  const validation = (0, _validateDateString.validateDateString)(date, options);
  if (validation.isValid) {
    return getValidDateResult(date, validation, options);
  } else {
    return getInvalidDateResult(options, validation.errorMessage);
  }
};
exports.extractAndValidateDateString = extractAndValidateDateString;
const getCurrentDateResult = options => {
  const {
    year,
    month,
    day
  } = (0, _getNowInCalendar.default)(options.calendar, options.timeZone);
  return {
    year,
    month,
    day,
    isValid: true
  };
};
const getValidDateResult = (date, validation, options) => {
  const {
    year,
    month,
    day,
    format
  } = (0, _extractDatePartsFromDateString.extractDatePartsFromDateString)(date);
  let result = {
    year,
    month,
    day,
    format,
    isValid: validation.isValid,
    warningMessage: validation.warningMessage,
    errorMessage: validation.errorMessage
  };
  if (options.calendar === 'ethiopic') {
    result = adjustForEthiopicCalendar(result);
  }
  return result;
};
const getInvalidDateResult = (options, errorMessage) => {
  const {
    year,
    month,
    day
  } = (0, _getNowInCalendar.default)(options.calendar, options.timeZone);
  return {
    year,
    month,
    day,
    isValid: false,
    errorMessage
  };
};
const adjustForEthiopicCalendar = result => {
  result.era = 'era1';
  result.eraYear = result.year;
  delete result.year;
  return result;
};
