// @dhis2-ui/table/build/cjs/data-table/table-elements/table-header-cell/table-header-cell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableHeaderCell = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = require("@dhis2/prop-types");
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes2 = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _tableHeaderCellStyles = _interopRequireDefault(require("./table-header-cell.styles.js"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const AUTO = 'auto';
const TableHeaderCell = exports.TableHeaderCell = /*#__PURE__*/(0, _react.forwardRef)((_ref, ref) => {
  let {
    active,
    align = 'left',
    backgroundColor,
    bordered,
    children,
    className,
    colSpan,
    dataTest = 'dhis2-uicore-tablecell',
    error,
    fixed,
    large,
    left = AUTO,
    muted,
    rowSpan,
    role,
    scope,
    staticStyle,
    top = AUTO,
    valid,
    width = AUTO,
    onClick,
    ...props
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("th", _extends({}, props, {
    ref: ref,
    colSpan: colSpan,
    rowSpan: rowSpan,
    onClick: onClick,
    "data-test": dataTest,
    role: role,
    scope: scope,
    className: `jsx-${_tableHeaderCellStyles.default.__hash}` + " " + _style.default.dynamic([["2588011372", [left, top, align, width, backgroundColor || _uiConstants.colors.grey200, backgroundColor || _uiConstants.colors.grey300, backgroundColor || _uiConstants.colors.grey300, backgroundColor || _uiConstants.colors.grey200]]]) + " " + ((0, _classnames.default)(className, {
      active,
      bordered,
      error,
      fixed,
      fixedHorizontally: fixed && left !== AUTO,
      large,
      muted,
      staticStyle,
      valid
    }) || "")
  }), children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: _tableHeaderCellStyles.default.__hash
  }, _tableHeaderCellStyles.default), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2588011372",
    dynamic: [left, top, align, width, backgroundColor || _uiConstants.colors.grey200, backgroundColor || _uiConstants.colors.grey300, backgroundColor || _uiConstants.colors.grey300, backgroundColor || _uiConstants.colors.grey200]
  }, [`th.__jsx-style-dynamic-selector{inset-inline-start:${left};top:${top};text-align:${align};width:${width};background-color:${backgroundColor || _uiConstants.colors.grey200};}`, `thead th.fixedHorizontally.__jsx-style-dynamic-selector{background-color:${backgroundColor || _uiConstants.colors.grey300};}`, `tbody>tr:hover>th.__jsx-style-dynamic-selector:not(.staticStyle),tfoot>tr:hover>th.__jsx-style-dynamic-selector:not(.staticStyle){background-color:${backgroundColor || _uiConstants.colors.grey300};}`, `tbody>tr:active>th.__jsx-style-dynamic-selector:not(.staticStyle){background-color:${backgroundColor || _uiConstants.colors.grey200};}`]));
});
TableHeaderCell.displayName = 'TableHeaderCell';
const stylePropType = (0, _propTypes.mutuallyExclusive)(['valid', 'error', 'muted'], _propTypes2.default.bool);
TableHeaderCell.propTypes = {
  /** To toggle border color, for example for editing */
  active: _propTypes2.default.bool,
  align: _propTypes2.default.oneOf(['left', 'center', 'right']),
  /** Sets background color of the cell. Disables dynamic background colors from active, hover, and selected states */
  backgroundColor: _propTypes2.default.string,
  bordered: _propTypes2.default.bool,
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  colSpan: _propTypes2.default.string,
  dataTest: _propTypes2.default.string,
  /** Mutually exclusive with muted and valid */
  error: stylePropType,
  fixed: _propTypes2.default.bool,
  large: _propTypes2.default.bool,
  /** Left or top required when fixed */
  left: (0, _propTypes.requiredIf)(props => props.fixed && !props.top, _propTypes2.default.string),
  /** Mutually exclusive with error and valid */
  muted: stylePropType,
  role: _propTypes2.default.string,
  rowSpan: _propTypes2.default.string,
  scope: _propTypes2.default.string,
  /** Surpress hover and active event styles */
  staticStyle: _propTypes2.default.bool,
  /** Left or top required when fixed */
  top: (0, _propTypes.requiredIf)(props => props.fixed && !props.left, _propTypes2.default.string),
  /** Mutually exclusive with error and muted */
  valid: stylePropType,
  width: _propTypes2.default.string,
  onClick: _propTypes2.default.func
};
