// @dhis2-ui/selector-bar/build/es/selector-bar-item/selector-bar-item.e2e.stories.js
import { SingleSelectField, SingleSelectOption } from '@dhis2-ui/select';
import React, { useState } from 'react';
import { SelectorBarItem } from './selector-bar-item.js';
const options = [{
  value: 'value 1',
  label: 'label 1'
}, {
  value: 'value 2',
  label: 'label 2'
}, {
  value: 'value 3',
  label: 'label 3'
}];
const createStateDecorator = args => {
  return fn => {
    const [selected, setSelected] = useState((args === null || args === void 0 ? void 0 : args.selected) || '');
    const [open, setOpen] = useState((args === null || args === void 0 ? void 0 : args.open) || false);
    return fn({
      selected,
      open,
      setSelected,
      setOpen
    });
  };
};
const createStory = () => (_, _ref) => {
  var _options$find;
  let {
    open,
    setOpen,
    selected,
    setSelected
  } = _ref;
  return /*#__PURE__*/React.createElement(SelectorBarItem, {
    label: "Selection bar item",
    noValueMessage: "No value message",
    open: open,
    setOpen: setOpen,
    value: ((_options$find = options.find(_ref2 => {
      let {
        value
      } = _ref2;
      return value === selected;
    })) === null || _options$find === void 0 ? void 0 : _options$find.label) || ''
  }, /*#__PURE__*/React.createElement(SingleSelectField, {
    selected: selected,
    onChange: _ref3 => {
      let {
        selected: nextSelected
      } = _ref3;
      setSelected(nextSelected);
      setOpen(false);
    }
  }, options.map(_ref4 => {
    let {
      value,
      label
    } = _ref4;
    return /*#__PURE__*/React.createElement(SingleSelectOption, {
      key: value,
      value: value,
      label: label
    });
  })));
};
export default {
  title: 'SelectorBarItem'
};
export const ClosedNoValue = createStory();
ClosedNoValue.decorators = [createStateDecorator()];
export const OpenNoValue = createStory();
OpenNoValue.decorators = [createStateDecorator({
  open: true
})];
export const ClosedWithValue = createStory();
ClosedWithValue.decorators = [createStateDecorator({
  selected: 'value 1'
})];
export const OpenWithValue = createStory();
OpenWithValue.decorators = [createStateDecorator({
  open: true,
  selected: 'value 1'
})];
