// @dhis2-ui/node/build/cjs/node.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithIcon = exports.WithComponent = exports.OpenWithOnClose = exports.OpenWithChildren = exports.ClosedWithOnOpen = exports.ClosedWithChildren = void 0;
var _react = _interopRequireDefault(require("react"));
var _node = require("./node.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onClose = window.Cypress && window.Cypress.cy.stub();
window.onOpen = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'Node'
};
const OpenWithOnClose = () => /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: true,
  onClose: window.onClose,
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component")
}, "Children");
exports.OpenWithOnClose = OpenWithOnClose;
const ClosedWithOnOpen = () => /*#__PURE__*/_react.default.createElement(_node.Node, {
  onOpen: window.onOpen,
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component")
}, "Children");
exports.ClosedWithOnOpen = ClosedWithOnOpen;
const ClosedWithChildren = () => /*#__PURE__*/_react.default.createElement(_node.Node, {
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component")
}, "I am a child");
exports.ClosedWithChildren = ClosedWithChildren;
const OpenWithChildren = () => /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: true,
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component")
}, "I am a child");
exports.OpenWithChildren = OpenWithChildren;
const WithComponent = () => /*#__PURE__*/_react.default.createElement(_node.Node, {
  component: /*#__PURE__*/_react.default.createElement("div", null, "I am a component")
}, "Children");
exports.WithComponent = WithComponent;
const WithIcon = () => /*#__PURE__*/_react.default.createElement(_node.Node, {
  component: /*#__PURE__*/_react.default.createElement("div", null, "Component"),
  icon: /*#__PURE__*/_react.default.createElement("div", null, "Icon")
}, "Children");
exports.WithIcon = WithIcon;
