// @dhis2-ui/node/build/cjs/node.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports._2LevelsOpen = exports.TextLeaves = exports.RTL = exports.MultipleRoots = exports.CustomIcon = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _checkbox = require("@dhis2-ui/checkbox");
var _loader = require("@dhis2-ui/loader");
var _react = _interopRequireDefault(require("react"));
var _node = require("./node.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const description = `
A tool for creating hierarchical, multi-level selectors.  Applied in the Organisation Unit Tree component, for example.

\`\`\`js
import { Node } from '@dhis2/ui'
\`\`\`
`;
const say = something => () => alert(something);
window.onOpen = (payload, event) => {
  console.log('onOpen payload', payload);
  console.log('onOpen event', event);
};
window.onClose = (payload, event) => {
  console.log('onClose payload', payload);
  console.log('onClose event', event);
};
const onOpen = function () {
  return window.onOpen(...arguments);
};
const onClose = function () {
  return window.onClose(...arguments);
};
var _default = exports.default = {
  title: 'Node',
  component: _node.Node,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const CustomIcon = args => /*#__PURE__*/_react.default.createElement(_node.Node, args);
exports.CustomIcon = CustomIcon;
CustomIcon.args = {
  icon: /*#__PURE__*/_react.default.createElement(_loader.CircularLoader, {
    small: true
  }),
  open: false,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label 1.1",
    value: "foobar",
    name: "l1.1",
    onChange: say('checkbox 1.1 clicked'),
    checked: false
  })
};
CustomIcon.storyName = 'Custom icon';
const MultipleRoots = args => /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_node.Node, _extends({}, args, {
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label 1.1",
    value: "foobar",
    name: "l1.1",
    onChange: say('checkbox 1.1 clicked'),
    checked: false
  })
}), /*#__PURE__*/_react.default.createElement("span", null, "Placeholder content")), /*#__PURE__*/_react.default.createElement(_node.Node, _extends({}, args, {
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label 1.2",
    value: "foobar",
    name: "l1.2",
    onChange: say('checkbox 1.2 clicked'),
    checked: false
  })
}), /*#__PURE__*/_react.default.createElement("span", null, "Placeholder content")));
exports.MultipleRoots = MultipleRoots;
MultipleRoots.args = {
  open: false,
  onOpen,
  onClose
};
MultipleRoots.storyName = 'Multiple roots';
const _2LevelsOpen = () => /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: true,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    value: "foobar",
    name: "l1.1",
    onChange: say('Check 1.1'),
    checked: false
  })
}, /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: true,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    value: "foobar",
    name: "l2.1",
    onChange: say('Check 2.1'),
    checked: false
  })
}, /*#__PURE__*/_react.default.createElement(_node.Node, {
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l3.1",
    value: "foobar",
    onChange: say('Check 3.1'),
    checked: false
  })
}), /*#__PURE__*/_react.default.createElement(_node.Node, {
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l3.2",
    onChange: say('Check 3.2'),
    value: "foobar",
    checked: false
  })
}), /*#__PURE__*/_react.default.createElement(_node.Node, {
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l3.3",
    onChange: say('Check 3.3'),
    checked: false,
    value: "foobar"
  })
})), /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: false,
  onOpen: onOpen,
  onClose: say('close tree 2.2'),
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l3.1",
    onChange: say('Check 2.2'),
    value: "foobar",
    checked: false
  })
}, /*#__PURE__*/_react.default.createElement(_node.Node, {
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l3.4",
    value: "foobar",
    onChange: say('Check 3.4'),
    checked: false
  })
}), /*#__PURE__*/_react.default.createElement(_node.Node, {
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l3.5",
    onChange: say('Check 3.5'),
    value: "foobar",
    checked: false
  })
}), /*#__PURE__*/_react.default.createElement(_node.Node, {
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l3.6",
    onChange: say('Check 3.6'),
    value: "foobar",
    checked: false
  })
})), /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: false,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l2.3",
    value: "foobar",
    onChange: say('Check 2.3'),
    checked: false
  })
}), /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: false,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l2.4",
    onChange: say('Check 2.4'),
    value: "foobar",
    checked: false
  })
}), /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: false,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l2.5",
    value: "foobar",
    onChange: say('Check 2.5'),
    checked: false
  })
}), /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: false,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    label: "Node label",
    name: "l2.5",
    value: "foobar",
    onChange: say('Check 2.5'),
    checked: false
  })
}, false && 'Foo'));
exports._2LevelsOpen = _2LevelsOpen;
_2LevelsOpen.storyName = '2 Levels open';
const TextLeaves = () => /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-3948139656"
}, /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: true,
  onOpen: onOpen,
  onClose: onClose,
  arrowTopOffset: "10px",
  component: /*#__PURE__*/_react.default.createElement("h2", {
    className: "jsx-3948139656"
  }, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr")
}, "Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.", /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-3948139656" + " " + "sub-tree sub-tree--open"
}, /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: true,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement("h2", {
    className: "jsx-3948139656"
  }, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr")
}, "Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.")), /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-3948139656" + " " + "sub-tree"
}, /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: false,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement("h2", {
    className: "jsx-3948139656"
  }, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr")
}, /*#__PURE__*/_react.default.createElement("span", {
  className: "jsx-3948139656"
}, "Dummy content"))), /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-3948139656" + " " + "sub-tree"
}, /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: false,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement("h2", {
    className: "jsx-3948139656"
  }, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr")
}, /*#__PURE__*/_react.default.createElement("span", {
  className: "jsx-3948139656"
}, "Dummy content"))), /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-3948139656" + " " + "sub-tree"
}, /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: false,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement("h2", {
    className: "jsx-3948139656"
  }, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr")
}, /*#__PURE__*/_react.default.createElement("span", {
  className: "jsx-3948139656"
}, "Dummy content"))), /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-3948139656" + " " + "sub-tree"
}, /*#__PURE__*/_react.default.createElement(_node.Node, {
  open: false,
  onOpen: onOpen,
  onClose: onClose,
  component: /*#__PURE__*/_react.default.createElement("h2", {
    className: "jsx-3948139656"
  }, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr")
}, /*#__PURE__*/_react.default.createElement("span", {
  className: "jsx-3948139656"
}, "Dummy content")))), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "3948139656"
}, ["div.jsx-3948139656{font-size:16px;line-height:24px;}", "h2.jsx-3948139656{font-size:24px;line-height:32px;margin:0 0 10px;}", ".sub-tree.jsx-3948139656{margin-top:10px;}", ".sub-tree--open.jsx-3948139656{margin-bottom:20px;}", "div.jsx-3948139656 .tree__arrow > span{padding-top:4px;}", "div.jsx-3948139656 .tree__arrow.open:after{top:17px;height:calc(100% - 28px);}"]));
exports.TextLeaves = TextLeaves;
TextLeaves.storyName = 'Text leaves';
const RTL = args => /*#__PURE__*/_react.default.createElement("div", {
  dir: "rtl"
}, /*#__PURE__*/_react.default.createElement(MultipleRoots, args));
exports.RTL = RTL;
