// @dhis2-ui/header-bar/build/cjs/profile-menu/profile-header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ProfileHeader = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _appRuntime = require("@dhis2/app-runtime");
var _userAvatar = require("@dhis2-ui/user-avatar");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _joinPath = require("../join-path.js");
var _index = _interopRequireDefault(require("../locales/index.js"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ProfileName = _ref => {
  let {
    children
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": "headerbar-profile-username",
    className: "jsx-2223023701"
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2223023701"
  }, ["div.jsx-2223023701{margin-bottom:3px;font-size:16px;line-height:19px;}"]));
};
ProfileName.propTypes = {
  children: _propTypes.default.string
};
const ProfileEmail = _ref2 => {
  let {
    children
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": "headerbar-profile-user-email",
    className: "jsx-1072768994"
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1072768994"
  }, ["div.jsx-1072768994{margin-bottom:6px;font-size:14px;line-height:16px;}"]));
};
ProfileEmail.propTypes = {
  children: _propTypes.default.string
};
const ProfileEdit = _ref3 => {
  let {
    children
  } = _ref3;
  const {
    baseUrl
  } = (0, _appRuntime.useConfig)();
  return /*#__PURE__*/_react.default.createElement("a", {
    href: (0, _joinPath.joinPath)(baseUrl, 'dhis-web-user-profile/#/profile'),
    "data-test": "headerbar-profile-edit-profile-link",
    className: "jsx-233684196"
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "233684196"
  }, ["a.jsx-233684196{color:rgba(0,0,0,0.87);font-size:12px;line-height:14px;-webkit-text-decoration:underline;text-decoration:underline;cursor:pointer;}"]));
};
ProfileEdit.propTypes = {
  children: _propTypes.default.string
};
const ProfileDetails = _ref4 => {
  let {
    name,
    email
  } = _ref4;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-4236878114"
  }, /*#__PURE__*/_react.default.createElement(ProfileName, null, name), /*#__PURE__*/_react.default.createElement(ProfileEmail, null, email), /*#__PURE__*/_react.default.createElement(ProfileEdit, null, _index.default.t('Edit profile')), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4236878114"
  }, ["div.jsx-4236878114{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;margin-inline-start:20px;color:#000;font-size:14px;font-weight:400;}"]));
};
ProfileDetails.propTypes = {
  email: _propTypes.default.string,
  name: _propTypes.default.string
};
const ProfileHeader = _ref5 => {
  let {
    name,
    email,
    avatarId
  } = _ref5;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-371352323"
  }, /*#__PURE__*/_react.default.createElement(_userAvatar.UserAvatar, {
    avatarId: avatarId,
    name: name,
    dataTest: "headerbar-profile-menu-icon",
    large: true
  }), /*#__PURE__*/_react.default.createElement(ProfileDetails, {
    name: name,
    email: email
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "371352323"
  }, ["div.jsx-371352323{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;margin-inline-start:24px;padding-top:20px;}"]));
};
exports.ProfileHeader = ProfileHeader;
ProfileHeader.propTypes = {
  avatarId: _propTypes.default.string,
  email: _propTypes.default.string,
  name: _propTypes.default.string
};
