// @dhis2-ui/header-bar/build/cjs/profile-menu/profile-menu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ProfileMenu = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _appRuntime = require("@dhis2/app-runtime");
var _uiConstants = require("@dhis2/ui-constants");
var _uiIcons = require("@dhis2/ui-icons");
var _card = require("@dhis2-ui/card");
var _center = require("@dhis2-ui/center");
var _divider = require("@dhis2-ui/divider");
var _layer = require("@dhis2-ui/layer");
var _loader = require("@dhis2-ui/loader");
var _menu = require("@dhis2-ui/menu");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _debugInfoMenuItem = require("../debug-info/debug-info-menu-item.js");
var _joinPath = require("../join-path.js");
var _index = _interopRequireDefault(require("../locales/index.js"));
var _profileHeader = require("./profile-header.js");
var _updateNotification = require("./update-notification.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const LoadingMask = () => /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  translucent: true,
  disablePortal: true,
  dataTest: "headerbar-profile-menu-loading-mask"
}, /*#__PURE__*/_react.default.createElement(_center.Center, null, /*#__PURE__*/_react.default.createElement(_loader.CircularLoader, null)));
const ProfileContents = _ref => {
  let {
    name,
    email,
    avatarId,
    helpUrl,
    hideProfileMenu,
    showDebugInfoModal
  } = _ref;
  const {
    baseUrl
  } = (0, _appRuntime.useConfig)();
  const [loading, setLoading] = (0, _react.useState)(false);
  return /*#__PURE__*/_react.default.createElement(_card.Card, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-2099675810"
  }, /*#__PURE__*/_react.default.createElement(_profileHeader.ProfileHeader, {
    name: name,
    email: email,
    avatarId: avatarId
  }), /*#__PURE__*/_react.default.createElement(_divider.Divider, {
    margin: "13px 0 7px 0"
  }), /*#__PURE__*/_react.default.createElement("ul", {
    "data-test": "headerbar-profile-menu",
    className: "jsx-2099675810"
  }, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    href: (0, _joinPath.joinPath)(baseUrl, 'dhis-web-user-profile/#/settings'),
    label: _index.default.t('Settings'),
    value: "settings",
    icon: /*#__PURE__*/_react.default.createElement(_uiIcons.IconSettings24, {
      color: _uiConstants.colors.grey700
    })
  }), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    href: (0, _joinPath.joinPath)(baseUrl, 'dhis-web-user-profile/#/account'),
    label: _index.default.t('Account'),
    value: "account",
    icon: /*#__PURE__*/_react.default.createElement(_uiIcons.IconUser24, {
      color: _uiConstants.colors.grey700
    })
  }), helpUrl && /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    href: helpUrl,
    label: _index.default.t('Help'),
    value: "help",
    icon: /*#__PURE__*/_react.default.createElement(_uiIcons.IconQuestion24, {
      color: _uiConstants.colors.grey700
    })
  }), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    href: (0, _joinPath.joinPath)(baseUrl, 'dhis-web-user-profile/#/aboutPage'),
    label: _index.default.t('About DHIS2'),
    value: "about",
    icon: /*#__PURE__*/_react.default.createElement(_uiIcons.IconInfo24, {
      color: _uiConstants.colors.grey700
    })
  }), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    href: (0, _joinPath.joinPath)(baseUrl, 'dhis-web-commons-security/logout.action')
    // NB: By MenuItem implementation, this callback
    // overwrites default navigation behavior but maintains
    // the href attribute
    ,
    onClick: async () => {
      setLoading(true);
      await (0, _appRuntime.clearSensitiveCaches)();
      setLoading(false);
      window.location.assign((0, _joinPath.joinPath)(baseUrl, 'dhis-web-commons-security/logout.action'));
    },
    label: _index.default.t('Logout'),
    value: "logout",
    icon: /*#__PURE__*/_react.default.createElement(_uiIcons.IconLogOut24, {
      color: _uiConstants.colors.grey700
    })
  }), /*#__PURE__*/_react.default.createElement(_menu.MenuDivider, {
    dense: true
  }), /*#__PURE__*/_react.default.createElement(_debugInfoMenuItem.DebugInfoMenuItem, {
    hideProfileMenu: hideProfileMenu,
    showDebugInfoModal: showDebugInfoModal
  }), /*#__PURE__*/_react.default.createElement(_updateNotification.UpdateNotification, {
    hideProfileMenu: hideProfileMenu
  }))), loading && /*#__PURE__*/_react.default.createElement(LoadingMask, null), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2099675810"
  }, ["div.jsx-2099675810{width:100%;padding:0;}", "ul.jsx-2099675810{padding:0;margin:0;}", "a.jsx-2099675810,a.jsx-2099675810:hover,a.jsx-2099675810:focus,a.jsx-2099675810:active,a.jsx-2099675810:visited{-webkit-text-decoration:none;text-decoration:none;display:block;}"]));
};
ProfileContents.propTypes = {
  hideProfileMenu: _propTypes.default.func.isRequired,
  showDebugInfoModal: _propTypes.default.func.isRequired,
  avatarId: _propTypes.default.string,
  email: _propTypes.default.string,
  helpUrl: _propTypes.default.string,
  name: _propTypes.default.string
};
const ProfileMenu = _ref2 => {
  let {
    ...props
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": "headerbar-profile-menu",
    className: "jsx-3924623997"
  }, /*#__PURE__*/_react.default.createElement(ProfileContents, props), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3924623997"
  }, ["div.jsx-3924623997{z-index:10000;position:absolute;inset-inline-end:4px;width:320px;}"]));
};
exports.ProfileMenu = ProfileMenu;
ProfileMenu.propTypes = {
  hideProfileMenu: _propTypes.default.func.isRequired,
  showDebugInfoModal: _propTypes.default.func.isRequired,
  avatarId: _propTypes.default.string,
  email: _propTypes.default.string,
  helpUrl: _propTypes.default.string,
  name: _propTypes.default.string
};
