// @dhis2-ui/notice-box/build/cjs/notice-box-message.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NoticeBoxMessage = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const NoticeBoxMessage = _ref => {
  let {
    children,
    dataTest
  } = _ref;
  if (!children) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["3568970766", [_uiConstants.colors.grey900]]])
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3568970766",
    dynamic: [_uiConstants.colors.grey900]
  }, [`div.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey900};font-size:14px;line-height:19px;}`]));
};
exports.NoticeBoxMessage = NoticeBoxMessage;
NoticeBoxMessage.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  children: _propTypes.default.node
};
