// @dhis2-ui/table/build/es/table/table-cell-head.js
import _JSXStyle from "styled-jsx/style";
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
const tableCellHeadStyles = ["th.jsx-3181834369{border-bottom:1px solid #e8edf2;font-size:14px;line-height:18px;padding:13px 12px;height:45px;}", "th.jsx-3181834369:dir(rtl){text-align:right;}", ".dense.jsx-3181834369{padding:9px 12px;height:36px;}"];
tableCellHeadStyles.__hash = "3181834369";
export const TableCellHead = _ref => {
  let {
    role,
    colSpan,
    rowSpan,
    dense,
    children,
    className,
    dataTest = 'dhis2-uicore-tablecellhead'
  } = _ref;
  return /*#__PURE__*/React.createElement("th", {
    colSpan: colSpan,
    rowSpan: rowSpan,
    "data-test": dataTest,
    role: role,
    className: `jsx-${tableCellHeadStyles.__hash}` + " " + (cx({
      dense
    }, className) || "")
  }, children, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: tableCellHeadStyles.__hash
  }, tableCellHeadStyles));
};
TableCellHead.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  colSpan: PropTypes.string,
  dataTest: PropTypes.string,
  /** Uses less padding and height for information-dense layouts */
  dense: PropTypes.bool,
  role: PropTypes.string,
  rowSpan: PropTypes.string
};
