// @dhis2-ui/transfer/build/es/picked-options.js
import _JSXStyle from "styled-jsx/style";
import { spacers } from '@dhis2/ui-constants';
import PropTypes from 'prop-types';
import React from 'react';
import { EndIntersectionDetector } from './end-intersection-detector.js';
export const PickedOptions = _ref => {
  let {
    children,
    dataTest,
    selectedEmptyComponent,
    pickedOptionsRef,
    onPickedEndReached
  } = _ref;
  return /*#__PURE__*/React.createElement("div", {
    "data-test": dataTest,
    ref: pickedOptionsRef,
    className: _JSXStyle.dynamic([["392419471", [spacers.dp4]]]) + " " + "container"
  }, /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["392419471", [spacers.dp4]]]) + " " + "content-container"
  }, !React.Children.count(children) && selectedEmptyComponent, children, onPickedEndReached && /*#__PURE__*/React.createElement(EndIntersectionDetector, {
    rootRef: pickedOptionsRef,
    onEndReached: onPickedEndReached
  })), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "392419471",
    dynamic: [spacers.dp4]
  }, [`.container.__jsx-style-dynamic-selector{padding:${spacers.dp4} 0;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;overflow-y:auto;}`, ".content-container.__jsx-style-dynamic-selector{position:relative;}"]));
};
PickedOptions.propTypes = {
  children: PropTypes.node.isRequired,
  dataTest: PropTypes.string.isRequired,
  pickedOptionsRef: PropTypes.shape({
    current: PropTypes.instanceOf(HTMLElement)
  }),
  selectedEmptyComponent: PropTypes.node,
  onPickedEndReached: PropTypes.func
};
