// @dhis2-ui/alert/build/cjs/alert-stack/alert-stack.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RTL = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _index = require("../alert-bar/index.js");
var _alertStack = require("./alert-stack.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const description = `
A container for Alert Bars.

_**Note:** The demos on this page may be slow._

\`\`\`js
import { AlertStack } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Alert Stack',
  component: _alertStack.AlertStack,
  // Use an iframe in docs to contain 'portal'
  parameters: {
    docs: {
      inlineStories: false,
      iframeHeight: '300px',
      description: {
        component: description
      }
    }
  }
};
const Default = args => /*#__PURE__*/_react.default.createElement(_alertStack.AlertStack, args, /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true
}, "First notification - I am at the bottom"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  critical: true
}, "Second notification"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  warning: true
}, "Third notification"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  success: true
}, "Fourth notification - I am at the top"));
exports.Default = Default;
const RTL = args => {
  (0, _react.useEffect)(() => {
    document.body.dir = 'rtl';
    return () => {
      document.body.dir = 'ltr';
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement("div", {
    dir: "rtl"
  }, /*#__PURE__*/_react.default.createElement(_alertStack.AlertStack, args, /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
    permanent: true
  }, "First notification - I am at the bottom"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
    permanent: true,
    critical: true
  }, "Second notification"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
    permanent: true,
    warning: true
  }, "Third notification"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
    permanent: true,
    success: true
  }, "Fourth notification - I am at the top")));
};
exports.RTL = RTL;
