// @dhis2-ui/file-input/build/es/file-input/file-input.e2e.stories.js
import React from 'react';
import { FileInput } from './index.js';
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
window.onChange = window.Cypress && window.Cypress.cy.stub();
const onChange = (payload, event) => {
  // NOTE: if files is not transformed into an array,
  // cypress will get an empty file list!
  window.onChange({
    ...payload,
    files: [...payload.files]
  }, event);
};
export default {
  title: 'FileInput'
};
export const WithOnChange = () => /*#__PURE__*/React.createElement(FileInput, {
  onChange: onChange,
  buttonLabel: "Upload file",
  name: "upload"
});
export const WithOnChangeAndMultiple = () => /*#__PURE__*/React.createElement(FileInput, {
  name: "upload",
  onChange: onChange,
  buttonLabel: "Upload files",
  multiple: true
});
export const WithInitialFocusAndOnBlur = () => /*#__PURE__*/React.createElement(FileInput, {
  buttonLabel: "Upload file",
  name: "upload",
  initialFocus: true,
  onBlur: window.onBlur
});
export const WithOnFocus = () => /*#__PURE__*/React.createElement(FileInput, {
  buttonLabel: "Upload file",
  name: "upload",
  onFocus: window.onFocus
});
