// @dhis2-ui/field/build/es/field/field.prod.stories.js
import { sharedPropTypes } from '@dhis2/ui-constants';
import { Input } from '@dhis2-ui/input';
import React from 'react';
import { Field } from './field.js';
const description = `
A useful container for form components, including a label, help text, and validation text.

\`\`\`js
import { Field } from '@dhis2/ui'
\`\`\`
`;
export default {
  title: 'Field',
  component: Field,
  argTypes: {
    valid: {
      ...sharedPropTypes.statusArgType
    },
    warning: {
      ...sharedPropTypes.statusArgType
    },
    error: {
      ...sharedPropTypes.statusArgType
    }
  },
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
export const Default = args => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Field, args, /*#__PURE__*/React.createElement(Input, {
  onChange: () => {
    console.log('Nothing happens');
  },
  name: "input",
  label: "An input"
})), /*#__PURE__*/React.createElement(Field, {
  helpText: "Help!",
  label: "Second field"
}, /*#__PURE__*/React.createElement(Input, {
  onChange: () => {
    console.log('Nothing happens');
  },
  name: "input2",
  label: "An second input"
})));
Default.args = {
  label: 'First field (change me with controls)',
  helpText: 'Help text!',
  validationText: "I'm validation text"
};
