// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/expanded.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithRootMainAndRootSubOrgUnit = exports.NoInitiallyExpandedPaths = exports.InitiallyExpandedPaths = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _common = require("./common.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const decorators = [(0, _common.createDecoratorStatefulMultiSelection)(), (0, _common.createDecoratorCustomDataProvider)()];
const NoInitiallyExpandedPaths = (_, _ref) => {
  let {
    onChange,
    selected
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    roots: "A0000000000",
    selected: selected,
    onChange: onChange
  });
};
exports.NoInitiallyExpandedPaths = NoInitiallyExpandedPaths;
NoInitiallyExpandedPaths.decorators = decorators;
const InitiallyExpandedPaths = (_, _ref2) => {
  let {
    onChange,
    selected
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    roots: "A0000000000",
    selected: selected,
    onChange: onChange,
    initiallyExpanded: ['/A0000000000']
  });
};
exports.InitiallyExpandedPaths = InitiallyExpandedPaths;
InitiallyExpandedPaths.decorators = decorators;
const WithRootMainAndRootSubOrgUnit = (_, _ref3) => {
  let {
    onChange,
    selected
  } = _ref3;
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    roots: ['A0000000000', 'A0000000001'],
    selected: selected,
    onChange: onChange
  });
};
exports.WithRootMainAndRootSubOrgUnit = WithRootMainAndRootSubOrgUnit;
WithRootMainAndRootSubOrgUnit.decorators = decorators;
