// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/no_selection.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NoSelectionRootOpened = exports.NoSelectionClosed = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _common = require("./common.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const NoSelectionClosed = () => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
  data: _common.dataProviderData
}, /*#__PURE__*/_react.default.createElement(_common.StatefulMultiSelectionWrapper, null, _ref => {
  let {
    onChange
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    disableSelection: true,
    roots: "A0000000000",
    onChange: onChange
  });
}));
exports.NoSelectionClosed = NoSelectionClosed;
const NoSelectionRootOpened = () => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
  data: _common.dataProviderData
}, /*#__PURE__*/_react.default.createElement(_common.StatefulMultiSelectionWrapper, null, _ref2 => {
  let {
    onChange
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    disableSelection: true,
    roots: "A0000000000",
    onChange: onChange,
    initiallyExpanded: ['/A0000000000']
  });
}));
exports.NoSelectionRootOpened = NoSelectionRootOpened;
