// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/sub_unit_as_root.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MultipleRootSubAndOneMainOrgUnit = void 0;
var _react = _interopRequireWildcard(require("react"));
var _index = require("../index.js");
var _common = require("./common.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const MultipleRootSubAndOneMainOrgUnit = (_, _ref) => {
  let {
    selected,
    onChange
  } = _ref;
  const [filter, setFilter] = (0, _react.useState)('');
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("input", {
    type: "input",
    onChange: e => setFilter(e.target.value)
  }), /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    filter: filter ? filter.split(',') : [],
    onChange: onChange,
    selected: selected,
    roots: ['A0000000000', 'A0000000001', 'A0000000002']
  }));
};
exports.MultipleRootSubAndOneMainOrgUnit = MultipleRootSubAndOneMainOrgUnit;
MultipleRootSubAndOneMainOrgUnit.decorators = [(0, _common.createDecoratorStatefulMultiSelection)(), (0, _common.createDecoratorCustomDataProvider)()];
