// @dhis2-ui/transfer/build/cjs/__e2e__/highlight-range-of-options.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.HasSelected = exports.HasOptions = exports.AllSelected = void 0;
var _react = _interopRequireDefault(require("react"));
var _transfer = require("../transfer.js");
var _options = require("./common/options.js");
var _statefulDecorator = require("./common/stateful-decorator.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'Transfer highlight range of options',
  decorators: [(0, _statefulDecorator.statefulDecorator)()]
};
const HasOptions = (_, _ref) => {
  let {
    onChange,
    selected
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    filterable: true,
    selected: selected,
    onChange: onChange,
    options: _options.options
  });
};
exports.HasOptions = HasOptions;
const HasSelected = (_, _ref2) => {
  let {
    onChange,
    selected
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    filterable: true,
    onChange: onChange,
    selected: selected,
    options: _options.options
  });
};
exports.HasSelected = HasSelected;
HasSelected.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.slice(0, 4).map(_ref3 => {
      let {
        value
      } = _ref3;
      return value;
    })
  })]
};
const AllSelected = (_, _ref4) => {
  let {
    onChange,
    selected
  } = _ref4;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    filterable: true,
    onChange: onChange,
    selected: selected,
    options: _options.options
  });
};
exports.AllSelected = AllSelected;
AllSelected.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.map(_ref5 => {
      let {
        value
      } = _ref5;
      return value;
    })
  })]
};
