// @dhis2-ui/transfer/build/cjs/__e2e__/reorder-with-buttons.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.HasSomeSelected = void 0;
var _react = _interopRequireDefault(require("react"));
var _transfer = require("../transfer.js");
var _options = require("./common/options.js");
var _statefulDecorator = require("./common/stateful-decorator.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'Transfer Reorder Buttons'
};
const HasSomeSelected = (_, _ref) => {
  let {
    selected,
    onChange
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    enableOrderChange: true,
    selected: selected,
    onChange: onChange,
    options: _options.options
  });
};
exports.HasSomeSelected = HasSomeSelected;
HasSomeSelected.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.slice(0, 3).map(_ref2 => {
      let {
        value
      } = _ref2;
      return value;
    })
  })]
};
