// @dhis2-ui/button/build/cjs/button/button.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnFocus = exports.WithOnClick = exports.WithInitialFocusAndOnBlur = void 0;
var _react = _interopRequireDefault(require("react"));
var _button = require("./button.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onClick = window.Cypress && window.Cypress.cy.stub();
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'Button'
};
const WithOnClick = () => /*#__PURE__*/_react.default.createElement(_button.Button, {
  name: "Button",
  value: "default",
  onClick: window.onClick
}, "Label me!");
exports.WithOnClick = WithOnClick;
const WithInitialFocusAndOnBlur = () => /*#__PURE__*/_react.default.createElement(_button.Button, {
  name: "Button",
  value: "default",
  initialFocus: true,
  onBlur: window.onBlur
}, "Label me!");
exports.WithInitialFocusAndOnBlur = WithInitialFocusAndOnBlur;
const WithOnFocus = () => /*#__PURE__*/_react.default.createElement(_button.Button, {
  name: "Button",
  value: "default",
  onFocus: window.onFocus
}, "Label me!");
exports.WithOnFocus = WithOnFocus;
