// @dhis2-ui/table/build/es/data-table/data-table.e2e.stories.js
import React from 'react';
import { DataTableBody } from './data-table-body.js';
import { DataTableCell } from './data-table-cell.js';
import { DataTableColumnHeader } from './data-table-column-header/data-table-column-header.js';
import { DataTableHead } from './data-table-head.js';
import { DataTableRow } from './data-table-row/data-table-row.js';
import { DataTable } from './data-table.js';
export default {
  title: 'DataTable',
  component: DataTable
};
export const FixedHeaderAndColumns = () => /*#__PURE__*/React.createElement(DataTable, {
  layout: "fixed",
  width: "1000px",
  scrollWidth: "500px",
  scrollHeight: "400px"
}, /*#__PURE__*/React.createElement(DataTableHead, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "Header fixed-left 1"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Header fixed-left 2"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/React.createElement(DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"))), /*#__PURE__*/React.createElement(DataTableBody, null, /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "scrolling top-left"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "scrolling top-right")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "...")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "...")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "...")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "...")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "...")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "...")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "...")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "...")), /*#__PURE__*/React.createElement(DataTableRow, null, /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/React.createElement(DataTableCell, null, "scrolling bottom-left"), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "..."), /*#__PURE__*/React.createElement(DataTableCell, null, "scrolling bottom-right"))));
