// @dhis2-ui/organisation-unit-tree/build/es/__stories__/development-stories.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import { DataProvider } from '@dhis2/app-runtime';
import React, { useState } from 'react';
import { OrganisationUnitTree } from '../index.js';
const DX_onChange = (selected, setSelected, singleSelection) => _ref => {
  let {
    id,
    path,
    checked
  } = _ref;
  console.log('onChange', {
    path,
    id,
    checked
  });
  const pathIndex = selected.indexOf(path);
  let nextSelected;
  if (checked) {
    nextSelected = singleSelection ? [path] : [...selected, path];
  } else {
    nextSelected = singleSelection ? [] : [...selected.slice(0, pathIndex), ...selected.slice(pathIndex + 1)];
  }
  console.log('> nextSelected', nextSelected);
  setSelected(nextSelected);
};
const Wrapper = props => {
  const [selected, setSelected] = useState([]);
  return /*#__PURE__*/React.createElement(OrganisationUnitTree, _extends({
    name: "Root org unit",
    roots: ['A0000000000'],
    selected: selected,
    onChange: DX_onChange(selected, setSelected, props.singleSelection),
    initiallyExpanded: ['A0000000001/A0000000002']
  }, props));
};
export const DxMultiSelection = () => /*#__PURE__*/React.createElement(Wrapper, null);
DxMultiSelection.storyName = 'DX: Multi selection';
export const DxSingleSelection = () => /*#__PURE__*/React.createElement(Wrapper, {
  singleSelection: true
});
DxSingleSelection.storyName = 'DX: Single selection';
export const DxNoSelection = () => /*#__PURE__*/React.createElement(Wrapper, {
  disableSelection: true
});
DxNoSelection.storyName = 'DX: No selection';
export const DxWithRealBackend = () => /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
  style: {
    marginBottom: 20,
    lineHeight: '28px'
  }
}, /*#__PURE__*/React.createElement("b", null, "This story doesn't work on netlify for some reason, just run it locally."), /*#__PURE__*/React.createElement("br", null), "You need to log in to", ' ', /*#__PURE__*/React.createElement("a", {
  href: "https://debug.dhis2.org/dev",
  target: "_blank",
  rel: "noopener noreferrer"
}, "https://debug.dhis2.org/dev"), /*#__PURE__*/React.createElement("br", null), "Make sure the", ' ', /*#__PURE__*/React.createElement("code", {
  style: {
    background: '#ccc'
  }
}, "localhost:[PORT]"), " is part of the accepted list:", ' ', /*#__PURE__*/React.createElement("a", {
  href: "https://debug.dhis2.org/dev/dhis-web-settings/#/access",
  target: "_blank",
  rel: "noopener noreferrer"
}, "Settings app / Access")), /*#__PURE__*/React.createElement(DataProvider, {
  baseUrl: "https://debug.dhis2.org/dev",
  apiVersion: ""
}, /*#__PURE__*/React.createElement(Wrapper, {
  roots: ['ImspTQPwCqd', 'O6uvpzGd5pu', 'fdc6uOvgoji']
})));
DxWithRealBackend.storyName = 'DX: With real backend';
