// @dhis2-ui/organisation-unit-tree/build/es/__stories__/loading-error-grandchild.js
import { CustomDataProvider } from '@dhis2/app-runtime';
import React from 'react';
import { OrganisationUnitTree } from '../index.js';
import { customData, onChange, onExpand, onCollapse, onChildrenLoaded } from './shared.js';
export const LoadingErrorGrandchild = () => /*#__PURE__*/React.createElement(CustomDataProvider, {
  data: {
    ...customData,
    organisationUnits: function () {
      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }
      const [, {
        id
      }] = args;
      if (id === 'A0000000003') {
        return Promise.reject('Loading org unit 4 and 5 failed');
      }
      return customData.organisationUnits(...args);
    }
  }
}, /*#__PURE__*/React.createElement(OrganisationUnitTree, {
  autoExpandLoadingError: true,
  name: "Root org unit",
  roots: ['A0000000000'],
  onChange: onChange,
  onExpand: onExpand,
  onCollapse: onCollapse,
  onChildrenLoaded: onChildrenLoaded,
  initiallyExpanded: ['/A0000000000/A0000000001']
}));
LoadingErrorGrandchild.storyName = 'Loading error grandchild';
