// @dhis2-ui/notice-box/build/cjs/notice-box-content.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NoticeBoxContent = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _noticeBoxMessage = require("./notice-box-message.js");
var _noticeBoxTitle = require("./notice-box-title.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const NoticeBoxContent = _ref => {
  let {
    children,
    dataTest = 'dhis2-uicore-noticebox-content',
    title
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["403435427", [_uiConstants.spacers.dp8]]])
  }, /*#__PURE__*/_react.default.createElement(_noticeBoxTitle.NoticeBoxTitle, {
    title: title,
    dataTest: `${dataTest}-title`
  }), /*#__PURE__*/_react.default.createElement(_noticeBoxMessage.NoticeBoxMessage, {
    dataTest: `${dataTest}-message`
  }, children), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "403435427",
    dynamic: [_uiConstants.spacers.dp8]
  }, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:${_uiConstants.spacers.dp8};padding-block-start:3px;}`]));
};
exports.NoticeBoxContent = NoticeBoxContent;
NoticeBoxContent.propTypes = {
  children: _propTypes.default.node,
  dataTest: _propTypes.default.string,
  title: _propTypes.default.string
};
