// @dhis2-ui/organisation-unit-tree/build/es/organisation-unit-node/label/single-selection-label.js
import _JSXStyle from "styled-jsx/style";
import { colors } from '@dhis2/ui-constants';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

/**
 * @param {Object} props
 * @param {string} props.label
 * @param {bool} [props.checked]
 * @param {bool} [props.loading]
 * @param {Function} [props.onChange]
 * @returns {React.Component}
 */
export const SingleSelectionLabel = _ref => {
  let {
    checked,
    children,
    onChange,
    loading
  } = _ref;
  return /*#__PURE__*/React.createElement("span", {
    onClick: event => {
      const payload = {
        checked: !checked
      };
      onChange(payload, event);
    },
    className: _JSXStyle.dynamic([["321503310", [colors.grey900, colors.teal700]]]) + " " + (cx({
      checked,
      loading
    }) || "")
  }, children, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "321503310",
    dynamic: [colors.grey900, colors.teal700]
  }, [`span.__jsx-style-dynamic-selector{background:transparent;border-radius:3px;color:${colors.grey900};cursor:pointer;display:inline-block;font-size:14px;line-height:24px;padding:0 5px;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;white-space:nowrap;}`, `.checked.__jsx-style-dynamic-selector{background:${colors.teal700};color:white;}`, ".loading.__jsx-style-dynamic-selector{cursor:auto;}"]));
};
SingleSelectionLabel.propTypes = {
  children: PropTypes.any.isRequired,
  checked: PropTypes.bool,
  loading: PropTypes.bool,
  onChange: PropTypes.func
};
