// @dhis2-ui/cover/build/cjs/cover.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithClickHandler = exports.WithChildren = exports.Blocking = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _react = _interopRequireDefault(require("react"));
var _cover = require("./cover.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onButtonClick = window.Cypress && window.Cypress.cy.stub();
window.onCover = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'Cover',
  component: _cover.Cover,
  decorators: [story => /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-1235544169"
  }, story(), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1235544169"
  }, ["div.jsx-1235544169{width:400px;height:400px;position:relative;border:1px dotted grey;}"]))]
};
const WithChildren = () => /*#__PURE__*/_react.default.createElement(_cover.Cover, null, /*#__PURE__*/_react.default.createElement("p", null, "I am a child"));
exports.WithChildren = WithChildren;
const Blocking = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
  "data-test": "button",
  onClick: window.onButtonClick
}, "Test"), /*#__PURE__*/_react.default.createElement(_cover.Cover, null));
exports.Blocking = Blocking;
const WithClickHandler = () => /*#__PURE__*/_react.default.createElement(_cover.Cover, {
  onClick: window.onCover
}, /*#__PURE__*/_react.default.createElement("button", {
  "data-test": "button",
  onClick: window.onButtonClick
}, "Test"));
exports.WithClickHandler = WithClickHandler;
