// @dhis2-ui/table/build/cjs/table/table-row.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableRow = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _tableContext = require("./table-context.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const tableRowStyles = [".zebraStriping.jsx-3124144591:nth-child(even){background:#fbfcfd;}"];
tableRowStyles.__hash = "3124144591";
const TableRow = _ref => {
  let {
    role,
    children,
    className,
    dataTest = 'dhis2-uicore-tablerow',
    suppressZebraStriping,
    ...rest
  } = _ref;
  const {
    suppressZebraStriping: suppressZebraStripingFromContext
  } = (0, _react.useContext)(_tableContext.TableContext);
  const zebraStriping = typeof suppressZebraStriping !== 'undefined' ? !suppressZebraStriping : !suppressZebraStripingFromContext;
  return /*#__PURE__*/_react.default.createElement("tr", _extends({
    "data-test": dataTest,
    role: role
  }, rest, {
    className: `jsx-${tableRowStyles.__hash}` + " " + (rest && rest.className != null && rest.className || (0, _classnames.default)(className, {
      zebraStriping
    }) || "")
  }), children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: tableRowStyles.__hash
  }, tableRowStyles));
};
exports.TableRow = TableRow;
TableRow.propTypes = {
  /** Should be `<TableCell>` or `<TableCellHead>` components */
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  role: _propTypes.default.string,
  /** Disables the default row striping for this row */
  suppressZebraStriping: _propTypes.default.bool
};
