// @dhis2-ui/menu/build/cjs/menu-section-header/menu-section-header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuSectionHeader = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _divider = require("@dhis2-ui/divider");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const MenuSectionHeader = _ref => {
  let {
    className,
    dataTest = 'dhis2-uicore-menusectionheader',
    dense,
    hideDivider,
    label
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("li", {
    "data-test": dataTest,
    className: _style.default.dynamic([["1160864328", [_uiConstants.colors.white, _uiConstants.spacers.dp8, _uiConstants.spacers.dp16, _uiConstants.spacers.dp8, _uiConstants.spacers.dp16, _uiConstants.colors.grey700, _uiConstants.spacers.dp8, _uiConstants.spacers.dp12, _uiConstants.spacers.dp4, _uiConstants.spacers.dp12]]]) + " " + ((0, _classnames.default)(className, {
      dense
    }) || "")
  }, !hideDivider && /*#__PURE__*/_react.default.createElement(_divider.Divider, {
    dense: dense
  }), /*#__PURE__*/_react.default.createElement("h6", {
    className: _style.default.dynamic([["1160864328", [_uiConstants.colors.white, _uiConstants.spacers.dp8, _uiConstants.spacers.dp16, _uiConstants.spacers.dp8, _uiConstants.spacers.dp16, _uiConstants.colors.grey700, _uiConstants.spacers.dp8, _uiConstants.spacers.dp12, _uiConstants.spacers.dp4, _uiConstants.spacers.dp12]]])
  }, label), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1160864328",
    dynamic: [_uiConstants.colors.white, _uiConstants.spacers.dp8, _uiConstants.spacers.dp16, _uiConstants.spacers.dp8, _uiConstants.spacers.dp16, _uiConstants.colors.grey700, _uiConstants.spacers.dp8, _uiConstants.spacers.dp12, _uiConstants.spacers.dp4, _uiConstants.spacers.dp12]
  }, [`li.__jsx-style-dynamic-selector{list-style:none;background-color:${_uiConstants.colors.white};-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;padding:0;line-height:0;}`, `h6.__jsx-style-dynamic-selector{margin:0;padding:${_uiConstants.spacers.dp8} ${_uiConstants.spacers.dp16} ${_uiConstants.spacers.dp8} ${_uiConstants.spacers.dp16};font-size:13px;line-height:16px;font-weight:500;color:${_uiConstants.colors.grey700};}`, `li.dense.__jsx-style-dynamic-selector h6.__jsx-style-dynamic-selector{padding:${_uiConstants.spacers.dp8} ${_uiConstants.spacers.dp12} ${_uiConstants.spacers.dp4} ${_uiConstants.spacers.dp12};}`]));
};
exports.MenuSectionHeader = MenuSectionHeader;
MenuSectionHeader.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  dense: _propTypes.default.bool,
  hideDivider: _propTypes.default.bool,
  label: _propTypes.default.node
};
