// @dhis2-ui/tooltip/build/cjs/tooltip.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tooltip = exports.TOOLTIP_OFFSET = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _popper = require("@dhis2-ui/popper");
var _portal = require("@dhis2-ui/portal");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const TOOLTIP_OFFSET = exports.TOOLTIP_OFFSET = 4;
const popperStyle = {
  styles: /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1892168270"
  }, [`.jsx-1892168270{z-index:${_uiConstants.layers.applicationTop};}`, "div[data-popper-reference-hidden=\"true\"].jsx-1892168270{visibility:hidden;}"]),
  className: "jsx-1892168270"
};
const offsetModifier = {
  name: 'offset',
  options: {
    offset: [0, TOOLTIP_OFFSET]
  }
};

/**
 * For some reason the intended effects of the 'hide' modifier work with or
 * without adding it to the popper... but it may be safe to include it anyway
 */
const hideModifier = {
  name: 'hide'
};
const flipModifier = {
  name: 'flip',
  options: {
    altBoundary: true
  }
};
const Tooltip = _ref => {
  let {
    children,
    className,
    closeDelay = 200,
    content,
    dataTest = 'dhis2-uicore-tooltip',
    maxWidth = 300,
    openDelay = 200,
    placement = 'top'
  } = _ref;
  const [open, setOpen] = (0, _react.useState)(false);
  const popperReference = (0, _react.useRef)();
  const openTimerRef = (0, _react.useRef)(null);
  const closeTimerRef = (0, _react.useRef)(null);
  const onMouseOver = () => {
    clearTimeout(closeTimerRef.current);
    openTimerRef.current = setTimeout(() => {
      setOpen(true);
    }, openDelay);
  };
  const onMouseOut = () => {
    clearTimeout(openTimerRef.current);
    closeTimerRef.current = setTimeout(() => {
      setOpen(false);
    }, closeDelay);
  };
  const onFocus = () => {
    clearTimeout(closeTimerRef.current);
    openTimerRef.current = setTimeout(() => {
      setOpen(true);
    }, openDelay);
  };
  const onBlur = () => {
    clearTimeout(openTimerRef.current);
    closeTimerRef.current = setTimeout(() => {
      setOpen(false);
    }, closeDelay);
  };
  const handleKeyDown = event => {
    if (event.key === 'Escape') {
      closeTimerRef.current = setTimeout(() => {
        setOpen(false);
      }, closeDelay);
    }
  };
  (0, _react.useEffect)(() => {
    document.addEventListener('keydown', handleKeyDown);
    return () => {
      clearTimeout(openTimerRef.current);
      clearTimeout(closeTimerRef.current);
      document.removeEventListener('keydown', handleKeyDown);
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, typeof children === 'function' ? children({
    onMouseOver: onMouseOver,
    onMouseOut: onMouseOut,
    onFocus: {
      onFocus
    },
    onBlur: {
      onBlur
    },
    ref: popperReference
  }) : /*#__PURE__*/_react.default.createElement("span", {
    onMouseOver: onMouseOver,
    onMouseOut: onMouseOut,
    onFocus: onFocus,
    onBlur: onBlur,
    ref: popperReference,
    tabIndex: 0,
    "aria-describedby": open ? 'tooltipContenDhis2Ui' : '',
    "data-test": `${dataTest}-reference`,
    className: _style.default.dynamic([["3515683788", [maxWidth, _uiConstants.colors.grey900, _uiConstants.colors.white]]])
  }, children), open && /*#__PURE__*/_react.default.createElement(_portal.Portal, null, /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    className: popperStyle.className,
    placement: placement,
    reference: popperReference,
    modifiers: [offsetModifier, flipModifier, hideModifier]
  }, /*#__PURE__*/_react.default.createElement("div", {
    id: "tooltipContenDhis2Ui",
    onMouseOver: onMouseOver,
    onMouseOut: onMouseOut,
    "data-test": `${dataTest}-content`,
    role: "tooltip",
    className: _style.default.dynamic([["3515683788", [maxWidth, _uiConstants.colors.grey900, _uiConstants.colors.white]]]) + " " + (className || "")
  }, content))), popperStyle.styles, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3515683788",
    dynamic: [maxWidth, _uiConstants.colors.grey900, _uiConstants.colors.white]
  }, [`div.__jsx-style-dynamic-selector{max-width:${maxWidth}px;}`, `div.__jsx-style-dynamic-selector{padding:4px 6px;background-color:${_uiConstants.colors.grey900};border-radius:3px;color:${_uiConstants.colors.white};font-size:13px;line-height:17px;}`]));
};
exports.Tooltip = Tooltip;
Tooltip.propTypes = {
  /** If child is a function, it's called with `{ onMouseOver, onMouseOut, ref }` args to apply to a reference element. If child is a node, it is wrapped in a `span` with the appropriate attributes and handlers. */
  children: _propTypes.default.oneOfType([_propTypes.default.node, _propTypes.default.func]),
  className: _propTypes.default.string,
  /** Time (in ms) until tooltip closes after mouse out */
  closeDelay: _propTypes.default.number,
  /** Content to display when the tooltip is open */
  content: _propTypes.default.node,
  dataTest: _propTypes.default.string,
  /** Max width of the tooltip in px */
  maxWidth: _propTypes.default.number,
  /** Time (in ms) until tooltip open after mouse over */
  openDelay: _propTypes.default.number,
  /** Where to place the tooltip relative to its reference */
  placement: _propTypes.default.oneOf(['top', 'right', 'bottom', 'left'])
};
