// @dhis2-ui/tooltip/build/cjs/tooltip.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ShiftingPosition = exports.ScrollingContainers = exports.ModalWithTooltip = exports.FlippedVertically = exports.Default = exports.AdjustedHorizontalPosition = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _modal = require("@dhis2-ui/modal");
var _react = _interopRequireDefault(require("react"));
var _tooltip = require("./tooltip.js");
var _tooltipProdStories = require("./tooltip.prod.stories.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const noPaddingStyles = ["#root{padding:0;}"];
noPaddingStyles.__hash = "249208768";
var _default = exports.default = {
  title: 'Tooltip',
  component: _tooltip.Tooltip
};
const Default = () => /*#__PURE__*/_react.default.createElement("p", {
  className: "jsx-4042709097 " + `jsx-${noPaddingStyles.__hash}`
}, "I am a ", /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
  content: "Short content"
}, "paragraph"), " that contains a tooltip.", /*#__PURE__*/_react.default.createElement(_style.default, {
  id: noPaddingStyles.__hash
}, noPaddingStyles), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "4042709097"
}, ["p.jsx-4042709097{margin-top:40px;}"]));
exports.Default = Default;
const FlippedVertically = () => /*#__PURE__*/_react.default.createElement("p", {
  className: "jsx-2658853518 " + `jsx-${noPaddingStyles.__hash}`
}, "I am a ", /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
  content: "Short content"
}, "paragraph"), " that contains a tooltip.", /*#__PURE__*/_react.default.createElement(_style.default, {
  id: noPaddingStyles.__hash
}, noPaddingStyles), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "2658853518"
}, ["div#root{padding-top:0;}", "p.jsx-2658853518{margin-top:0;}"]));
exports.FlippedVertically = FlippedVertically;
const AdjustedHorizontalPosition = () => /*#__PURE__*/_react.default.createElement("p", {
  className: "jsx-4042709097 " + `jsx-${noPaddingStyles.__hash}`
}, "I am a", ' ', /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
  content: "Longer content will force the tooltip popper to shift"
}, "paragraph"), ' ', "that contains a tooltip.", /*#__PURE__*/_react.default.createElement(_style.default, {
  id: noPaddingStyles.__hash
}, noPaddingStyles), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "4042709097"
}, ["p.jsx-4042709097{margin-top:40px;}"]));
exports.AdjustedHorizontalPosition = AdjustedHorizontalPosition;
const ShiftingPosition = () => /*#__PURE__*/_react.default.createElement("p", {
  className: "jsx-823038592 " + `jsx-${noPaddingStyles.__hash}`
}, "I am a ", /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
  content: "Short content"
}, "paragraph"), " that contains a tooltip.", /*#__PURE__*/_react.default.createElement(_style.default, {
  id: noPaddingStyles.__hash
}, noPaddingStyles), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "823038592"
}, ["html,body,div#storybook-root{position:relative;height:40px;max-height:40px;overflow:hidden;}"]));
exports.ShiftingPosition = ShiftingPosition;
const ModalWithTooltip = () => /*#__PURE__*/_react.default.createElement(_modal.Modal, null, /*#__PURE__*/_react.default.createElement("p", {
  className: "jsx-3185463749"
}, "I am a ", /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
  content: "Short content"
}, "paragraph"), " that contains a tooltip."), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "3185463749"
}, ["[data-test='dhis2-uicore-popper']{pointer-events:all;}"]));
exports.ModalWithTooltip = ModalWithTooltip;
const ScrollingContainers = exports.ScrollingContainers = _tooltipProdStories.HidesWhenOutOfFrame.bind({});
ScrollingContainers.args = {
  content: 'Tooltip content'
};
ScrollingContainers.decorators = [story => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    marginTop: '150px'
  }
}, story())];
