// @dhis2-ui/file-input/build/cjs/file-list/file-list-item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FileListItem = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _uiIcons = require("@dhis2/ui-icons");
var _loader = require("@dhis2-ui/loader");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const FileListItem = _ref => {
  let {
    className,
    label,
    onRemove,
    removeText,
    loading,
    onCancel,
    cancelText,
    dataTest = 'dhis2-uicore-filelistitem'
  } = _ref;
  const handleKeyDown = event => {
    if (!onRemove) {
      return;
    }
    if (event.key === 'Enter' || event.key === 'Backspace' || event.key === 'Delete') {
      onRemove({}, event);
    }
  };
  return /*#__PURE__*/_react.default.createElement("p", {
    "data-test": dataTest,
    className: _style.default.dynamic([["513536609", [_uiConstants.spacers.dp4, _uiConstants.colors.grey900, _uiConstants.colors.grey700, _uiConstants.colors.red700, _uiConstants.colors.red800, _uiConstants.theme.focus]]]) + " " + ((0, _classnames.default)('selected-file', className) || "")
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: _style.default.dynamic([["513536609", [_uiConstants.spacers.dp4, _uiConstants.colors.grey900, _uiConstants.colors.grey700, _uiConstants.colors.red700, _uiConstants.colors.red800, _uiConstants.theme.focus]]]) + " " + "icon"
  }, loading ? /*#__PURE__*/_react.default.createElement(_loader.CircularLoader, {
    extrasmall: true
  }) : /*#__PURE__*/_react.default.createElement(_uiIcons.IconAttachment16, {
    color: _uiConstants.colors.grey700
  })), /*#__PURE__*/_react.default.createElement("span", {
    className: _style.default.dynamic([["513536609", [_uiConstants.spacers.dp4, _uiConstants.colors.grey900, _uiConstants.colors.grey700, _uiConstants.colors.red700, _uiConstants.colors.red800, _uiConstants.theme.focus]]]) + " " + "text"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: _style.default.dynamic([["513536609", [_uiConstants.spacers.dp4, _uiConstants.colors.grey900, _uiConstants.colors.grey700, _uiConstants.colors.red700, _uiConstants.colors.red800, _uiConstants.theme.focus]]]) + " " + "label"
  }, label), loading && onCancel && cancelText && /*#__PURE__*/_react.default.createElement("span", {
    onClick: event => onCancel({}, event),
    "data-test": `${dataTest}-cancel`,
    className: _style.default.dynamic([["513536609", [_uiConstants.spacers.dp4, _uiConstants.colors.grey900, _uiConstants.colors.grey700, _uiConstants.colors.red700, _uiConstants.colors.red800, _uiConstants.theme.focus]]]) + " " + "action"
  }, cancelText), !loading && /*#__PURE__*/_react.default.createElement("span", {
    onClick: event => onRemove({}, event),
    "data-test": `${dataTest}-remove`,
    tabIndex: 0,
    onKeyDown: handleKeyDown,
    className: _style.default.dynamic([["513536609", [_uiConstants.spacers.dp4, _uiConstants.colors.grey900, _uiConstants.colors.grey700, _uiConstants.colors.red700, _uiConstants.colors.red800, _uiConstants.theme.focus]]]) + " " + "action"
  }, removeText)), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "513536609",
    dynamic: [_uiConstants.spacers.dp4, _uiConstants.colors.grey900, _uiConstants.colors.grey700, _uiConstants.colors.red700, _uiConstants.colors.red800, _uiConstants.theme.focus]
  }, [`p.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin:0;padding-top:${_uiConstants.spacers.dp4};}`, "span.__jsx-style-dynamic-selector{display:inline-block;}", ".icon.__jsx-style-dynamic-selector{margin-inline-end:4px;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-top:1px;}", ".text.__jsx-style-dynamic-selector{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;-webkit-flex-shrink:1;-ms-flex-negative:1;flex-shrink:1;}", `.label.__jsx-style-dynamic-selector{font-size:14px;color:${_uiConstants.colors.grey900};word-break:break-all;}`, ".label.__jsx-style-dynamic-selector::after{content:' ';display:inline-block;width:12px;}", `.action.__jsx-style-dynamic-selector{font-size:12px;line-height:12px;margin-top:0;-webkit-text-decoration:underline;text-decoration:underline;cursor:pointer;color:${_uiConstants.colors.grey700};}`, `.action.__jsx-style-dynamic-selector:hover{color:${_uiConstants.colors.red700};}`, `.action.__jsx-style-dynamic-selector:active{color:${_uiConstants.colors.red800};}`, `.action.__jsx-style-dynamic-selector:focus{outline:3px solid ${_uiConstants.theme.focus};outline-offset:2px;}`]));
};
exports.FileListItem = FileListItem;
FileListItem.propTypes = {
  onRemove: _propTypes.default.func.isRequired,
  cancelText: _propTypes.default.string,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  label: _propTypes.default.string,
  loading: _propTypes.default.bool,
  removeText: _propTypes.default.string,
  onCancel: _propTypes.default.func
};
