// @dhis2-ui/table/build/cjs/data-table/table-elements/table-data-cell/table-data-cell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableDataCell = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = require("@dhis2/prop-types");
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes2 = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _tableDataCellStyles = _interopRequireDefault(require("./table-data-cell.styles.js"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const rtlCorrespondingAlignments = {
  left: 'right',
  right: 'left',
  center: 'center'
};
const TableDataCell = exports.TableDataCell = /*#__PURE__*/(0, _react.forwardRef)((_ref, ref) => {
  var _rtlCorrespondingAlig;
  let {
    active,
    align = 'left',
    backgroundColor,
    bordered,
    children,
    className,
    colSpan,
    dataTest = 'dhis2-uicore-tabledatacel',
    error,
    large,
    left = 'auto',
    muted,
    role,
    rowSpan,
    scope,
    staticStyle,
    valid,
    width = 'auto',
    onClick,
    ...props
  } = _ref;
  const rtlAlign = (_rtlCorrespondingAlig = rtlCorrespondingAlignments[align]) !== null && _rtlCorrespondingAlig !== void 0 ? _rtlCorrespondingAlig : align;
  return /*#__PURE__*/_react.default.createElement("td", _extends({}, props, {
    ref: ref,
    colSpan: colSpan,
    rowSpan: rowSpan,
    onClick: onClick,
    "data-test": dataTest,
    role: role,
    scope: scope,
    className: `jsx-${_tableDataCellStyles.default.__hash}` + " " + _style.default.dynamic([["836058623", [left, align, width, backgroundColor || _uiConstants.colors.white, rtlAlign, backgroundColor || _uiConstants.colors.teal100, backgroundColor || _uiConstants.colors.grey100, backgroundColor || _uiConstants.colors.grey200, backgroundColor || '#cdeae8']]]) + " " + ((0, _classnames.default)(className, {
      active,
      bordered,
      error,
      large,
      muted,
      staticStyle,
      valid
    }) || "")
  }), children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: _tableDataCellStyles.default.__hash
  }, _tableDataCellStyles.default), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "836058623",
    dynamic: [left, align, width, backgroundColor || _uiConstants.colors.white, rtlAlign, backgroundColor || _uiConstants.colors.teal100, backgroundColor || _uiConstants.colors.grey100, backgroundColor || _uiConstants.colors.grey200, backgroundColor || '#cdeae8']
  }, [`td.__jsx-style-dynamic-selector{inset-inline-start:${left};text-align:${align};width:${width};background-color:${backgroundColor || _uiConstants.colors.white};}`, `.__jsx-style-dynamic-selector:dir(rtl){text-align:${rtlAlign};}`, `tr.selected>td.__jsx-style-dynamic-selector{background-color:${backgroundColor || _uiConstants.colors.teal100};}`, `tr:hover>td.__jsx-style-dynamic-selector:not(.staticStyle){background-color:${backgroundColor || _uiConstants.colors.grey100};}`, `tr:active>td.__jsx-style-dynamic-selector:not(.staticStyle){background-color:${backgroundColor || _uiConstants.colors.grey200};}`, `tr.selected:hover>td.__jsx-style-dynamic-selector:not(.staticStyle){background-color:${backgroundColor || '#cdeae8'};}`]));
});
TableDataCell.displayName = 'TableDataCell';
const stylePropType = (0, _propTypes.mutuallyExclusive)(['valid', 'error', 'muted'], _propTypes2.default.bool);
TableDataCell.propTypes = {
  /** To toggle border color, for example for editing */
  active: _propTypes2.default.bool,
  align: _propTypes2.default.oneOf(['left', 'center', 'right']),
  /** Sets background color of the cell. Disables dynamic background colors from active, hover, and selected states */
  backgroundColor: _propTypes2.default.string,
  bordered: _propTypes2.default.bool,
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  colSpan: _propTypes2.default.string,
  dataTest: _propTypes2.default.string,
  /** Mutually exclusive with muted and valid */
  error: stylePropType,
  large: _propTypes2.default.bool,
  /** Required when fixed */
  left: (0, _propTypes.requiredIf)(props => props.fixed, _propTypes2.default.string),
  /** Mutually exclusive with error and valid */
  muted: stylePropType,
  role: _propTypes2.default.string,
  rowSpan: _propTypes2.default.string,
  scope: _propTypes2.default.string,
  /** Surpress hover and active event styles */
  staticStyle: _propTypes2.default.bool,
  /** Mutually exclusive with error and muted */
  valid: stylePropType,
  width: _propTypes2.default.string,
  onClick: _propTypes2.default.func
};
