// @dhis2-ui/field/build/cjs/field-group/field-group.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithSwitch = exports.WithRadio = exports.WithLabel = exports.WithCheckbox = exports.HelpAndValidationTexts = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _checkbox = require("@dhis2-ui/checkbox");
var _radio = require("@dhis2-ui/radio");
var _switch = require("@dhis2-ui/switch");
var _react = _interopRequireDefault(require("react"));
var _fieldGroup = require("./field-group.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
Wraps a group of form components like Radios, Checkboxes, or Switches. The FieldGroup wraps the form controls in a FieldSet and a Field component to group them and add a label, help text, and/or validation text.

\`\`\`js
import { FieldGroup } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Field Group',
  component: _fieldGroup.FieldGroup,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    valid: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: {
      ..._uiConstants.sharedPropTypes.statusArgType
    }
  }
};
const WithCheckbox = args => /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, args, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "third",
  label: "Third"
}));
exports.WithCheckbox = WithCheckbox;
const WithRadio = args => /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, args, /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  value: "first",
  label: "First"
}), /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  value: "second",
  label: "Second",
  checked: true
}), /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  value: "third",
  label: "Third"
}), /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  value: "fourth",
  label: "Fourth"
}));
exports.WithRadio = WithRadio;
const WithSwitch = args => /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, args, /*#__PURE__*/_react.default.createElement(_switch.Switch, {
  value: "first",
  label: "First"
}), /*#__PURE__*/_react.default.createElement(_switch.Switch, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/_react.default.createElement(_switch.Switch, {
  value: "third",
  label: "Third"
}), /*#__PURE__*/_react.default.createElement(_switch.Switch, {
  value: "fourth",
  label: "Fourth"
}));
exports.WithSwitch = WithSwitch;
const WithLabel = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, args, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "third",
  label: "Third"
})), /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, {
  label: "I am a required field",
  required: true
}, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "third",
  label: "Third"
})));
exports.WithLabel = WithLabel;
WithLabel.args = {
  label: 'I am a label/legend'
};
const HelpAndValidationTexts = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, args, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "third",
  label: "Third"
})), /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, {
  label: "I am a legend",
  helpText: "I am disabled",
  disabled: true
}, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "first",
  label: "First",
  disabled: true
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "second",
  label: "Second",
  disabled: true
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "third",
  label: "Third",
  disabled: true
})), /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, {
  label: "I am a legend",
  valid: true,
  validationText: "I am valid"
}, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "second",
  label: "Second",
  checked: true
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "third",
  label: "Third",
  checked: true
})), /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, {
  label: "I am a legend",
  name: "warning",
  warning: true,
  validationText: "I have a warning"
}, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "third",
  label: "Third"
})), /*#__PURE__*/_react.default.createElement(_fieldGroup.FieldGroup, {
  label: "I am a legend",
  error: true,
  validationText: "I have an error"
}, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
  value: "third",
  label: "Third"
})));
exports.HelpAndValidationTexts = HelpAndValidationTexts;
HelpAndValidationTexts.args = {
  label: 'I am a field',
  helpText: 'I am help text!'
};
