// @dhis2-ui/tab/build/cjs/tab/tab.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnClickAndDisabled = exports.WithOnClick = exports.WithIcon = exports.WithChildren = void 0;
var _react = _interopRequireDefault(require("react"));
var _tab = require("./tab.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onClick = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'Tab'
};
const WithOnClick = () => /*#__PURE__*/_react.default.createElement(_tab.Tab, {
  onClick: window.onClick
}, "Tab A");
exports.WithOnClick = WithOnClick;
const WithChildren = () => /*#__PURE__*/_react.default.createElement(_tab.Tab, null, "I am a child");
exports.WithChildren = WithChildren;
const WithIcon = () => /*#__PURE__*/_react.default.createElement(_tab.Tab, {
  icon: /*#__PURE__*/_react.default.createElement("div", null, "Icon")
}, "Children");
exports.WithIcon = WithIcon;
const WithOnClickAndDisabled = () => /*#__PURE__*/_react.default.createElement(_tab.Tab, {
  onClick: window.onClick,
  disabled: true
}, "Tab A");
exports.WithOnClickAndDisabled = WithOnClickAndDisabled;
