// @dhis2-ui/calendar/build/cjs/__e2e__/calendar-input.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "EthiopicWithAmharic", {
  enumerable: true,
  get: function () {
    return _calendarInputProdStories.EthiopicWithAmharic;
  }
});
Object.defineProperty(exports, "EthiopicWithEnglish", {
  enumerable: true,
  get: function () {
    return _calendarInputProdStories.EthiopicWithEnglish;
  }
});
Object.defineProperty(exports, "GregorianWithArabic", {
  enumerable: true,
  get: function () {
    return _calendarInputProdStories.GregorianWithArabic;
  }
});
Object.defineProperty(exports, "GregorianWithEnglish", {
  enumerable: true,
  get: function () {
    return _calendarInputProdStories.GregorianWithEnglish;
  }
});
Object.defineProperty(exports, "IslamicWithArabic", {
  enumerable: true,
  get: function () {
    return _calendarInputProdStories.IslamicWithArabic;
  }
});
Object.defineProperty(exports, "NepaliWithEnglish", {
  enumerable: true,
  get: function () {
    return _calendarInputProdStories.NepaliWithEnglish;
  }
});
Object.defineProperty(exports, "NepaliWithNepali", {
  enumerable: true,
  get: function () {
    return _calendarInputProdStories.NepaliWithNepali;
  }
});
exports.default = exports.TestCalendarWithClearButton = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _calendarInputProdStories = require("../stories/calendar-input.prod.stories.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'CalendarInputTesting',
  component: _index.CalendarInput
};
const TestCalendarWithClearButton = () => {
  var _params$get, _params$get2;
  const params = new URLSearchParams(location.search);
  const calendar = (_params$get = params.get('calendar')) !== null && _params$get !== void 0 ? _params$get : 'gregory';
  const initialDate = (_params$get2 = params.get('initialDate')) !== null && _params$get2 !== void 0 ? _params$get2 : null;
  return /*#__PURE__*/_react.default.createElement(_calendarInputProdStories.CalendarWithClearButton, {
    calendar: calendar,
    date: initialDate
  });
};
exports.TestCalendarWithClearButton = TestCalendarWithClearButton;
