// @dhis2-ui/organisation-unit-tree/build/cjs/organisation-unit-tree.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Collapsed: true,
  Expanded: true,
  CustomExpandedImperativeOpen: true,
  MultipleRoots: true,
  CustomNodeLabel: true,
  FilteredRoot: true,
  Filtered: true,
  SelectedMultiple: true,
  Indeterminate: true,
  SingleSelection: true,
  NoSelection: true,
  Highlighted: true,
  ForceReloadAll: true,
  ForceReloadOneUnit: true,
  ReplaceRoots: true,
  Loading: true,
  RootLoading: true,
  RootError: true,
  LoadingErrorGrandchild: true,
  RTL: true
};
Object.defineProperty(exports, "Collapsed", {
  enumerable: true,
  get: function () {
    return _collapsed.Collapsed;
  }
});
Object.defineProperty(exports, "CustomExpandedImperativeOpen", {
  enumerable: true,
  get: function () {
    return _customExpandedImperativeOpen.CustomExpandedImperativeOpen;
  }
});
Object.defineProperty(exports, "CustomNodeLabel", {
  enumerable: true,
  get: function () {
    return _customNodeLabel.CustomNodeLabel;
  }
});
Object.defineProperty(exports, "Expanded", {
  enumerable: true,
  get: function () {
    return _expanded.Expanded;
  }
});
Object.defineProperty(exports, "Filtered", {
  enumerable: true,
  get: function () {
    return _filtered.Filtered;
  }
});
Object.defineProperty(exports, "FilteredRoot", {
  enumerable: true,
  get: function () {
    return _filteredRoot.FilteredRoot;
  }
});
Object.defineProperty(exports, "ForceReloadAll", {
  enumerable: true,
  get: function () {
    return _forceReloadAll.ForceReloadAll;
  }
});
Object.defineProperty(exports, "ForceReloadOneUnit", {
  enumerable: true,
  get: function () {
    return _forceReloadOneUnit.ForceReloadOneUnit;
  }
});
Object.defineProperty(exports, "Highlighted", {
  enumerable: true,
  get: function () {
    return _highlighted.Highlighted;
  }
});
Object.defineProperty(exports, "Indeterminate", {
  enumerable: true,
  get: function () {
    return _indeterminate.Indeterminate;
  }
});
Object.defineProperty(exports, "Loading", {
  enumerable: true,
  get: function () {
    return _loading.Loading;
  }
});
Object.defineProperty(exports, "LoadingErrorGrandchild", {
  enumerable: true,
  get: function () {
    return _loadingErrorGrandchild.LoadingErrorGrandchild;
  }
});
Object.defineProperty(exports, "MultipleRoots", {
  enumerable: true,
  get: function () {
    return _multipleRoots.MultipleRoots;
  }
});
Object.defineProperty(exports, "NoSelection", {
  enumerable: true,
  get: function () {
    return _noSelection.NoSelection;
  }
});
Object.defineProperty(exports, "RTL", {
  enumerable: true,
  get: function () {
    return _rtl.RTL;
  }
});
Object.defineProperty(exports, "ReplaceRoots", {
  enumerable: true,
  get: function () {
    return _replaceRoots.ReplaceRoots;
  }
});
Object.defineProperty(exports, "RootError", {
  enumerable: true,
  get: function () {
    return _rootError.RootError;
  }
});
Object.defineProperty(exports, "RootLoading", {
  enumerable: true,
  get: function () {
    return _rootLoading.RootLoading;
  }
});
Object.defineProperty(exports, "SelectedMultiple", {
  enumerable: true,
  get: function () {
    return _selectedMultiple.SelectedMultiple;
  }
});
Object.defineProperty(exports, "SingleSelection", {
  enumerable: true,
  get: function () {
    return _singleSelection.SingleSelection;
  }
});
exports.default = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireDefault(require("react"));
var _shared = require("./__stories__/shared.js");
var _index = require("./index.js");
var _collapsed = require("./__stories__/collapsed.js");
var _expanded = require("./__stories__/expanded.js");
var _customExpandedImperativeOpen = require("./__stories__/custom-expanded-imperative-open.js");
var _multipleRoots = require("./__stories__/multiple-roots.js");
var _customNodeLabel = require("./__stories__/custom-node-label.js");
var _filteredRoot = require("./__stories__/filtered-root.js");
var _filtered = require("./__stories__/filtered.js");
var _selectedMultiple = require("./__stories__/selected-multiple.js");
var _indeterminate = require("./__stories__/indeterminate.js");
var _singleSelection = require("./__stories__/single-selection.js");
var _noSelection = require("./__stories__/no-selection.js");
var _highlighted = require("./__stories__/highlighted.js");
var _forceReloadAll = require("./__stories__/force-reload-all.js");
var _forceReloadOneUnit = require("./__stories__/force-reload-one-unit.js");
var _replaceRoots = require("./__stories__/replace-roots.js");
var _loading = require("./__stories__/loading.js");
var _rootLoading = require("./__stories__/root-loading.js");
var _rootError = require("./__stories__/root-error.js");
var _loadingErrorGrandchild = require("./__stories__/loading-error-grandchild.js");
var _rtl = require("./__stories__/rtl.js");
var _developmentStories = require("./__stories__/development-stories.js");
Object.keys(_developmentStories).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _developmentStories[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _developmentStories[key];
    }
  });
});
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const subtitle = 'Display, manipulate and select organization units displayed in a hierarchical tree';
const description = `
This is a complex, controlled component. It needs access to an App Runtime data provider to fetch org unit data.

Several props require arrays of org. unit paths (referred to as \`orgUnitPathPropType[]\` in the table below). Take a look at the \`initiallyExpanded\` and \`filter\` props in the example to see an example of the paths format.

Example:

\`\`\`js
import { OrganisationUnitTree } from '@dhis2/ui'

const orgUnitTree = (
    <OrganisationUnitTree
        name="Root org unit"
        roots="A0000000000"
        onChange={onChange}
        onExpand={onExpand}
        onCollapse={onCollapse}
        onChildrenLoaded={onChildrenLoaded}
        // Notice the format of the org unit paths
        initiallyExpanded={['/A0000000000/A0000000001']}
        filter={['/A0000000000/A0000000001/A0000000003']}
    />
)
\`\`\`

`;
var _default = exports.default = {
  title: 'Organisation Unit Tree',
  component: _index.OrganisationUnitTree,
  decorators: [fn => /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
    data: _shared.customData
  }, fn())],
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  }
};
