// @dhis2-ui/select/build/cjs/single-select/single-select.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithRTL = exports.WithPrefixAndSelection = exports.WithPrefix = exports.WithPlaceholderAndSelection = exports.WithPlaceholder = exports.WithOptionsLoadingAndLoadingText = exports.WithOptionsAndOnChange = exports.WithOptionsAndLoading = exports.WithOptionsAndDisabled = exports.WithOptionsAndASelection = exports.WithOptionsASelectionAndDisabled = exports.WithOnFocus = exports.WithOnBlur = exports.WithMoreThanThreeOptionsAndA100PxMaxHeight = exports.WithMoreThanTenOptions = exports.WithInvalidOptions = exports.WithInvalidFilterableOptions = exports.WithInitialFocus = exports.WithFilterField = exports.WithDisabledOptionAndOnChange = exports.WithCustomOptionsAndOnChange = exports.WithClearButtonSelectionAndOnChange = exports.ShiftedIntoView = exports.FlippedPosition = exports.EmptyWithEmptyText = exports.EmptyWithEmptyComponent = exports.Empty = exports.Dense = exports.DefaultPosition = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _index = require("../index.js");
var _index2 = require("./index.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
Use a select component wherever the user needs to make a selection of one or more options from a list of 6 or more options. If there are less than 6 options to choose from and space permits, use checkboxes for multiple selection and radio buttons for single selection. If the user needs to make a complex selection with a specific ordering, use a transfer instead.

Learn more about using Selects at [Design System: Select](https://github.com/dhis2/design-system/blob/master/molecules/select.md).

\`\`\`js
import { SingleSelect, SingleSelectOption } from '@dhis2/ui'
\`\`\`

_**Note:** Due to demo limitations on this page, only one representative example is rendered here. For more (interactive) examples, see individual stories in the 'Canvas' tab._
`;
const eventHandler = handlerName => (payload, event) => {
  console.log(`${handlerName} payload`, payload);
  console.log(`${handlerName} event`, event);
};
const onChange = eventHandler('onChange');
const onFocus = eventHandler('onFocus');
const onBlur = eventHandler('onBlur');
const CustomSingleSelectOption = _ref => {
  let {
    label,
    onClick
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    onClick: e => onClick({}, e)
  }, label);
};
CustomSingleSelectOption.propTypes = {
  label: _propTypes.default.string,
  onClick: _propTypes.default.func
};
const requiredIfArgType = {
  table: {
    type: {
      summary: 'string'
    },
    control: {
      type: 'string'
    }
  }
};
var _default = exports.default = {
  title: 'Single Select',
  component: _index2.SingleSelect,
  parameters: {
    docs: {
      description: {
        component: description
      },
      // Disable stories in docs page by default, then use one representative examples
      disable: true,
      // Use iframe to keep dropdown menu inside story for docs page
      inlineStories: false,
      iframeHeight: '300px'
    }
  },
  // Use 'onChange' as a default arg, otherwise components throw an error when interacted with
  // (Maybe this could be handled in the component - as a required prop)
  args: {
    className: 'select',
    onChange
  },
  argTypes: {
    valid: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    clearText: {
      ...requiredIfArgType
    },
    noMatchText: {
      ...requiredIfArgType
    }
  }
};
const WithOptionsTemplate = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}));
const EmptyTemplate = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args);
const WithOptionsAndOnChange = exports.WithOptionsAndOnChange = WithOptionsTemplate.bind({});
WithOptionsAndOnChange.storyName = 'With options and onChange';
const WithOptionsAndASelection = exports.WithOptionsAndASelection = WithOptionsTemplate.bind({});
WithOptionsAndASelection.args = {
  selected: '1'
};
// Enable this story as the primary for docs as a representative example
WithOptionsAndASelection.parameters = {
  docs: {
    disable: false
  }
};
WithOptionsAndASelection.storyName = 'With options and a selection';
const WithOnFocus = exports.WithOnFocus = WithOptionsTemplate.bind({});
WithOnFocus.args = {
  onFocus
};
WithOnFocus.storyName = 'With on focus';
const WithOnBlur = exports.WithOnBlur = WithOptionsTemplate.bind({});
WithOnBlur.args = {
  onBlur
};
WithOnBlur.storyName = 'With onBlur';
const WithCustomOptionsAndOnChange = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(CustomSingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(CustomSingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(CustomSingleSelectOption, {
  value: "3",
  label: "option three"
}));
exports.WithCustomOptionsAndOnChange = WithCustomOptionsAndOnChange;
WithCustomOptionsAndOnChange.storyName = 'With custom options and onChange';
const WithInvalidOptions = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement("div", null, "invalid one"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement("div", null, "invalid two"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement("div", null, "invalid three"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), null, undefined, false);
exports.WithInvalidOptions = WithInvalidOptions;
const WithInvalidFilterableOptions = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement("div", null, "invalid one"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement("div", null, "invalid two"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement("div", null, "invalid three"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}));
exports.WithInvalidFilterableOptions = WithInvalidFilterableOptions;
WithInvalidFilterableOptions.args = {
  filterable: true,
  noMatchText: 'No match found'
};
const WithInitialFocus = exports.WithInitialFocus = EmptyTemplate.bind({});
WithInitialFocus.args = {
  initialFocus: true
};
WithInitialFocus.parameters = {
  docs: {
    disable: true
  }
};
const Dense = exports.Dense = WithOptionsTemplate.bind({});
Dense.args = {
  dense: true,
  placeholder: 'Dense sized select'
};
Dense.storyName = 'Dense';
const Empty = exports.Empty = EmptyTemplate.bind({});
const EmptyWithEmptyText = exports.EmptyWithEmptyText = EmptyTemplate.bind({});
EmptyWithEmptyText.args = {
  empty: 'Custom empty text'
};
const EmptyWithEmptyComponent = exports.EmptyWithEmptyComponent = EmptyTemplate.bind({});
EmptyWithEmptyComponent.args = {
  empty: /*#__PURE__*/_react.default.createElement("div", {
    className: "custom-empty"
  }, "Custom empty component")
};
const WithOptionsAndLoading = exports.WithOptionsAndLoading = WithOptionsTemplate.bind({});
WithOptionsAndLoading.args = {
  loading: true
};
const WithOptionsLoadingAndLoadingText = exports.WithOptionsLoadingAndLoadingText = WithOptionsTemplate.bind({});
WithOptionsLoadingAndLoadingText.args = {
  ...WithOptionsAndLoading.args,
  loadingText: 'Loading options'
};
WithOptionsLoadingAndLoadingText.storyName = 'With options, loading and loading text';
const WithMoreThanTenOptions = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
}));
exports.WithMoreThanTenOptions = WithMoreThanTenOptions;
const WithMoreThanThreeOptionsAndA100PxMaxHeight = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
}));
exports.WithMoreThanThreeOptionsAndA100PxMaxHeight = WithMoreThanThreeOptionsAndA100PxMaxHeight;
WithMoreThanThreeOptionsAndA100PxMaxHeight.args = {
  maxHeight: '100px'
};
WithMoreThanThreeOptionsAndA100PxMaxHeight.storyName = 'With more than three options and a 100px max-height';
const WithOptionsAndDisabled = exports.WithOptionsAndDisabled = WithOptionsTemplate.bind({});
WithOptionsAndDisabled.args = {
  disabled: true
};
WithOptionsAndDisabled.storyName = 'With options and disabled';
const WithOptionsASelectionAndDisabled = exports.WithOptionsASelectionAndDisabled = WithOptionsTemplate.bind({});
WithOptionsASelectionAndDisabled.args = {
  ...WithOptionsAndDisabled.args,
  ...WithOptionsAndASelection.args
};
WithOptionsASelectionAndDisabled.storyName = 'With options, a selection and disabled';
const WithPrefix = exports.WithPrefix = WithOptionsTemplate.bind({});
WithPrefix.args = {
  prefix: 'Prefix text',
  selected: '1'
};
const WithPrefixAndSelection = exports.WithPrefixAndSelection = WithOptionsTemplate.bind({});
WithPrefixAndSelection.args = {
  ...WithPrefix.args,
  ...WithOptionsAndASelection.args
};
const WithRTL = args => {
  // as options are rendered in Portal, the body dir (of the iframe) needs to be set to 'rtl'
  (0, _react.useEffect)(() => {
    document.body.dir = 'rtl';
    return () => {
      document.body.dir = 'ltr';
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement("div", {
    dir: "rtl"
  }, /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
    value: "1",
    label: "\u0627\u0644\u062E\u064A\u0627\u0631 1"
  }), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
    value: "2",
    label: "\u0627\u0644\u062E\u064A\u0627\u0631 2"
  }), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
    value: "3",
    label: "\u0627\u0644\u062E\u064A\u0627\u0631 3"
  })));
};
exports.WithRTL = WithRTL;
WithRTL.args = {
  prefix: 'نص البادئة',
  ...WithOptionsAndASelection.args
};
WithRTL.storyName = 'RTL Text';
const WithPlaceholder = exports.WithPlaceholder = WithOptionsTemplate.bind({});
WithPlaceholder.args = {
  placeholder: 'Placeholder text'
};
const WithPlaceholderAndSelection = exports.WithPlaceholderAndSelection = WithOptionsTemplate.bind({});
WithPlaceholderAndSelection.args = {
  ...WithPlaceholder.args,
  ...WithOptionsAndASelection.args
};
const WithDisabledOptionAndOnChange = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "disabled option",
  disabled: true
}));
exports.WithDisabledOptionAndOnChange = WithDisabledOptionAndOnChange;
WithDisabledOptionAndOnChange.storyName = 'With disabled option and onChange';
const WithClearButtonSelectionAndOnChange = exports.WithClearButtonSelectionAndOnChange = WithOptionsTemplate.bind({});
WithClearButtonSelectionAndOnChange.args = {
  ...WithOptionsAndASelection.args,
  clearable: true,
  clearText: 'Clear'
};
WithClearButtonSelectionAndOnChange.storyName = 'With clear button, selection and onChange';
const WithFilterField = exports.WithFilterField = WithOptionsTemplate.bind({});
WithFilterField.args = {
  ...WithInvalidFilterableOptions.args
};
const DefaultPosition = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
})), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "2006483503"
}, ["#root{margin-bottom:2000px;}"]));
exports.DefaultPosition = DefaultPosition;
const FlippedPosition = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
})), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "1520050670"
}, ["html,body,#root{position:relative;height:500px;max-height:500px;}", "#root{padding-top:400px !important;}"]));
exports.FlippedPosition = FlippedPosition;
const ShiftedIntoView = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
})), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "1828862170"
}, ["html,body,#root{position:relative;height:300px !important;max-height:300px;}", "#root{padding-top:130px !important;}"]));
exports.ShiftedIntoView = ShiftedIntoView;
