// @dhis2-ui/chip/build/cjs/chip.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Chip = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _content = require("./content.js");
var _icon = require("./icon.js");
var _remove = require("./remove.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const DEFAULT_INLINE_MARGIN = '4';
const Chip = _ref => {
  var _ref2, _ref3, _ref4, _ref5;
  let {
    selected,
    dense,
    disabled,
    dragging,
    overflow,
    className,
    children,
    onRemove,
    onClick,
    icon,
    dataTest = 'dhis2-uicore-chip',
    marginBottom = 4,
    marginLeft,
    marginRight,
    marginTop = 4,
    marginInlineStart,
    marginInlineEnd
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("span", {
    onClick: e => {
      if (!disabled && onClick) {
        onClick({}, e);
      }
    },
    "data-test": dataTest,
    className: _style.default.dynamic([["3780556490", [_uiConstants.colors.grey200, _uiConstants.colors.grey900, _uiConstants.colors.grey300, _uiConstants.theme.secondary600, _uiConstants.colors.white]], ["1687525377", [marginBottom && `margin-bottom: ${marginBottom}px;`, (_ref2 = marginInlineStart !== null && marginInlineStart !== void 0 ? marginInlineStart : marginLeft) !== null && _ref2 !== void 0 ? _ref2 : DEFAULT_INLINE_MARGIN, (_ref3 = marginInlineEnd !== null && marginInlineEnd !== void 0 ? marginInlineEnd : marginRight) !== null && _ref3 !== void 0 ? _ref3 : DEFAULT_INLINE_MARGIN, marginTop && `margin-top: ${marginTop}px`]]]) + " " + ((0, _classnames.default)(className, {
      selected,
      dense,
      disabled,
      dragging
    }) || "")
  }, /*#__PURE__*/_react.default.createElement(_icon.Icon, {
    icon: icon,
    dataTest: `${dataTest}-icon`
  }), /*#__PURE__*/_react.default.createElement(_content.Content, {
    overflow: overflow
  }, children), /*#__PURE__*/_react.default.createElement(_remove.Remove, {
    onRemove: onRemove,
    dataTest: `${dataTest}-remove`
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3780556490",
    dynamic: [_uiConstants.colors.grey200, _uiConstants.colors.grey900, _uiConstants.colors.grey300, _uiConstants.theme.secondary600, _uiConstants.colors.white]
  }, [`span.__jsx-style-dynamic-selector{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:32px;border-radius:16px;background-color:${_uiConstants.colors.grey200};font-size:14px;line-height:16px;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;color:${_uiConstants.colors.grey900};}`, ".dense.__jsx-style-dynamic-selector{height:24px;font-size:13px;line-height:15px;}", `span.__jsx-style-dynamic-selector:hover{background-color:${_uiConstants.colors.grey300};}`, `.selected.__jsx-style-dynamic-selector{background-color:${_uiConstants.theme.secondary600};font-weight:500;}`, ".selected.__jsx-style-dynamic-selector:hover{background-color:#00695c;}", `.selected.__jsx-style-dynamic-selector,.selected.__jsx-style-dynamic-selector .icon.__jsx-style-dynamic-selector,.selected.__jsx-style-dynamic-selector .remove-icon.__jsx-style-dynamic-selector{color:${_uiConstants.colors.white};}`, ".disabled.__jsx-style-dynamic-selector{cursor:not-allowed;opacity:0.3;}", ".disabled.__jsx-style-dynamic-selector .remove-icon.__jsx-style-dynamic-selector{pointer-events:none;}", ".dragging.__jsx-style-dynamic-selector{box-shadow:0 3px 1px -2px rgba(0,0,0,0.2), 0 2px 2px 0 rgba(0,0,0,0.14), 0 1px 5px 0 rgba(0,0,0,0.12);}"]), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1687525377",
    dynamic: [marginBottom && `margin-bottom: ${marginBottom}px;`, (_ref2 = marginInlineStart !== null && marginInlineStart !== void 0 ? marginInlineStart : marginLeft) !== null && _ref2 !== void 0 ? _ref2 : DEFAULT_INLINE_MARGIN, (_ref3 = marginInlineEnd !== null && marginInlineEnd !== void 0 ? marginInlineEnd : marginRight) !== null && _ref3 !== void 0 ? _ref3 : DEFAULT_INLINE_MARGIN, marginTop && `margin-top: ${marginTop}px`]
  }, [`span.__jsx-style-dynamic-selector{${marginBottom && `margin-bottom: ${marginBottom}px;`} margin-inline-start:${(_ref4 = marginInlineStart !== null && marginInlineStart !== void 0 ? marginInlineStart : marginLeft) !== null && _ref4 !== void 0 ? _ref4 : DEFAULT_INLINE_MARGIN}px;margin-inline-end:${(_ref5 = marginInlineEnd !== null && marginInlineEnd !== void 0 ? marginInlineEnd : marginRight) !== null && _ref5 !== void 0 ? _ref5 : DEFAULT_INLINE_MARGIN}px;${marginTop && `margin-top: ${marginTop}px`};}`]));
};
exports.Chip = Chip;
Chip.propTypes = {
  children: _propTypes.default.any,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  dense: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  dragging: _propTypes.default.bool,
  icon: _propTypes.default.element,
  /** `margin-bottom` value, applied in `px`  */
  marginBottom: _propTypes.default.number,
  /** `margin-inline-end` value, applied in `px`  */
  marginInlineEnd: _propTypes.default.number,
  /** `margin-inline-start` value, applied in `px`  */
  marginInlineStart: _propTypes.default.number,
  /** `margin-inline-start` value, applied in `px`  */
  marginLeft: _propTypes.default.number,
  /** `margin-inline-end` value, applied in `px`  */
  marginRight: _propTypes.default.number,
  /** `margin-top` value, applied in `px`  */
  marginTop: _propTypes.default.number,
  overflow: _propTypes.default.bool,
  selected: _propTypes.default.bool,
  onClick: _propTypes.default.func,
  onRemove: _propTypes.default.func
};
