// @dhis2-ui/chip/build/cjs/chip.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnRemove = exports.WithOnClick = exports.WithIcon = exports.WithChildren = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _chip = require("./chip.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onClick = window.Cypress && window.Cypress.cy.stub();
window.onRemove = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'Chip'
};
const Default = () => /*#__PURE__*/_react.default.createElement(_chip.Chip, null, "Message");
exports.Default = Default;
const WithOnClick = () => /*#__PURE__*/_react.default.createElement(_chip.Chip, {
  onClick: window.onClick
}, "Chippy");
exports.WithOnClick = WithOnClick;
const WithOnRemove = () => /*#__PURE__*/_react.default.createElement(_chip.Chip, {
  onRemove: window.onRemove
}, "Chipmunk");
exports.WithOnRemove = WithOnRemove;
const WithChildren = () => /*#__PURE__*/_react.default.createElement(_chip.Chip, null, "I am a child");
exports.WithChildren = WithChildren;
const WithIcon = () => /*#__PURE__*/_react.default.createElement(_chip.Chip, {
  icon: /*#__PURE__*/_react.default.createElement("div", null, "Icon")
}, "Message");
exports.WithIcon = WithIcon;
