// @dhis2-ui/header-bar/build/cjs/profile-menu/update-notification.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UpdateNotification = UpdateNotification;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _appRuntime = require("@dhis2/app-runtime");
var _uiConstants = require("@dhis2/ui-constants");
var _menu = require("@dhis2-ui/menu");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _headerBarContext = require("../header-bar-context.js");
var _index = _interopRequireDefault(require("../locales/index.js"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function UpdateNotification(_ref) {
  let {
    hideProfileMenu
  } = _ref;
  const {
    appName
  } = (0, _appRuntime.useConfig)();
  const {
    updateAvailable,
    onApplyAvailableUpdate
  } = (0, _headerBarContext.useHeaderBarContext)();
  const onClick = () => {
    hideProfileMenu();
    onApplyAvailableUpdate === null || onApplyAvailableUpdate === void 0 ? void 0 : onApplyAvailableUpdate();
  };
  const updateNotificationLabel = /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["4135170305", [_uiConstants.colors.blue600]]]) + " " + "root"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["4135170305", [_uiConstants.colors.blue600]]]) + " " + "badge"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["4135170305", [_uiConstants.colors.blue600]]]) + " " + "spacer"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["4135170305", [_uiConstants.colors.blue600]]]) + " " + "message"
  }, appName ? _index.default.t('New {{appName}} version available', {
    appName
  }) : _index.default.t('New app version available'), /*#__PURE__*/_react.default.createElement("br", {
    className: _style.default.dynamic([["4135170305", [_uiConstants.colors.blue600]]])
  }), _index.default.t('Click to reload')), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4135170305",
    dynamic: [_uiConstants.colors.blue600]
  }, [".root.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;font-size:14px;line-height:17px;}", `.badge.__jsx-style-dynamic-selector{display:inline-block;width:12px;height:12px;margin:0 8px;border-radius:6px;background-color:${_uiConstants.colors.blue600};}`, ".spacer.__jsx-style-dynamic-selector{display:inline-block;width:8px;}", ".message.__jsx-style-dynamic-selector{display:inline-block;}"]));
  return updateAvailable ? /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    dense: true,
    onClick: onClick,
    label: updateNotificationLabel,
    dataTest: "dhis2-ui-headerbar-updatenotification"
  }) : null;
}
UpdateNotification.propTypes = {
  hideProfileMenu: _propTypes.default.func.isRequired
};
