// @dhis2-ui/transfer/build/es/__e2e__/disabled-transfer-buttons.e2e.stories.js
import React from 'react';
import { Transfer } from '../transfer.js';
import { options } from './common/options.js';
import { statefulDecorator } from './common/stateful-decorator.js';
export default {
  title: 'Transfer Disabled Transfer Buttons',
  decorators: [statefulDecorator()]
};
export const NoOptions = (_, _ref) => {
  let {
    selected,
    onChange
  } = _ref;
  return /*#__PURE__*/React.createElement(Transfer, {
    selected: selected,
    onChange: onChange,
    options: []
  });
};
export const HasOptions = (_, _ref2) => {
  let {
    selected,
    onChange
  } = _ref2;
  return /*#__PURE__*/React.createElement(Transfer, {
    selected: selected,
    onChange: onChange,
    options: options
  });
};
export const SomeOptionsSelected = (_, _ref3) => {
  let {
    selected,
    onChange
  } = _ref3;
  return /*#__PURE__*/React.createElement(Transfer, {
    selected: selected,
    onChange: onChange,
    options: options
  });
};
export const OnlyDisabledSourceOptions = (_, _ref4) => {
  let {
    selected,
    onChange
  } = _ref4;
  return /*#__PURE__*/React.createElement(Transfer, {
    selected: selected,
    onChange: onChange,
    options: [{
      ...options[0],
      disabled: true
    }]
  });
};
SomeOptionsSelected.story = {
  decorators: [statefulDecorator({
    initialState: options.slice(0, 4).map(_ref5 => {
      let {
        value
      } = _ref5;
      return value;
    })
  })]
};
export const AllOptionsSelected = (_, _ref6) => {
  let {
    selected,
    onChange
  } = _ref6;
  return /*#__PURE__*/React.createElement(Transfer, {
    selected: selected,
    onChange: onChange,
    options: options
  });
};
AllOptionsSelected.story = {
  decorators: [statefulDecorator({
    initialState: options.map(_ref7 => {
      let {
        value
      } = _ref7;
      return value;
    })
  })]
};
