// @dhis2-ui/transfer/build/es/__e2e__/notify_at_end_of_list.e2e.stories.js
import React from 'react';
import { Transfer } from '../transfer.js';
import { options } from './common/options.js';
import { statefulDecorator } from './common/stateful-decorator.js';
export default {
  title: 'Transfer End Of List',
  decorators: [statefulDecorator()]
};
window.onEndReached = window.Cypress ? window.Cypress.cy.stub() : () => console.log('onEndReached');
window.onEndReachedPicked = window.Cypress ? window.Cypress.cy.stub() : () => console.log('onEndReachedPicked');
export const FullSourceList = (_, _ref) => {
  let {
    selected,
    onChange
  } = _ref;
  return /*#__PURE__*/React.createElement(Transfer, {
    options: options,
    selected: selected,
    onChange: onChange,
    onEndReached: window.onEndReached
  });
};
export const FullPickedList = (_, _ref2) => {
  let {
    selected,
    onChange
  } = _ref2;
  return /*#__PURE__*/React.createElement(Transfer, {
    options: options,
    selected: selected,
    onChange: onChange,
    onEndReachedPicked: window.onEndReachedPicked
  });
};
FullPickedList.story = {
  decorators: [statefulDecorator({
    initialState: options.map(_ref3 => {
      let {
        value
      } = _ref3;
      return value;
    })
  })]
};
export const PartialSourceList = (_, _ref4) => {
  let {
    selected,
    onChange
  } = _ref4;
  return /*#__PURE__*/React.createElement(Transfer, {
    options: options.slice(0, 4),
    selected: selected,
    onChange: onChange,
    onEndReached: window.onEndReached
  });
};
export const PartialPickedList = (_, _ref5) => {
  let {
    selected,
    onChange
  } = _ref5;
  return /*#__PURE__*/React.createElement(Transfer, {
    options: options,
    selected: selected,
    onChange: onChange,
    onEndReachedPicked: window.onEndReachedPicked
  });
};
PartialPickedList.story = {
  decorators: [statefulDecorator({
    initialState: options.slice(0, 4).map(_ref6 => {
      let {
        value
      } = _ref6;
      return value;
    })
  })]
};
