// @dhis2-ui/checkbox/build/es/checkbox/checkbox.e2e.stories.js
import React from 'react';
import { Checkbox } from './index.js';
window.onClick = window.Cypress && window.Cypress.cy.stub();
window.onChange = window.Cypress && window.Cypress.cy.stub();
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'Checkbox'
};
export const WithOnChange = () => /*#__PURE__*/React.createElement(Checkbox, {
  name: "Ex",
  label: "Checkbox",
  value: "default",
  onChange: window.onChange
});
export const WithInitialFocusAndOnBlur = () => /*#__PURE__*/React.createElement(Checkbox, {
  initialFocus: true,
  name: "Ex",
  label: "Checkbox",
  value: "default",
  onBlur: window.onBlur
});
export const WithOnFocus = () => /*#__PURE__*/React.createElement(Checkbox, {
  name: "Ex",
  label: "Checkbox",
  value: "default",
  onFocus: window.onFocus
});
export const DisabledWithOnClick = () => /*#__PURE__*/React.createElement(Checkbox, {
  name: "Ex",
  label: "Checkbox",
  value: "default",
  disabled: true,
  onClick: window.onClick
});
export const WithLabel = () => /*#__PURE__*/React.createElement(Checkbox, {
  name: "Ex",
  label: "The label",
  value: "default"
});
export const WithInitialFocus = () => /*#__PURE__*/React.createElement(Checkbox, {
  name: "Ex",
  label: "The label",
  value: "default",
  initialFocus: true
});
export const IndeterminateProp = () => /*#__PURE__*/React.createElement(Checkbox, {
  name: "Ex",
  label: "The label",
  value: "default",
  indeterminate: true
});
export const NoIndeterminateProp = () => /*#__PURE__*/React.createElement(Checkbox, {
  name: "Ex",
  label: "The label",
  value: "default"
});
