// @dhis2-ui/table/build/cjs/data-table/data-table-row/expand-handle-cell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ExpandHandleCell = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiIcons = require("@dhis2/ui-icons");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = require("../table-elements/index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const pointer = {
  styles: /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "927553068"
  }, ["td.jsx-927553068{cursor:pointer;}"]),
  className: "jsx-927553068"
};
const ExpandHandleCell = _ref => {
  let {
    expanded,
    onClick
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_index.TableDataCell, {
    className: (0, _classnames.default)(pointer.className, {
      expanded
    }),
    onClick: () => onClick({
      expanded: !expanded
    }),
    width: "48px"
  }, expanded ? /*#__PURE__*/_react.default.createElement(_uiIcons.IconChevronUp24, null) : /*#__PURE__*/_react.default.createElement(_uiIcons.IconChevronDown24, null), pointer.styles);
};
exports.ExpandHandleCell = ExpandHandleCell;
ExpandHandleCell.propTypes = {
  expanded: _propTypes.default.bool,
  onClick: _propTypes.default.func
};
