// @dhis2-ui/table/build/cjs/data-table/data-table-row/expanded-row.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ExpandedRow = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = require("../table-elements/index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const {
  className: cellClassName,
  styles
} = {
  styles: /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "626485122"
  }, [`tr.expanded:hover+tr>td.jsx-626485122{background-color:${_uiConstants.colors.grey100};}`, `tr.expanded:active+tr>td.jsx-626485122{background-color:${_uiConstants.colors.grey200};}`, "tr.selected.expanded:hover+tr.jsx-626485122>td.jsx-626485122{background-color:#cdeae8;}"]),
  className: "jsx-626485122"
};
const ExpandedRow = _ref => {
  let {
    children,
    className,
    colSpan,
    dataTest,
    selected,
    setIsHoveringExpandedContent
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_index.TableRow, {
    onMouseOver: () => setIsHoveringExpandedContent(true),
    onMouseOut: () => setIsHoveringExpandedContent(false),
    className: (0, _classnames.default)({
      [`${className}-expandedrow`]: !!className
    }),
    selected: selected,
    dataTest: dataTest && `${dataTest}-expandedrow`
  }, /*#__PURE__*/_react.default.createElement(_index.TableDataCell, {
    className: cellClassName,
    colSpan: colSpan
  }, children, styles));
};
exports.ExpandedRow = ExpandedRow;
ExpandedRow.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  colSpan: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  selected: _propTypes.default.bool,
  setIsHoveringExpandedContent: _propTypes.default.func
};
