// @dhis2-ui/field/build/cjs/field/field.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _input = require("@dhis2-ui/input");
var _react = _interopRequireDefault(require("react"));
var _field = require("./field.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
A useful container for form components, including a label, help text, and validation text.

\`\`\`js
import { Field } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Field',
  component: _field.Field,
  argTypes: {
    valid: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: {
      ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: {
      ..._uiConstants.sharedPropTypes.statusArgType
    }
  },
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Default = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_field.Field, args, /*#__PURE__*/_react.default.createElement(_input.Input, {
  onChange: () => {
    console.log('Nothing happens');
  },
  name: "input",
  label: "An input"
})), /*#__PURE__*/_react.default.createElement(_field.Field, {
  helpText: "Help!",
  label: "Second field"
}, /*#__PURE__*/_react.default.createElement(_input.Input, {
  onChange: () => {
    console.log('Nothing happens');
  },
  name: "input2",
  label: "An second input"
})));
exports.Default = Default;
Default.args = {
  label: 'First field (change me with controls)',
  helpText: 'Help text!',
  validationText: "I'm validation text"
};
