// @dhis2-ui/switch/build/es/switch/switch.e2e.stories.js
import React from 'react';
import { Switch } from './index.js';
window.onChange = window.Cypress && window.Cypress.cy.stub();
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'Switch'
};
export const WithOnChange = () => /*#__PURE__*/React.createElement(Switch, {
  name: "Ex",
  label: "Switch",
  value: "default",
  onChange: window.onChange
});
export const WithInitialFocusAndOnBlur = () => /*#__PURE__*/React.createElement(Switch, {
  initialFocus: true,
  name: "Ex",
  label: "Switch",
  value: "default",
  onBlur: window.onBlur
});
export const WithOnFocus = () => /*#__PURE__*/React.createElement(Switch, {
  name: "Ex",
  label: "Switch",
  value: "default",
  onFocus: window.onFocus
});
export const WithDisabled = () => /*#__PURE__*/React.createElement(Switch, {
  name: "Ex",
  label: "Switch",
  value: "default",
  disabled: true
});
export const WithLabel = () => /*#__PURE__*/React.createElement(Switch, {
  name: "Ex",
  label: "The label",
  value: "default"
});
export const WithInitialFocus = () => /*#__PURE__*/React.createElement(Switch, {
  name: "Ex",
  label: "The label",
  value: "default",
  initialFocus: true
});
