// @dhis2-ui/table/build/cjs/data-table/data-table-column-header/filter-handle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FilterHandle = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _uiIcons = require("@dhis2/ui-icons");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = _interopRequireDefault(require("../../locales/index.js"));
var _index2 = require("../table-elements/index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const FilterHandle = _ref => {
  let {
    active,
    name,
    onClick
  } = _ref;
  const filterIconColor = active ? _uiConstants.colors.blue700 : _uiConstants.colors.grey600;
  const clickHandler = onClick ? event => {
    onClick({
      name,
      show: !active
    }, event);
  } : undefined;
  return /*#__PURE__*/_react.default.createElement(_index2.TableHeaderCellAction, {
    onClick: clickHandler,
    title: _index.default.t('Toggle filter')
  }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconSearch16, {
    color: filterIconColor
  }));
};
exports.FilterHandle = FilterHandle;
FilterHandle.propTypes = {
  active: _propTypes.default.bool,
  name: _propTypes.default.string,
  onClick: _propTypes.default.func
};
