// @dhis2-ui/button/build/cjs/button-strip/button-strip.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ButtonStrip = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = require("@dhis2/prop-types");
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes2 = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ButtonStrip = _ref => {
  let {
    className,
    children,
    middle,
    end,
    dataTest = 'dhis2-uicore-buttonstrip'
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["3148822533", [_uiConstants.spacers.dp8]]]) + " " + ((0, _classnames.default)(className, {
      start: !middle && !end,
      middle,
      end
    }) || "")
  }, _react.Children.map(children, child => /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["3148822533", [_uiConstants.spacers.dp8]]]) + " " + "box"
  }, child)), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3148822533",
    dynamic: [_uiConstants.spacers.dp8]
  }, ["div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", "div.middle.__jsx-style-dynamic-selector{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}", "div.end.__jsx-style-dynamic-selector{-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;}", `.box.__jsx-style-dynamic-selector{margin-inline-start:${_uiConstants.spacers.dp8};}`, ".box.__jsx-style-dynamic-selector:first-child{margin-inline-start:0;}"]));
};
exports.ButtonStrip = ButtonStrip;
const alignmentPropType = (0, _propTypes.mutuallyExclusive)(['middle', 'end'], _propTypes2.default.bool);
ButtonStrip.propTypes = {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  dataTest: _propTypes2.default.string,
  /** Horizontal alignment for buttons. Mutually exclusive with `middle` prop */
  end: alignmentPropType,
  /** Horizontal alignment. Mutually exclusive with `end` prop */
  middle: alignmentPropType
};
