// @dhis2-ui/modal/build/es/modal/modal.prod.stories.js
import _JSXStyle from "styled-jsx/style";
import { sharedPropTypes } from '@dhis2/ui-constants';
import { Box } from '@dhis2-ui/box';
import { Button, ButtonStrip } from '@dhis2-ui/button';
import { Center as CenteredContent } from '@dhis2-ui/center';
import { InputField } from '@dhis2-ui/input';
import { CircularLoader } from '@dhis2-ui/loader';
import { FlyoutMenu, MenuDivider, MenuItem, MenuSectionHeader } from '@dhis2-ui/menu';
import { NoticeBox } from '@dhis2-ui/notice-box';
import { SingleSelect, SingleSelectOption } from '@dhis2-ui/select';
import { Tooltip } from '@dhis2-ui/tooltip';
import React, { useEffect, useState } from 'react';
import { ModalActions, ModalContent, ModalTitle } from '../index.js';
import { Modal } from './modal.js';
const description = `
_**Note**: For performance reasons, only one representative example is shown here. For more (interactive) examples, see individual stories in the Canvas tab._

A modal focuses user attention on a single task or piece of information inside a container. A modal blocks the rest of the application until it is dismissed.

#### Usage

A modal should be used to focus user attention on a single task or piece of information. Modals take over the entire screen and should be used sparingly to avoid interrupting a user's flow too often.

Use a modal in the following cases:

- to collect short, focussed user input that is blocking progress
- to present critical information that a user needs to acknowledge before continuing
- to ask the user to confirm a destructive action that cannot be undone

Do not use a modal in the following cases:

- to display unrelated or non-critical information; use an alert bar or a notice box instead
- to display complex, workflows that span multiple screens; navigate to a new, full page in the app instead

#### Children

Modals should be used with children \`<ModalTitle>\` (optional), \`<ModalContent>\` (required), and \`<ModalActions>\` (recommended)

See more about modal usage, including alignment and sizing, at [Design System: Modals](https://github.com/dhis2/design-system/blob/master/molecules/modal.md#usage).

\`\`\`js
import { Modal } from '@dhis2/ui'
\`\`\`
`;
const say = something => () => alert(something);
window.onClose = (payload, event) => {
  console.log('onClose payload', payload);
  console.log('onClose event', event);
};
const onClose = function () {
  return window.onClose(...arguments);
};
export default {
  title: 'Modal',
  component: Modal,
  parameters: {
    docs: {
      // Use iframes to contain modals in docs page (otherwise chaos ensues)
      inlineStories: false,
      iframeHeight: '500px',
      /**
       * Due to iframes being very slow, disable stories on the docs page by default and
       * make one representative story as the primary ('SmallTitleContentAction')
       */
      disable: true,
      description: {
        component: description
      }
    }
  },
  argTypes: {
    small: {
      ...sharedPropTypes.sizeArgType
    },
    large: {
      ...sharedPropTypes.sizeArgType
    },
    position: {
      ...sharedPropTypes.insideAlignmentArgType
    }
  }
};
export const DefaultContent = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."));
DefaultContent.args = {
  onClose: onClose
};
DefaultContent.storyName = 'Default: Content';
export const AlignmentMiddle = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."));
AlignmentMiddle.args = {
  onClose,
  position: 'middle'
};
AlignmentMiddle.storyName = 'Alignment: Middle';
export const AlignmentBottom = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."));
AlignmentBottom.args = {
  onClose,
  position: 'bottom'
};
AlignmentBottom.storyName = 'Position: Bottom';
export const SmallTitleContentAction = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a small modal with title, content and primary action"), /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
// Have this be the primary story on the docs page
SmallTitleContentAction.parameters = {
  docs: {
    disable: false,
    source: {
      type: 'dynamic'
    }
  }
};
SmallTitleContentAction.args = {
  small: true,
  onClose
};
SmallTitleContentAction.storyName = 'Small: Title, Content, Action';
export const MediumTitleContentAction = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a medium modal with title, content and primary action"), /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
MediumTitleContentAction.storyName = 'Medium: Title, Content, Action';
export const LargeTitleContentPrimary = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a large modal with title, content and primary action"), /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
LargeTitleContentPrimary.args = {
  large: true
};
LargeTitleContentPrimary.storyName = 'Large: Title, Content, Primary';
export const FluidTitleContentPrimary = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a modal using custom dimensions, with title, content and primary action"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement("div", {
  style: {
    display: 'flex',
    flexFlow: 'row wrap'
  }
}, /*#__PURE__*/React.createElement("div", {
  style: {
    width: '600px',
    backgroundColor: '#fea'
  }
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/React.createElement("div", {
  style: {
    width: '300px',
    backgroundColor: '#eaf'
  }
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."))), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
FluidTitleContentPrimary.args = {
  fluid: true,
  position: 'top'
};
FluidTitleContentPrimary.storyName = 'Fluid (Custom sizes): Title, Content, Primary';
export const SmallContentPrimary = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
SmallContentPrimary.args = {
  small: true
};
SmallContentPrimary.storyName = 'Small: Content & Primary';
export const SmallDestructivePrimary = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));
SmallDestructivePrimary.args = {
  small: true
};
SmallDestructivePrimary.storyName = 'Small: Destructive Primary';
export const SmallClickableScreenCover = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a modal with clickable screen cover"), /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  primary: true,
  onClick: say('Button primary')
}, "Primary action"))));
SmallClickableScreenCover.args = {
  small: true,
  onClose: args => {
    onClose(args);
    say('Clickable screen cover')();
  }
};
SmallClickableScreenCover.storyName = 'Small: Clickable screen cover';
export const TopScrollable = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a modal with scrollable content"), /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));
TopScrollable.args = {
  ...SmallClickableScreenCover.args
};
TopScrollable.storyName = 'Top: scrollable';
export const MiddleScrollable = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a modal with scrollable content"), /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));
MiddleScrollable.args = {
  ...TopScrollable.args,
  position: 'middle'
};
MiddleScrollable.storyName = 'Middle: scrollable';
export const BottomScrollable = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a modal with scrollable content"), /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));
BottomScrollable.args = {
  ...TopScrollable.args,
  position: 'bottom'
};
BottomScrollable.storyName = 'Bottom: scrollable';
export const SmallLongTitle = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This headline should break into multiple lines because it's way too long for one!"), /*#__PURE__*/React.createElement(ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum."), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
SmallLongTitle.args = {
  small: true
};
SmallLongTitle.storyName = 'Small: Long title';
export const LargeWithSelectComponent = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "Select opens on top of the Modal"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(SingleSelect, null, /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
}))), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
LargeWithSelectComponent.args = {
  large: true
};
LargeWithSelectComponent.storyName = 'Large: with Select component';
export const ModalWithErrorState = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "Update profile"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(InputField, {
  label: "First name",
  required: true,
  error: true,
  validationText: "First name can't be empty.",
  inputWidth: "320px"
}), /*#__PURE__*/React.createElement(InputField, {
  label: "Last name",
  value: "Olefeme",
  required: true,
  inputWidth: "320px"
}), /*#__PURE__*/React.createElement(Box, {
  marginTop: "16px"
}, /*#__PURE__*/React.createElement(NoticeBox, {
  error: true
}, "There is a problem with this form."))), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  secondary: true
}, "Cancel"), /*#__PURE__*/React.createElement(Button, {
  disabled: true,
  primary: true
}, "Save changes"))));
ModalWithErrorState.storyName = 'With error state';
export const ModalWithLoadingState = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "Modal title"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(Box, {
  minHeight: "240px"
}, /*#__PURE__*/React.createElement(CenteredContent, null, /*#__PURE__*/React.createElement(CircularLoader, null)))), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  secondary: true
}, "Cancel"), /*#__PURE__*/React.createElement(Button, {
  primary: true
}, "Save changes"))));
ModalWithLoadingState.storyName = 'With loading state';
export const LargeModalWithMoreNestedModals = args => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "Select opens on top of the Modal - Level 1"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(SingleSelect, null, /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
}))), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))), /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "Select opens on top of the Modal - Level 2"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(SingleSelect, null, /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
}))), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))), /*#__PURE__*/React.createElement(Modal, {
  small: true
}, /*#__PURE__*/React.createElement(ModalTitle, null, "Select opens on top of the Modal - Level 3"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(SingleSelect, null, /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
})), /*#__PURE__*/React.createElement(Tooltip, {
  content: "Some extra info"
}, _ref => {
  let {
    ref,
    onMouseOver,
    onMouseOut
  } = _ref;
  return /*#__PURE__*/React.createElement("span", {
    ref: ref,
    onMouseOver: onMouseOver,
    onMouseOut: onMouseOut,
    className: "jsx-2364308943"
  }, /*#__PURE__*/React.createElement("button", {
    className: "jsx-2364308943"
  }, "Tooltip on button"), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "2364308943"
  }, ["span.jsx-2364308943{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}"]));
}), /*#__PURE__*/React.createElement(FlyoutMenu, null, /*#__PURE__*/React.createElement(MenuSectionHeader, {
  label: "Section with sub-menus"
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 2"
}, /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 2 a"
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 2 b"
}, /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 2 b i"
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 2 b ii"
})), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 2 c"
})), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 3"
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 4"
}, /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 4 a"
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 4 b"
}, /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 4 b i"
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 4 b ii"
})), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 4 c"
})), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 5"
}), /*#__PURE__*/React.createElement(MenuSectionHeader, {
  label: "Section with dividers between menu items"
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/React.createElement(MenuDivider, null), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 2"
}), /*#__PURE__*/React.createElement(MenuDivider, null), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 3"
}))), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))));
LargeModalWithMoreNestedModals.args = {
  large: true
};
LargeModalWithMoreNestedModals.storyName = 'Large: modal with more nested modals';
const StatefuleComponent = () => {
  const [counter, setCounter] = useState(0);
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("p", null, "Current counter: ", counter), /*#__PURE__*/React.createElement("button", {
    onClick: () => setCounter(counter + 1)
  }, "Add 1 to counter"));
};
export const ModalThatHidesWithStatefulComponens = () => {
  const [render, setRender] = useState(false);
  const [hide, setHide] = useState(false);
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(ButtonStrip, null, /*#__PURE__*/React.createElement(Button, {
    onClick: () => setHide(false),
    disabled: !render
  }, "Show the hidden modal"), /*#__PURE__*/React.createElement(Button, {
    onClick: () => setRender(true),
    disabled: render
  }, "Render Modal")), render && /*#__PURE__*/React.createElement(Modal, {
    hide: hide
  }, /*#__PURE__*/React.createElement(ModalTitle, null, "Can be hidden"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(StatefuleComponent, null)), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
    end: true
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: () => setRender(false),
    secondary: true
  }, "Close modal"), /*#__PURE__*/React.createElement(Button, {
    onClick: () => setHide(true),
    primary: true
  }, "Hide modal")))));
};
export const FluidTop = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a modal using custom dimensions, with title, content and primary action"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(Box, {
  width: "666px",
  height: "666px"
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.")), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
FluidTop.args = {
  fluid: true,
  position: 'top'
};
FluidTop.storyName = 'Fluid (Top)';
export const FluidMiddle = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a modal using custom dimensions, with title, content and primary action"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(Box, {
  width: "666px",
  height: "666px"
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.")), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
FluidMiddle.args = {
  fluid: true,
  position: 'middle'
};
FluidMiddle.storyName = 'Fluid (Middle)';
export const FluidBottom = args => /*#__PURE__*/React.createElement(Modal, args, /*#__PURE__*/React.createElement(ModalTitle, null, "This is a modal using custom dimensions, with title, content and primary action"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(Box, {
  width: "666px",
  height: "666px"
}, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.")), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
  end: true
}, /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/React.createElement(Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));
FluidBottom.args = {
  fluid: true,
  position: 'bottom'
};
FluidBottom.storyName = 'Fluid (Bottom)';
export const RTL = args => {
  useEffect(() => {
    document.body.dir = 'rtl';
    return () => {
      document.body.dir = 'ltr';
    };
  }, []);
  return /*#__PURE__*/React.createElement("div", {
    dir: "rtl"
  }, /*#__PURE__*/React.createElement(SmallTitleContentAction, args));
};
RTL.args = {
  small: true,
  onClose
};
