// @dhis2-ui/pagination/build/cjs/pagination.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithoutPageSummary = exports.WithoutPageSizeSelect = exports.WithoutGoToPageSelect = exports.WithoutAnySelect = exports.WithCustomPageSummary = exports.RTL = exports.PagerAtLastPage = exports.PagerAtFirstPage = exports.NoTotalAtLastPageWithoutPageLength = exports.NoTotalAtLastPage = exports.NoTotal = exports.NarrowWidth = exports.MediumWidth = exports.Disabled = exports.Default = void 0;
var _d2I18n = _interopRequireDefault(require("@dhis2/d2-i18n"));
var _react = _interopRequireDefault(require("react"));
var pagers = _interopRequireWildcard(require("./__fixtures__/index.js"));
var _pagination = require("./pagination.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const subtitle = 'Allows navigation through data displayed over several pages';
const description = `
Pagination allows data to be split in pages. Paging large amounts of data avoids overwhelming users and should always be used wherever a lot of data is displayed. Pagination controls allow a user to browse through a set of data or navigate to a specific page depending on the type of pagination used.

**Do not rely on pagination for navigating datasets. A user should be able to search within, sort and filter datasets too, rather than needing to click through many pages looking for the right data item.**

\`\`\`js
import { Pagination } from '@dhis2/ui'
\`\`\`

_**Note**: Dropdown menus may not display properly on this page. View these demos in the 'Canvas' tab._
`;
const logOnPageChange = page => {
  console.log(`Now navigate to page ${page}...`);
};
const logOnPageSizeChange = pageSize => {
  console.log(`Now change page size to ${pageSize}...`);
};
var _default = exports.default = {
  title: 'Pagination',
  component: _pagination.Pagination,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  // Default args for stories
  args: {
    onPageChange: logOnPageChange,
    onPageSizeChange: logOnPageSizeChange
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_pagination.Pagination, args);
const Default = exports.Default = Template.bind({});
Default.args = {
  ...pagers.atTenthPage
};
const Disabled = exports.Disabled = Template.bind({});
Disabled.args = {
  ...pagers.atTenthPage,
  disabled: true
};
const PagerAtFirstPage = exports.PagerAtFirstPage = Template.bind({});
PagerAtFirstPage.args = {
  ...pagers.atFirstPage
};
const PagerAtLastPage = exports.PagerAtLastPage = Template.bind({});
PagerAtLastPage.args = {
  ...pagers.atLastPage
};
const NoTotal = exports.NoTotal = Template.bind({});
NoTotal.args = {
  ...pagers.noTotal
};
const NoTotalAtLastPage = exports.NoTotalAtLastPage = Template.bind({});
NoTotalAtLastPage.args = {
  ...pagers.noTotalAtLastPage,
  pageLength: 26
};
const NoTotalAtLastPageWithoutPageLength = exports.NoTotalAtLastPageWithoutPageLength = Template.bind({});
NoTotalAtLastPageWithoutPageLength.args = {
  ...pagers.noTotalAtLastPage
};
const WithoutPageSizeSelect = exports.WithoutPageSizeSelect = Template.bind({});
WithoutPageSizeSelect.args = {
  ...pagers.atTenthPage,
  hidePageSizeSelect: true,
  hidePageSelect: false,
  hidePageSummary: false
};
const WithoutGoToPageSelect = exports.WithoutGoToPageSelect = Template.bind({});
WithoutGoToPageSelect.args = {
  ...pagers.atTenthPage,
  hidePageSelect: true
};
const WithoutPageSummary = exports.WithoutPageSummary = Template.bind({});
WithoutPageSummary.args = {
  ...pagers.atTenthPage,
  hidePageSummary: true
};
const WithCustomPageSummary = exports.WithCustomPageSummary = Template.bind({});
WithCustomPageSummary.args = {
  ...pagers.atTenthPage,
  pageSummaryText: interpolationObject => _d2I18n.default.t('You are at page {{page}} showing items {{firstItem}}-{{lastItem}}, but there are {{pageCount}} pages and {{total}} items', interpolationObject)
};
const WithoutAnySelect = exports.WithoutAnySelect = Template.bind({});
WithoutAnySelect.args = {
  ...pagers.atTenthPage,
  ...WithoutGoToPageSelect.args,
  ...WithoutPageSizeSelect.args
};
const InBox = _ref => {
  let {
    boxWidth,
    ...args
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: boxWidth
    }
  }, /*#__PURE__*/_react.default.createElement(_pagination.Pagination, args));
};
const MediumWidth = exports.MediumWidth = InBox.bind({});
MediumWidth.args = {
  ...pagers.atTenthPage,
  boxWidth: 500
};
const NarrowWidth = exports.NarrowWidth = InBox.bind({});
NarrowWidth.args = {
  ...pagers.atTenthPage,
  boxWidth: 100
};
const RTL = args => /*#__PURE__*/_react.default.createElement("div", {
  dir: "rtl"
}, /*#__PURE__*/_react.default.createElement(_pagination.Pagination, args));
exports.RTL = RTL;
RTL.args = {
  ...pagers.atTenthPage
};
