// @dhis2-ui/header-bar/build/cjs/__e2e__/stories/with-debug-info-edge-cases.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithUnknownInstanceVersion = exports.WithUnknownAppVersion = exports.WithUnknownAppNameAndVersion = exports.WithUnknownAppName = void 0;
var _react = _interopRequireDefault(require("react"));
var _headerBar = require("../../header-bar.js");
var _common = require("./common.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const WithUnknownInstanceVersion = () => /*#__PURE__*/_react.default.createElement(_headerBar.HeaderBar, null);
exports.WithUnknownInstanceVersion = WithUnknownInstanceVersion;
WithUnknownInstanceVersion.decorators = [(0, _common.createDecoratorCustomDataProviderHeaderBar)(), (0, _common.createDecoratorProvider)({
  ..._common.providerConfig,
  systemInfo: {
    ..._common.providerConfig.systemInfo,
    version: undefined
  }
})];
const WithUnknownAppVersion = () => /*#__PURE__*/_react.default.createElement(_headerBar.HeaderBar, null);
exports.WithUnknownAppVersion = WithUnknownAppVersion;
WithUnknownAppVersion.decorators = [(0, _common.createDecoratorCustomDataProviderHeaderBar)(), (0, _common.createDecoratorProvider)({
  ..._common.providerConfig,
  appVersion: undefined
})];
const WithUnknownAppName = () => /*#__PURE__*/_react.default.createElement(_headerBar.HeaderBar, null);
exports.WithUnknownAppName = WithUnknownAppName;
WithUnknownAppName.decorators = [(0, _common.createDecoratorCustomDataProviderHeaderBar)(), (0, _common.createDecoratorProvider)({
  ..._common.providerConfig,
  appName: undefined
})];
const WithUnknownAppNameAndVersion = () => /*#__PURE__*/_react.default.createElement(_headerBar.HeaderBar, null);
exports.WithUnknownAppNameAndVersion = WithUnknownAppNameAndVersion;
WithUnknownAppNameAndVersion.decorators = [(0, _common.createDecoratorCustomDataProviderHeaderBar)(), (0, _common.createDecoratorProvider)({
  ..._common.providerConfig,
  appName: undefined,
  appVersion: undefined
})];
