// @dhis2-ui/intersection-detector/build/es/intersection-detector.e2e.stories.js
import React, { useRef } from 'react';
import { IntersectionDetector } from './intersection-detector.js';
window.onChange = window.Cypress ? window.Cypress.cy.stub() : () => null;
export default {
  title: 'IntersectionDetector'
};
export const OutOfView = () => {
  const rootRef = useRef();
  return /*#__PURE__*/React.createElement("div", {
    ref: rootRef,
    "data-test": "scroll-container",
    style: {
      width: 200,
      height: 300,
      overflow: 'auto'
    }
  }, /*#__PURE__*/React.createElement("div", {
    // spacer to push indicator out of the intersecting area
    style: {
      width: 200,
      height: 310,
      overflow: 'auto',
      background: 'blue'
    }
  }), /*#__PURE__*/React.createElement(IntersectionDetector, {
    rootRef: rootRef,
    onChange: window.onChange
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      background: 'red',
      height: 200
    }
  })));
};
export const InView = () => {
  const rootRef = useRef();
  return /*#__PURE__*/React.createElement("div", {
    ref: rootRef,
    "data-test": "scroll-container",
    style: {
      width: 200,
      height: 300,
      overflow: 'auto'
    }
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      background: 'red',
      height: 200
    }
  }, /*#__PURE__*/React.createElement(IntersectionDetector, {
    rootRef: rootRef,
    onChange: window.onChange
  })), /*#__PURE__*/React.createElement("div", {
    // spacer to push indicator out of the intersecting area
    style: {
      width: 200,
      height: 310,
      overflow: 'auto',
      background: 'blue'
    }
  }));
};
