// @dhis2-ui/tab/build/es/tab-bar/tab-bar.js
import PropTypes from 'prop-types';
import React from 'react';
import { ScrollBar } from './scroll-bar.js';
import { Tabs } from './tabs.js';
const TabBar = _ref => {
  let {
    fixed,
    children,
    className,
    scrollable,
    dataTest = 'dhis2-uicore-tabbar'
  } = _ref;
  if (scrollable) {
    return /*#__PURE__*/React.createElement("div", {
      className: className,
      "data-test": dataTest
    }, /*#__PURE__*/React.createElement(ScrollBar, {
      dataTest: `${dataTest}-scrollbar`
    }, /*#__PURE__*/React.createElement(Tabs, {
      fixed: fixed,
      dataTest: `${dataTest}-tabs`
    }, children)));
  }
  return /*#__PURE__*/React.createElement("div", {
    className: className,
    "data-test": dataTest
  }, /*#__PURE__*/React.createElement(Tabs, {
    fixed: fixed,
    dataTest: `${dataTest}-tabs`
  }, children));
};
TabBar.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  /** Fixed tabs fill the width of their container. If false (i.e. fluid), tabs take up an amount of space defined by the tab name. Fluid tabs should be used most of the time. */
  fixed: PropTypes.bool,
  /** Enables horizontal scrolling for many tabs that don't fit the width of the container */
  scrollable: PropTypes.bool
};
export { TabBar };
