// @dhis2-ui/menu/build/cjs/menu-divider/menu-divider.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Dense = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _menuDivider = require("./menu-divider.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
Items in a menu can be split into separate sections by using dividers. Group relevant menu items together to help the user understand the options quickly. A divider can be used alone. If using a MenuSectionHeader, a divider will be automatically included. Try not to group single menu items together. An exception to this is a critical destructive menu item, like 'Delete', which can be separated from other menu items.

\`\`\`js
import { MenuDivider } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Menu Divider',
  component: _menuDivider.MenuDivider,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_index.Menu, null, /*#__PURE__*/_react.default.createElement(_index.MenuItem, {
  label: "Item above divider"
}), /*#__PURE__*/_react.default.createElement(_menuDivider.MenuDivider, args), /*#__PURE__*/_react.default.createElement(_index.MenuItem, {
  label: "Item below divider"
}));
const Default = exports.Default = Template.bind({});
const Dense = exports.Dense = Template.bind({});
Dense.args = {
  dense: true
};
