// @dhis2-ui/text-area/build/cjs/text-area/text-area.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithOnFocus = exports.WithOnChange = exports.WithInitialFocusAndOnBlur = exports.WithInitialFocus = exports.WithDisabled = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onChange = window.Cypress && window.Cypress.cy.stub();
window.onBlur = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
var _default = exports.default = {
  title: 'TextArea'
};
const WithOnChange = () => /*#__PURE__*/_react.default.createElement(_index.TextArea, {
  onChange: window.onChange,
  name: "textarea"
});
exports.WithOnChange = WithOnChange;
const WithInitialFocusAndOnBlur = () => /*#__PURE__*/_react.default.createElement(_index.TextArea, {
  initialFocus: true,
  name: "textarea",
  onBlur: window.onBlur
});
exports.WithInitialFocusAndOnBlur = WithInitialFocusAndOnBlur;
const WithOnFocus = () => /*#__PURE__*/_react.default.createElement(_index.TextArea, {
  name: "textarea",
  onFocus: window.onFocus
});
exports.WithOnFocus = WithOnFocus;
const WithInitialFocus = () => /*#__PURE__*/_react.default.createElement(_index.TextArea, {
  name: "textarea",
  initialFocus: true
});
exports.WithInitialFocus = WithInitialFocus;
const WithDisabled = () => /*#__PURE__*/_react.default.createElement(_index.TextArea, {
  name: "textarea",
  disabled: true
});
exports.WithDisabled = WithDisabled;
