// @dhis2-ui/sharing-dialog/build/es/modal/modal.js
import { ButtonStrip, Button } from '@dhis2-ui/button';
import { Modal as Dhis2Modal, ModalTitle, ModalContent, ModalActions } from '@dhis2-ui/modal';
import PropTypes from 'prop-types';
import React from 'react';
import i18n from '../locales/index.js';
export const Modal = _ref => {
  let {
    onClose,
    name,
    children,
    dataTest = 'dhis2-uicore-sharingdialog-modal'
  } = _ref;
  const title = name ? i18n.t('Sharing and access: {{- objectName}}', {
    objectName: name,
    nsSeparator: '|'
  }) : i18n.t('Sharing and access');
  return /*#__PURE__*/React.createElement(Dhis2Modal, {
    large: true,
    position: "top",
    onClose: onClose,
    dataTest: dataTest
  }, /*#__PURE__*/React.createElement(ModalTitle, null, title), /*#__PURE__*/React.createElement(ModalContent, null, children), /*#__PURE__*/React.createElement(ModalActions, null, /*#__PURE__*/React.createElement(ButtonStrip, {
    end: true
  }, /*#__PURE__*/React.createElement(Button, {
    type: "button",
    secondary: true,
    onClick: onClose
  }, i18n.t('Close')))));
};
Modal.propTypes = {
  children: PropTypes.node.isRequired,
  dataTest: PropTypes.string,
  name: PropTypes.string,
  onClose: PropTypes.func
};
