// @dhis2/ui-forms/build/cjs/formDecorator.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.formDecorator = void 0;
var _button = require("@dhis2-ui/button");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
const formProps = {
  onSubmit: values => {
    console.log('++++++++++++++++\n', 'Form was submitted with values:\n', values, '\n----------------');
  },
  mutators: {}
};
class FormWithSpyAndSubmit extends _react.default.Component {
  constructor() {
    super(...arguments);
    _defineProperty(this, "state", {
      cypressProps: {}
    });
    _defineProperty(this, "updateCypressProps", updateObj => {
      const cypressProps = {
        ...this.state.cypressProps,
        ...updateObj
      };
      this.setState({
        cypressProps
      });
    });
    _defineProperty(this, "clearCypressProps", () => {
      this.setState({
        cypressProps: {}
      });
    });
  }
  componentDidMount() {
    window.updateCypressProps = this.updateCypressProps;
    window.clearCypressProps = this.clearCypressProps;
    this.forceUpdate();
  }
  componentWillUnmount() {
    delete window.updateCypressProps;
    delete window.clearCypressProps;
  }
  render() {
    return /*#__PURE__*/_react.default.createElement(_reactFinalForm.Form, formProps, formRenderProps => /*#__PURE__*/_react.default.createElement("form", {
      onSubmit: formRenderProps.handleSubmit
    }, this.props.renderChildren({
      formRenderProps,
      cypressProps: this.state.cypressProps
    }), /*#__PURE__*/_react.default.createElement(_button.Button, {
      primary: true,
      type: "submit"
    }, "Submit"), /*#__PURE__*/_react.default.createElement(_reactFinalForm.FormSpy, null, _ref => {
      let {
        values
      } = _ref;
      window.formValues = values;
      return /*#__PURE__*/_react.default.createElement("span", {
        className: "form-spy-internal"
      });
    })));
  }
}
FormWithSpyAndSubmit.propTypes = {
  renderChildren: _propTypes.default.func.isRequired
};
const formDecorator = fn => /*#__PURE__*/_react.default.createElement(FormWithSpyAndSubmit, {
  renderChildren: fn
});
exports.formDecorator = formDecorator;
