// @dhis2-ui/pagination/build/es/page-size-select.js
import _JSXStyle from "styled-jsx/style";
import { spacers } from '@dhis2/ui-constants';
import { SingleSelect, SingleSelectOption } from '@dhis2-ui/select';
import PropTypes from 'prop-types';
import React from 'react';

// TODO: i18n translate
const translate = (prop, interpolationObject) => {
  if (typeof prop === 'function') {
    return prop(interpolationObject);
  }
  return prop;
};
const PageSizeSelect = _ref => {
  let {
    dataTest,
    disabled,
    pageSizeSelectText,
    pageSize,
    pageSizes,
    onChange
  } = _ref;
  return /*#__PURE__*/React.createElement("div", {
    "data-test": `${dataTest}-pagesize`,
    className: _JSXStyle.dynamic([["3539890457", [spacers.dp12]]])
  }, /*#__PURE__*/React.createElement(SingleSelect, {
    dense: true,
    disabled: disabled,
    selected: pageSize.toString(),
    onChange: _ref2 => {
      let {
        selected
      } = _ref2;
      return onChange(parseInt(selected, 10));
    },
    className: "select",
    dataTest: `${dataTest}-pagesize-select`,
    prefix: translate(pageSizeSelectText)
  }, pageSizes.map(length => /*#__PURE__*/React.createElement(SingleSelectOption, {
    key: length,
    value: length,
    label: length
  }))), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "3539890457",
    dynamic: [spacers.dp12]
  }, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;min-height:32px;margin-inline-end:${spacers.dp12};-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}`]));
};
PageSizeSelect.propTypes = {
  dataTest: PropTypes.string.isRequired,
  pageSize: PropTypes.number.isRequired,
  pageSizeSelectText: PropTypes.oneOfType([PropTypes.string, PropTypes.func]).isRequired,
  pageSizes: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool
};
export { PageSizeSelect };
