// @dhis2-ui/chip/build/cjs/content.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Content = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Content = _ref => {
  let {
    children,
    overflow
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("span", {
    className: _style.default.dynamic([["765109298", [_uiConstants.spacers.dp12]]]) + " " + ((0, _classnames.default)({
      overflow
    }) || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "765109298",
    dynamic: [_uiConstants.spacers.dp12]
  }, [`span.__jsx-style-dynamic-selector{margin:0 ${_uiConstants.spacers.dp12};color:inherit;white-space:nowrap;}`, ".overflow.__jsx-style-dynamic-selector{max-width:150px;overflow:hidden;text-overflow:ellipsis;}"]));
};
exports.Content = Content;
Content.propTypes = {
  children: _propTypes.default.any,
  overflow: _propTypes.default.bool
};
