// @dhis2-ui/chip/build/cjs/remove.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Remove = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function CancelOutline(_ref) {
  let {
    className
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 16 16",
    className: "jsx-2898942518" + " " + (className || "")
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M11.2928932,3.29289322 L12.7071068,4.70710678 L9.41389322,7.99989322 L12.7071068,11.2928932 L11.2928932,12.7071068 L7.99989322,9.41389322 L4.70710678,12.7071068 L3.29289322,11.2928932 L6.58589322,7.99989322 L3.29289322,4.70710678 L4.70710678,3.29289322 L7.99989322,6.58589322 L11.2928932,3.29289322 Z",
    className: "jsx-2898942518"
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2898942518"
  }, ["svg.jsx-2898942518{fill:inherit;height:16px;width:16px;vertical-align:middle;pointer-events:none;}"]));
}
CancelOutline.propTypes = {
  className: _propTypes.default.string
};
const containerStyle = ["span.jsx-2824632002{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:20px;width:20px;margin-inline-end:4px;border-radius:12px;margin-inline-start:-8px;}", `span.jsx-2824632002:hover{background:${_uiConstants.colors.grey400};}`];
containerStyle.__hash = "2824632002";
const removeIcon = {
  styles: /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2658726446"
  }, [`svg.jsx-2658726446{fill:${_uiConstants.colors.grey600};height:16px;width:16px;cursor:pointer;opacity:1;pointer-events:all;}`]),
  className: "jsx-2658726446"
};
const Remove = _ref2 => {
  let {
    onRemove,
    dataTest
  } = _ref2;
  if (!onRemove) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement("span", {
    onClick: e => {
      e.stopPropagation(); // stop onRemove from triggering onClick on container
      onRemove({}, e);
    },
    "data-test": dataTest,
    className: `jsx-${containerStyle.__hash}`
  }, /*#__PURE__*/_react.default.createElement(CancelOutline, {
    className: removeIcon.className
  }), removeIcon.styles, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: containerStyle.__hash
  }, containerStyle));
};
exports.Remove = Remove;
Remove.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  onRemove: _propTypes.default.func
};
