// @dhis2-ui/sharing-dialog/build/cjs/autocomplete/menu-popup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuPopup = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _card = require("@dhis2-ui/card");
var _layer = require("@dhis2-ui/layer");
var _popper = require("@dhis2-ui/popper");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const MenuPopup = _ref => {
  let {
    children,
    dataTest,
    maxHeight = '280px',
    menuWidth,
    onClick,
    menuRef
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_layer.Layer, {
    onBackdropClick: onClick,
    transparent: true
  }, /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    reference: menuRef,
    placement: "bottom",
    observeReferenceResize: true
  }, /*#__PURE__*/_react.default.createElement("div", {
    "data-test": `${dataTest}-menuwrapper`,
    className: _style.default.dynamic([["1686499909", [menuWidth, maxHeight]]]) + " " + "card"
  }, /*#__PURE__*/_react.default.createElement(_card.Card, null, children))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1686499909",
    dynamic: [menuWidth, maxHeight]
  }, [`.card.__jsx-style-dynamic-selector{width:${menuWidth};max-height:${maxHeight};}`]));
};
exports.MenuPopup = MenuPopup;
MenuPopup.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  menuRef: _propTypes.default.object.isRequired,
  menuWidth: _propTypes.default.string.isRequired,
  children: _propTypes.default.node,
  maxHeight: _propTypes.default.string,
  onClick: _propTypes.default.func
};
