// @dhis2-ui/field/build/es/field-set/field-set.prod.stories.js
import { Help } from '@dhis2-ui/help';
import { Legend } from '@dhis2-ui/legend';
import { Radio } from '@dhis2-ui/radio';
import React from 'react';
import { Field } from '../index.js';
import { FieldSet } from './field-set.js';
const description = `
A container for grouping several Field components together. Use a \`<Legend>\` component to add helpful guiding text.

\`\`\`js
import { FieldSet } from '@dhis2/ui'
\`\`\`
`;
const onChange = () => {
  console.log('Radio was clicked, nothing else will happen');
};
export default {
  title: 'Field Set',
  component: FieldSet,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
export const Default = args => /*#__PURE__*/React.createElement(FieldSet, args, "It renders something in a fieldset element without the browser styles");
export const UsageExampleARadioButtonGroupWithErrorStatus = args => /*#__PURE__*/React.createElement(FieldSet, args, /*#__PURE__*/React.createElement(Legend, {
  required: true
}, "Choose an option"), /*#__PURE__*/React.createElement(Field, null, /*#__PURE__*/React.createElement(Radio, {
  onChange: onChange,
  name: "radio",
  value: "first",
  label: "First",
  error: true
})), /*#__PURE__*/React.createElement(Field, null, /*#__PURE__*/React.createElement(Radio, {
  onChange: onChange,
  name: "radio",
  value: "second",
  label: "Second",
  error: true
})), /*#__PURE__*/React.createElement(Field, null, /*#__PURE__*/React.createElement(Radio, {
  onChange: onChange,
  name: "radio",
  value: "third",
  label: "Third",
  error: true
})), /*#__PURE__*/React.createElement(Help, {
  error: true
}, "You really have to choose something!"));
UsageExampleARadioButtonGroupWithErrorStatus.storyName = 'Usage example - a radio button group with error status';
