// @dhis2-ui/loader/build/cjs/linear-loader/linear-loader.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RTL = exports.OverlayPage = exports.OverlayComponent = exports.Determinate = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _center = require("@dhis2-ui/center");
var _cover = require("@dhis2-ui/cover");
var _layer = require("@dhis2-ui/layer");
var _react = _interopRequireDefault(require("react"));
var _index = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const subtitle = `Used to inform the user that an element or page is in a loading state`;
const description = `
Use loading indicators whenever a component or application takes longer than 700ms to load. After this time a loader should be displayed so that the user can understand what is happening: loading is in progress. Consider that without a loading indicator a user would be unsure of their current status, so they are important UI elements.

A linear loader is used when the loading time is can be measured and displayed as a percentage.

\`\`\`js
import { LinearLoader } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Linear Loader',
  component: _index.LinearLoader,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    margin: {
      table: {
        defaultValue: {
          summary: '12px'
        }
      }
    }
  }
};
const Determinate = args => /*#__PURE__*/_react.default.createElement(_index.LinearLoader, args);
exports.Determinate = Determinate;
Determinate.args = {
  amount: 60,
  'aria-label': 'Determinate Loader'
};
const OverlayPage = args => /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.blocking,
  translucent: true
}, /*#__PURE__*/_react.default.createElement(_center.Center, null, /*#__PURE__*/_react.default.createElement(_index.LinearLoader, args)));
exports.OverlayPage = OverlayPage;
OverlayPage.args = {
  amount: 30,
  'aria-label': 'Loader with Overlay Page'
};
OverlayPage.parameters = {
  docs: {
    inlineStories: false
  }
};
const OverlayComponent = args => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    width: '400px',
    height: '400px'
  }
}, /*#__PURE__*/_react.default.createElement(_cover.Cover, {
  translucent: true
}, /*#__PURE__*/_react.default.createElement(_center.Center, null, /*#__PURE__*/_react.default.createElement(_index.LinearLoader, args))));
exports.OverlayComponent = OverlayComponent;
OverlayComponent.args = {
  amount: 80,
  'aria-label': 'Loader with Overlay Component'
};
const RTL = args => /*#__PURE__*/_react.default.createElement("div", {
  dir: "rtl"
}, /*#__PURE__*/_react.default.createElement(_index.LinearLoader, args));
exports.RTL = RTL;
RTL.args = {
  amount: 60
};
