// @dhis2-ui/divider/build/cjs/divider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Divider = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const flipMargin = margin => {
  const splitMargin = margin.split(/\s+/);
  if ((splitMargin === null || splitMargin === void 0 ? void 0 : splitMargin.length) === 4) {
    return [splitMargin[0], splitMargin[3], splitMargin[2], splitMargin[1]].join(' ');
  }
  return margin;
};
const Divider = _ref => {
  let {
    className,
    dataTest = 'dhis2-uicore-divider',
    dense,
    margin = `${_uiConstants.spacers.dp8} 0`
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    role: "separator",
    className: _style.default.dynamic([["2418722155", [_uiConstants.colors.grey300]], ["3247041537", [dense ? `${_uiConstants.spacers.dp4} 0` : margin, dense ? `${_uiConstants.spacers.dp4} 0` : flipMargin(margin)]]]) + " " + (className || "")
  }, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2418722155",
    dynamic: [_uiConstants.colors.grey300]
  }, [`div.__jsx-style-dynamic-selector{display:inline-block;width:100%;height:1px;background-color:${_uiConstants.colors.grey300};}`]), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3247041537",
    dynamic: [dense ? `${_uiConstants.spacers.dp4} 0` : margin, dense ? `${_uiConstants.spacers.dp4} 0` : flipMargin(margin)]
  }, [`div.__jsx-style-dynamic-selector{margin:${dense ? `${_uiConstants.spacers.dp4} 0` : margin};}`, `.__jsx-style-dynamic-selector:dir(rtl){margin:${dense ? `${_uiConstants.spacers.dp4} 0` : flipMargin(margin)};}`]));
};
exports.Divider = Divider;
Divider.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  dense: _propTypes.default.bool,
  margin: _propTypes.default.string
};
