// @dhis2-ui/divider/build/cjs/divider.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RTLUnevenMargin = exports.Margin = exports.Dense = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _divider = require("./divider.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
A light divider to separate content.

\`\`\`js
import { Divider } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Divider',
  component: _divider.Divider,
  argTypes: {
    margin: {
      table: {
        defaultValue: {
          summary: '8px 0'
        }
      }
    }
  },
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", null, "Content above"), /*#__PURE__*/_react.default.createElement(_divider.Divider, args), /*#__PURE__*/_react.default.createElement("div", null, "Content below"));
const Default = exports.Default = Template.bind({});
const Dense = exports.Dense = Template.bind({});
Dense.args = {
  dense: true
};
const Margin = exports.Margin = Template.bind({});
Margin.args = {
  margin: '20px 20px 20px 20px'
};
const RTLUnevenMargin = args => /*#__PURE__*/_react.default.createElement("div", {
  dir: "rtl"
}, /*#__PURE__*/_react.default.createElement(Template, args));
exports.RTLUnevenMargin = RTLUnevenMargin;
RTLUnevenMargin.args = {
  margin: '10px 5px 40px 50px'
};
