// @dhis2-ui/table/build/cjs/data-table/table-elements/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function () {
    return _table.Table;
  }
});
Object.defineProperty(exports, "TableBody", {
  enumerable: true,
  get: function () {
    return _tableBody.TableBody;
  }
});
Object.defineProperty(exports, "TableDataCell", {
  enumerable: true,
  get: function () {
    return _tableDataCell.TableDataCell;
  }
});
Object.defineProperty(exports, "TableFoot", {
  enumerable: true,
  get: function () {
    return _tableFoot.TableFoot;
  }
});
Object.defineProperty(exports, "TableHead", {
  enumerable: true,
  get: function () {
    return _tableHead.TableHead;
  }
});
Object.defineProperty(exports, "TableHeaderCell", {
  enumerable: true,
  get: function () {
    return _tableHeaderCell.TableHeaderCell;
  }
});
Object.defineProperty(exports, "TableHeaderCellAction", {
  enumerable: true,
  get: function () {
    return _tableHeaderCellAction.TableHeaderCellAction;
  }
});
Object.defineProperty(exports, "TableRow", {
  enumerable: true,
  get: function () {
    return _tableRow.TableRow;
  }
});
Object.defineProperty(exports, "TableScrollBox", {
  enumerable: true,
  get: function () {
    return _tableScrollBox.TableScrollBox;
  }
});
Object.defineProperty(exports, "TableToolbar", {
  enumerable: true,
  get: function () {
    return _tableToolbar.TableToolbar;
  }
});
var _table = require("./table.js");
var _tableBody = require("./table-body.js");
var _tableDataCell = require("./table-data-cell/table-data-cell.js");
var _tableHeaderCell = require("./table-header-cell/table-header-cell.js");
var _tableHeaderCellAction = require("./table-header-cell-action.js");
var _tableFoot = require("./table-foot.js");
var _tableHead = require("./table-head.js");
var _tableRow = require("./table-row.js");
var _tableScrollBox = require("./table-scroll-box.js");
var _tableToolbar = require("./table-toolbar.js");
