// @dhis2-ui/table/build/cjs/data-table/table-elements/table.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.FixedHeaderAndColumns = void 0;
var _react = _interopRequireDefault(require("react"));
var _tableBody = require("./table-body.js");
var _tableDataCell = require("./table-data-cell/table-data-cell.js");
var _tableHead = require("./table-head.js");
var _tableHeaderCell = require("./table-header-cell/table-header-cell.js");
var _tableRow = require("./table-row.js");
var _tableScrollBox = require("./table-scroll-box.js");
var _table = require("./table.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'Table',
  component: _table.Table
};
const FixedHeaderAndColumns = () => /*#__PURE__*/_react.default.createElement(_tableScrollBox.TableScrollBox, {
  maxHeight: "350px",
  maxWidth: "500px"
}, /*#__PURE__*/_react.default.createElement(_table.Table, {
  layout: "fixed",
  borderless: true,
  width: "1000px"
}, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "Header fixed-left 1"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Header fixed-left 2"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Header fixed"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "scrolling top-left"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "scrolling top-right")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "...")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "...")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "...")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "...")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "...")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "...")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "...")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "...")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "scrolling bottom-left"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "..."), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "scrolling bottom-right")))));
exports.FixedHeaderAndColumns = FixedHeaderAndColumns;
