// @dhis2-ui/table/build/cjs/data-table/table-elements/table-toolbar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableToolbar = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const TableToolbar = exports.TableToolbar = /*#__PURE__*/(0, _react.forwardRef)((_ref, ref) => {
  let {
    children,
    className,
    dataTest = 'dhis2-uicore-tabletoolbar',
    position = 'top',
    ...props
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", _extends({}, props, {
    "data-test": dataTest,
    ref: ref,
    className: _style.default.dynamic([["247244135", [_uiConstants.colors.grey300, _uiConstants.spacers.dp12]]]) + " " + ((0, _classnames.default)(className, position) || "")
  }), children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "247244135",
    dynamic: [_uiConstants.colors.grey300, _uiConstants.spacers.dp12]
  }, [`div.__jsx-style-dynamic-selector{width:100%;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border:1px solid ${_uiConstants.colors.grey300};padding:${_uiConstants.spacers.dp12};}`, "div.top.__jsx-style-dynamic-selector{border-bottom:none;}", "div.bottom.__jsx-style-dynamic-selector{border-top:none;}"]));
});
TableToolbar.displayName = 'TableToolbar';
TableToolbar.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  position: _propTypes.default.oneOf(['top', 'bottom'])
};
