// @dhis2-ui/table/build/cjs/data-table/table-elements/table-header-cell-action.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableHeaderCellAction = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const TableHeaderCellAction = exports.TableHeaderCellAction = /*#__PURE__*/(0, _react.forwardRef)((_ref, ref) => {
  let {
    onClick,
    children,
    className,
    dataTest = 'dhis2-uicore-tableheadercellaction',
    title,
    ...props
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("button", _extends({}, props, {
    "data-test": dataTest,
    onClick: onClick,
    title: title,
    ref: ref,
    className: _style.default.dynamic([["4225768774", [_uiConstants.colors.grey400]]]) + " " + (className || "")
  }), children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4225768774",
    dynamic: [_uiConstants.colors.grey400]
  }, ["button.__jsx-style-dynamic-selector{border:none;padding:0;background:transparent;width:24px;height:24px;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;cursor:pointer;border-radius:4px;margin-inline-start:2px;}", `button.__jsx-style-dynamic-selector:hover,button.__jsx-style-dynamic-selector:focus-visible{background:${_uiConstants.colors.grey400};}`, "button.__jsx-style-dynamic-selector:active,button.__jsx-style-dynamic-selector:focus{outline:none;}"]));
});
TableHeaderCellAction.displayName = 'TableHeaderCellAction';
TableHeaderCellAction.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  title: _propTypes.default.string,
  onClick: _propTypes.default.func
};
