// @dhis2-ui/layer/build/cjs/layer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Layer = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = require("@dhis2/prop-types");
var _portal = require("@dhis2-ui/portal");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes2 = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Layer = _ref => {
  let {
    children,
    className,
    dataTest = 'dhis2-uicore-layer',
    disablePortal,
    level = 'auto',
    onBackdropClick,
    onClick,
    position = 'fixed',
    translucent
  } = _ref;
  const resolvedOnClick = onBackdropClick || onClick;
  return /*#__PURE__*/_react.default.createElement(_portal.Portal, {
    disable: disablePortal
  }, /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: "jsx-4284133831 " + _style.default.dynamic([["1893150038", [level]]]) + " " + ((0, _classnames.default)('layer', className, position, {
      translucent
    }) || "")
  }, resolvedOnClick && /*#__PURE__*/_react.default.createElement("div", {
    onClick: event => resolvedOnClick({}, event),
    className: "jsx-4284133831 " + _style.default.dynamic([["1893150038", [level]]]) + " " + "backdrop"
  }), children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1893150038",
    dynamic: [level]
  }, [`div.__jsx-style-dynamic-selector{z-index:${level};}`]), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4284133831"
  }, ["div.jsx-4284133831{-webkit-inset-block-start:0;-ms-intb-rlock-start:0;inset-block-start:0;inset-inline-start:0;min-height:100vh;min-width:100vw;}", "div.fixed.jsx-4284133831{position:fixed;height:100vh;width:100vw;}", "div.absolute.jsx-4284133831{position:absolute;height:100%;width:100%;}", "div.translucent.jsx-4284133831{background-color:rgba(33,43,54,0.4);}", "div.backdrop.jsx-4284133831{position:absolute;inset:0;z-index:-1;}"])));
};
exports.Layer = Layer;
Layer.propTypes = {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  dataTest: _propTypes2.default.string,
  /** Disable the Portal, useful for nesting layers */
  disablePortal: _propTypes2.default.bool,
  /** Z-index level */
  level: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string]),
  position: _propTypes2.default.oneOf(['absolute', 'fixed']),
  /** Adds a semi-transparent background */
  translucent: _propTypes2.default.bool,
  /** Backdrop click handler */
  onBackdropClick: _propTypes2.default.func,
  /** Click handler - DEPRECATED */
  onClick: (0, _propTypes.deprecated)(_propTypes2.default.func, 'Please use "onBackdropClick" instead')
};
