// @dhis2-ui/calendar/build/es/stories/calendar.prod.stories.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React from 'react';
import { Calendar } from '../calendar/calendar.js';
import { CalendarStoryWrapper } from './calendar-story-wrapper.js';
const subtitle = `[Experimental] Calendar component is useful for creating a variety of calendars including Ethiopic, Islamic etc..`;
const description = `
Use a Calendar where there is a need to let the user select a date.

Note that it requires a parent, like [Box](../?path=/docs/layout-box--default), to define its size.

\`\`\`js
import { Calendar } from '@dhis2/ui'
\`\`\`
`;
export default {
  title: 'Calendar',
  component: Calendar,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    calendar: {
      control: 'select',
      options: ['gregory', 'islamic', 'nepali', 'ethiopic', 'persian', 'indian']
    },
    weekDayFormat: {
      control: 'select',
      options: ['long', 'short', 'narrow']
    },
    locale: {
      control: 'text'
    },
    numberingSystem: {
      control: 'select',
      options: ['latn', 'arab', 'ethi']
    }
  }
};
export const Basic = args => {
  return /*#__PURE__*/React.createElement(Calendar, args);
};
Basic.args = {
  onDateSelect: date => console.log(date),
  calendar: 'gregory',
  weekDayFormat: 'narrow',
  locale: 'en'
};
export const Ethiopic = args => {
  return /*#__PURE__*/React.createElement(CalendarStoryWrapper, _extends({
    calendar: "ethiopic",
    locale: "am-et",
    numberingSystem: "ethi",
    date: "2014-04-05" // 13th of October 2022
    ,
    timeZone: "Africa/Khartoum"
  }, args));
};
export const Nepali = args => {
  return /*#__PURE__*/React.createElement(CalendarStoryWrapper, _extends({
    calendar: "nepali",
    locale: "ne-NP",
    date: "2079-06-29" // 13th of October 2022
    ,
    timeZone: "Africa/Khartoum"
  }, args));
};
export const WithAnyCalendar = args => {
  return /*#__PURE__*/React.createElement(CalendarStoryWrapper, _extends({
    calendar: "gregory",
    locale: "en-GB"
  }, args));
};
