// @dhis2-ui/calendar/build/es/stories/calendar-story-wrapper.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import { constants } from '@dhis2/multi-calendar-dates';
import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { Calendar } from '../calendar/calendar.js';
const {
  calendars,
  numberingSystems
} = constants;
export const CalendarStoryWrapper = props => {
  var _selectedDate$calenda;
  const {
    calendar = 'gregory',
    locale,
    timeZone,
    dir,
    component: Component = Calendar,
    date,
    weekDayFormat = 'narrow'
  } = props;
  const [selectedCalendar, setSelectedCalendar] = useState(calendar);
  const [selectedNumberingSystem, setSelectedNumberingSystem] = useState();
  const [selectedDirection, setSelectedDirection] = useState(dir);
  const [selectedWeekFormat, setWeekDayFormat] = useState(weekDayFormat);
  const [selectedLocale, setLocale] = useState(locale);
  const [selectedDate, setSelectedDate] = useState({
    calendarDateString: date
  });
  const changeCalendar = _ref => {
    let {
      target: {
        value
      }
    } = _ref;
    setSelectedCalendar(value);
  };
  const changeNumberingSystem = _ref2 => {
    let {
      target: {
        value
      }
    } = _ref2;
    if (value === '-1') {
      setSelectedNumberingSystem(null);
    } else {
      setSelectedNumberingSystem(value);
    }
  };
  const changeDirection = _ref3 => {
    let {
      target: {
        value
      }
    } = _ref3;
    setSelectedDirection(value);
  };
  const changeWeekFormat = _ref4 => {
    let {
      target: {
        value
      }
    } = _ref4;
    setWeekDayFormat(value);
  };
  const onLocaleChanged = _ref5 => {
    let {
      target: {
        value
      }
    } = _ref5;
    setLocale(value);
  };
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    style: {
      fontSize: '0.8em',
      marginBottom: 20,
      gap: 5,
      display: 'flex',
      alignItems: 'center'
    }
  }, /*#__PURE__*/React.createElement("div", null, "Options"), /*#__PURE__*/React.createElement("input", {
    defaultValue: locale,
    placeholder: "locale",
    onBlur: onLocaleChanged
  }), /*#__PURE__*/React.createElement("select", {
    onChange: changeCalendar,
    value: selectedCalendar
  }, /*#__PURE__*/React.createElement("option", {
    disabled: true,
    key: calendar,
    value: "-1"
  }, "Select calendar"), calendars.map(calendar => {
    return /*#__PURE__*/React.createElement("option", {
      value: calendar,
      key: calendar
    }, calendar);
  })), /*#__PURE__*/React.createElement("select", {
    onChange: changeNumberingSystem,
    value: selectedNumberingSystem
  }, /*#__PURE__*/React.createElement("option", {
    disabled: true,
    key: calendar,
    value: "-1"
  }, "Select numbering system"), numberingSystems.map(system => {
    return /*#__PURE__*/React.createElement("option", {
      value: system,
      key: system
    }, system);
  })), /*#__PURE__*/React.createElement("select", {
    onChange: changeDirection,
    value: selectedDirection
  }, /*#__PURE__*/React.createElement("option", {
    disabled: true,
    value: "-1"
  }, "Select direction"), /*#__PURE__*/React.createElement("option", {
    value: "ltr"
  }, "Left-To-Right"), /*#__PURE__*/React.createElement("option", {
    value: "rtl"
  }, "Right-To-Left")), /*#__PURE__*/React.createElement("select", {
    onChange: changeWeekFormat,
    value: selectedWeekFormat
  }, /*#__PURE__*/React.createElement("option", {
    disabled: true,
    value: "-1"
  }, "Select format"), /*#__PURE__*/React.createElement("option", {
    value: "narrow"
  }, "narrow"), /*#__PURE__*/React.createElement("option", {
    value: "short"
  }, "short"), /*#__PURE__*/React.createElement("option", {
    value: "long"
  }, "long"))), /*#__PURE__*/React.createElement(Component, _extends({}, props, {
    calendar: selectedCalendar,
    dir: selectedDirection,
    locale: selectedLocale,
    date: selectedDate.calendarDateString,
    onDateSelect: date => {
      setSelectedDate(date);
    },
    timeZone: timeZone,
    weekDayFormat: selectedWeekFormat,
    numberingSystem: selectedNumberingSystem
  })), /*#__PURE__*/React.createElement("div", {
    style: {
      marginTop: 10,
      gap: 5,
      fontSize: '12px',
      display: 'flex',
      flexDirection: 'column'
    }
  }, /*#__PURE__*/React.createElement("div", null, selectedDate && /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("label", null, "calendar date: "), /*#__PURE__*/React.createElement("span", {
    "data-test": "storybook-calendar-result"
  }, selectedDate.calendarDateString)), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("label", null, "iso date: "), /*#__PURE__*/React.createElement("span", {
    "data-test": "storybook-calendar-result-iso"
  }, (_selectedDate$calenda = selectedDate.calendarDate) === null || _selectedDate$calenda === void 0 ? void 0 : _selectedDate$calenda.withCalendar('iso8601').toLocaleString('en-GB', {
    dateStyle: 'long'
  }))), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("label", null, "callback:"), JSON.stringify(selectedDate, null, 2))))));
};
CalendarStoryWrapper.propTypes = {
  calendar: PropTypes.string.isRequired,
  component: PropTypes.elementType.isRequired,
  date: PropTypes.string,
  dir: PropTypes.oneOf(['ltr', 'rtl']),
  locale: PropTypes.string,
  timeZone: PropTypes.string,
  weekDayFormat: PropTypes.string
};
