// @dhis2-ui/logo/build/cjs/logo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LogoWhite = exports.LogoIconWhite = exports.LogoIcon = exports.Logo = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _logoIconSvg = require("./logo-icon-svg.js");
var _logoSvg = require("./logo-svg.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/*
 * These are official colors for dhis2 logos.
 *
 * They are isolated here and not in @dhis2/ui-constants as they should not be
 * shared with any other components.
 *
 * https://github.com/dhis2/dhis2-identity
 *
 */
const blue = '#0080d4';
const white = '#ffffff';
const dark = '#212225';
const LogoIcon = _ref => {
  let {
    className,
    dataTest = 'dhis2-uicore-logoicon'
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_logoIconSvg.LogoIconSvg, {
    iconColor: blue,
    className: className,
    dataTest: dataTest
  });
};
exports.LogoIcon = LogoIcon;
LogoIcon.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
const LogoIconWhite = _ref2 => {
  let {
    className,
    dataTest = 'dhis2-uicore-logoiconwhite'
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement(_logoIconSvg.LogoIconSvg, {
    iconColor: white,
    className: className,
    dataTest: dataTest
  });
};
exports.LogoIconWhite = LogoIconWhite;
LogoIconWhite.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
const Logo = _ref3 => {
  let {
    className,
    dataTest = 'dhis2-uicore-logo'
  } = _ref3;
  return /*#__PURE__*/_react.default.createElement(_logoSvg.LogoSvg, {
    iconColor: blue,
    textColor: dark,
    className: className,
    dataTest: dataTest
  });
};
exports.Logo = Logo;
Logo.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
const LogoWhite = _ref4 => {
  let {
    className,
    dataTest = 'dhis2-uicore-logowhite'
  } = _ref4;
  return /*#__PURE__*/_react.default.createElement(_logoSvg.LogoSvg, {
    iconColor: white,
    textColor: white,
    className: className,
    dataTest: dataTest
  });
};
exports.LogoWhite = LogoWhite;
LogoWhite.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
