// @dhis2-ui/file-input/build/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "FileInput", {
  enumerable: true,
  get: function () {
    return _index.FileInput;
  }
});
Object.defineProperty(exports, "FileInputField", {
  enumerable: true,
  get: function () {
    return _index3.FileInputField;
  }
});
Object.defineProperty(exports, "FileInputFieldWithList", {
  enumerable: true,
  get: function () {
    return _index4.FileInputFieldWithList;
  }
});
Object.defineProperty(exports, "FileList", {
  enumerable: true,
  get: function () {
    return _index2.FileList;
  }
});
Object.defineProperty(exports, "FileListItem", {
  enumerable: true,
  get: function () {
    return _index2.FileListItem;
  }
});
Object.defineProperty(exports, "FileListPlaceholder", {
  enumerable: true,
  get: function () {
    return _index2.FileListPlaceholder;
  }
});
var _index = require("./file-input/index.js");
var _index2 = require("./file-list/index.js");
var _index3 = require("./file-input-field/index.js");
var _index4 = require("./file-input-field-with-list/index.js");
