// @dhis2-ui/organisation-unit-tree/build/es/__e2e__/displaying_loading_error.js
import { CustomDataProvider } from '@dhis2/app-runtime';
import React from 'react';
import { OrganisationUnitTree } from '../index.js';
import { StatefulMultiSelectionWrapper, dataProviderData } from './common.js';
const dataProviderDataWithError = {
  organisationUnits: function () {
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    const [, {
      id
    }] = args;
    if (id === 'A0000000001') {
      return new Promise((resolve, reject) => setTimeout(() => {
        reject(new Error('Foobar custom error message'));
      }, 100));
    }
    return dataProviderData.organisationUnits(...args);
  }
};
export const A0000000001LoadingError = () => /*#__PURE__*/React.createElement(CustomDataProvider, {
  data: dataProviderDataWithError
}, /*#__PURE__*/React.createElement(StatefulMultiSelectionWrapper, null, () => /*#__PURE__*/React.createElement(OrganisationUnitTree, {
  roots: "A0000000000",
  onChange: () => null
})));
export const A0000000001LoadingErrorAutoexpand = () => /*#__PURE__*/React.createElement(CustomDataProvider, {
  data: dataProviderDataWithError
}, /*#__PURE__*/React.createElement(StatefulMultiSelectionWrapper, null, () => /*#__PURE__*/React.createElement(OrganisationUnitTree, {
  autoExpandLoadingError: true,
  roots: "A0000000000",
  onChange: () => null
})));
