// @dhis2-ui/organisation-unit-tree/build/es/__e2e__/expanded.js
import React from 'react';
import { OrganisationUnitTree } from '../index.js';
import { createDecoratorCustomDataProvider, createDecoratorStatefulMultiSelection } from './common.js';
const decorators = [createDecoratorStatefulMultiSelection(), createDecoratorCustomDataProvider()];
export const NoInitiallyExpandedPaths = (_, _ref) => {
  let {
    onChange,
    selected
  } = _ref;
  return /*#__PURE__*/React.createElement(OrganisationUnitTree, {
    roots: "A0000000000",
    selected: selected,
    onChange: onChange
  });
};
NoInitiallyExpandedPaths.decorators = decorators;
export const InitiallyExpandedPaths = (_, _ref2) => {
  let {
    onChange,
    selected
  } = _ref2;
  return /*#__PURE__*/React.createElement(OrganisationUnitTree, {
    roots: "A0000000000",
    selected: selected,
    onChange: onChange,
    initiallyExpanded: ['/A0000000000']
  });
};
InitiallyExpandedPaths.decorators = decorators;
export const WithRootMainAndRootSubOrgUnit = (_, _ref3) => {
  let {
    onChange,
    selected
  } = _ref3;
  return /*#__PURE__*/React.createElement(OrganisationUnitTree, {
    roots: ['A0000000000', 'A0000000001'],
    selected: selected,
    onChange: onChange
  });
};
WithRootMainAndRootSubOrgUnit.decorators = decorators;
