// @dhis2-ui/organisation-unit-tree/build/es/__e2e__/no_selection.js
import { CustomDataProvider } from '@dhis2/app-runtime';
import React from 'react';
import { OrganisationUnitTree } from '../index.js';
import { StatefulMultiSelectionWrapper, dataProviderData } from './common.js';
export const NoSelectionClosed = () => /*#__PURE__*/React.createElement(CustomDataProvider, {
  data: dataProviderData
}, /*#__PURE__*/React.createElement(StatefulMultiSelectionWrapper, null, _ref => {
  let {
    onChange
  } = _ref;
  return /*#__PURE__*/React.createElement(OrganisationUnitTree, {
    disableSelection: true,
    roots: "A0000000000",
    onChange: onChange
  });
}));
export const NoSelectionRootOpened = () => /*#__PURE__*/React.createElement(CustomDataProvider, {
  data: dataProviderData
}, /*#__PURE__*/React.createElement(StatefulMultiSelectionWrapper, null, _ref2 => {
  let {
    onChange
  } = _ref2;
  return /*#__PURE__*/React.createElement(OrganisationUnitTree, {
    disableSelection: true,
    roots: "A0000000000",
    onChange: onChange,
    initiallyExpanded: ['/A0000000000']
  });
}));
