// @dhis2-ui/organisation-unit-tree/build/es/__e2e__/path_based_filtering.js
import { CustomDataProvider } from '@dhis2/app-runtime';
import React from 'react';
import { OrganisationUnitTree } from '../index.js';
import { StatefulMultiSelectionWrapper, dataProviderData } from './common.js';
window.allUnits = [];
for (let i = 0; i < 7; ++i) {
  window.allUnits.push(`A000000000${i}`);
}
export const FilteredBy3LevelPath = () => /*#__PURE__*/React.createElement(CustomDataProvider, {
  data: dataProviderData
}, /*#__PURE__*/React.createElement(StatefulMultiSelectionWrapper, null, _ref => {
  let {
    onChange
  } = _ref;
  return /*#__PURE__*/React.createElement(OrganisationUnitTree, {
    roots: "A0000000000",
    onChange: onChange,
    initiallyExpanded: ['/A0000000000', '/A0000000000/A0000000001', '/A0000000000/A0000000002'],
    filter: ['/A0000000000/A0000000001/A0000000003']
  });
}));
export const FilteredBy3LevelPathAnd2LevelPath = () => /*#__PURE__*/React.createElement(CustomDataProvider, {
  data: dataProviderData
}, /*#__PURE__*/React.createElement(StatefulMultiSelectionWrapper, null, _ref2 => {
  let {
    onChange
  } = _ref2;
  return /*#__PURE__*/React.createElement(OrganisationUnitTree, {
    roots: "A0000000000",
    onChange: onChange,
    initiallyExpanded: ['/A0000000000', '/A0000000000/A0000000001', '/A0000000000/A0000000002'],
    filter: ['/A0000000000/A0000000001/A0000000003', '/A0000000000/A0000000002']
  });
}));
