// @dhis2-ui/organisation-unit-tree/build/es/__e2e__/single_selection.js
import { CustomDataProvider } from '@dhis2/app-runtime';
import React from 'react';
import { OrganisationUnitTree } from '../index.js';
import { StatefulMultiSelectionWrapper, dataProviderData } from './common.js';
const data = {
  organisationUnits: function () {
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    const [, {
      id
    }] = args;
    if (id === 'A0000000000') {
      return dataProviderData.organisationUnits(...args).then(response => ({
        ...response,
        children: []
      }));
    }
    if (id === 'A0000000001') {
      return dataProviderData.organisationUnits(...args).then(response => ({
        ...response,
        path: '/A0000000001',
        children: []
      }));
    }
    return Promise.resolve({});
  }
};
export const SingleSelection = () => /*#__PURE__*/React.createElement(CustomDataProvider, {
  data: data
}, /*#__PURE__*/React.createElement(StatefulMultiSelectionWrapper, null, _ref => {
  let {
    selected,
    onChange
  } = _ref;
  return /*#__PURE__*/React.createElement(OrganisationUnitTree, {
    singleSelection: true,
    roots: ['A0000000000', 'A0000000001'],
    onChange: onChange,
    selected: selected
  });
}));
