// @dhis2-ui/tag/build/cjs/tag.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithoutIcon = exports.WithText = exports.WithIcon = exports.WithDefaultMaxWidth = exports.WithCustomMaxWidth = void 0;
var _react = _interopRequireDefault(require("react"));
var _tag = require("./tag.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'Tag'
};
const WithoutIcon = () => /*#__PURE__*/_react.default.createElement(_tag.Tag, null, "Default");
exports.WithoutIcon = WithoutIcon;
const WithIcon = () => /*#__PURE__*/_react.default.createElement(_tag.Tag, {
  icon: /*#__PURE__*/_react.default.createElement("span", null, "Icon")
}, "Default");
exports.WithIcon = WithIcon;
const WithText = () => /*#__PURE__*/_react.default.createElement(_tag.Tag, null, "Text content");
exports.WithText = WithText;
const WithDefaultMaxWidth = () => /*#__PURE__*/_react.default.createElement(_tag.Tag, null, "This is a lot of content that should exceed the default max width of the tag component quite significantly so testing is still works when the default value of the max width is changed");
exports.WithDefaultMaxWidth = WithDefaultMaxWidth;
const WithCustomMaxWidth = () => /*#__PURE__*/_react.default.createElement(_tag.Tag, {
  maxWidth: "30px"
}, "Text content");
exports.WithCustomMaxWidth = WithCustomMaxWidth;
