// @dhis2-ui/layer/build/cjs/layer.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithClickHandler = exports.NestedLowerLevels = exports.NestedHigherLevels = exports.NestedHigherLevelEndsOnTop = exports.LevelsAreRespectedWhenNesting = exports.InequalSiblings = exports.EqualSiblings = exports.Default = exports.Blocking = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _react = _interopRequireDefault(require("react"));
var _layer = require("./layer.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
window.onButtonClick = window.Cypress && window.Cypress.cy.stub();
window.onLayerClick = window.Cypress && window.Cypress.cy.stub();
const createNamedLayerClick = name => () => {
  window.onLayerClick(name);
};
var _default = exports.default = {
  title: 'Layer',
  component: _layer.Layer
};
const Default = () => /*#__PURE__*/_react.default.createElement(_layer.Layer, null, /*#__PURE__*/_react.default.createElement("p", null, "I am a child"));
exports.Default = Default;
const Blocking = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
  onClick: window.onButtonClick
}, "Test"), /*#__PURE__*/_react.default.createElement(_layer.Layer, null));
exports.Blocking = Blocking;
const WithClickHandler = () => /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  onBackdropClick: window.onLayerClick
}, /*#__PURE__*/_react.default.createElement("button", {
  onClick: window.onButtonClick
}, "Test"));
exports.WithClickHandler = WithClickHandler;
const EqualSiblings = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  onBackdropClick: createNamedLayerClick('first')
}), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  onBackdropClick: createNamedLayerClick('second')
}));
exports.EqualSiblings = EqualSiblings;
const InequalSiblings = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.alert,
  onBackdropClick: createNamedLayerClick('alert')
}), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.blocking,
  onBackdropClick: createNamedLayerClick('blocking')
}), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.applicationTop,
  onBackdropClick: createNamedLayerClick('applicationTop')
}));
exports.InequalSiblings = InequalSiblings;
const NestedLowerLevels = () => /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.alert,
  dataTest: "alert",
  onBackdropClick: createNamedLayerClick('alert')
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.blocking,
  dataTest: "blocking",
  disablePortal: true,
  onBackdropClick: createNamedLayerClick('blocking')
}));
exports.NestedLowerLevels = NestedLowerLevels;
const NestedHigherLevels = () => /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.blocking,
  dataTest: "blocking",
  onBackdropClick: createNamedLayerClick('blocking')
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.alert,
  dataTest: "alert",
  onBackdropClick: createNamedLayerClick('alert')
}));
exports.NestedHigherLevels = NestedHigherLevels;
const LevelsAreRespectedWhenNesting = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: 1000
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: 1000
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: 1000,
  onBackdropClick: createNamedLayerClick('1000')
}))), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: 1001,
  onBackdropClick: createNamedLayerClick('1001')
}));
exports.LevelsAreRespectedWhenNesting = LevelsAreRespectedWhenNesting;
const NestedHigherLevelEndsOnTop = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.applicationTop
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.alert,
  onBackdropClick: createNamedLayerClick('alert')
})), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.blocking,
  onBackdropClick: createNamedLayerClick('blocking')
}));
exports.NestedHigherLevelEndsOnTop = NestedHigherLevelEndsOnTop;
