// @dhis2-ui/loader/build/cjs/circular-loader/circular-loader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CircularLoader = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const CircularLoader = _ref => {
  let {
    small,
    large,
    extrasmall,
    invert,
    className,
    dataTest = 'dhis2-uicore-circularloader',
    'aria-label': ariaLabel
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    role: "progressbar",
    "data-test": dataTest,
    "aria-label": ariaLabel,
    className: _style.default.dynamic([["3803726117", [_uiConstants.colors.blue600, _uiConstants.colors.white]]]) + " " + ((0, _classnames.default)(className, {
      small,
      large,
      extrasmall,
      invert
    }) || "")
  }, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3803726117",
    dynamic: [_uiConstants.colors.blue600, _uiConstants.colors.white]
  }, [`div.__jsx-style-dynamic-selector{width:48px;height:48px;border:6px solid rgba(110,122,138,0.15);border-bottom-color:${_uiConstants.colors.blue600};border-radius:50%;-webkit-animation:rotation-__jsx-style-dynamic-selector 1s linear infinite;animation:rotation-__jsx-style-dynamic-selector 1s linear infinite;}`, "@-webkit-keyframes rotation-__jsx-style-dynamic-selector{0%{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}100%{-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}", "@keyframes rotation-__jsx-style-dynamic-selector{0%{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}100%{-webkit-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}", ".small.__jsx-style-dynamic-selector{width:24px;height:24px;border-width:4px;}", ".large.__jsx-style-dynamic-selector{width:72px;height:72px;border-width:8px;}", ".extrasmall.__jsx-style-dynamic-selector{width:16px;height:16px;border-width:2px;}", `.invert.__jsx-style-dynamic-selector{border-color:rgba(33,41,52,0.5);border-bottom-color:${_uiConstants.colors.white};}`]));
};
exports.CircularLoader = CircularLoader;
CircularLoader.propTypes = {
  'aria-label': _propTypes.default.string,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  extrasmall: _uiConstants.sharedPropTypes.sizePropType,
  invert: _propTypes.default.bool,
  large: _uiConstants.sharedPropTypes.sizePropType,
  small: _uiConstants.sharedPropTypes.sizePropType
};
