// @dhis2-ui/segmented-control/build/cjs/segmented-control.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.LongLabels = exports.DisabledOption = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _segmentedControl = require("./segmented-control.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const logger = _ref => {
  let {
    value
  } = _ref;
  return console.log(`Selected value: ${value}`);
};
var _default = exports.default = {
  title: 'Segmented Control',
  component: _segmentedControl.SegmentedControl,
  parameters: {
    componentSubtitle: 'Allows selection between related options'
  },
  args: {
    options: [{
      label: 'Dog',
      value: 'DOG'
    }, {
      label: 'Cat',
      value: 'CAT'
    }, {
      label: 'Giraffe',
      value: 'GIRAFFE'
    }],
    selected: 'DOG',
    onChange: logger,
    ariaLabel: 'Segmented control label'
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement(_segmentedControl.SegmentedControl, args);
const Default = exports.Default = Template.bind({});
const DisabledOption = exports.DisabledOption = Template.bind({});
DisabledOption.args = {
  options: [{
    label: 'One',
    value: 'ONE'
  }, {
    label: 'Two',
    value: 'TWO',
    disabled: true
  }, {
    label: 'Three',
    value: 'THREE'
  }],
  selected: 'THREE'
};
const LongLabels = exports.LongLabels = Template.bind({});
LongLabels.args = {
  options: [{
    label: 'Program configuration, security, and distribution',
    value: 'PROGRAM'
  }, {
    label: 'Two',
    value: 'TWO'
  }, {
    label: 'Three',
    value: 'THREE'
  }],
  selected: 'PROGRAM'
};
