// @dhis2-ui/cover/build/cjs/cover.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithClickHandler = exports.WithCenteredContentCircularLoader = exports.Translucent = exports.Default = void 0;
var _center = require("@dhis2-ui/center");
var _loader = require("@dhis2-ui/loader");
var _react = _interopRequireDefault(require("react"));
var _cover = require("./cover.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
Covers sibling components.  Useful for covering a component while it is loading.

\`\`\`js
import { Cover } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Component Cover',
  component: _cover.Cover,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    width: '400px',
    height: '200px',
    position: 'relative',
    border: '1px dotted grey'
  }
}, /*#__PURE__*/_react.default.createElement(_cover.Cover, args), /*#__PURE__*/_react.default.createElement("h1", null, "Text behind the cover"), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum"));
const Default = exports.Default = Template.bind({});
const Translucent = exports.Translucent = Template.bind({});
Translucent.args = {
  translucent: true
};
const WithClickHandler = exports.WithClickHandler = Template.bind({});
WithClickHandler.args = {
  onClick: () => alert('Cover was clicked')
};
const WithCenteredContentCircularLoader = exports.WithCenteredContentCircularLoader = Template.bind({});
WithCenteredContentCircularLoader.args = {
  translucent: true,
  children: /*#__PURE__*/_react.default.createElement(_center.Center, null, /*#__PURE__*/_react.default.createElement(_loader.CircularLoader, null))
};
