// @dhis2-ui/css/build/es/css-reset/css-reset.prod.stories.js
import React from 'react';
import { CssReset } from './index.js';
const description = `
A tool for adding a global normalization stylesheet into the DOM that applies DHIS2 styles.

- https://github.com/necolas/normalize.css
- https://www.paulirish.com/2012/box-sizing-border-box-ftw/

\`\`\`js
import { CssReset } from '@dhis2/ui'
\`\`\`
`;
const App = _ref => {
  let {
    children
  } = _ref;
  return /*#__PURE__*/React.createElement("div", null, children);
};
export default {
  title: 'CSS Reset',
  component: CssReset,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
export const Default = args => /*#__PURE__*/React.createElement(App, null, /*#__PURE__*/React.createElement(CssReset, args), /*#__PURE__*/React.createElement("p", null, "The ", /*#__PURE__*/React.createElement("code", null, "CssReset"), " component injects a global normalization stylesheet into the DOM that sets the DHIS2 application defaults."), /*#__PURE__*/React.createElement("p", null, "This also sets the ", /*#__PURE__*/React.createElement("code", null, "font-family"), " on the ", /*#__PURE__*/React.createElement("code", null, "body"), ' ', "element to the DHIS2 font, which allows it to trickle down to the components as well."), /*#__PURE__*/React.createElement("p", null, "Just include the component in your application, preferably as early as possible to avoid FOUC.", ' '));
