// @dhis2-ui/organisation-unit-tree/build/es/organisation-unit-tree/use-root-org-data/use-root-org-data.js
import { useDataQuery } from '@dhis2/app-runtime';
import { useMemo } from 'react';
import { patchMissingDisplayName } from './patch-missing-display-name.js';
export const createRootQuery = ids => ids.reduce((query, id) => ({
  ...query,
  [id]: {
    id,
    resource: `organisationUnits`,
    params: _ref => {
      let {
        isUserDataViewFallback
      } = _ref;
      return {
        isUserDataViewFallback,
        fields: ['displayName', 'path', 'id']
      };
    }
  }
}), {});

/**
 * @param {string[]} ids
 * @param {Object} [options]
 * @param {boolean} [options.withChildren]
 * @param {boolean} [options.isUserDataViewFallback]
 * @returns {Object}
 */
export const useRootOrgData = function (ids) {
  let {
    isUserDataViewFallback
  } = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
  const query = useMemo(() => createRootQuery(ids), [ids]);
  const variables = {
    isUserDataViewFallback
  };
  const rootOrgUnits = useDataQuery(query, {
    variables
  });
  const {
    called,
    loading,
    error,
    data,
    refetch
  } = rootOrgUnits;
  const patchedData = useMemo(() => {
    return data ? patchMissingDisplayName(data) : data;
  }, [data]);
  return {
    called,
    loading,
    error: error || null,
    data: patchedData || null,
    refetch
  };
};
