// @dhis2-ui/select/build/es/multi-select/multi-select.e2e.stories.js
import _JSXStyle from "styled-jsx/style";
import { Button } from '@dhis2-ui/button';
import { Modal, ModalTitle, ModalContent } from '@dhis2-ui/modal';
import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { MultiSelectOption } from '../index.js';
import { MultiSelect } from './multi-select.js';
window.onChange = window.Cypress && window.Cypress.cy.stub();
window.onFocus = window.Cypress && window.Cypress.cy.stub();
window.onBlur = window.Cypress && window.Cypress.cy.stub();
const CustomMultiSelectOption = _ref => {
  let {
    label,
    onClick
  } = _ref;
  return /*#__PURE__*/React.createElement("div", {
    onClick: e => onClick({}, e)
  }, label);
};
CustomMultiSelectOption.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func
};
const options = [{
  name: 'Hello',
  id: '1'
}, {
  name: 'world',
  id: '2'
}, {
  name: 'foo',
  id: '3'
}, {
  name: 'bar',
  id: '4'
}];
export default {
  title: 'MultiSelect'
};
export const WithOptions = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithOptionsAndOnChange = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  onChange: window.onChange
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithOnFocus = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  onFocus: window.onFocus
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithOnBlur = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  onBlur: window.onBlur
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithCustomOptionsAndOnChange = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  onChange: window.onChange
}, /*#__PURE__*/React.createElement(CustomMultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(CustomMultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(CustomMultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithInvalidOptions = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select"
}, /*#__PURE__*/React.createElement("div", null, "invalid one"), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement("div", null, "invalid two"), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement("div", null, "invalid three"), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}), null, undefined, false);
export const WithInvalidFilterableOptions = () => /*#__PURE__*/React.createElement(MultiSelect, {
  filterable: true,
  className: "select",
  noMatchText: "No match"
}, /*#__PURE__*/React.createElement("div", null, "invalid one"), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement("div", null, "invalid two"), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement("div", null, "invalid three"), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithInitialFocus = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  initialFocus: true
});
export const Empty = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select"
});
export const EmptyWithEmptyText = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  empty: "Custom empty text"
});
export const EmptyWithEmptyComponent = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  empty: /*#__PURE__*/React.createElement("div", {
    className: "custom-empty"
  }, "Custom empty component")
});
export const WithOptionsAndLoading = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  loading: true
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithOptionsLoadingAndLoadingText = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  loadingText: "Loading options",
  loading: true
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithMoreThanTenOptions = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "12",
  label: "option twelve"
}));
export const WithMoreThanThreeOptionsAndA100pxMaxHeight = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  maxHeight: "100px"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "12",
  label: "option twelve"
}));
export const WithOptionsASelectionAndDisabled = () => /*#__PURE__*/React.createElement(MultiSelect, {
  disabled: true,
  className: "select",
  selected: ['1']
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithOptionsAndDisabled = () => /*#__PURE__*/React.createElement(MultiSelect, {
  disabled: true,
  className: "select"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithPrefix = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  prefix: "Prefix text"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithPrefixAndSelection = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  prefix: "Prefix text",
  selected: ['1']
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithPlaceholder = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  placeholder: "Placeholder text"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithPlaceholderAndSelection = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  selected: ['1'],
  placeholder: "Placeholder text"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithDisabledOptionAndOnChange = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  onChange: window.onChange
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "4",
  label: "disabled option",
  disabled: true
}));
export const WithOptionsAndASelection = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  selected: ['1']
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithOptionsASelectionAndOnChange = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  selected: ['1'],
  onChange: window.onChange
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithOptionsAndMultipleSelections = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  selected: ['1', '2']
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithClearButtonSelectionAndOnChange = () => /*#__PURE__*/React.createElement(MultiSelect, {
  clearable: true,
  clearText: "Clear",
  className: "select",
  selected: ['1'],
  onChange: window.onChange
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithFilterField = () => /*#__PURE__*/React.createElement(MultiSelect, {
  filterable: true,
  noMatchText: "No match found",
  className: "select"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const DefaultPosition = () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "12",
  label: "option twelve"
})), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "2883393180"
}, ["#storybook-root{height:2000px !important;}"]));
export const FlippedPosition = () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "12",
  label: "option twelve"
})), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "979451016"
}, ["html,body,#storybook-root{position:relative;height:500px;max-height:500px;}", "#storybook-root{padding-top:400px !important;}"]));
export const ShiftedIntoView = () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select"
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "12",
  label: "option twelve"
})), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "438540012"
}, ["html,body,#storybook-root{position:relative;height:300px !important;max-height:300px;}", "#storybook-root{padding-top:130px !important;}"]));
export const WithDuplicateSelectedOptionValues = () => /*#__PURE__*/React.createElement(MultiSelect, {
  className: "select",
  selected: ['1']
}, /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "1",
  label: "option one a"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/React.createElement(MultiSelectOption, {
  value: "3",
  label: "option three"
}));
export const WithOptionsThatCanBeAddedToTheInput = () => {
  const [values, setValues] = React.useState([]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(MultiSelect, {
    className: "select",
    selected: values,
    onChange: _ref2 => {
      let {
        selected
      } = _ref2;
      window.onChange && window.onChange({
        selected
      });
      setValues(selected);
    }
  }, /*#__PURE__*/React.createElement(MultiSelectOption, {
    value: "1",
    label: "option one"
  }), /*#__PURE__*/React.createElement(MultiSelectOption, {
    value: "2",
    label: "option two"
  }), /*#__PURE__*/React.createElement(MultiSelectOption, {
    value: "3",
    label: "option three"
  })), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "4252233386"
  }, ["#storybook-root{height:2000px !important;}"]));
};
export const MenuWidthChanging = () => {
  const [toggle, setToggle] = useState(false);
  return /*#__PURE__*/React.createElement("div", {
    className: "App"
  }, /*#__PURE__*/React.createElement(Modal, {
    dataTest: 'myModal'
  }, /*#__PURE__*/React.createElement(ModalTitle, null, "Modal"), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: 'flex'
    }
  }, toggle && /*#__PURE__*/React.createElement("div", {
    style: {
      padding: '30px',
      border: '1px solid green'
    },
    className: "toggler"
  }, "Stuff"), /*#__PURE__*/React.createElement("div", {
    style: {
      flexGrow: 1
    }
  }, /*#__PURE__*/React.createElement(MultiSelect, {
    onChange: _ref3 => {
      let {
        selected
      } = _ref3;
      return console.log('size changed to ' + selected);
    }
  }, options.map(_ref4 => {
    let {
      name,
      id
    } = _ref4;
    return /*#__PURE__*/React.createElement(MultiSelectOption, {
      key: id,
      value: id,
      label: name
    });
  }))), toggle && /*#__PURE__*/React.createElement("div", {
    style: {
      padding: '30px',
      border: '1px solid green'
    },
    className: "toggler"
  }, "Stuff")), /*#__PURE__*/React.createElement(Button, {
    onClick: () => setToggle(!toggle)
  }, "Toggle"))));
};
