// @dhis2-ui/selector-bar/build/cjs/selector-bar/__stories__/common.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.workflows = exports.getOrganisationUnitData = exports.decoratorCommonStyles = exports.dataSets = exports.dataProviderData = exports.createStatefulDecorator = exports.createDecoratorCustomDataProvider = exports.OrgUnitSelect = exports.MenuSelect = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _appRuntime = require("@dhis2/app-runtime");
var _uiConstants = require("@dhis2/ui-constants");
var _menu = require("@dhis2-ui/menu");
var _organisationUnitTree = require("@dhis2-ui/organisation-unit-tree");
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const decoratorCommonStyles = fn => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, fn(), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "3746816729",
  dynamic: [_uiConstants.colors.grey100]
}, [`body{background:${_uiConstants.colors.grey100};padding:0 !important;}`, "div#root{padding:0;}"]));
exports.decoratorCommonStyles = decoratorCommonStyles;
const getOrganisationUnitData = id => {
  let data;
  if (id === 'A0000000000') {
    data = {
      id: 'A0000000000',
      path: '/A0000000000',
      displayName: 'Org Unit 1',
      children: [{
        id: 'A0000000001',
        path: '/A0000000000/A0000000001',
        children: [{
          id: 'A0000000003'
        }, {
          id: 'A0000000004'
        }],
        displayName: 'Org Unit 2'
      }, {
        id: 'A0000000002',
        path: '/A0000000000/A0000000002',
        children: [],
        displayName: 'Org Unit 3'
      }, {
        id: 'A0000000006',
        path: '/A0000000000/A0000000006',
        children: [],
        displayName: 'Org Unit 7'
      }]
    };
  }
  if (id === 'A0000000001') {
    data = {
      id: 'A0000000001',
      path: '/A0000000000/A0000000001',
      displayName: 'Org Unit 2',
      children: [{
        id: 'A0000000003',
        path: '/A0000000000/A0000000001/A0000000003',
        children: [],
        displayName: 'Org Unit 4'
      }, {
        id: 'A0000000004',
        path: '/A0000000000/A0000000001/A0000000004',
        children: [],
        displayName: 'Org Unit 5'
      }]
    };
  }
  if (id === 'A0000000002') {
    data = {
      displayName: 'Org Unit 3',
      id: 'A0000000002',
      path: '/A0000000000/A0000000002',
      children: []
    };
  }
  if (id === 'A0000000003') {
    data = {
      displayName: 'Org Unit 4',
      id: 'A0000000003',
      path: '/A0000000000/A0000000001/A0000000003',
      children: [{
        id: 'A0000000007',
        path: '/A0000000000/A0000000001/A0000000003/A0000000007',
        children: [],
        displayName: 'Org Unit 8'
      }]
    };
  }
  if (id === 'A0000000004') {
    data = {
      displayName: 'Org Unit 5',
      id: 'A0000000004',
      path: '/A0000000000/A0000000001/A0000000004',
      children: []
    };
  }
  if (id === 'A0000000006') {
    data = {
      displayName: 'Org Unit 7',
      id: 'A0000000006',
      path: '/A0000000000/A0000000006',
      children: []
    };
  }
  if (id === 'A0000000007') {
    data = {
      displayName: 'Org Unit 8',
      id: 'A0000000007',
      path: '/A0000000000/A0000000001/A0000000003/A0000000007',
      children: []
    };
  }
  return data;
};
exports.getOrganisationUnitData = getOrganisationUnitData;
const dataProviderData = exports.dataProviderData = {
  organisationUnits: (_, _ref) => {
    let {
      id
    } = _ref;
    const data = getOrganisationUnitData(id);
    if (!data) {
      return Promise.reject(new Error('404 - Org unit not found'));
    }
    return Promise.resolve(data);
  }
};
const createDecoratorCustomDataProvider = args => {
  const data = (args === null || args === void 0 ? void 0 : args.data) || dataProviderData;
  return Story => {
    window.dataProviderData = data;
    return /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
      data: data
    }, /*#__PURE__*/_react.default.createElement(Story, null));
  };
};
exports.createDecoratorCustomDataProvider = createDecoratorCustomDataProvider;
const createStatefulDecorator = args => {
  return fn => {
    const [workflow, setWorkflow] = (0, _react.useState)((args === null || args === void 0 ? void 0 : args.workflow) || null);
    const [workflowOpen, setWorkflowOpen] = (0, _react.useState)((args === null || args === void 0 ? void 0 : args.workflowOpen) || false);
    const [dataSet, setDataSet] = (0, _react.useState)((args === null || args === void 0 ? void 0 : args.dataSet) || null);
    const [dataSetOpen, setDataSetOpen] = (0, _react.useState)((args === null || args === void 0 ? void 0 : args.dataSetOpen) || false);
    const [orgUnit, setOrgUnit] = (0, _react.useState)((args === null || args === void 0 ? void 0 : args.orgUnit) || null);
    const [orgUnitOpen, setOrgUnitOpen] = (0, _react.useState)((args === null || args === void 0 ? void 0 : args.orgUnitOpen) || false);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, fn({
      workflow,
      setWorkflow,
      workflowOpen,
      setWorkflowOpen,
      dataSet,
      setDataSet,
      dataSetOpen,
      setDataSetOpen,
      orgUnit,
      setOrgUnit,
      orgUnitOpen,
      setOrgUnitOpen
    }));
  };
};
exports.createStatefulDecorator = createStatefulDecorator;
const workflows = exports.workflows = [{
  value: 'm<4y',
  label: 'Mortality < 4 years'
}, {
  value: 'm<5y',
  label: 'Mortality < 5 years'
}];
const dataSets = exports.dataSets = [{
  value: 'data-set-1',
  label: 'Data set 1'
}, {
  value: 'data-set-2',
  label: 'Data set 2'
}, {
  value: 'data-set-3',
  label: 'Data set 3'
}];
const MenuSelect = _ref2 => {
  let {
    values,
    selected,
    onChange
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 400
    }
  }, /*#__PURE__*/_react.default.createElement(_menu.Menu, null, values.map(_ref3 => {
    let {
      value,
      label
    } = _ref3;
    return /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
      key: value,
      label: label,
      active: selected === value,
      onClick: () => onChange({
        selected: value
      })
    });
  })));
};
exports.MenuSelect = MenuSelect;
const OrgUnitSelect = _ref4 => {
  let {
    onChange,
    selected
  } = _ref4;
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 400,
      minHeight: 400,
      maxHeight: '70vh'
    }
  }, /*#__PURE__*/_react.default.createElement(_organisationUnitTree.OrganisationUnitTree, {
    singleSelection: true,
    onChange: onChange,
    roots: ['A0000000000'],
    selected: selected
  }));
};
exports.OrgUnitSelect = OrgUnitSelect;
