// @dhis2-ui/selector-bar/build/cjs/selector-bar/__stories__/rtl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RTL = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _button = require("@dhis2-ui/button");
var _react = _interopRequireDefault(require("react"));
var _index = require("../../index.js");
var _common = require("./common.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const RHSContent = () => /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-876392447" + " " + "rhs-content"
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  small: true
}, "Options"), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "876392447"
}, [".rhs-content.jsx-876392447{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;height:40px;padding:0 16px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}"]));
const RTL = (_, _ref) => {
  let {
    workflow,
    setWorkflow,
    workflowOpen,
    setWorkflowOpen,
    orgUnit,
    setOrgUnit,
    orgUnitOpen,
    setOrgUnitOpen
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    dir: "rtl"
  }, /*#__PURE__*/_react.default.createElement(_index.SelectorBar, {
    disableClearSelections: !workflow || !orgUnit,
    onClearSelectionClick: () => {
      setWorkflow(null);
      setOrgUnit(null);
    },
    additionalContent: /*#__PURE__*/_react.default.createElement(RHSContent, null)
  }, /*#__PURE__*/_react.default.createElement(_index.SelectorBarItem, {
    disabled: false,
    label: "Workflow",
    value: workflow === null || workflow === void 0 ? void 0 : workflow.label,
    noValueMessage: "Choose a workflow",
    open: workflowOpen,
    setOpen: setWorkflowOpen,
    onClearSelectionClick: () => {
      setWorkflow(null);
    }
  }, /*#__PURE__*/_react.default.createElement(_common.MenuSelect, {
    values: _common.workflows,
    selected: workflow === null || workflow === void 0 ? void 0 : workflow.value,
    onChange: _ref2 => {
      let {
        selected
      } = _ref2;
      setWorkflow(_common.workflows.find(currentWorkflow => selected === currentWorkflow.value));
      setWorkflowOpen(false);
    }
  })), /*#__PURE__*/_react.default.createElement(_index.SelectorBarItem, {
    disabled: !workflow,
    label: "Org unit",
    value: orgUnit === null || orgUnit === void 0 ? void 0 : orgUnit.displayName,
    noValueMessage: "Choose an organisation unit",
    open: orgUnitOpen,
    setOpen: setOrgUnitOpen,
    onClearSelectionClick: () => {
      setOrgUnit(null);
    }
  }, /*#__PURE__*/_react.default.createElement(_common.OrgUnitSelect, {
    selected: orgUnit ? [orgUnit.path] : [],
    onChange: (nextOrgUnit, evt) => {
      evt.stopPropagation();
      setOrgUnit(nextOrgUnit);
      setOrgUnitOpen(false);
    }
  }))));
};
exports.RTL = RTL;
RTL.decorators = [_common.decoratorCommonStyles, (0, _common.createStatefulDecorator)(), (0, _common.createDecoratorCustomDataProvider)()];
