// @dhis2-ui/selector-bar/build/cjs/selector-bar/__stories__/with-multiple-rows.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithMultipleRows = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _organisationUnitTree = require("@dhis2-ui/organisation-unit-tree");
var _react = _interopRequireDefault(require("react"));
var _index = require("../../index.js");
var _common = require("./common.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const WithMultipleRows = (_, _ref) => {
  let {
    workflow,
    setWorkflow,
    workflowOpen,
    setWorkflowOpen,
    dataSet,
    setDataSet,
    dataSetOpen,
    setDataSetOpen,
    orgUnit,
    setOrgUnit,
    orgUnitOpen,
    setOrgUnitOpen
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-3203152640" + " " + "narrow-container"
  }, /*#__PURE__*/_react.default.createElement(_index.SelectorBar, {
    disableClearSelections: true,
    onClearSelectionClick: () => null
  }, /*#__PURE__*/_react.default.createElement(_index.SelectorBarItem, {
    disabled: false,
    label: "Workflow",
    value: workflow === null || workflow === void 0 ? void 0 : workflow.label,
    noValueMessage: "Choose a workflow",
    open: workflowOpen,
    setOpen: setWorkflowOpen
  }, /*#__PURE__*/_react.default.createElement(_common.MenuSelect, {
    values: _common.workflows,
    selected: workflow === null || workflow === void 0 ? void 0 : workflow.value,
    onChange: _ref2 => {
      let {
        selected
      } = _ref2;
      setWorkflow(_common.workflows.find(currentWorkflow => selected === currentWorkflow.value));
      setWorkflowOpen(false);
    }
  })), /*#__PURE__*/_react.default.createElement(_index.SelectorBarItem, {
    disabled: !workflow,
    label: "Data set",
    value: dataSet === null || dataSet === void 0 ? void 0 : dataSet.label,
    noValueMessage: "Choose a data set",
    open: dataSetOpen,
    setOpen: setDataSetOpen
  }, /*#__PURE__*/_react.default.createElement(_common.MenuSelect, {
    values: _common.dataSets,
    selected: dataSet === null || dataSet === void 0 ? void 0 : dataSet.value,
    onChange: _ref3 => {
      let {
        selected
      } = _ref3;
      setDataSet(_common.dataSets.find(currentWorkflow => selected === currentWorkflow.value));
      setDataSetOpen(false);
    }
  })), /*#__PURE__*/_react.default.createElement(_index.SelectorBarItem, {
    disabled: !workflow && !dataSet,
    label: "Org unit",
    value: orgUnit === null || orgUnit === void 0 ? void 0 : orgUnit.displayName,
    noValueMessage: "Choose an organisation unit",
    open: orgUnitOpen,
    setOpen: setOrgUnitOpen
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 400,
      minHeight: 400,
      maxHeight: '70vh'
    },
    className: "jsx-3203152640"
  }, /*#__PURE__*/_react.default.createElement(_organisationUnitTree.OrganisationUnitTree, {
    singleSelection: true,
    onChange: (nextOrgUnit, evt) => {
      evt.stopPropagation();
      setOrgUnit(nextOrgUnit);
      setOrgUnitOpen(false);
    },
    roots: ['A0000000000'],
    selected: orgUnit ? [orgUnit.path] : []
  })))), /*#__PURE__*/_react.default.createElement("p", {
    style: {
      padding: '0 16px',
      textAlign: 'center'
    },
    className: "jsx-3203152640"
  }, "This story shows the selection bar component with multiple rows when a workflow has been selected."), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3203152640"
  }, [".narrow-container.jsx-3203152640{margin:0 auto;width:100%;max-width:600px;}"]));
};
exports.WithMultipleRows = WithMultipleRows;
WithMultipleRows.decorators = [_common.decoratorCommonStyles, (0, _common.createStatefulDecorator)({
  workflow: _common.workflows[0]
}), (0, _common.createDecoratorCustomDataProvider)()];
