// @dhis2-ui/transfer/build/cjs/container.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Container = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Container = _ref => {
  let {
    children,
    dataTest,
    className,
    height
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["3332822660", [height]]]) + " " + (className || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3332822660",
    dynamic: [height]
  }, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;height:${height};}`]));
};
exports.Container = Container;
Container.propTypes = {
  height: _propTypes.default.string.isRequired,
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
