// @dhis2-ui/transfer/build/cjs/filter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Filter = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _field = require("@dhis2-ui/field");
var _input = require("@dhis2-ui/input");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Filter = _ref => {
  let {
    dataTest,
    filter,
    onChange,
    label,
    placeholder
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["3679907342", [_uiConstants.spacers.dp8, _uiConstants.spacers.dp8]]])
  }, /*#__PURE__*/_react.default.createElement(_field.Field, {
    label: label,
    name: dataTest,
    dataTest: `${dataTest}-field`
  }, /*#__PURE__*/_react.default.createElement(_input.Input, {
    dataTest: `${dataTest}-input`,
    type: "search",
    placeholder: placeholder,
    value: filter,
    onChange: onChange
  })), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3679907342",
    dynamic: [_uiConstants.spacers.dp8, _uiConstants.spacers.dp8]
  }, [`div.__jsx-style-dynamic-selector{padding-bottom:${_uiConstants.spacers.dp8};}`, `div.__jsx-style-dynamic-selector:first-child{padding-top:${_uiConstants.spacers.dp8};}`]));
};
exports.Filter = Filter;
Filter.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  filter: _propTypes.default.string.isRequired,
  onChange: _propTypes.default.func.isRequired,
  label: _propTypes.default.string,
  placeholder: _propTypes.default.string
};
