// @dhis2-ui/transfer/build/cjs/left-side.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LeftSide = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = require("./common/index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const LeftSide = _ref => {
  let {
    children,
    dataTest,
    width
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["4143353612", [_index.backgroundColor, _index.borderRadius, _index.borderColor, width]]])
  }, children,
  /**
   * Flex basis 0px to make sure right and left side
   * always have the same width
   */
  '', /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4143353612",
    dynamic: [_index.backgroundColor, _index.borderRadius, _index.borderColor, width]
  }, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;background-color:${backgroundColor};border-radius:${borderRadius};border:1px solid ${borderColor};min-height:240px;max-width:100%;width:${width};}`]));
};
exports.LeftSide = LeftSide;
LeftSide.propTypes = {
  width: _propTypes.default.string.isRequired,
  children: _propTypes.default.node,
  dataTest: _propTypes.default.string
};
