// @dhis2-ui/sharing-dialog/build/cjs/sharing-dialog.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Visualization = exports.Dashboard = void 0;
var _appRuntime = require("@dhis2/app-runtime");
var _react = _interopRequireDefault(require("react"));
var _sharingDialog = require("./sharing-dialog.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'Sharing Dialog',
  component: _sharingDialog.SharingDialog
};
const config = {
  baseUrl: 'http://dhis2.test',
  apiVersion: 38
};
const Visualization = () => /*#__PURE__*/_react.default.createElement(_appRuntime.Provider, {
  config: config
}, /*#__PURE__*/_react.default.createElement(_sharingDialog.SharingDialog, {
  type: "visualization",
  id: "id"
}));
exports.Visualization = Visualization;
const Dashboard = () => /*#__PURE__*/_react.default.createElement(_appRuntime.Provider, {
  config: config
}, /*#__PURE__*/_react.default.createElement(_sharingDialog.SharingDialog, {
  type: "dashboard",
  id: "id"
}));
exports.Dashboard = Dashboard;
