// @dhis2-ui/box/build/es/box.js
import _JSXStyle from "styled-jsx/style";
import PropTypes from 'prop-types';
import React from 'react';
export const Box = _ref => {
  let {
    overflow,
    height,
    minHeight,
    maxHeight,
    width,
    minWidth,
    maxWidth,
    marginTop,
    children,
    dataTest = 'dhis2-uicore-box',
    className
  } = _ref;
  return /*#__PURE__*/React.createElement("div", {
    "data-test": dataTest,
    className: _JSXStyle.dynamic([["3369228338", [marginTop ? `margin-top: ${marginTop};` : '', height ? `height: ${height};` : '', minHeight ? `min-height: ${minHeight};` : '', maxHeight ? `max-height: ${maxHeight};` : '', width ? `width: ${width};` : '', minWidth ? `min-width: ${minWidth};` : '', maxWidth ? `max-width: ${maxWidth};` : '', overflow ? `overflow: ${overflow};` : '']]]) + " " + (className || "")
  }, children, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "3369228338",
    dynamic: [marginTop ? `margin-top: ${marginTop};` : '', height ? `height: ${height};` : '', minHeight ? `min-height: ${minHeight};` : '', maxHeight ? `max-height: ${maxHeight};` : '', width ? `width: ${width};` : '', minWidth ? `min-width: ${minWidth};` : '', maxWidth ? `max-width: ${maxWidth};` : '', overflow ? `overflow: ${overflow};` : '']
  }, [`div.__jsx-style-dynamic-selector{${marginTop ? `margin-top: ${marginTop};` : ''} ${height ? `height: ${height};` : ''} ${minHeight ? `min-height: ${minHeight};` : ''} ${maxHeight ? `max-height: ${maxHeight};` : ''} ${width ? `width: ${width};` : ''} ${minWidth ? `min-width: ${minWidth};` : ''} ${maxWidth ? `max-width: ${maxWidth};` : ''} ${overflow ? `overflow: ${overflow};` : ''};}`]));
};
Box.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  height: PropTypes.string,
  marginTop: PropTypes.string,
  maxHeight: PropTypes.string,
  maxWidth: PropTypes.string,
  minHeight: PropTypes.string,
  minWidth: PropTypes.string,
  overflow: PropTypes.string,
  width: PropTypes.string
};
