// @dhis2-ui/transfer/build/cjs/common/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ADD_MODE", {
  enumerable: true,
  get: function () {
    return _modes.ADD_MODE;
  }
});
Object.defineProperty(exports, "RANGE_MODE", {
  enumerable: true,
  get: function () {
    return _modes.RANGE_MODE;
  }
});
Object.defineProperty(exports, "REPLACE_MODE", {
  enumerable: true,
  get: function () {
    return _modes.REPLACE_MODE;
  }
});
Object.defineProperty(exports, "backgroundColor", {
  enumerable: true,
  get: function () {
    return _styles.backgroundColor;
  }
});
Object.defineProperty(exports, "borderColor", {
  enumerable: true,
  get: function () {
    return _styles.borderColor;
  }
});
Object.defineProperty(exports, "borderRadius", {
  enumerable: true,
  get: function () {
    return _styles.borderRadius;
  }
});
Object.defineProperty(exports, "findOptionIndex", {
  enumerable: true,
  get: function () {
    return _findOptionIndex.findOptionIndex;
  }
});
Object.defineProperty(exports, "getModeByModifierKey", {
  enumerable: true,
  get: function () {
    return _getModeByModifierKey.getModeByModifierKey;
  }
});
Object.defineProperty(exports, "isOption", {
  enumerable: true,
  get: function () {
    return _isOption.isOption;
  }
});
Object.defineProperty(exports, "toggleValue", {
  enumerable: true,
  get: function () {
    return _toggleValue.toggleValue;
  }
});
var _modes = require("./modes.js");
var _styles = require("./styles.js");
var _findOptionIndex = require("./find-option-index.js");
var _getModeByModifierKey = require("./get-mode-by-modifier-key.js");
var _isOption = require("./is-option.js");
var _toggleValue = require("./toggle-value.js");
