// @dhis2-ui/select/build/cjs/single-select-field/single-select-field.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithoutOptions = exports.WithValidationText = exports.WithLoading = exports.WithLabelAndRequiredStatus = exports.WithLabel = exports.WithHiddenParent = exports.WithHelpText = exports.WithFilterable = exports.WithClearableAndSelectedOption = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _index2 = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'SingleSelectField'
};
const WithLabel = () => /*#__PURE__*/_react.default.createElement(_index2.SingleSelectField, {
  label: "The label"
}, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "three"
}));
exports.WithLabel = WithLabel;
const WithHelpText = () => /*#__PURE__*/_react.default.createElement(_index2.SingleSelectField, {
  helpText: "The help text"
}, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "three"
}));
exports.WithHelpText = WithHelpText;
const WithValidationText = () => /*#__PURE__*/_react.default.createElement(_index2.SingleSelectField, {
  validationText: "The validation text"
}, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "three"
}));
exports.WithValidationText = WithValidationText;
const WithLabelAndRequiredStatus = () => /*#__PURE__*/_react.default.createElement(_index2.SingleSelectField, {
  label: "The label",
  required: true
}, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "three"
}));
exports.WithLabelAndRequiredStatus = WithLabelAndRequiredStatus;
const WithClearableAndSelectedOption = () => /*#__PURE__*/_react.default.createElement(_index2.SingleSelectField, {
  selected: "1",
  clearable: true
}, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "One"
}));
exports.WithClearableAndSelectedOption = WithClearableAndSelectedOption;
const WithFilterable = () => /*#__PURE__*/_react.default.createElement(_index2.SingleSelectField, {
  filterable: true
}, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "One"
}));
exports.WithFilterable = WithFilterable;
const WithLoading = () => /*#__PURE__*/_react.default.createElement(_index2.SingleSelectField, {
  loading: true
});
exports.WithLoading = WithLoading;
const WithoutOptions = () => /*#__PURE__*/_react.default.createElement(_index2.SingleSelectField, null);
exports.WithoutOptions = WithoutOptions;
const WithHiddenParent = () => {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-3657609608" + " " + "container"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-3657609608" + " " + "hoverSpan"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-3657609608"
  }, "Hover over me"), /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-3657609608" + " " + "hiddenSelect"
  }, /*#__PURE__*/_react.default.createElement(_index2.SingleSelectField, {
    dataTest: "hoverable-select-field",
    label: "Choose something",
    onChange: () => {}
  }, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
    value: "1",
    label: "one"
  }), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
    value: "2",
    label: "two"
  }))))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3657609608"
  }, [".container.jsx-3657609608{width:100%;height:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;font-size:1rem;}", ".hiddenSelect.jsx-3657609608{display:none;}", ".hoverSpan.jsx-3657609608:hover .hiddenSelect.jsx-3657609608{display:block;}"]));
};
exports.WithHiddenParent = WithHiddenParent;
