// @dhis2/multi-calendar-dates/build/cjs/period-calculation/generate-fixed-periods/generate-fixed-periods-daily.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _polyfill = require("@js-temporal/polyfill");
var _index = require("../../utils/index");
var _index2 = require("../daily-periods/index");
const generateFixedPeriodsDaily = _ref => {
  let {
    year,
    calendar,
    endsBefore: _endsBefore,
    locale
  } = _ref;
  const endsBefore = _endsBefore ? (0, _index.fromAnyDate)({
    calendar,
    date: _endsBefore
  }) : null;
  const day = _polyfill.Temporal.PlainDate.from({
    year,
    month: 1,
    day: 1,
    calendar
  });
  const days = [];
  for (let i = 0; i < day.daysInYear; i++) {
    const nextDay = day.add({
      days: i
    });
    if (endsBefore && _polyfill.Temporal.PlainDate.compare(nextDay, endsBefore) > -1) {
      break;
    }
    const period = (0, _index2.buildDailyFixedPeriod)({
      date: nextDay,
      calendar,
      locale: locale
    });
    days.push(period);
  }
  return days;
};
var _default = generateFixedPeriodsDaily;
exports.default = _default;
