// @dhis2/multi-calendar-dates/build/cjs/period-calculation/generate-fixed-periods/generate-fixed-periods-weekly.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _d2I18n = _interopRequireDefault(require("@dhis2/d2-i18n"));
var _polyfill = require("@js-temporal/polyfill");
var _index = require("../../utils/index");
var _doesPeriodEndBefore = _interopRequireDefault(require("./does-period-end-before"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const Days = {
  Monday: 1,
  Tuesday: 2,
  Wednesday: 3,
  Thursday: 4,
  Friday: 5,
  Saturday: 6,
  Sunday: 7
};
// Does not need a `locale` as we're displaying the month as number in the
// displayName/name
const generateFixedPeriodsWeekly = _ref => {
  let {
    year,
    calendar,
    periodType,
    startingDay,
    endsBefore
  } = _ref;
  const startingDayToUse = getStartingDay(periodType, startingDay);
  let date = getStartingDate({
    year,
    calendar,
    periodType,
    startingDay: startingDayToUse
  });
  const days = [];
  let i = 1;
  const daysToAdd = periodType === 'BIWEEKLY' ? 13 : 6;
  do {
    const endofWeek = date.add({
      days: daysToAdd
    });
    if (endsBefore && (0, _doesPeriodEndBefore.default)({
      period: {
        startDate: (0, _index.formatDate)(date),
        endDate: (0, _index.formatDate)(endofWeek)
      },
      date: endsBefore,
      calendar
    })) {
      break;
    }
    const value = buildValue({
      periodType,
      startingDay: startingDayToUse,
      year,
      weekIndex: i
    });
    if (!(endofWeek.year === year + 1 && endofWeek.day >= 4)) {
      const name = buildLabel({
        periodType,
        date,
        nextWeek: endofWeek,
        weekIndex: i
      });
      days.push({
        periodType,
        id: value,
        iso: value,
        name,
        displayName: name,
        startDate: (0, _index.formatDate)(date),
        endDate: (0, _index.formatDate)(endofWeek)
      });
    }
    date = (0, _index.fromAnyDate)({
      date: endofWeek,
      calendar
    }).add({
      days: 1
    });
    i++;
  } while (date.year === year); // important to have the condition after since the very first day can be in the previous year
  return days;
};
const getStartingDay = (periodType, startingDay) => {
  switch (periodType) {
    case 'WEEKLYMON':
      return Days.Monday;
    case 'WEEKLYTUE':
      return Days.Tuesday;
    case 'WEEKLYWED':
      return Days.Wednesday;
    case 'WEEKLYTHU':
      return Days.Thursday;
    case 'WEEKLYFRI':
      return Days.Friday;
    case 'WEEKLYSAT':
      return Days.Saturday;
    case 'WEEKLYSUN':
      return Days.Sunday;
    case 'WEEKLY':
    case 'BIWEEKLY':
    default:
      return startingDay || Days.Monday;
  }
};
const getStartingDate = options => {
  const {
    year,
    calendar,
    startingDay
  } = options;

  // first week in every year has the 4th in the first month
  const fourthOfFirstMonth = _polyfill.Temporal.PlainDate.from({
    year,
    month: 1,
    day: 4,
    calendar
  });
  const dayDiff = fourthOfFirstMonth.dayOfWeek - startingDay;
  if (dayDiff > 0) {
    return fourthOfFirstMonth.subtract({
      days: dayDiff
    });
  } else if (dayDiff < 0) {
    return fourthOfFirstMonth.subtract({
      days: dayDiff + 7
    });
  }
  return fourthOfFirstMonth;
};
const DaysKeys = {
  1: '',
  // Monday is the default and has no key
  2: 'Tue',
  3: 'Wed',
  4: 'Thu',
  5: 'Fri',
  6: 'Sat',
  7: 'Sun'
};
const buildValue = _ref2 => {
  let {
    startingDay,
    periodType,
    year,
    weekIndex
  } = _ref2;
  const periodKey = periodType === 'BIWEEKLY' ? 'BiW' : 'W';
  return `${year}${DaysKeys[startingDay]}${periodKey}${weekIndex}`;
};
const buildLabel = _ref3 => {
  let {
    periodType,
    date,
    nextWeek,
    weekIndex
  } = _ref3;
  const {
    year,
    month,
    day
  } = date;
  const {
    year: nextYear,
    month: nextMonth,
    day: nextDay
  } = nextWeek;
  const prefix = periodType === 'BIWEEKLY' ? _d2I18n.default.t('Bi-Week') : _d2I18n.default.t('Week');
  const label = `${prefix} ${weekIndex} - ${year}-${(0, _index.padWithZeroes)(month)}-${(0, _index.padWithZeroes)(day)} - ${nextYear}-${(0, _index.padWithZeroes)(nextMonth)}-${(0, _index.padWithZeroes)(nextDay)}`;
  return label;
};
var _default = generateFixedPeriodsWeekly;
exports.default = _default;
