// @dhis2-ui/user-avatar/build/cjs/image-avatar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useAvatarImgSrc = exports.ImageAvatar = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _appRuntime = require("@dhis2/app-runtime");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const useAvatarImgSrc = avatarId => {
  const {
    baseUrl
  } = (0, _appRuntime.useConfig)();
  return avatarId ? [baseUrl, 'api/fileResources', avatarId, 'data'].filter(part => !!part).map(part => part.replace(/^\/+|\/+$/g, '')).join('/') : null;
};
exports.useAvatarImgSrc = useAvatarImgSrc;
const ImageAvatar = _ref => {
  let {
    avatarId,
    dataTest
  } = _ref;
  const src = useAvatarImgSrc(avatarId);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("img", {
    src: src,
    alt: "user avatar",
    "data-test": dataTest,
    className: "jsx-3329187909"
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3329187909"
  }, ["img.jsx-3329187909{width:100%;height:100%;border-radius:50%;object-fit:cover;}"]));
};
exports.ImageAvatar = ImageAvatar;
ImageAvatar.propTypes = {
  avatarId: _propTypes.default.string.isRequired,
  dataTest: _propTypes.default.string
};
