// @dhis2-ui/user-avatar/build/cjs/text-avatar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getInitials = exports.TextAvatar = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const getInitials = name => {
  const nameParts = name.split(' ');
  let initials = nameParts.shift().charAt(0);
  if (nameParts.length) {
    initials += nameParts.pop().charAt(0);
  }
  return initials;
};
exports.getInitials = getInitials;
const TextAvatar = _ref => {
  let {
    name,
    dataTest = 'dhis2-uicore-textavatar',
    extralarge,
    extrasmall,
    large,
    medium,
    small
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    "aria-label": name,
    className: _style.default.dynamic([["1018031843", [_uiConstants.colors.grey800, _uiConstants.colors.grey050]]]) + " " + "text-avatar"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["1018031843", [_uiConstants.colors.grey800, _uiConstants.colors.grey050]]]) + " " + ((0, _classnames.default)('text-avatar-initials', {
      extrasmall,
      small,
      medium,
      large,
      extralarge
    }) || "")
  }, getInitials(name)), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1018031843",
    dynamic: [_uiConstants.colors.grey800, _uiConstants.colors.grey050]
  }, [`.text-avatar.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;width:100%;height:100%;overflow:hidden;border-radius:50%;background-color:${_uiConstants.colors.grey800}cc;color:${_uiConstants.colors.grey050};cursor:pointer;}`, ".text-avatar-initials.__jsx-style-dynamic-selector{font-size:80%;font-weight:500;-webkit-letter-spacing:1px;-moz-letter-spacing:1px;-ms-letter-spacing:1px;letter-spacing:1px;text-align:center;text-transform:uppercase;}", ".text-avatar-initials.extrasmall.__jsx-style-dynamic-selector{font-size:7px;}", ".text-avatar-initials.small.__jsx-style-dynamic-selector{font-size:9px;}", ".text-avatar-initials.medium.__jsx-style-dynamic-selector{font-size:14px;}", ".text-avatar-initials.large.__jsx-style-dynamic-selector{font-size:22px;}", ".text-avatar-initials.extralarge.__jsx-style-dynamic-selector{font-size:32px;}"]));
};
exports.TextAvatar = TextAvatar;
TextAvatar.propTypes = {
  name: _propTypes.default.string.isRequired,
  dataTest: _propTypes.default.string,
  extralarge: _uiConstants.sharedPropTypes.sizePropType,
  extrasmall: _uiConstants.sharedPropTypes.sizePropType,
  large: _uiConstants.sharedPropTypes.sizePropType,
  medium: _uiConstants.sharedPropTypes.sizePropType,
  small: _uiConstants.sharedPropTypes.sizePropType
};
