// @dhis2-ui/user-avatar/build/cjs/user-avatar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UserAvatar = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _imageAvatar = require("./image-avatar.js");
var _textAvatar = require("./text-avatar.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/**
An avatar is a visual icon that represents a user.

Use an avatar to give extra information when a user is mentioned or displayed in DHIS2. The avatar shows a user uploaded photograph or initials. The avatar is intended to give context and help to identify different users. An avatar is usually shown alongside the user name, but can be used alone to show a visual hint of a user.

See specification: [Design System](https://github.com/dhis2/design-system/blob/master/atoms/avatar.md)

```js
import { UserAvatar } from '@dhis2/ui'
```
*/

const UserAvatar = _ref => {
  let {
    name,
    avatarId,
    className,
    dataTest = 'dhis2-uicore-useravatar',
    extralarge,
    extrasmall,
    large,
    medium,
    small,
    ariaLabel
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    "aria-label": ariaLabel || name,
    "aria-hidden": !name,
    className: "jsx-3366500546" + " " + ((0, _classnames.default)(className, {
      extrasmall,
      small,
      medium,
      large,
      extralarge
    }) || "")
  }, avatarId ? /*#__PURE__*/_react.default.createElement(_imageAvatar.ImageAvatar, {
    avatarId: avatarId,
    dataTest: `${dataTest}-image`
  }) : /*#__PURE__*/_react.default.createElement(_textAvatar.TextAvatar, {
    name: name,
    dataTest: `${dataTest}-text`,
    extrasmall: extrasmall,
    small: small,
    medium: medium,
    large: large,
    extralarge: extralarge
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3366500546"
  }, ["div.jsx-3366500546{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}", "div.extrasmall.jsx-3366500546{height:16px;width:16px;}", "div.small.jsx-3366500546{height:24px;width:24px;}", "div.jsx-3366500546,div.medium.jsx-3366500546{height:36px;width:36px;}", "div.large.jsx-3366500546{height:48px;width:48px;}", "div.extralarge.jsx-3366500546{height:72px;width:72px;}"]));
};
exports.UserAvatar = UserAvatar;
UserAvatar.propTypes = {
  // Name could stop being required if we implement an
  // SVG fallback with a `IconUser24`.
  // This has been discussed and deferred
  name: _propTypes.default.string.isRequired,
  ariaLabel: _propTypes.default.string,
  avatarId: _propTypes.default.string,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  extralarge: _uiConstants.sharedPropTypes.sizePropType,
  extrasmall: _uiConstants.sharedPropTypes.sizePropType,
  large: _uiConstants.sharedPropTypes.sizePropType,
  medium: _uiConstants.sharedPropTypes.sizePropType,
  small: _uiConstants.sharedPropTypes.sizePropType
};
