// @dhis2-ui/file-input/build/es/file-input-field-with-list/file-input-field-with-list.e2e.stories.js
import React from 'react';
import { FileInputFieldWithList } from './index.js';
window.onChange = window.Cypress && window.Cypress.cy.stub();
const singleFileArray = [new File([], 'dummy.txt')];
window.duplicateFileConstructorArgs = [[], 'duplicate.md', {
  lastModified: 1581348175357,
  type: 'text/markdown'
}];
const multipleFileArray = [new File([], 'test1.md'), new File(...window.duplicateFileConstructorArgs), new File([], 'test2.md')];
const onChange = (payload, event) => {
  // NOTE: if files is not transformed into an array,
  // cypress will get an empty file list!
  window.onChange({
    ...payload,
    files: [...payload.files]
  }, event);
};
export default {
  title: 'FileInputFieldWithList'
};
export const SingleFileWithFile = () => /*#__PURE__*/React.createElement(FileInputFieldWithList, {
  buttonLabel: "Upload file",
  name: "upload",
  files: singleFileArray,
  removeText: "remove",
  onChange: onChange
});
export const MultipleFilesEmpty = () => /*#__PURE__*/React.createElement(FileInputFieldWithList, {
  buttonLabel: "Upload file",
  multiple: true,
  name: "upload",
  removeText: "remove",
  onChange: onChange
});
export const MultipleFilesWithFiles = () => /*#__PURE__*/React.createElement(FileInputFieldWithList, {
  buttonLabel: "Upload file",
  multiple: true,
  name: "upload",
  files: multipleFileArray,
  removeText: "remove",
  onChange: onChange
});
export const WithFileAndDefaultTexts = () => /*#__PURE__*/React.createElement(FileInputFieldWithList, {
  name: "upload",
  files: singleFileArray,
  onChange: onChange
});
export const WithDefaultTexts = () => /*#__PURE__*/React.createElement(FileInputFieldWithList, {
  name: "upload",
  onChange: onChange
});
