// @dhis2-ui/modal/build/cjs/modal-title/modal-title.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalTitle = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const ModalTitle = _ref => {
  let {
    children,
    dataTest = 'dhis2-uicore-modaltitle'
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("h1", {
    "data-test": dataTest,
    className: _style.default.dynamic([["4169842822", [_uiConstants.colors.grey900, _uiConstants.spacers.dp12]]]) + " " + ((0, _classnames.default)('title') || "")
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4169842822",
    dynamic: [_uiConstants.colors.grey900, _uiConstants.spacers.dp12]
  }, [`h1.__jsx-style-dynamic-selector{-webkit-order:1;-ms-flex-order:1;order:1;-webkit-align-self:flex-start;-ms-flex-item-align:start;align-self:flex-start;color:${_uiConstants.colors.grey900};font-size:20px;font-weight:500;line-height:24px;margin:0 0 ${_uiConstants.spacers.dp12};}`]));
};
exports.ModalTitle = ModalTitle;
ModalTitle.propTypes = {
  children: _propTypes.default.node,
  dataTest: _propTypes.default.string
};
