// @dhis2/ui-constants/build/cjs/theme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.theme = void 0;
var _colors = require("./colors.js");
const theme = exports.theme = {
  /* theme */
  fonts: 'Roboto, sans-serif',
  /*primary*/
  primary900: _colors.colors.blue900,
  primary800: _colors.colors.blue800,
  primary700: _colors.colors.blue700,
  primary600: _colors.colors.blue600,
  primary500: _colors.colors.blue500,
  primary400: _colors.colors.blue400,
  primary300: _colors.colors.blue300,
  primary200: _colors.colors.blue200,
  primary100: _colors.colors.blue100,
  primary050: _colors.colors.blue050,
  /*secondary*/
  secondary900: _colors.colors.teal900,
  secondary800: _colors.colors.teal800,
  secondary700: _colors.colors.teal700,
  secondary600: _colors.colors.teal600,
  secondary500: _colors.colors.teal500,
  secondary400: _colors.colors.teal400,
  secondary300: _colors.colors.teal300,
  secondary200: _colors.colors.teal200,
  secondary100: _colors.colors.teal100,
  secondary050: _colors.colors.teal050,
  /*status*/
  default: _colors.colors.grey700,
  error: _colors.colors.red500,
  valid: _colors.colors.blue600,
  warning: _colors.colors.yellow500,
  disabled: _colors.colors.grey600,
  focus: _colors.colors.blue600
};
