// @dhis2-ui/popper/build/cjs/popper.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.VirtualElementAsReference = exports.TopStart = exports.TopEnd = exports.Top = exports.RightStart = exports.RightEnd = exports.Right = exports.ReactRefAsReference = exports.LeftStart = exports.LeftEnd = exports.Left = exports.DOMNodeAsReference = exports.BottomStart = exports.BottomEnd = exports.Bottom = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _popper = require("./popper.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const PopperPlacement = _ref => {
  let {
    placement
  } = _ref;
  const ref = (0, _react.useRef)();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-305551169" + " " + "box"
  }, /*#__PURE__*/_react.default.createElement("div", {
    ref: ref,
    className: "jsx-305551169" + " " + "reference-element"
  }, "Reference element"), /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    reference: ref,
    placement: placement
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-305551169" + " " + "popper-content"
  }, "Popper")), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "305551169"
  }, [".box.jsx-305551169{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;width:400px;height:400px;margin-bottom:1000px;background-color:aliceblue;}", ".reference-element.jsx-305551169{width:100px;height:50px;background-color:cadetblue;text-align:center;padding:6px;}", ".popper-content.jsx-305551169{width:80px;height:30px;background-color:lightblue;text-align:center;padding:6px;}"]));
};
PopperPlacement.propTypes = {
  placement: _propTypes.default.string
};
var _default = exports.default = {
  title: 'Popper',
  component: _popper.Popper
};
const Top = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "top"
});
exports.Top = Top;
const TopStart = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "top-start"
});
exports.TopStart = TopStart;
const TopEnd = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "top-end"
});
exports.TopEnd = TopEnd;
const Bottom = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "bottom"
});
exports.Bottom = Bottom;
const BottomStart = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "bottom-start"
});
exports.BottomStart = BottomStart;
const BottomEnd = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "bottom-end"
});
exports.BottomEnd = BottomEnd;
const Right = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "right"
});
exports.Right = Right;
const RightStart = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "right-start"
});
exports.RightStart = RightStart;
const RightEnd = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "right-end"
});
exports.RightEnd = RightEnd;
const Left = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "left"
});
exports.Left = Left;
const LeftStart = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "left-start"
});
exports.LeftStart = LeftStart;
const LeftEnd = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "left-end"
});
exports.LeftEnd = LeftEnd;
const ReactRefAsReference = () => {
  const ref = (0, _react.useRef)();
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
    ref: ref
  }, "Reference"), /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    placement: "bottom-start",
    reference: ref
  }, "Popper"));
};
exports.ReactRefAsReference = ReactRefAsReference;
const DOMNodeAsReference = () => {
  const [node, setNode] = (0, _react.useState)(null);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
    ref: setNode
  }, "Reference"), /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    placement: "bottom-start",
    reference: node
  }, "Popper"));
};
exports.DOMNodeAsReference = DOMNodeAsReference;
const VirtualElementAsReference = () => {
  const virtualElement = {
    getBoundingClientRect: () => ({
      top: 200,
      left: 200,
      bottom: 'auto',
      right: 'auto',
      width: 0,
      height: 0
    })
  };
  return /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    placement: "bottom-start",
    reference: virtualElement
  }, "Popper");
};
exports.VirtualElementAsReference = VirtualElementAsReference;
