// @dhis2-ui/popper/build/cjs/popper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Popper = void 0;
var _uiConstants = require("@dhis2/ui-constants");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _reactPopper = require("react-popper");
var _getReferenceElement = require("./get-reference-element.js");
var _modifiers = require("./modifiers.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const flipPlacement = placement => {
  if (placement.startsWith('right')) {
    return placement.replace('right', 'left');
  }
  if (placement.startsWith('left')) {
    return placement.replace('left', 'right');
  }
  return placement;
};

// Stable object reference
const staticArray = [];
const Popper = _ref => {
  let {
    children,
    className,
    dataTest = 'dhis2-uicore-popper',
    modifiers = staticArray,
    observePopperResize,
    observeReferenceResize,
    onFirstUpdate,
    placement = 'auto',
    reference,
    strategy
  } = _ref;
  const referenceElement = (0, _getReferenceElement.getReferenceElement)(reference);
  const [popperElement, setPopperElement] = (0, _react.useState)(null);
  const deduplicatedModifiers = (0, _react.useMemo)(() => (0, _modifiers.deduplicateModifiers)(modifiers, {
    observePopperResize,
    observeReferenceResize
  }), [modifiers, observePopperResize, observeReferenceResize]);
  const {
    styles,
    attributes
  } = (0, _reactPopper.usePopper)(referenceElement, popperElement, {
    strategy,
    onFirstUpdate,
    placement: document.documentElement.dir === 'rtl' ? flipPlacement(placement) : placement,
    modifiers: deduplicatedModifiers
  });
  (0, _react.useEffect)(() => {
    if (popperElement) {
      var _popperElement$firstE;
      popperElement === null || popperElement === void 0 ? void 0 : (_popperElement$firstE = popperElement.firstElementChild) === null || _popperElement$firstE === void 0 ? void 0 : _popperElement$firstE.focus();
    }
  }, [popperElement]);
  return /*#__PURE__*/_react.default.createElement("div", _extends({
    className: className,
    "data-test": dataTest,
    ref: setPopperElement,
    style: styles.popper
  }, attributes.popper, {
    tabIndex: 0
  }), children);
};

// Prop names follow the names here: https://popper.js.org/docs/v2/constructors/
exports.Popper = Popper;
Popper.propTypes = {
  /** Content inside the Popper */
  children: _propTypes.default.node.isRequired,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  /** A property of the `createPopper` options. See [popper docs](https://popper.js.org/docs/v2/constructors/) */
  modifiers: _propTypes.default.arrayOf(_propTypes.default.shape({
    name: _propTypes.default.string,
    options: _propTypes.default.object
  })),
  /** Makes the Popper update position when the **Popper content** changes size */
  observePopperResize: _propTypes.default.bool,
  /** Makes the Popper update position when the **reference element** changes size */
  observeReferenceResize: _propTypes.default.bool,
  /** A property of the `createPopper` options. See [popper docs](https://popper.js.org/docs/v2/constructors/) */
  placement: _uiConstants.sharedPropTypes.popperPlacementPropType,
  /** A React ref, DOM node, or [virtual element](https://popper.js.org/docs/v2/virtual-elements/) for the popper to position itself against */
  reference: _uiConstants.sharedPropTypes.popperReferencePropType,
  /** A property of the `createPopper` options. See [popper docs](https://popper.js.org/docs/v2/constructors/) */
  strategy: _propTypes.default.oneOf(['absolute', 'fixed']),
  // defaults to 'absolute'
  /** A property of the `createPopper` options. See [popper docs](https://popper.js.org/docs/v2/constructors/) */
  onFirstUpdate: _propTypes.default.func
};
