// @dhis2-ui/calendar/build/cjs/calendar-input/calendar-input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CalendarInput = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _button = require("@dhis2-ui/button");
var _card = require("@dhis2-ui/card");
var _input = require("@dhis2-ui/input");
var _layer = require("@dhis2-ui/layer");
var _popper = require("@dhis2-ui/popper");
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireWildcard(require("react"));
var _calendar = require("../calendar/calendar.js");
var _index = _interopRequireDefault(require("../locales/index.js"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const offsetModifier = {
  name: 'offset',
  options: {
    offset: [0, 2]
  }
};
const CalendarInput = function () {
  let {
    onDateSelect,
    calendar,
    date,
    dir,
    locale,
    numberingSystem,
    weekDayFormat,
    timeZone,
    width,
    cellSize,
    clearable,
    dataTest = 'dhis2-uiwidgets-calendar-inputfield',
    ...rest
  } = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  const ref = (0, _react.useRef)();
  const [open, setOpen] = (0, _react.useState)(false);
  const calendarProps = _react.default.useMemo(() => {
    const onDateSelectWrapper = selectedDate => {
      setOpen(false);
      onDateSelect === null || onDateSelect === void 0 ? void 0 : onDateSelect(selectedDate);
    };
    return {
      onDateSelect: onDateSelectWrapper,
      calendar,
      date,
      dir,
      locale,
      numberingSystem,
      weekDayFormat,
      timeZone,
      width,
      cellSize
    };
  }, [calendar, cellSize, date, dir, locale, numberingSystem, onDateSelect, timeZone, weekDayFormat, width]);
  const onFocus = () => {
    setOpen(true);
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    ref: ref,
    className: "jsx-862452676" + " " + "calendar-input-wrapper"
  }, /*#__PURE__*/_react.default.createElement(_input.InputField, _extends({
    label: _index.default.t('Pick a date')
  }, rest, {
    dataTest: dataTest,
    type: "text",
    onFocus: onFocus,
    value: date
  })), clearable && /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-862452676" + " " + ((0, _classnames.default)('calendar-clear-button', {
      // ToDo: this is a workaround to show the clear button in the correct place when an icon is shown.
      // Long-term, we should abstract and share the logic multi-select uses for the input-wrapper
      // https://dhis2.atlassian.net/browse/DHIS2-14848
      'with-icon': rest.valid || rest.error || rest.warning || rest.loading,
      'with-dense-wrapper': rest.dense
    }) || "")
  }, /*#__PURE__*/_react.default.createElement(_button.Button, {
    dataTest: "calendar-clear-button",
    secondary: true,
    small: true,
    onClick: () => calendarProps.onDateSelect(null),
    type: "button"
  }, _index.default.t('Clear')))), open && /*#__PURE__*/_react.default.createElement(_layer.Layer, {
    onBackdropClick: () => {
      setOpen(false);
    }
  }, /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    reference: ref,
    placement: "bottom-start",
    modifiers: [offsetModifier]
  }, /*#__PURE__*/_react.default.createElement(_card.Card, null, /*#__PURE__*/_react.default.createElement(_calendar.Calendar, _extends({}, calendarProps, {
    date: date
  }))))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "862452676"
  }, [".calendar-input-wrapper.jsx-862452676{position:relative;}", ".calendar-clear-button.jsx-862452676{position:absolute;inset-inline-end:6px;-webkit-inset-block-start:27px;-ms-intb-rlock-start:27px;inset-block-start:27px;}", ".calendar-clear-button.with-icon.jsx-862452676{inset-inline-end:36px;}", ".calendar-clear-button.with-dense-wrapper.jsx-862452676{-webkit-inset-block-start:23px;-ms-intb-rlock-start:23px;inset-block-start:23px;}"]));
};
exports.CalendarInput = CalendarInput;
CalendarInput.propTypes = {
  ..._calendar.CalendarProps,
  ..._input.InputFieldProps
};
