// @dhis2-ui/button/build/es/dropdown-button/dropdown-button.e2e.stories.js
import React from 'react';
import { DropdownButton } from './index.js';
window.onClick = window.Cypress && window.Cypress.cy.stub();
export default {
  title: 'DropdownButton'
};
export const Default = () => /*#__PURE__*/React.createElement(DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Content")
}, "Button");
export const WithOnClick = () => /*#__PURE__*/React.createElement(DropdownButton, {
  name: "Button",
  value: "default",
  onClick: window.onClick,
  component: /*#__PURE__*/React.createElement("div", null, "Content")
}, "Button");
export const WithChildren = () => /*#__PURE__*/React.createElement(DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Component")
}, "I am a child");
export const WithComponent = () => /*#__PURE__*/React.createElement(DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "I am a component")
});
export const WithIcon = () => /*#__PURE__*/React.createElement(DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "I am a component"),
  icon: 'Icon'
});
export const WithInitialFocus = () => /*#__PURE__*/React.createElement(DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Content"),
  initialFocus: true
});
export const DisabledWithOnClick = () => /*#__PURE__*/React.createElement(DropdownButton, {
  name: "Button",
  value: "default",
  component: /*#__PURE__*/React.createElement("div", null, "Content"),
  onClick: window.onClick,
  disabled: true
});
