// @dhis2-ui/tooltip/build/cjs/tooltip.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RTLLeft = exports.PlacementRight = exports.PlacementLeft = exports.PlacementBottom = exports.HidesWhenOutOfFrame = exports.DefaultPlacementTop = exports.CustomElementViaTagProp = exports.CustomComponent = exports.CustomBuiltInComponent = exports.ConfigurableOpenAndCloseDelays = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _react = _interopRequireWildcard(require("react"));
var _tooltip = require("./tooltip.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const subtitle = 'Displayed when a user hovers over the element';
const description = `
Tooltips only display when a user hovers over an element. Do not use tooltips for critical or important information, users may not find the information or it may completely unavailable to mobile users. Information provided in tooltips should be supplementary or provide helpful context. Icon buttons can use tooltips to inform the user of their action. Never put important information or actions inside a tooltip.

- Common usage of a tooltip is to expand on the displayed information when the user hovers over the element.
- Do not place actions inside a tooltip, they would be hidden from the user and difficult to click.
- Only text can be displayed in a tooltip. A popover can be used for rich information.
- Limit the text inside a tooltip to a single, short sentence.
- Do not repeat information in a tooltip, provide extra relevant, useful information.

By default the tooltip should display above the hovered element. Alternatively, a tooltip may be displayed underneath or to the side of an element if there is limited space.

\`\`\`js
import { Tooltip } from '@dhis2/ui'
\`\`\`

_**Note**: The tooltips may not be placed correctly on this page. View the demos in the 'Canvas' tab for correct placement._
`;
var _default = exports.default = {
  title: 'Tooltip',
  component: _tooltip.Tooltip,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      },
      source: {
        type: 'code'
      }
    }
  },
  decorators: [story => /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: 500,
      height: 200
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 400,
      height: 100
    }
  }, story()))],
  args: {
    content: 'Some tooltip content here'
  }
};
const Template = args => /*#__PURE__*/_react.default.createElement("p", null, "Mouse over ", /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, args, "[these words]"), " to open a tooltip.");
const DefaultPlacementTop = exports.DefaultPlacementTop = Template.bind({});
const PlacementRight = exports.PlacementRight = Template.bind({});
PlacementRight.args = {
  placement: 'right'
};
PlacementRight.parameters = {
  docs: {
    description: {
      story: `_**Note:** The tooltips may not be placed correctly on this page. View the demos in the 'Canvas' tab for correct placement._`
    }
  }
};
const PlacementBottom = exports.PlacementBottom = Template.bind({});
PlacementBottom.args = {
  placement: 'bottom'
};
const PlacementLeft = exports.PlacementLeft = Template.bind({});
PlacementLeft.args = {
  placement: 'left'
};
const ConfigurableOpenAndCloseDelays = args => /*#__PURE__*/_react.default.createElement("p", null, /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, args, "The tooltip that opens when this content is moused over opens with 0 open or close delay"));
exports.ConfigurableOpenAndCloseDelays = ConfigurableOpenAndCloseDelays;
ConfigurableOpenAndCloseDelays.args = {
  content: 'Some extra info',
  openDelay: 0,
  closeDelay: 0
};
const CustomElementViaTagProp = () => {
  return /*#__PURE__*/_react.default.createElement("p", null, "I am a", ' ', /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
    content: "Some extra info",
    tag: "em"
  }, "paragraph"), ' ', "that contains a tooltip.");
};
exports.CustomElementViaTagProp = CustomElementViaTagProp;
const CustomBuiltInComponent = args => {
  return /*#__PURE__*/_react.default.createElement("p", null, "Mouse over", ' ', /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, args, _ref => {
    let {
      ref,
      onMouseOver,
      onMouseOut
    } = _ref;
    return /*#__PURE__*/_react.default.createElement("span", {
      ref: ref,
      onMouseOver: onMouseOver,
      onMouseOut: onMouseOut,
      className: "jsx-1344832130"
    }, "these words", /*#__PURE__*/_react.default.createElement(_style.default, {
      id: "1344832130"
    }, ["span.jsx-1344832130{color:green;-webkit-text-decoration:underline;text-decoration:underline;}"]));
  }), ' ', "to open a tooltip.");
};
exports.CustomBuiltInComponent = CustomBuiltInComponent;
const CustomComponent = args => {
  return /*#__PURE__*/_react.default.createElement("p", null, "Mouse over", ' ', /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, args, _ref2 => {
    let {
      ref,
      onMouseOver,
      onMouseOut
    } = _ref2;
    return /*#__PURE__*/_react.default.createElement("span", {
      ref: ref,
      onMouseOver: onMouseOver,
      onMouseOut: onMouseOut,
      className: "jsx-588712602"
    }, /*#__PURE__*/_react.default.createElement("button", {
      className: "jsx-588712602"
    }, "this button"), /*#__PURE__*/_react.default.createElement(_style.default, {
      id: "588712602"
    }, ["span.jsx-588712602{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}"]));
  }), ' ', "to open a tooltip.");
};
exports.CustomComponent = CustomComponent;
const HidesWhenOutOfFrame = args => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    height: '250px',
    padding: '1rem',
    overflow: 'scroll',
    border: '1px solid #43cbcb'
  },
  "data-test": "hiding-and-flipping-container"
}, /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, args, /*#__PURE__*/_react.default.createElement("div", {
  style: {
    height: '80px',
    border: '1px solid grey'
  }
}, "I am a rectangle that contains a tooltip that hides when I'm obscured from view")), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"), /*#__PURE__*/_react.default.createElement("p", null, "I'm an extra paragraph"));
exports.HidesWhenOutOfFrame = HidesWhenOutOfFrame;
const RTLLeft = args => {
  (0, _react.useEffect)(() => {
    document.documentElement.setAttribute('dir', 'rtl');
    return () => {
      document.documentElement.setAttribute('dir', 'ltr');
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement("div", {
    dir: "rtl"
  }, /*#__PURE__*/_react.default.createElement(Template, args));
};
exports.RTLLeft = RTLLeft;
RTLLeft.args = {
  placement: 'left'
};
