// @dhis2-ui/menu/build/cjs/menu/menu.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SideBarMenuRTL = exports.SideBarMenu = exports.Dense = exports.Default = exports.AutoHideFirstSectionHeaderDivider = void 0;
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _index2 = require("./index.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
Use menus to provide access to options and actions where space is limited and displaying all the options would be impractical. For example, providing access to a range of actions for every dashboard item displayed. Containing all those actions in menus keeps the page manageable.

The menu component is flexible in where it can be used and its contents can be flexible too. However, the most common use case is a menu containing menu items.

Make sure the menu item labels are short and easy to understand. One word is often enough to describe an action or option. Do not use sentences as labels. Some examples of good menu item labels:

- "Save"
- "Open as map"
- "Export PDF"
- "Duplicate"

Typical children are Menu Items, Menu Dividers, and Menu Section Headers.

\`\`\`js
import { Menu } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Menu',
  component: _index2.Menu,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Default = args => /*#__PURE__*/_react.default.createElement(_index2.Menu, args, /*#__PURE__*/_react.default.createElement(_index.MenuItem, {
  label: "Menu item"
}), /*#__PURE__*/_react.default.createElement(_index.MenuItem, {
  label: "Menu item"
}));
exports.Default = Default;
const Dense = args => /*#__PURE__*/_react.default.createElement(_index2.Menu, args, /*#__PURE__*/_react.default.createElement(_index.MenuItem, {
  label: "Menu item"
}), /*#__PURE__*/_react.default.createElement(_index.MenuItem, {
  label: "Menu item"
}));
exports.Dense = Dense;
Dense.args = {
  dense: true
};
Dense.parameters = {
  docs: {
    description: {
      story: 'Menus are available in regular or dense sizes. Use dense menus in data-heavy applications used by users comfortable with technology. Use regular menus in apps that are less complex or have few controls.'
    }
  }
};
const AutoHideFirstSectionHeaderDivider = args => /*#__PURE__*/_react.default.createElement(_index2.Menu, args, /*#__PURE__*/_react.default.createElement(_index.MenuSectionHeader, {
  label: "First - no divider above"
}), /*#__PURE__*/_react.default.createElement(_index.MenuSectionHeader, {
  label: "Second - with divider above"
}));
exports.AutoHideFirstSectionHeaderDivider = AutoHideFirstSectionHeaderDivider;
const SideBarMenu = args => /*#__PURE__*/_react.default.createElement("main", {
  style: {
    display: 'flex',
    height: '100%',
    border: '1px solid grey'
  }
}, /*#__PURE__*/_react.default.createElement("aside", {
  style: {
    width: 200,
    height: '100%',
    flexGrow: 0
  }
}, /*#__PURE__*/_react.default.createElement(_index2.Menu, args, /*#__PURE__*/_react.default.createElement(_index.MenuItem, {
  label: "Menu item"
}), /*#__PURE__*/_react.default.createElement(_index.MenuItem, {
  label: "Menu item"
}))), /*#__PURE__*/_react.default.createElement("section", {
  style: {
    backgroundColor: '#f3ffff',
    flexGrow: 1,
    padding: 20,
    borderInlineStart: '1px solid grey'
  }
}, "Main content"));
exports.SideBarMenu = SideBarMenu;
const SideBarMenuRTL = args => /*#__PURE__*/_react.default.createElement("div", {
  dir: "rtl",
  style: {
    height: '100%'
  }
}, /*#__PURE__*/_react.default.createElement(SideBarMenu, args));
exports.SideBarMenuRTL = SideBarMenuRTL;
