// @dhis2-ui/field/build/cjs/field-set/field-set.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.UsageExampleARadioButtonGroupWithErrorStatus = exports.Default = void 0;
var _help = require("@dhis2-ui/help");
var _legend = require("@dhis2-ui/legend");
var _radio = require("@dhis2-ui/radio");
var _react = _interopRequireDefault(require("react"));
var _index = require("../index.js");
var _fieldSet = require("./field-set.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const description = `
A container for grouping several Field components together. Use a \`<Legend>\` component to add helpful guiding text.

\`\`\`js
import { FieldSet } from '@dhis2/ui'
\`\`\`
`;
const onChange = () => {
  console.log('Radio was clicked, nothing else will happen');
};
var _default = exports.default = {
  title: 'Field Set',
  component: _fieldSet.FieldSet,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Default = args => /*#__PURE__*/_react.default.createElement(_fieldSet.FieldSet, args, "It renders something in a fieldset element without the browser styles");
exports.Default = Default;
const UsageExampleARadioButtonGroupWithErrorStatus = args => /*#__PURE__*/_react.default.createElement(_fieldSet.FieldSet, args, /*#__PURE__*/_react.default.createElement(_legend.Legend, {
  required: true
}, "Choose an option"), /*#__PURE__*/_react.default.createElement(_index.Field, null, /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  onChange: onChange,
  name: "radio",
  value: "first",
  label: "First",
  error: true
})), /*#__PURE__*/_react.default.createElement(_index.Field, null, /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  onChange: onChange,
  name: "radio",
  value: "second",
  label: "Second",
  error: true
})), /*#__PURE__*/_react.default.createElement(_index.Field, null, /*#__PURE__*/_react.default.createElement(_radio.Radio, {
  onChange: onChange,
  name: "radio",
  value: "third",
  label: "Third",
  error: true
})), /*#__PURE__*/_react.default.createElement(_help.Help, {
  error: true
}, "You really have to choose something!"));
exports.UsageExampleARadioButtonGroupWithErrorStatus = UsageExampleARadioButtonGroupWithErrorStatus;
UsageExampleARadioButtonGroupWithErrorStatus.storyName = 'Usage example - a radio button group with error status';
