// @dhis2-ui/selector-bar/build/cjs/selector-bar-item/selector-bar-item.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.OpenWithValue = exports.OpenNoValue = exports.ClosedWithValue = exports.ClosedNoValue = void 0;
var _select = require("@dhis2-ui/select");
var _react = _interopRequireWildcard(require("react"));
var _selectorBarItem = require("./selector-bar-item.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const options = [{
  value: 'value 1',
  label: 'label 1'
}, {
  value: 'value 2',
  label: 'label 2'
}, {
  value: 'value 3',
  label: 'label 3'
}];
const createStateDecorator = args => {
  return fn => {
    const [selected, setSelected] = (0, _react.useState)((args === null || args === void 0 ? void 0 : args.selected) || '');
    const [open, setOpen] = (0, _react.useState)((args === null || args === void 0 ? void 0 : args.open) || false);
    return fn({
      selected,
      open,
      setSelected,
      setOpen
    });
  };
};
const createStory = () => (_, _ref) => {
  var _options$find;
  let {
    open,
    setOpen,
    selected,
    setSelected
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_selectorBarItem.SelectorBarItem, {
    label: "Selection bar item",
    noValueMessage: "No value message",
    open: open,
    setOpen: setOpen,
    value: ((_options$find = options.find(_ref2 => {
      let {
        value
      } = _ref2;
      return value === selected;
    })) === null || _options$find === void 0 ? void 0 : _options$find.label) || ''
  }, /*#__PURE__*/_react.default.createElement(_select.SingleSelectField, {
    selected: selected,
    onChange: _ref3 => {
      let {
        selected: nextSelected
      } = _ref3;
      setSelected(nextSelected);
      setOpen(false);
    }
  }, options.map(_ref4 => {
    let {
      value,
      label
    } = _ref4;
    return /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
      key: value,
      value: value,
      label: label
    });
  })));
};
var _default = exports.default = {
  title: 'SelectorBarItem'
};
const ClosedNoValue = exports.ClosedNoValue = createStory();
ClosedNoValue.decorators = [createStateDecorator()];
const OpenNoValue = exports.OpenNoValue = createStory();
OpenNoValue.decorators = [createStateDecorator({
  open: true
})];
const ClosedWithValue = exports.ClosedWithValue = createStory();
ClosedWithValue.decorators = [createStateDecorator({
  selected: 'value 1'
})];
const OpenWithValue = exports.OpenWithValue = createStory();
OpenWithValue.decorators = [createStateDecorator({
  open: true,
  selected: 'value 1'
})];
