// @dhis2-ui/menu/build/es/menu-section-header/menu-section-header.prod.stories.js
import PropTypes from 'prop-types';
import React from 'react';
import { Menu, MenuItem } from '../index.js';
import { MenuSectionHeader } from './menu-section-header.js';
const description = `
Items in a menu can be split into separate sections by using Dividers. Group relevant menu items together to help the user understand the options quickly. A Divider can be used alone. If using a Menu Section Header, a divider will be automatically included. Try not to group single menu items together. An exception to this is a critical destructive menu item, like 'Delete', which can be separated from other menu items.

\`\`\`js
import { MenuSectionHeader } from '@dhis2/ui'
\`\`\`
`;
export default {
  title: 'Menu Section Header',
  component: MenuSectionHeader,
  args: {
    label: 'Section header'
  },
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Template = _ref => {
  let {
    menuArgs,
    ...args
  } = _ref;
  return /*#__PURE__*/React.createElement(Menu, menuArgs, /*#__PURE__*/React.createElement(MenuItem, {
    label: "Menu item above"
  }), /*#__PURE__*/React.createElement(MenuSectionHeader, args), /*#__PURE__*/React.createElement(MenuItem, {
    label: "Menu item below"
  }));
};
Template.propTypes = {
  menuArgs: PropTypes.shape()
};
export const Default = Template.bind({});
export const Dense = Template.bind({});
Dense.args = {
  menuArgs: {
    dense: true
  },
  dense: true
};
export const WithoutDivider = Template.bind({});
WithoutDivider.args = {
  hideDivider: true
};
export const TopOfList = args => /*#__PURE__*/React.createElement(Menu, null, /*#__PURE__*/React.createElement(MenuSectionHeader, args), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: "Item 2"
}));
TopOfList.args = {
  label: 'Top of list (so <Menu> hides divider)'
};
TopOfList.parameters = {
  docs: {
    description: {
      story: 'When the Section Header is the first child of a `<Menu>`, the Menu parent automatically applies the `hideDivider` prop.'
    }
  }
};
