// @dhis2-ui/header-bar/build/es/__e2e__/stories/with-debug-info-edge-cases.js
import React from 'react';
import { HeaderBar } from '../../header-bar.js';
import { createDecoratorCustomDataProviderHeaderBar, createDecoratorProvider, providerConfig } from './common.js';
export const WithUnknownInstanceVersion = () => /*#__PURE__*/React.createElement(HeaderBar, null);
WithUnknownInstanceVersion.decorators = [createDecoratorCustomDataProviderHeaderBar(), createDecoratorProvider({
  ...providerConfig,
  systemInfo: {
    ...providerConfig.systemInfo,
    version: undefined
  }
})];
export const WithUnknownAppVersion = () => /*#__PURE__*/React.createElement(HeaderBar, null);
WithUnknownAppVersion.decorators = [createDecoratorCustomDataProviderHeaderBar(), createDecoratorProvider({
  ...providerConfig,
  appVersion: undefined
})];
export const WithUnknownAppName = () => /*#__PURE__*/React.createElement(HeaderBar, null);
WithUnknownAppName.decorators = [createDecoratorCustomDataProviderHeaderBar(), createDecoratorProvider({
  ...providerConfig,
  appName: undefined
})];
export const WithUnknownAppNameAndVersion = () => /*#__PURE__*/React.createElement(HeaderBar, null);
WithUnknownAppNameAndVersion.decorators = [createDecoratorCustomDataProviderHeaderBar(), createDecoratorProvider({
  ...providerConfig,
  appName: undefined,
  appVersion: undefined
})];
