// @dhis2-ui/organisation-unit-tree/build/cjs/organisation-unit-node/label/disabled-selection-label.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DisabledSelectionLabel = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _singleSelectionLabel = require("./single-selection-label.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/**
 * @param {Object} props
 * @param {string} props.label
 * @param {Function} props.onToggleOpen
 * @param {bool} [props.loading]
 * @returns {React.Component}
 */
const DisabledSelectionLabel = _ref => {
  let {
    children,
    loading,
    onToggleOpen
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_singleSelectionLabel.SingleSelectionLabel, {
    checked: false,
    loading: loading,
    onChange: onToggleOpen
  }, children);
};
exports.DisabledSelectionLabel = DisabledSelectionLabel;
DisabledSelectionLabel.propTypes = {
  children: _propTypes.default.any.isRequired,
  onToggleOpen: _propTypes.default.func.isRequired,
  loading: _propTypes.default.bool
};
