// @dhis2-ui/label/build/cjs/label.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Label = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _required = require("@dhis2-ui/required");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const styles = [`label.jsx-2904627559{display:block;box-sizing:border-box;font-size:14px;line-height:19px;color:${_uiConstants.colors.grey900};padding:0;}`, ".disabled.jsx-2904627559{cursor:not-allowed;}"];
styles.__hash = "2904627559";
const constructClassName = _ref => {
  let {
    disabled,
    className
  } = _ref;
  return (0, _classnames.default)(className, {
    disabled: disabled
  });
};
const Label = _ref2 => {
  let {
    htmlFor,
    children,
    required,
    disabled,
    className,
    dataTest = 'dhis2-uicore-label'
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement("label", {
    htmlFor: htmlFor,
    "data-test": dataTest,
    className: `jsx-${styles.__hash}` + " " + (constructClassName({
      className,
      disabled
    }) || "")
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: `jsx-${styles.__hash}`
  }, children), required && /*#__PURE__*/_react.default.createElement(_required.Required, {
    dataTest: `${dataTest}-required`
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: styles.__hash
  }, styles));
};
exports.Label = Label;
Label.propTypes = {
  children: _propTypes.default.string,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  disabled: _propTypes.default.bool,
  htmlFor: _propTypes.default.string,
  required: _propTypes.default.bool
};
