// @dhis2-ui/select/build/cjs/multi-select-option/multi-select-option.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MultiSelectOption = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _uiConstants = require("@dhis2/ui-constants");
var _checkbox = require("@dhis2-ui/checkbox");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
// Padding has to be set on the label, so that the entire area is clickable
const {
  styles,
  className: checkboxClassname
} = {
  styles: /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2013783366"
  }, [`.jsx-2013783366{padding:${_uiConstants.spacers.dp8} ${_uiConstants.spacers.dp12};}`]),
  className: "jsx-2013783366"
};
const MultiSelectOption = _ref => {
  let {
    label,
    active,
    disabled,
    onClick,
    className,
    dataTest = 'dhis2-uicore-multiselectoption',
    value
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    "data-value": value,
    "data-label": label,
    className: _style.default.dynamic([["3813980210", [_uiConstants.colors.grey200]]]) + " " + ((0, _classnames.default)(className, {
      disabled
    }) || "")
  }, /*#__PURE__*/_react.default.createElement(_checkbox.Checkbox, {
    name: label,
    className: checkboxClassname,
    checked: active,
    label: label,
    onChange: onClick,
    disabled: disabled,
    dense: true
  }), styles, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3813980210",
    dynamic: [_uiConstants.colors.grey200]
  }, [`div.__jsx-style-dynamic-selector:hover{background-color:${_uiConstants.colors.grey200};}`, "div.disabled.__jsx-style-dynamic-selector:hover{background-color:initial;}"]));
};
exports.MultiSelectOption = MultiSelectOption;
MultiSelectOption.propTypes = {
  label: _propTypes.default.string.isRequired,
  value: _propTypes.default.string.isRequired,
  active: _propTypes.default.bool,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  disabled: _propTypes.default.bool,
  onClick: _propTypes.default.func
};
