// @dhis2-ui/field/build/es/field-group/field-group.prod.stories.js
import { sharedPropTypes } from '@dhis2/ui-constants';
import { Checkbox } from '@dhis2-ui/checkbox';
import { Radio } from '@dhis2-ui/radio';
import { Switch } from '@dhis2-ui/switch';
import React from 'react';
import { FieldGroup } from './field-group.js';
const description = `
Wraps a group of form components like Radios, Checkboxes, or Switches. The FieldGroup wraps the form controls in a FieldSet and a Field component to group them and add a label, help text, and/or validation text.

\`\`\`js
import { FieldGroup } from '@dhis2/ui'
\`\`\`
`;
export default {
  title: 'Field Group',
  component: FieldGroup,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    valid: {
      ...sharedPropTypes.statusArgType
    },
    warning: {
      ...sharedPropTypes.statusArgType
    },
    error: {
      ...sharedPropTypes.statusArgType
    }
  }
};
export const WithCheckbox = args => /*#__PURE__*/React.createElement(FieldGroup, args, /*#__PURE__*/React.createElement(Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "third",
  label: "Third"
}));
export const WithRadio = args => /*#__PURE__*/React.createElement(FieldGroup, args, /*#__PURE__*/React.createElement(Radio, {
  value: "first",
  label: "First"
}), /*#__PURE__*/React.createElement(Radio, {
  value: "second",
  label: "Second",
  checked: true
}), /*#__PURE__*/React.createElement(Radio, {
  value: "third",
  label: "Third"
}), /*#__PURE__*/React.createElement(Radio, {
  value: "fourth",
  label: "Fourth"
}));
export const WithSwitch = args => /*#__PURE__*/React.createElement(FieldGroup, args, /*#__PURE__*/React.createElement(Switch, {
  value: "first",
  label: "First"
}), /*#__PURE__*/React.createElement(Switch, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/React.createElement(Switch, {
  value: "third",
  label: "Third"
}), /*#__PURE__*/React.createElement(Switch, {
  value: "fourth",
  label: "Fourth"
}));
export const WithLabel = args => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(FieldGroup, args, /*#__PURE__*/React.createElement(Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "third",
  label: "Third"
})), /*#__PURE__*/React.createElement(FieldGroup, {
  label: "I am a required field",
  required: true
}, /*#__PURE__*/React.createElement(Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "third",
  label: "Third"
})));
WithLabel.args = {
  label: 'I am a label/legend'
};
export const HelpAndValidationTexts = args => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(FieldGroup, args, /*#__PURE__*/React.createElement(Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "third",
  label: "Third"
})), /*#__PURE__*/React.createElement(FieldGroup, {
  label: "I am a legend",
  helpText: "I am disabled",
  disabled: true
}, /*#__PURE__*/React.createElement(Checkbox, {
  value: "first",
  label: "First",
  disabled: true
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "second",
  label: "Second",
  disabled: true
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "third",
  label: "Third",
  disabled: true
})), /*#__PURE__*/React.createElement(FieldGroup, {
  label: "I am a legend",
  valid: true,
  validationText: "I am valid"
}, /*#__PURE__*/React.createElement(Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "second",
  label: "Second",
  checked: true
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "third",
  label: "Third",
  checked: true
})), /*#__PURE__*/React.createElement(FieldGroup, {
  label: "I am a legend",
  name: "warning",
  warning: true,
  validationText: "I have a warning"
}, /*#__PURE__*/React.createElement(Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "third",
  label: "Third"
})), /*#__PURE__*/React.createElement(FieldGroup, {
  label: "I am a legend",
  error: true,
  validationText: "I have an error"
}, /*#__PURE__*/React.createElement(Checkbox, {
  value: "first",
  label: "First"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "second",
  label: "Second"
}), /*#__PURE__*/React.createElement(Checkbox, {
  value: "third",
  label: "Third"
})));
HelpAndValidationTexts.args = {
  label: 'I am a field',
  helpText: 'I am help text!'
};
