// @dhis2-ui/header-bar/build/cjs/debug-info/debug-info-modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DebugInfoModal = DebugInfoModal;
var _appRuntime = require("@dhis2/app-runtime");
var _button = require("@dhis2-ui/button");
var _modal = require("@dhis2-ui/modal");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = _interopRequireDefault(require("../locales/index.js"));
var _debugInfoTable = require("./debug-info-table.js");
var _useDebugInfo = require("./use-debug-info.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function DebugInfoModal(_ref) {
  let {
    onClose
  } = _ref;
  const debugInfo = (0, _useDebugInfo.useFormattedDebugInfo)();
  const {
    show: showClipboardAlert
  } = (0, _appRuntime.useAlert)('Debug information copied to clipboard', {
    duration: 3000
  });
  const copyDebugInfo = () => {
    navigator.clipboard.writeText(debugInfo);
    onClose();
    showClipboardAlert();
  };
  return /*#__PURE__*/_react.default.createElement(_modal.Modal, {
    position: "middle",
    dataTest: "dhis2-ui-headerbar-debuginfomodal"
  }, /*#__PURE__*/_react.default.createElement(_modal.ModalTitle, null, _index.default.t('Debug info')), /*#__PURE__*/_react.default.createElement(_modal.ModalContent, null, /*#__PURE__*/_react.default.createElement(_debugInfoTable.DebugInfoTable, null)), /*#__PURE__*/_react.default.createElement(_modal.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
    end: true
  }, /*#__PURE__*/_react.default.createElement(_button.Button, {
    onClick: () => onClose()
  }, _index.default.t('Close')), /*#__PURE__*/_react.default.createElement(_button.Button, {
    primary: true,
    onClick: copyDebugInfo,
    dataTest: "dhis2-ui-headerbar-debuginfomodal-copybutton"
  }, _index.default.t('Copy debug info')))));
}
DebugInfoModal.propTypes = {
  onClose: _propTypes.default.func.isRequired
};
