// @dhis2-ui/menu/build/cjs/flyout-menu/flyout-menu.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithVariousChildren = exports.WithSubMenus = exports.WithCustomMenuItem = exports.MaxWidth = exports.MaxHeight = exports.DropDownMenu = exports.Dense = exports.Default = void 0;
var _uiIcons = require("@dhis2/ui-icons");
var _layer = require("@dhis2-ui/layer");
var _menu = require("@dhis2-ui/menu");
var _popper = require("@dhis2-ui/popper");
var _react = _interopRequireWildcard(require("react"));
var _flyoutMenu = require("./flyout-menu.js");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const description = `
Use menus to provide access to options and actions where space is limited and displaying all the options would be impractical. For example, providing access to a range of actions for every dashboard item displayed. Containing all those actions in menus keeps the page manageable.

The menu component is flexible in where it can be used and its contents can be flexible too. However, the most common use case is a menu containing menu items.

Make sure the menu item labels are short and easy to understand. One word is often enough to describe an action or option. Do not use sentences as labels. Some examples of good menu item labels:

- "Save"
- "Open as map"
- "Export PDF"
- "Duplicate"

See more about how to use menus at the [design system](https://github.com/dhis2/design-system/blob/master/molecules/menu.md).

\`\`\`js
import { FlyoutMenu } from 'dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Flyout Menu',
  component: _flyoutMenu.FlyoutMenu,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
const Default = args => /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, args, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}));
exports.Default = Default;
const Dense = exports.Dense = Default.bind({});
Dense.args = {
  dense: true
};
const MaxHeight = args => /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, args, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 3"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 5"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 6"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 7"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 8"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 9"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 10"
}));
exports.MaxHeight = MaxHeight;
MaxHeight.args = {
  maxHeight: '250px'
};
const MaxWidth = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, null, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Short item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Short item 2"
})), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, null, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1 - with a lot of text and using a default maxWidth value of 380px"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 - with a lot of text and using a default maxWidth value of 380px"
})), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, args, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: `Item 1 - with a lot of text and using a custom maxWidth value of
                ${args.maxWidth}`
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: `Item 2 - with a lot of text and using a custom maxWidth value of
                ${args.maxWidth}`
})));
exports.MaxWidth = MaxWidth;
MaxWidth.args = {
  maxWidth: '300px'
};
const WithSubMenus = args => /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, args, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 a"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b i"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b ii"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 c"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 3"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 a"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b i"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b ii"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 c"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 5"
}));
exports.WithSubMenus = WithSubMenus;
WithSubMenus.parameters = {
  docs: {
    description: {
      story: 'See this demo in the Canvas tab for proper alignment of sub menus.'
    }
  }
};
const WithVariousChildren = args => /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, args, /*#__PURE__*/_react.default.createElement(_menu.MenuSectionHeader, {
  label: "Section with sub-menus"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 a"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b i"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b ii"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 c"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 3"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 a"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b i"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b ii"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 c"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 5"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuSectionHeader, {
  label: "Section with dividers between menu items"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuDivider, null), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuDivider, null), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 3"
}));
exports.WithVariousChildren = WithVariousChildren;
WithVariousChildren.parameters = {
  docs: {
    description: {
      story: 'See this demo in the Canvas tab for proper alignment of sub menus.'
    }
  }
};
const DropDownMenu = args => {
  const ref = (0, _react.useRef)();
  const [open, setOpen] = (0, _react.useState)(false);
  const toggle = () => setOpen(!open);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
    ref: ref,
    onClick: toggle
  }, "Open menu \xA0\xA0 ", /*#__PURE__*/_react.default.createElement(_uiIcons.IconChevronDown16, null)), open && /*#__PURE__*/_react.default.createElement(_layer.Layer, {
    onBackdropClick: toggle
  }, /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    reference: ref,
    placement: "bottom-start"
  }, /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, _extends({}, args, {
    closeMenu: toggle
  }), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    label: "Item 1"
  }), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    label: "Item 2"
  })))));
};
exports.DropDownMenu = DropDownMenu;
DropDownMenu.parameters = {
  docs: {
    description: {
      story: 'View this demo in the canvas to see the dropdown menu.'
    },
    source: {
      type: 'code'
    }
  }
};
const WithCustomMenuItem = args => {
  // You should not create custom components in the render cycle
  // this is just for demo purposes
  const PopupWindowMenuItem = _ref => {
    let {
      to,
      children,
      ...rest
    } = _ref;
    const HEIGHT = 1000;
    const WIDTH = 1400;
    const centerY = (window.screen.height - HEIGHT) / 2;
    const centerX = (window.screen.width - WIDTH) / 2;
    const onClick = () => window.open(to, 'Popup', ['menubar=no', 'location=no', 'resizable=no', 'scrollbars=no', 'status=no', `width=${WIDTH}`, `height=${HEIGHT}`, `top=${centerY}`, `left=${centerX}`].join());
    return /*#__PURE__*/_react.default.createElement(_menu.MenuItem, _extends({
      onClick: onClick,
      label: children
    }, rest));
  };
  return /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, args, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    label: "A normal menu item"
  }), /*#__PURE__*/_react.default.createElement(PopupWindowMenuItem, {
    to: "http://dhis2.org"
  }, "A custom menu item (popup window)"));
};
exports.WithCustomMenuItem = WithCustomMenuItem;
