// @dhis2-ui/menu/build/cjs/flyout-menu/flyout-menu.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithChildren = exports.TogglesSubMenus = exports.ShiftIntoView = exports.FlippedPosition = exports.DefaultPosition = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _menu = require("@dhis2-ui/menu");
var _react = _interopRequireDefault(require("react"));
var _flyoutMenu = require("./flyout-menu.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'FlyoutMenu',
  component: _flyoutMenu.FlyoutMenu
};
const MenuItemSubMenuPositions = () => /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, null, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}, /*#__PURE__*/_react.default.createElement("div", {
  style: {
    padding: 30
  }
}, "SubMenu 1")));
const WithChildren = () => /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, null, "I am a child");
exports.WithChildren = WithChildren;
const TogglesSubMenus = () => /*#__PURE__*/_react.default.createElement(_flyoutMenu.FlyoutMenu, null, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}, "SubMenu 1"), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}, "SubMenu 2"));
exports.TogglesSubMenus = TogglesSubMenus;
const DefaultPosition = () => /*#__PURE__*/_react.default.createElement(MenuItemSubMenuPositions, null);
exports.DefaultPosition = DefaultPosition;
const FlippedPosition = () => /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-2297277891"
}, /*#__PURE__*/_react.default.createElement(MenuItemSubMenuPositions, null), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "2297277891"
}, ["div.jsx-2297277891{text-align:right;}"]));
exports.FlippedPosition = FlippedPosition;
const ShiftIntoView = () => /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-3052292650"
}, /*#__PURE__*/_react.default.createElement(MenuItemSubMenuPositions, null), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "3052292650"
}, ["html,body,#root{position:relative;width:200px;max-width:200px;overflow:hidden;}", "#root{border:3px dashed grey;}"]));
exports.ShiftIntoView = ShiftIntoView;
