// @dhis2-ui/css/build/es/css-variables/css-variables.prod.stories.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React from 'react';
import { CssVariables } from './index.js';
const description = `
A utility for adding DHIS2 theme variables to global CSS variables.

\`\`\`js
import { CssVariables } from '@dhis2/ui'
\`\`\`
`;
const App = _ref => {
  let {
    children
  } = _ref;
  return /*#__PURE__*/React.createElement("div", null, children);
};
export default {
  title: 'CSS Variables',
  component: CssVariables,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
export const AllVariables = args => /*#__PURE__*/React.createElement(App, null, /*#__PURE__*/React.createElement(CssVariables, args), /*#__PURE__*/React.createElement("p", null, "The sections of the theme that should be inserted can be toggled with flags, which allows the theme variables to be used as regular CSS custom properties. So this", ' ', /*#__PURE__*/React.createElement("span", {
  style: {
    color: 'var(--colors-red500)'
  }
}, "text"), " uses the vanilla CSS", ' ', /*#__PURE__*/React.createElement("span", {
  style: {
    color: 'var(--colors-blue500)'
  }
}, "custom properties"), ' ', "set by the CssVariables component."));
AllVariables.args = {
  colors: true,
  theme: true,
  layers: true,
  spacers: true,
  elevations: true
};
export const NoVariables = args => /*#__PURE__*/React.createElement(App, null, /*#__PURE__*/React.createElement(CssVariables, args), /*#__PURE__*/React.createElement("p", null, "By default no custom properties are inserted."));
export const Elevations = args => /*#__PURE__*/React.createElement(App, null, /*#__PURE__*/React.createElement(CssVariables, _extends({
  elevations: true
}, args)), /*#__PURE__*/React.createElement("div", {
  style: {
    display: 'flex',
    gap: '8px'
  }
}, /*#__PURE__*/React.createElement("div", {
  style: {
    background: 'white',
    boxShadow: 'var(--elevations-e100',
    display: 'inline-block',
    padding: '16px',
    marginRight: '16px',
    fontFamily: 'monospace'
  }
}, "e100"), /*#__PURE__*/React.createElement("div", {
  style: {
    background: 'white',
    boxShadow: 'var(--elevations-e200',
    display: 'inline-block',
    padding: '16px',
    marginRight: '16px',
    fontFamily: 'monospace'
  }
}, "e200"), /*#__PURE__*/React.createElement("div", {
  style: {
    background: 'white',
    boxShadow: 'var(--elevations-e300',
    display: 'inline-block',
    padding: '16px',
    marginRight: '16px',
    fontFamily: 'monospace'
  }
}, "e300"), /*#__PURE__*/React.createElement("div", {
  style: {
    background: 'white',
    boxShadow: 'var(--elevations-e400',
    display: 'inline-block',
    padding: '16px',
    marginRight: '16px',
    fontFamily: 'monospace'
  }
}, "e400")));
