// @dhis2-ui/css/build/es/css-variables/css-variables.e2e.stories.js
import React from 'react';
import { CssVariables } from './index.js';
export default {
  title: 'CssVariables'
};
export const WithColors = () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(CssVariables, {
  colors: true
}), /*#__PURE__*/React.createElement("div", {
  id: "custom",
  style: {
    backgroundColor: 'var(--colors-blue900)'
  }
}));
export const WithTheme = () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(CssVariables, {
  theme: true
}), /*#__PURE__*/React.createElement("div", {
  id: "custom",
  style: {
    backgroundColor: 'var(--theme-primary900)'
  }
}));
export const WithLayers = () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(CssVariables, {
  layers: true
}), /*#__PURE__*/React.createElement("div", {
  id: "custom",
  style: {
    zIndex: 'var(--layers-alert)',
    position: 'absolute'
  }
}));
export const WithSpacers = () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(CssVariables, {
  spacers: true
}), /*#__PURE__*/React.createElement("div", {
  id: "custom",
  style: {
    margin: 'var(--spacers-dp4)'
  }
}));
export const WithElevations = () => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(CssVariables, {
  elevations: true
}), /*#__PURE__*/React.createElement("div", {
  id: "custom",
  style: {
    boxShadow: 'var(--elevations-e100)'
  }
}));
