// @dhis2-ui/calendar/build/cjs/stories/calendar.prod.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithAnyCalendar = exports.Nepali = exports.Ethiopic = exports.Basic = void 0;
var _react = _interopRequireDefault(require("react"));
var _calendar = require("../calendar/calendar.js");
var _calendarStoryWrapper = require("./calendar-story-wrapper.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const subtitle = `[Experimental] Calendar component is useful for creating a variety of calendars including Ethiopic, Islamic etc..`;
const description = `
Use a Calendar where there is a need to let the user select a date.

Note that it requires a parent, like [Box](../?path=/docs/layout-box--default), to define its size.

\`\`\`js
import { Calendar } from '@dhis2/ui'
\`\`\`
`;
var _default = exports.default = {
  title: 'Calendar',
  component: _calendar.Calendar,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    calendar: {
      control: 'select',
      options: ['gregory', 'islamic', 'nepali', 'ethiopic', 'persian', 'indian']
    },
    weekDayFormat: {
      control: 'select',
      options: ['long', 'short', 'narrow']
    },
    locale: {
      control: 'text'
    },
    numberingSystem: {
      control: 'select',
      options: ['latn', 'arab', 'ethi']
    }
  }
};
const Basic = args => {
  return /*#__PURE__*/_react.default.createElement(_calendar.Calendar, args);
};
exports.Basic = Basic;
Basic.args = {
  onDateSelect: date => console.log(date),
  calendar: 'gregory',
  weekDayFormat: 'narrow',
  locale: 'en'
};
const Ethiopic = args => {
  return /*#__PURE__*/_react.default.createElement(_calendarStoryWrapper.CalendarStoryWrapper, _extends({
    calendar: "ethiopic",
    locale: "am-et",
    numberingSystem: "ethi",
    date: "2014-04-05" // 13th of October 2022
    ,
    timeZone: "Africa/Khartoum"
  }, args));
};
exports.Ethiopic = Ethiopic;
const Nepali = args => {
  return /*#__PURE__*/_react.default.createElement(_calendarStoryWrapper.CalendarStoryWrapper, _extends({
    calendar: "nepali",
    locale: "ne-NP",
    date: "2079-06-29" // 13th of October 2022
    ,
    timeZone: "Africa/Khartoum"
  }, args));
};
exports.Nepali = Nepali;
const WithAnyCalendar = args => {
  return /*#__PURE__*/_react.default.createElement(_calendarStoryWrapper.CalendarStoryWrapper, _extends({
    calendar: "gregory",
    locale: "en-GB"
  }, args));
};
exports.WithAnyCalendar = WithAnyCalendar;
