// @dhis2-ui/table/build/cjs/data-table/data-table.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.FixedHeaderAndColumns = void 0;
var _react = _interopRequireDefault(require("react"));
var _dataTableBody = require("./data-table-body.js");
var _dataTableCell = require("./data-table-cell.js");
var _dataTableColumnHeader = require("./data-table-column-header/data-table-column-header.js");
var _dataTableHead = require("./data-table-head.js");
var _dataTableRow = require("./data-table-row/data-table-row.js");
var _dataTable = require("./data-table.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'DataTable',
  component: _dataTable.DataTable
};
const FixedHeaderAndColumns = () => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, {
  layout: "fixed",
  width: "1000px",
  scrollWidth: "500px",
  scrollHeight: "400px"
}, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "Header fixed-left 1"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Header fixed-left 2"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Header fixed"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "scrolling top-left"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "scrolling top-right")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "...")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "...")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "...")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "...")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "...")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "...")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "...")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "...")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "fixed"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "scrolling bottom-left"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "..."), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "scrolling bottom-right"))));
exports.FixedHeaderAndColumns = FixedHeaderAndColumns;
