// @dhis2-ui/table/build/cjs/data-table/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "DataTable", {
  enumerable: true,
  get: function () {
    return _dataTable.DataTable;
  }
});
Object.defineProperty(exports, "DataTableBody", {
  enumerable: true,
  get: function () {
    return _dataTableBody.DataTableBody;
  }
});
Object.defineProperty(exports, "DataTableCell", {
  enumerable: true,
  get: function () {
    return _dataTableCell.DataTableCell;
  }
});
Object.defineProperty(exports, "DataTableColumnHeader", {
  enumerable: true,
  get: function () {
    return _dataTableColumnHeader.DataTableColumnHeader;
  }
});
Object.defineProperty(exports, "DataTableFoot", {
  enumerable: true,
  get: function () {
    return _dataTableFoot.DataTableFoot;
  }
});
Object.defineProperty(exports, "DataTableHead", {
  enumerable: true,
  get: function () {
    return _dataTableHead.DataTableHead;
  }
});
Object.defineProperty(exports, "DataTableRow", {
  enumerable: true,
  get: function () {
    return _dataTableRow.DataTableRow;
  }
});
Object.defineProperty(exports, "DataTableToolbar", {
  enumerable: true,
  get: function () {
    return _dataTableToolbar.DataTableToolbar;
  }
});
var _dataTable = require("./data-table.js");
var _dataTableBody = require("./data-table-body.js");
var _dataTableCell = require("./data-table-cell.js");
var _dataTableColumnHeader = require("./data-table-column-header/data-table-column-header.js");
var _dataTableFoot = require("./data-table-foot.js");
var _dataTableHead = require("./data-table-head.js");
var _dataTableRow = require("./data-table-row/data-table-row.js");
var _dataTableToolbar = require("./data-table-toolbar.js");
