// @dhis2-ui/select/build/cjs/multi-select/input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Input = void 0;
var _style = _interopRequireDefault(require("styled-jsx/style"));
var _propTypes = require("@dhis2/prop-types");
var _uiConstants = require("@dhis2/ui-constants");
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes2 = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _index = require("../select/index.js");
var _selectionList = require("./selection-list.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Input = _ref => {
  let {
    selected,
    onChange,
    clearable,
    clearText,
    placeholder,
    dataTest,
    prefix,
    options,
    className,
    disabled,
    inputMaxHeight = '100px'
  } = _ref;
  const hasSelection = selected.length > 0;
  const onClear = e => {
    const data = {
      selected: []
    };
    e.stopPropagation();
    onChange(data, e);
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["64009541", [_uiConstants.colors.grey900]], ["1331770409", [inputMaxHeight]]]) + " " + ((0, _classnames.default)('root', className) || "")
  }, /*#__PURE__*/_react.default.createElement(_index.InputPrefix, {
    prefix: prefix,
    dataTest: `${dataTest}-prefix`
  }), !hasSelection && !prefix && /*#__PURE__*/_react.default.createElement(_index.InputPlaceholder, {
    placeholder: placeholder,
    dataTest: `${dataTest}-placeholder`
  }), hasSelection && /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["64009541", [_uiConstants.colors.grey900]], ["1331770409", [inputMaxHeight]]]) + " " + "root-input"
  }, /*#__PURE__*/_react.default.createElement(_selectionList.SelectionList, {
    selected: selected,
    onChange: onChange,
    options: options,
    disabled: disabled
  })), hasSelection && clearable && !disabled && /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["64009541", [_uiConstants.colors.grey900]], ["1331770409", [inputMaxHeight]]]) + " " + "root-right"
  }, /*#__PURE__*/_react.default.createElement(_index.InputClearButton, {
    onClear: onClear,
    clearText: clearText,
    dataTest: `${dataTest}-clear`
  })), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "64009541",
    dynamic: [_uiConstants.colors.grey900]
  }, [`.root.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;color:${_uiConstants.colors.grey900};font-size:14px;line-height:16px;}`, ".root-input.__jsx-style-dynamic-selector{overflow-y:auto;-webkit-flex:1;-ms-flex:1;flex:1;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-flow:wrap;-ms-flex-flow:wrap;flex-flow:wrap;gap:4px;}", ".root-right.__jsx-style-dynamic-selector{margin-inline-start:auto;}"]), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "1331770409",
    dynamic: [inputMaxHeight]
  }, [`.root-input.__jsx-style-dynamic-selector{max-height:${inputMaxHeight};}`]));
};
exports.Input = Input;
Input.propTypes = {
  dataTest: _propTypes2.default.string.isRequired,
  className: _propTypes2.default.string,
  clearText: (0, _propTypes.requiredIf)(props => props.clearable, _propTypes2.default.string),
  clearable: _propTypes2.default.bool,
  disabled: _propTypes2.default.bool,
  inputMaxHeight: _propTypes2.default.string,
  options: _propTypes2.default.node,
  placeholder: _propTypes2.default.string,
  prefix: _propTypes2.default.string,
  selected: _propTypes2.default.arrayOf(_propTypes2.default.string),
  onChange: _propTypes2.default.func
};
