// @dhis2/ui-forms/build/cjs/TextAreaFieldFF/TextAreaFieldFF.e2e.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Required = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactFinalForm = require("react-final-form");
var _formDecorator = require("../formDecorator.js");
var _index = require("../validators/index.js");
var _TextAreaFieldFF = require("./TextAreaFieldFF.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = {
  title: 'TextArea',
  decorators: [_formDecorator.formDecorator]
};
const Default = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _TextAreaFieldFF.TextAreaFieldFF,
  name: "comment",
  label: "Do you have any comments?"
});
exports.Default = Default;
const Required = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  required: true,
  name: "comment",
  component: _TextAreaFieldFF.TextAreaFieldFF,
  validate: _index.hasValue,
  label: "Do you have any comments?"
});
exports.Required = Required;
